# coding: utf-8

"""
    IGVF Project API

    No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)

    The version of the OpenAPI document: 52.0.2
    Generated by OpenAPI Generator (https://openapi-generator.tech)

    Do not edit the class manually.
"""  # noqa: E501

import warnings
from pydantic import validate_call, Field, StrictFloat, StrictStr, StrictInt
from typing import Any, Dict, List, Optional, Tuple, Union
from typing_extensions import Annotated

from pydantic import Field, StrictBool, StrictBytes, StrictFloat, StrictInt, StrictStr, field_validator
from typing import Any, Dict, List, Optional, Union
from typing_extensions import Annotated
from igvf_client.models.access_key_results import AccessKeyResults
from igvf_client.models.alignment_file_results import AlignmentFileResults
from igvf_client.models.analysis_set_results import AnalysisSetResults
from igvf_client.models.analysis_step_results import AnalysisStepResults
from igvf_client.models.analysis_step_version_results import AnalysisStepVersionResults
from igvf_client.models.assay_term_results import AssayTermResults
from igvf_client.models.auxiliary_set_results import AuxiliarySetResults
from igvf_client.models.award_results import AwardResults
from igvf_client.models.biomarker_results import BiomarkerResults
from igvf_client.models.configuration_file_results import ConfigurationFileResults
from igvf_client.models.construct_library_set_results import ConstructLibrarySetResults
from igvf_client.models.crispr_modification_results import CrisprModificationResults
from igvf_client.models.curated_set_results import CuratedSetResults
from igvf_client.models.degron_modification_results import DegronModificationResults
from igvf_client.models.document_results import DocumentResults
from igvf_client.models.gene_location import GeneLocation
from igvf_client.models.gene_results import GeneResults
from igvf_client.models.genome_browser_annotation_file_results import GenomeBrowserAnnotationFileResults
from igvf_client.models.human_donor_results import HumanDonorResults
from igvf_client.models.image_file_results import ImageFileResults
from igvf_client.models.image_results import ImageResults
from igvf_client.models.in_vitro_system_results import InVitroSystemResults
from igvf_client.models.institutional_certificate_results import InstitutionalCertificateResults
from igvf_client.models.item import Item
from igvf_client.models.item_type import ItemType
from igvf_client.models.lab_results import LabResults
from igvf_client.models.locus import Locus
from igvf_client.models.matrix_file_results import MatrixFileResults
from igvf_client.models.measurement_set_results import MeasurementSetResults
from igvf_client.models.model_file_results import ModelFileResults
from igvf_client.models.model_set_results import ModelSetResults
from igvf_client.models.multiplexed_sample_results import MultiplexedSampleResults
from igvf_client.models.open_reading_frame_results import OpenReadingFrameResults
from igvf_client.models.page_results import PageResults
from igvf_client.models.phenotype_term_results import PhenotypeTermResults
from igvf_client.models.phenotypic_feature_results import PhenotypicFeatureResults
from igvf_client.models.platform_term_results import PlatformTermResults
from igvf_client.models.prediction_set_results import PredictionSetResults
from igvf_client.models.primary_cell_results import PrimaryCellResults
from igvf_client.models.publication_results import PublicationResults
from igvf_client.models.reference_file_results import ReferenceFileResults
from igvf_client.models.rodent_donor_results import RodentDonorResults
from igvf_client.models.sample_term_results import SampleTermResults
from igvf_client.models.search_results import SearchResults
from igvf_client.models.sequence_file_results import SequenceFileResults
from igvf_client.models.signal_file_results import SignalFileResults
from igvf_client.models.software_results import SoftwareResults
from igvf_client.models.software_version_results import SoftwareVersionResults
from igvf_client.models.source_results import SourceResults
from igvf_client.models.tabular_file_results import TabularFileResults
from igvf_client.models.technical_sample_results import TechnicalSampleResults
from igvf_client.models.tissue_results import TissueResults
from igvf_client.models.treatment_results import TreatmentResults
from igvf_client.models.user_results import UserResults
from igvf_client.models.whole_organism_results import WholeOrganismResults
from igvf_client.models.workflow_results import WorkflowResults

from igvf_client.api_client import ApiClient, RequestSerialized
from igvf_client.api_response import ApiResponse
from igvf_client.rest import RESTResponseType


class IgvfApi:
    """NOTE: This class is auto generated by OpenAPI Generator
    Ref: https://openapi-generator.tech

    Do not edit the class manually.
    """

    def __init__(self, api_client=None) -> None:
        if api_client is None:
            api_client = ApiClient.get_default()
        self.api_client = api_client


    @validate_call
    def access_keys(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        access_key_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by access_key_id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        secret_access_key_hash: Annotated[Optional[List[StrictStr]], Field(description="Filter by secret_access_key_hash")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        user: Annotated[Optional[List[StrictStr]], Field(description="Filter by user")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AccessKeyResults:
        """List items in the AccessKey collection.

        Collection endpoint that accepts various query parameters to filter and sort AccessKey items. Supports filtering on fields within AccessKey items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param access_key_id: Filter by access_key_id
        :type access_key_id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param secret_access_key_hash: Filter by secret_access_key_hash
        :type secret_access_key_hash: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param user: Filter by user
        :type user: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._access_keys_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            access_key_id=access_key_id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            notes=notes,
            secret_access_key_hash=secret_access_key_hash,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            user=user,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AccessKeyResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def access_keys_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        access_key_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by access_key_id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        secret_access_key_hash: Annotated[Optional[List[StrictStr]], Field(description="Filter by secret_access_key_hash")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        user: Annotated[Optional[List[StrictStr]], Field(description="Filter by user")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AccessKeyResults]:
        """List items in the AccessKey collection.

        Collection endpoint that accepts various query parameters to filter and sort AccessKey items. Supports filtering on fields within AccessKey items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param access_key_id: Filter by access_key_id
        :type access_key_id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param secret_access_key_hash: Filter by secret_access_key_hash
        :type secret_access_key_hash: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param user: Filter by user
        :type user: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._access_keys_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            access_key_id=access_key_id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            notes=notes,
            secret_access_key_hash=secret_access_key_hash,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            user=user,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AccessKeyResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def access_keys_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        access_key_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by access_key_id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        secret_access_key_hash: Annotated[Optional[List[StrictStr]], Field(description="Filter by secret_access_key_hash")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        user: Annotated[Optional[List[StrictStr]], Field(description="Filter by user")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AccessKey collection.

        Collection endpoint that accepts various query parameters to filter and sort AccessKey items. Supports filtering on fields within AccessKey items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param access_key_id: Filter by access_key_id
        :type access_key_id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param secret_access_key_hash: Filter by secret_access_key_hash
        :type secret_access_key_hash: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param user: Filter by user
        :type user: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._access_keys_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            access_key_id=access_key_id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            notes=notes,
            secret_access_key_hash=secret_access_key_hash,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            user=user,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AccessKeyResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _access_keys_serialize(
        self,
        query,
        limit,
        sort,
        id,
        access_key_id,
        aliases,
        creation_timestamp,
        description,
        notes,
        secret_access_key_hash,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        user,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'access_key_id': 'multi',
            'aliases': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'notes': 'multi',
            'secret_access_key_hash': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'user': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if access_key_id is not None:
            
            _query_params.append(('access_key_id', access_key_id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if secret_access_key_hash is not None:
            
            _query_params.append(('secret_access_key_hash', secret_access_key_hash))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if user is not None:
            
            _query_params.append(('user', user))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/access-keys/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def alignment_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        filtered: Annotated[Optional[List[StrictBool]], Field(description="Filter by filtered")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        redacted: Annotated[Optional[List[StrictBool]], Field(description="Filter by redacted")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AlignmentFileResults:
        """List items in the AlignmentFile collection.

        Collection endpoint that accepts various query parameters to filter and sort AlignmentFile items. Supports filtering on fields within AlignmentFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param filtered: Filter by filtered
        :type filtered: List[bool]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param redacted: Filter by redacted
        :type redacted: List[bool]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._alignment_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            filtered=filtered,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            redacted=redacted,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AlignmentFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def alignment_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        filtered: Annotated[Optional[List[StrictBool]], Field(description="Filter by filtered")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        redacted: Annotated[Optional[List[StrictBool]], Field(description="Filter by redacted")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AlignmentFileResults]:
        """List items in the AlignmentFile collection.

        Collection endpoint that accepts various query parameters to filter and sort AlignmentFile items. Supports filtering on fields within AlignmentFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param filtered: Filter by filtered
        :type filtered: List[bool]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param redacted: Filter by redacted
        :type redacted: List[bool]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._alignment_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            filtered=filtered,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            redacted=redacted,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AlignmentFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def alignment_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        filtered: Annotated[Optional[List[StrictBool]], Field(description="Filter by filtered")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        redacted: Annotated[Optional[List[StrictBool]], Field(description="Filter by redacted")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AlignmentFile collection.

        Collection endpoint that accepts various query parameters to filter and sort AlignmentFile items. Supports filtering on fields within AlignmentFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param filtered: Filter by filtered
        :type filtered: List[bool]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param redacted: Filter by redacted
        :type redacted: List[bool]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._alignment_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            filtered=filtered,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            redacted=redacted,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AlignmentFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _alignment_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        anvil_url,
        assay_titles,
        assembly,
        award_id,
        award_component,
        collections,
        content_md5sum,
        content_summary,
        content_type,
        controlled_access,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        filtered,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        redacted,
        reference_files,
        release_timestamp,
        revoke_detail,
        s3_uri,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        transcriptome_annotation,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'anvil_url': 'multi',
            'assay_titles': 'multi',
            'assembly': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_summary': 'multi',
            'content_type': 'multi',
            'controlled_access': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'filtered': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'redacted': 'multi',
            'reference_files': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'transcriptome_annotation': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if anvil_url is not None:
            
            _query_params.append(('anvil_url', anvil_url))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if assembly is not None:
            
            _query_params.append(('assembly', assembly))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_summary is not None:
            
            _query_params.append(('content_summary', content_summary))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if controlled_access is not None:
            
            _query_params.append(('controlled_access', controlled_access))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if filtered is not None:
            
            _query_params.append(('filtered', filtered))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if redacted is not None:
            
            _query_params.append(('redacted', redacted))
            
        if reference_files is not None:
            
            _query_params.append(('reference_files', reference_files))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if transcriptome_annotation is not None:
            
            _query_params.append(('transcriptome_annotation', transcriptome_annotation))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/alignment-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def analysis_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_image_data_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by external_image_data_url")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        functional_assay_mechanisms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by functional_assay_mechanisms.@id")] = None,
        functional_assay_mechanisms_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_id")] = None,
        functional_assay_mechanisms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_name")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.@id")] = None,
        input_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.accession")] = None,
        input_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by input_file_sets.aliases")] = None,
        input_file_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.file_set_type")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_summary")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AnalysisSetResults:
        """List items in the AnalysisSet collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisSet items. Supports filtering on fields within AnalysisSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_image_data_url: Filter by external_image_data_url
        :type external_image_data_url: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param functional_assay_mechanisms_id: Filter by functional_assay_mechanisms.@id
        :type functional_assay_mechanisms_id: List[str]
        :param functional_assay_mechanisms_term_id: Filter by functional_assay_mechanisms.term_id
        :type functional_assay_mechanisms_term_id: List[str]
        :param functional_assay_mechanisms_term_name: Filter by functional_assay_mechanisms.term_name
        :type functional_assay_mechanisms_term_name: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets_id: Filter by input_file_sets.@id
        :type input_file_sets_id: List[str]
        :param input_file_sets_accession: Filter by input_file_sets.accession
        :type input_file_sets_accession: List[str]
        :param input_file_sets_aliases: Filter by input_file_sets.aliases
        :type input_file_sets_aliases: List[str]
        :param input_file_sets_file_set_type: Filter by input_file_sets.file_set_type
        :type input_file_sets_file_set_type: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_summary: Filter by sample_summary
        :type sample_summary: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_image_data_url=external_image_data_url,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            functional_assay_mechanisms_id=functional_assay_mechanisms_id,
            functional_assay_mechanisms_term_id=functional_assay_mechanisms_term_id,
            functional_assay_mechanisms_term_name=functional_assay_mechanisms_term_name,
            input_file_set_for=input_file_set_for,
            input_file_sets_id=input_file_sets_id,
            input_file_sets_accession=input_file_sets_accession,
            input_file_sets_aliases=input_file_sets_aliases,
            input_file_sets_file_set_type=input_file_sets_file_set_type,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_summary=sample_summary,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def analysis_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_image_data_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by external_image_data_url")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        functional_assay_mechanisms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by functional_assay_mechanisms.@id")] = None,
        functional_assay_mechanisms_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_id")] = None,
        functional_assay_mechanisms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_name")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.@id")] = None,
        input_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.accession")] = None,
        input_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by input_file_sets.aliases")] = None,
        input_file_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.file_set_type")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_summary")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AnalysisSetResults]:
        """List items in the AnalysisSet collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisSet items. Supports filtering on fields within AnalysisSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_image_data_url: Filter by external_image_data_url
        :type external_image_data_url: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param functional_assay_mechanisms_id: Filter by functional_assay_mechanisms.@id
        :type functional_assay_mechanisms_id: List[str]
        :param functional_assay_mechanisms_term_id: Filter by functional_assay_mechanisms.term_id
        :type functional_assay_mechanisms_term_id: List[str]
        :param functional_assay_mechanisms_term_name: Filter by functional_assay_mechanisms.term_name
        :type functional_assay_mechanisms_term_name: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets_id: Filter by input_file_sets.@id
        :type input_file_sets_id: List[str]
        :param input_file_sets_accession: Filter by input_file_sets.accession
        :type input_file_sets_accession: List[str]
        :param input_file_sets_aliases: Filter by input_file_sets.aliases
        :type input_file_sets_aliases: List[str]
        :param input_file_sets_file_set_type: Filter by input_file_sets.file_set_type
        :type input_file_sets_file_set_type: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_summary: Filter by sample_summary
        :type sample_summary: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_image_data_url=external_image_data_url,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            functional_assay_mechanisms_id=functional_assay_mechanisms_id,
            functional_assay_mechanisms_term_id=functional_assay_mechanisms_term_id,
            functional_assay_mechanisms_term_name=functional_assay_mechanisms_term_name,
            input_file_set_for=input_file_set_for,
            input_file_sets_id=input_file_sets_id,
            input_file_sets_accession=input_file_sets_accession,
            input_file_sets_aliases=input_file_sets_aliases,
            input_file_sets_file_set_type=input_file_sets_file_set_type,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_summary=sample_summary,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def analysis_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_image_data_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by external_image_data_url")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        functional_assay_mechanisms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by functional_assay_mechanisms.@id")] = None,
        functional_assay_mechanisms_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_id")] = None,
        functional_assay_mechanisms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_name")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.@id")] = None,
        input_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.accession")] = None,
        input_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by input_file_sets.aliases")] = None,
        input_file_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.file_set_type")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_summary")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AnalysisSet collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisSet items. Supports filtering on fields within AnalysisSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_image_data_url: Filter by external_image_data_url
        :type external_image_data_url: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param functional_assay_mechanisms_id: Filter by functional_assay_mechanisms.@id
        :type functional_assay_mechanisms_id: List[str]
        :param functional_assay_mechanisms_term_id: Filter by functional_assay_mechanisms.term_id
        :type functional_assay_mechanisms_term_id: List[str]
        :param functional_assay_mechanisms_term_name: Filter by functional_assay_mechanisms.term_name
        :type functional_assay_mechanisms_term_name: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets_id: Filter by input_file_sets.@id
        :type input_file_sets_id: List[str]
        :param input_file_sets_accession: Filter by input_file_sets.accession
        :type input_file_sets_accession: List[str]
        :param input_file_sets_aliases: Filter by input_file_sets.aliases
        :type input_file_sets_aliases: List[str]
        :param input_file_sets_file_set_type: Filter by input_file_sets.file_set_type
        :type input_file_sets_file_set_type: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_summary: Filter by sample_summary
        :type sample_summary: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_image_data_url=external_image_data_url,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            functional_assay_mechanisms_id=functional_assay_mechanisms_id,
            functional_assay_mechanisms_term_id=functional_assay_mechanisms_term_id,
            functional_assay_mechanisms_term_name=functional_assay_mechanisms_term_name,
            input_file_set_for=input_file_set_for,
            input_file_sets_id=input_file_sets_id,
            input_file_sets_accession=input_file_sets_accession,
            input_file_sets_aliases=input_file_sets_aliases,
            input_file_sets_file_set_type=input_file_sets_file_set_type,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_summary=sample_summary,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _analysis_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        assay_titles,
        award_id,
        award_component,
        award_contact_pi_id,
        award_contact_pi_title,
        award_title,
        collections,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        donors_id,
        donors_accession,
        donors_aliases,
        donors_sex,
        donors_status,
        donors_taxa,
        external_image_data_url,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_creation_timestamp,
        files_file_format,
        files_file_size,
        files_href,
        files_s3_uri,
        files_sequencing_platform,
        files_submitted_file_name,
        files_upload_status,
        functional_assay_mechanisms_id,
        functional_assay_mechanisms_term_id,
        functional_assay_mechanisms_term_name,
        input_file_set_for,
        input_file_sets_id,
        input_file_sets_accession,
        input_file_sets_aliases,
        input_file_sets_file_set_type,
        lab_id,
        lab_title,
        notes,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_summary,
        samples_id,
        samples_accession,
        samples_aliases,
        samples_cell_fate_change_treatments,
        samples_classifications,
        samples_construct_library_sets,
        samples_disease_terms_id,
        samples_disease_terms_term_name,
        samples_modifications,
        samples_sample_terms_id,
        samples_sample_terms_aliases,
        samples_sample_terms_status,
        samples_sample_terms_summary,
        samples_sample_terms_term_name,
        samples_status,
        samples_summary,
        samples_targeted_sample_term_id,
        samples_targeted_sample_term_term_name,
        samples_taxa,
        samples_treatments,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'assay_titles': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.contact_pi.@id': 'multi',
            'award.contact_pi.title': 'multi',
            'award.title': 'multi',
            'collections': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'donors.@id': 'multi',
            'donors.accession': 'multi',
            'donors.aliases': 'multi',
            'donors.sex': 'multi',
            'donors.status': 'multi',
            'donors.taxa': 'multi',
            'external_image_data_url': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.creation_timestamp': 'multi',
            'files.file_format': 'multi',
            'files.file_size': 'multi',
            'files.href': 'multi',
            'files.s3_uri': 'multi',
            'files.sequencing_platform': 'multi',
            'files.submitted_file_name': 'multi',
            'files.upload_status': 'multi',
            'functional_assay_mechanisms.@id': 'multi',
            'functional_assay_mechanisms.term_id': 'multi',
            'functional_assay_mechanisms.term_name': 'multi',
            'input_file_set_for': 'multi',
            'input_file_sets.@id': 'multi',
            'input_file_sets.accession': 'multi',
            'input_file_sets.aliases': 'multi',
            'input_file_sets.file_set_type': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_summary': 'multi',
            'samples.@id': 'multi',
            'samples.accession': 'multi',
            'samples.aliases': 'multi',
            'samples.cell_fate_change_treatments': 'multi',
            'samples.classifications': 'multi',
            'samples.construct_library_sets': 'multi',
            'samples.disease_terms.@id': 'multi',
            'samples.disease_terms.term_name': 'multi',
            'samples.modifications': 'multi',
            'samples.sample_terms.@id': 'multi',
            'samples.sample_terms.aliases': 'multi',
            'samples.sample_terms.status': 'multi',
            'samples.sample_terms.summary': 'multi',
            'samples.sample_terms.term_name': 'multi',
            'samples.status': 'multi',
            'samples.summary': 'multi',
            'samples.targeted_sample_term.@id': 'multi',
            'samples.targeted_sample_term.term_name': 'multi',
            'samples.taxa': 'multi',
            'samples.treatments': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_contact_pi_id is not None:
            
            _query_params.append(('award.contact_pi.@id', award_contact_pi_id))
            
        if award_contact_pi_title is not None:
            
            _query_params.append(('award.contact_pi.title', award_contact_pi_title))
            
        if award_title is not None:
            
            _query_params.append(('award.title', award_title))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors_id is not None:
            
            _query_params.append(('donors.@id', donors_id))
            
        if donors_accession is not None:
            
            _query_params.append(('donors.accession', donors_accession))
            
        if donors_aliases is not None:
            
            _query_params.append(('donors.aliases', donors_aliases))
            
        if donors_sex is not None:
            
            _query_params.append(('donors.sex', donors_sex))
            
        if donors_status is not None:
            
            _query_params.append(('donors.status', donors_status))
            
        if donors_taxa is not None:
            
            _query_params.append(('donors.taxa', donors_taxa))
            
        if external_image_data_url is not None:
            
            _query_params.append(('external_image_data_url', external_image_data_url))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_creation_timestamp is not None:
            
            _query_params.append(('files.creation_timestamp', files_creation_timestamp))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_file_size is not None:
            
            _query_params.append(('files.file_size', files_file_size))
            
        if files_href is not None:
            
            _query_params.append(('files.href', files_href))
            
        if files_s3_uri is not None:
            
            _query_params.append(('files.s3_uri', files_s3_uri))
            
        if files_sequencing_platform is not None:
            
            _query_params.append(('files.sequencing_platform', files_sequencing_platform))
            
        if files_submitted_file_name is not None:
            
            _query_params.append(('files.submitted_file_name', files_submitted_file_name))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if functional_assay_mechanisms_id is not None:
            
            _query_params.append(('functional_assay_mechanisms.@id', functional_assay_mechanisms_id))
            
        if functional_assay_mechanisms_term_id is not None:
            
            _query_params.append(('functional_assay_mechanisms.term_id', functional_assay_mechanisms_term_id))
            
        if functional_assay_mechanisms_term_name is not None:
            
            _query_params.append(('functional_assay_mechanisms.term_name', functional_assay_mechanisms_term_name))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if input_file_sets_id is not None:
            
            _query_params.append(('input_file_sets.@id', input_file_sets_id))
            
        if input_file_sets_accession is not None:
            
            _query_params.append(('input_file_sets.accession', input_file_sets_accession))
            
        if input_file_sets_aliases is not None:
            
            _query_params.append(('input_file_sets.aliases', input_file_sets_aliases))
            
        if input_file_sets_file_set_type is not None:
            
            _query_params.append(('input_file_sets.file_set_type', input_file_sets_file_set_type))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_summary is not None:
            
            _query_params.append(('sample_summary', sample_summary))
            
        if samples_id is not None:
            
            _query_params.append(('samples.@id', samples_id))
            
        if samples_accession is not None:
            
            _query_params.append(('samples.accession', samples_accession))
            
        if samples_aliases is not None:
            
            _query_params.append(('samples.aliases', samples_aliases))
            
        if samples_cell_fate_change_treatments is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments', samples_cell_fate_change_treatments))
            
        if samples_classifications is not None:
            
            _query_params.append(('samples.classifications', samples_classifications))
            
        if samples_construct_library_sets is not None:
            
            _query_params.append(('samples.construct_library_sets', samples_construct_library_sets))
            
        if samples_disease_terms_id is not None:
            
            _query_params.append(('samples.disease_terms.@id', samples_disease_terms_id))
            
        if samples_disease_terms_term_name is not None:
            
            _query_params.append(('samples.disease_terms.term_name', samples_disease_terms_term_name))
            
        if samples_modifications is not None:
            
            _query_params.append(('samples.modifications', samples_modifications))
            
        if samples_sample_terms_id is not None:
            
            _query_params.append(('samples.sample_terms.@id', samples_sample_terms_id))
            
        if samples_sample_terms_aliases is not None:
            
            _query_params.append(('samples.sample_terms.aliases', samples_sample_terms_aliases))
            
        if samples_sample_terms_status is not None:
            
            _query_params.append(('samples.sample_terms.status', samples_sample_terms_status))
            
        if samples_sample_terms_summary is not None:
            
            _query_params.append(('samples.sample_terms.summary', samples_sample_terms_summary))
            
        if samples_sample_terms_term_name is not None:
            
            _query_params.append(('samples.sample_terms.term_name', samples_sample_terms_term_name))
            
        if samples_status is not None:
            
            _query_params.append(('samples.status', samples_status))
            
        if samples_summary is not None:
            
            _query_params.append(('samples.summary', samples_summary))
            
        if samples_targeted_sample_term_id is not None:
            
            _query_params.append(('samples.targeted_sample_term.@id', samples_targeted_sample_term_id))
            
        if samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('samples.targeted_sample_term.term_name', samples_targeted_sample_term_term_name))
            
        if samples_taxa is not None:
            
            _query_params.append(('samples.taxa', samples_taxa))
            
        if samples_treatments is not None:
            
            _query_params.append(('samples.treatments', samples_treatments))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/analysis-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def analysis_step_versions(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        analysis_step_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step.@id")] = None,
        analysis_step_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step.name")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        software_versions_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions.@id")] = None,
        software_versions_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions.name")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AnalysisStepVersionResults:
        """List items in the AnalysisStepVersion collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisStepVersion items. Supports filtering on fields within AnalysisStepVersion items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param analysis_step_id: Filter by analysis_step.@id
        :type analysis_step_id: List[str]
        :param analysis_step_name: Filter by analysis_step.name
        :type analysis_step_name: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param software_versions_id: Filter by software_versions.@id
        :type software_versions_id: List[str]
        :param software_versions_name: Filter by software_versions.name
        :type software_versions_name: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_step_versions_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            analysis_step_id=analysis_step_id,
            analysis_step_name=analysis_step_name,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            software_versions_id=software_versions_id,
            software_versions_name=software_versions_name,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisStepVersionResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def analysis_step_versions_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        analysis_step_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step.@id")] = None,
        analysis_step_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step.name")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        software_versions_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions.@id")] = None,
        software_versions_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions.name")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AnalysisStepVersionResults]:
        """List items in the AnalysisStepVersion collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisStepVersion items. Supports filtering on fields within AnalysisStepVersion items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param analysis_step_id: Filter by analysis_step.@id
        :type analysis_step_id: List[str]
        :param analysis_step_name: Filter by analysis_step.name
        :type analysis_step_name: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param software_versions_id: Filter by software_versions.@id
        :type software_versions_id: List[str]
        :param software_versions_name: Filter by software_versions.name
        :type software_versions_name: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_step_versions_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            analysis_step_id=analysis_step_id,
            analysis_step_name=analysis_step_name,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            software_versions_id=software_versions_id,
            software_versions_name=software_versions_name,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisStepVersionResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def analysis_step_versions_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        analysis_step_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step.@id")] = None,
        analysis_step_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step.name")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        software_versions_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions.@id")] = None,
        software_versions_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions.name")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AnalysisStepVersion collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisStepVersion items. Supports filtering on fields within AnalysisStepVersion items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param analysis_step_id: Filter by analysis_step.@id
        :type analysis_step_id: List[str]
        :param analysis_step_name: Filter by analysis_step.name
        :type analysis_step_name: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param software_versions_id: Filter by software_versions.@id
        :type software_versions_id: List[str]
        :param software_versions_name: Filter by software_versions.name
        :type software_versions_name: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_step_versions_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            analysis_step_id=analysis_step_id,
            analysis_step_name=analysis_step_name,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            software_versions_id=software_versions_id,
            software_versions_name=software_versions_name,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisStepVersionResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _analysis_step_versions_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        analysis_step_id,
        analysis_step_name,
        award_id,
        award_component,
        creation_timestamp,
        description,
        lab_id,
        lab_title,
        notes,
        release_timestamp,
        software_versions_id,
        software_versions_name,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'analysis_step.@id': 'multi',
            'analysis_step.name': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'software_versions.@id': 'multi',
            'software_versions.name': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if analysis_step_id is not None:
            
            _query_params.append(('analysis_step.@id', analysis_step_id))
            
        if analysis_step_name is not None:
            
            _query_params.append(('analysis_step.name', analysis_step_name))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if software_versions_id is not None:
            
            _query_params.append(('software_versions.@id', software_versions_id))
            
        if software_versions_name is not None:
            
            _query_params.append(('software_versions.name', software_versions_name))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/analysis-step-versions/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def analysis_steps(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        analysis_step_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_types")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        input_content_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_content_types")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        output_content_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by output_content_types")] = None,
        parents_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by parents.@id")] = None,
        parents_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by parents.title")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        step_label: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by step_label")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        workflow_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow.@id")] = None,
        workflow_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow.accession")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AnalysisStepResults:
        """List items in the AnalysisStep collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisStep items. Supports filtering on fields within AnalysisStep items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param analysis_step_types: Filter by analysis_step_types
        :type analysis_step_types: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param input_content_types: Filter by input_content_types
        :type input_content_types: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param output_content_types: Filter by output_content_types
        :type output_content_types: List[str]
        :param parents_id: Filter by parents.@id
        :type parents_id: List[str]
        :param parents_title: Filter by parents.title
        :type parents_title: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param step_label: Filter by step_label
        :type step_label: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param workflow_id: Filter by workflow.@id
        :type workflow_id: List[str]
        :param workflow_accession: Filter by workflow.accession
        :type workflow_accession: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_steps_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            analysis_step_types=analysis_step_types,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            input_content_types=input_content_types,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            output_content_types=output_content_types,
            parents_id=parents_id,
            parents_title=parents_title,
            release_timestamp=release_timestamp,
            status=status,
            step_label=step_label,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            workflow_id=workflow_id,
            workflow_accession=workflow_accession,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisStepResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def analysis_steps_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        analysis_step_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_types")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        input_content_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_content_types")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        output_content_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by output_content_types")] = None,
        parents_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by parents.@id")] = None,
        parents_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by parents.title")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        step_label: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by step_label")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        workflow_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow.@id")] = None,
        workflow_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow.accession")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AnalysisStepResults]:
        """List items in the AnalysisStep collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisStep items. Supports filtering on fields within AnalysisStep items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param analysis_step_types: Filter by analysis_step_types
        :type analysis_step_types: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param input_content_types: Filter by input_content_types
        :type input_content_types: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param output_content_types: Filter by output_content_types
        :type output_content_types: List[str]
        :param parents_id: Filter by parents.@id
        :type parents_id: List[str]
        :param parents_title: Filter by parents.title
        :type parents_title: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param step_label: Filter by step_label
        :type step_label: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param workflow_id: Filter by workflow.@id
        :type workflow_id: List[str]
        :param workflow_accession: Filter by workflow.accession
        :type workflow_accession: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_steps_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            analysis_step_types=analysis_step_types,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            input_content_types=input_content_types,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            output_content_types=output_content_types,
            parents_id=parents_id,
            parents_title=parents_title,
            release_timestamp=release_timestamp,
            status=status,
            step_label=step_label,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            workflow_id=workflow_id,
            workflow_accession=workflow_accession,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisStepResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def analysis_steps_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        analysis_step_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_types")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        input_content_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_content_types")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        output_content_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by output_content_types")] = None,
        parents_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by parents.@id")] = None,
        parents_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by parents.title")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        step_label: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by step_label")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        workflow_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow.@id")] = None,
        workflow_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow.accession")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AnalysisStep collection.

        Collection endpoint that accepts various query parameters to filter and sort AnalysisStep items. Supports filtering on fields within AnalysisStep items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param analysis_step_types: Filter by analysis_step_types
        :type analysis_step_types: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param input_content_types: Filter by input_content_types
        :type input_content_types: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param output_content_types: Filter by output_content_types
        :type output_content_types: List[str]
        :param parents_id: Filter by parents.@id
        :type parents_id: List[str]
        :param parents_title: Filter by parents.title
        :type parents_title: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param step_label: Filter by step_label
        :type step_label: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param workflow_id: Filter by workflow.@id
        :type workflow_id: List[str]
        :param workflow_accession: Filter by workflow.accession
        :type workflow_accession: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._analysis_steps_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            analysis_step_types=analysis_step_types,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            input_content_types=input_content_types,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            output_content_types=output_content_types,
            parents_id=parents_id,
            parents_title=parents_title,
            release_timestamp=release_timestamp,
            status=status,
            step_label=step_label,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            workflow_id=workflow_id,
            workflow_accession=workflow_accession,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AnalysisStepResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _analysis_steps_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        analysis_step_types,
        award_id,
        award_component,
        creation_timestamp,
        description,
        input_content_types,
        lab_id,
        lab_title,
        name,
        notes,
        output_content_types,
        parents_id,
        parents_title,
        release_timestamp,
        status,
        step_label,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        uuid,
        workflow_id,
        workflow_accession,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'analysis_step_types': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'input_content_types': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'output_content_types': 'multi',
            'parents.@id': 'multi',
            'parents.title': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'step_label': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'uuid': 'multi',
            'workflow.@id': 'multi',
            'workflow.accession': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if analysis_step_types is not None:
            
            _query_params.append(('analysis_step_types', analysis_step_types))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if input_content_types is not None:
            
            _query_params.append(('input_content_types', input_content_types))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if output_content_types is not None:
            
            _query_params.append(('output_content_types', output_content_types))
            
        if parents_id is not None:
            
            _query_params.append(('parents.@id', parents_id))
            
        if parents_title is not None:
            
            _query_params.append(('parents.title', parents_title))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if step_label is not None:
            
            _query_params.append(('step_label', step_label))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if workflow_id is not None:
            
            _query_params.append(('workflow.@id', workflow_id))
            
        if workflow_accession is not None:
            
            _query_params.append(('workflow.accession', workflow_accession))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/analysis-steps/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def assay_terms(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        assay_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_slims")] = None,
        category_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by category_slims")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        objective_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by objective_slims")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        preferred_assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by preferred_assay_titles")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AssayTermResults:
        """List items in the AssayTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort AssayTerm items. Supports filtering on fields within AssayTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param assay_slims: Filter by assay_slims
        :type assay_slims: List[str]
        :param category_slims: Filter by category_slims
        :type category_slims: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param objective_slims: Filter by objective_slims
        :type objective_slims: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param preferred_assay_titles: Filter by preferred_assay_titles
        :type preferred_assay_titles: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._assay_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            assay_slims=assay_slims,
            category_slims=category_slims,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            objective_slims=objective_slims,
            ontology=ontology,
            preferred_assay_titles=preferred_assay_titles,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AssayTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def assay_terms_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        assay_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_slims")] = None,
        category_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by category_slims")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        objective_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by objective_slims")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        preferred_assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by preferred_assay_titles")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AssayTermResults]:
        """List items in the AssayTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort AssayTerm items. Supports filtering on fields within AssayTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param assay_slims: Filter by assay_slims
        :type assay_slims: List[str]
        :param category_slims: Filter by category_slims
        :type category_slims: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param objective_slims: Filter by objective_slims
        :type objective_slims: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param preferred_assay_titles: Filter by preferred_assay_titles
        :type preferred_assay_titles: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._assay_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            assay_slims=assay_slims,
            category_slims=category_slims,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            objective_slims=objective_slims,
            ontology=ontology,
            preferred_assay_titles=preferred_assay_titles,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AssayTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def assay_terms_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        assay_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_slims")] = None,
        category_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by category_slims")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        objective_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by objective_slims")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        preferred_assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by preferred_assay_titles")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AssayTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort AssayTerm items. Supports filtering on fields within AssayTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param assay_slims: Filter by assay_slims
        :type assay_slims: List[str]
        :param category_slims: Filter by category_slims
        :type category_slims: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param objective_slims: Filter by objective_slims
        :type objective_slims: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param preferred_assay_titles: Filter by preferred_assay_titles
        :type preferred_assay_titles: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._assay_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            assay_slims=assay_slims,
            category_slims=category_slims,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            objective_slims=objective_slims,
            ontology=ontology,
            preferred_assay_titles=preferred_assay_titles,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AssayTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _assay_terms_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        ancestors,
        assay_slims,
        category_slims,
        creation_timestamp,
        deprecated_ntr_terms,
        description,
        is_a,
        name,
        notes,
        objective_slims,
        ontology,
        preferred_assay_titles,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        synonyms,
        term_id,
        term_name,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'ancestors': 'multi',
            'assay_slims': 'multi',
            'category_slims': 'multi',
            'creation_timestamp': 'multi',
            'deprecated_ntr_terms': 'multi',
            'description': 'multi',
            'is_a': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'objective_slims': 'multi',
            'ontology': 'multi',
            'preferred_assay_titles': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'synonyms': 'multi',
            'term_id': 'multi',
            'term_name': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if ancestors is not None:
            
            _query_params.append(('ancestors', ancestors))
            
        if assay_slims is not None:
            
            _query_params.append(('assay_slims', assay_slims))
            
        if category_slims is not None:
            
            _query_params.append(('category_slims', category_slims))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if deprecated_ntr_terms is not None:
            
            _query_params.append(('deprecated_ntr_terms', deprecated_ntr_terms))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if is_a is not None:
            
            _query_params.append(('is_a', is_a))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if objective_slims is not None:
            
            _query_params.append(('objective_slims', objective_slims))
            
        if ontology is not None:
            
            _query_params.append(('ontology', ontology))
            
        if preferred_assay_titles is not None:
            
            _query_params.append(('preferred_assay_titles', preferred_assay_titles))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if synonyms is not None:
            
            _query_params.append(('synonyms', synonyms))
            
        if term_id is not None:
            
            _query_params.append(('term_id', term_id))
            
        if term_name is not None:
            
            _query_params.append(('term_name', term_name))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/assay-terms/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def auxiliary_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        barcode_map: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        measurement_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.@id")] = None,
        measurement_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.accession")] = None,
        measurement_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by measurement_sets.aliases")] = None,
        measurement_sets_preferred_assay_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.preferred_assay_title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AuxiliarySetResults:
        """List items in the AuxiliarySet collection.

        Collection endpoint that accepts various query parameters to filter and sort AuxiliarySet items. Supports filtering on fields within AuxiliarySet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param barcode_map: Filter by barcode_map
        :type barcode_map: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param measurement_sets_id: Filter by measurement_sets.@id
        :type measurement_sets_id: List[str]
        :param measurement_sets_accession: Filter by measurement_sets.accession
        :type measurement_sets_accession: List[str]
        :param measurement_sets_aliases: Filter by measurement_sets.aliases
        :type measurement_sets_aliases: List[str]
        :param measurement_sets_preferred_assay_title: Filter by measurement_sets.preferred_assay_title
        :type measurement_sets_preferred_assay_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._auxiliary_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            barcode_map=barcode_map,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            measurement_sets_id=measurement_sets_id,
            measurement_sets_accession=measurement_sets_accession,
            measurement_sets_aliases=measurement_sets_aliases,
            measurement_sets_preferred_assay_title=measurement_sets_preferred_assay_title,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AuxiliarySetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def auxiliary_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        barcode_map: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        measurement_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.@id")] = None,
        measurement_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.accession")] = None,
        measurement_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by measurement_sets.aliases")] = None,
        measurement_sets_preferred_assay_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.preferred_assay_title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AuxiliarySetResults]:
        """List items in the AuxiliarySet collection.

        Collection endpoint that accepts various query parameters to filter and sort AuxiliarySet items. Supports filtering on fields within AuxiliarySet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param barcode_map: Filter by barcode_map
        :type barcode_map: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param measurement_sets_id: Filter by measurement_sets.@id
        :type measurement_sets_id: List[str]
        :param measurement_sets_accession: Filter by measurement_sets.accession
        :type measurement_sets_accession: List[str]
        :param measurement_sets_aliases: Filter by measurement_sets.aliases
        :type measurement_sets_aliases: List[str]
        :param measurement_sets_preferred_assay_title: Filter by measurement_sets.preferred_assay_title
        :type measurement_sets_preferred_assay_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._auxiliary_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            barcode_map=barcode_map,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            measurement_sets_id=measurement_sets_id,
            measurement_sets_accession=measurement_sets_accession,
            measurement_sets_aliases=measurement_sets_aliases,
            measurement_sets_preferred_assay_title=measurement_sets_preferred_assay_title,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AuxiliarySetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def auxiliary_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        barcode_map: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        measurement_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.@id")] = None,
        measurement_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.accession")] = None,
        measurement_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by measurement_sets.aliases")] = None,
        measurement_sets_preferred_assay_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by measurement_sets.preferred_assay_title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the AuxiliarySet collection.

        Collection endpoint that accepts various query parameters to filter and sort AuxiliarySet items. Supports filtering on fields within AuxiliarySet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param barcode_map: Filter by barcode_map
        :type barcode_map: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param measurement_sets_id: Filter by measurement_sets.@id
        :type measurement_sets_id: List[str]
        :param measurement_sets_accession: Filter by measurement_sets.accession
        :type measurement_sets_accession: List[str]
        :param measurement_sets_aliases: Filter by measurement_sets.aliases
        :type measurement_sets_aliases: List[str]
        :param measurement_sets_preferred_assay_title: Filter by measurement_sets.preferred_assay_title
        :type measurement_sets_preferred_assay_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._auxiliary_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            barcode_map=barcode_map,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            measurement_sets_id=measurement_sets_id,
            measurement_sets_accession=measurement_sets_accession,
            measurement_sets_aliases=measurement_sets_aliases,
            measurement_sets_preferred_assay_title=measurement_sets_preferred_assay_title,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AuxiliarySetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _auxiliary_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        award_contact_pi_id,
        award_contact_pi_title,
        award_title,
        barcode_map,
        collections,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        donors_id,
        donors_accession,
        donors_aliases,
        donors_sex,
        donors_status,
        donors_taxa,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_creation_timestamp,
        files_file_format,
        files_file_size,
        files_href,
        files_s3_uri,
        files_sequencing_platform,
        files_submitted_file_name,
        files_upload_status,
        input_file_set_for,
        lab_id,
        lab_title,
        measurement_sets_id,
        measurement_sets_accession,
        measurement_sets_aliases,
        measurement_sets_preferred_assay_title,
        notes,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        samples_id,
        samples_accession,
        samples_aliases,
        samples_cell_fate_change_treatments,
        samples_classifications,
        samples_construct_library_sets,
        samples_disease_terms_id,
        samples_disease_terms_term_name,
        samples_modifications,
        samples_sample_terms_id,
        samples_sample_terms_aliases,
        samples_sample_terms_status,
        samples_sample_terms_summary,
        samples_sample_terms_term_name,
        samples_status,
        samples_summary,
        samples_targeted_sample_term_id,
        samples_targeted_sample_term_term_name,
        samples_taxa,
        samples_treatments,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        url,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.contact_pi.@id': 'multi',
            'award.contact_pi.title': 'multi',
            'award.title': 'multi',
            'barcode_map': 'multi',
            'collections': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'donors.@id': 'multi',
            'donors.accession': 'multi',
            'donors.aliases': 'multi',
            'donors.sex': 'multi',
            'donors.status': 'multi',
            'donors.taxa': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.creation_timestamp': 'multi',
            'files.file_format': 'multi',
            'files.file_size': 'multi',
            'files.href': 'multi',
            'files.s3_uri': 'multi',
            'files.sequencing_platform': 'multi',
            'files.submitted_file_name': 'multi',
            'files.upload_status': 'multi',
            'input_file_set_for': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'measurement_sets.@id': 'multi',
            'measurement_sets.accession': 'multi',
            'measurement_sets.aliases': 'multi',
            'measurement_sets.preferred_assay_title': 'multi',
            'notes': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'samples.@id': 'multi',
            'samples.accession': 'multi',
            'samples.aliases': 'multi',
            'samples.cell_fate_change_treatments': 'multi',
            'samples.classifications': 'multi',
            'samples.construct_library_sets': 'multi',
            'samples.disease_terms.@id': 'multi',
            'samples.disease_terms.term_name': 'multi',
            'samples.modifications': 'multi',
            'samples.sample_terms.@id': 'multi',
            'samples.sample_terms.aliases': 'multi',
            'samples.sample_terms.status': 'multi',
            'samples.sample_terms.summary': 'multi',
            'samples.sample_terms.term_name': 'multi',
            'samples.status': 'multi',
            'samples.summary': 'multi',
            'samples.targeted_sample_term.@id': 'multi',
            'samples.targeted_sample_term.term_name': 'multi',
            'samples.taxa': 'multi',
            'samples.treatments': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'url': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_contact_pi_id is not None:
            
            _query_params.append(('award.contact_pi.@id', award_contact_pi_id))
            
        if award_contact_pi_title is not None:
            
            _query_params.append(('award.contact_pi.title', award_contact_pi_title))
            
        if award_title is not None:
            
            _query_params.append(('award.title', award_title))
            
        if barcode_map is not None:
            
            _query_params.append(('barcode_map', barcode_map))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors_id is not None:
            
            _query_params.append(('donors.@id', donors_id))
            
        if donors_accession is not None:
            
            _query_params.append(('donors.accession', donors_accession))
            
        if donors_aliases is not None:
            
            _query_params.append(('donors.aliases', donors_aliases))
            
        if donors_sex is not None:
            
            _query_params.append(('donors.sex', donors_sex))
            
        if donors_status is not None:
            
            _query_params.append(('donors.status', donors_status))
            
        if donors_taxa is not None:
            
            _query_params.append(('donors.taxa', donors_taxa))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_creation_timestamp is not None:
            
            _query_params.append(('files.creation_timestamp', files_creation_timestamp))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_file_size is not None:
            
            _query_params.append(('files.file_size', files_file_size))
            
        if files_href is not None:
            
            _query_params.append(('files.href', files_href))
            
        if files_s3_uri is not None:
            
            _query_params.append(('files.s3_uri', files_s3_uri))
            
        if files_sequencing_platform is not None:
            
            _query_params.append(('files.sequencing_platform', files_sequencing_platform))
            
        if files_submitted_file_name is not None:
            
            _query_params.append(('files.submitted_file_name', files_submitted_file_name))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if measurement_sets_id is not None:
            
            _query_params.append(('measurement_sets.@id', measurement_sets_id))
            
        if measurement_sets_accession is not None:
            
            _query_params.append(('measurement_sets.accession', measurement_sets_accession))
            
        if measurement_sets_aliases is not None:
            
            _query_params.append(('measurement_sets.aliases', measurement_sets_aliases))
            
        if measurement_sets_preferred_assay_title is not None:
            
            _query_params.append(('measurement_sets.preferred_assay_title', measurement_sets_preferred_assay_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if samples_id is not None:
            
            _query_params.append(('samples.@id', samples_id))
            
        if samples_accession is not None:
            
            _query_params.append(('samples.accession', samples_accession))
            
        if samples_aliases is not None:
            
            _query_params.append(('samples.aliases', samples_aliases))
            
        if samples_cell_fate_change_treatments is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments', samples_cell_fate_change_treatments))
            
        if samples_classifications is not None:
            
            _query_params.append(('samples.classifications', samples_classifications))
            
        if samples_construct_library_sets is not None:
            
            _query_params.append(('samples.construct_library_sets', samples_construct_library_sets))
            
        if samples_disease_terms_id is not None:
            
            _query_params.append(('samples.disease_terms.@id', samples_disease_terms_id))
            
        if samples_disease_terms_term_name is not None:
            
            _query_params.append(('samples.disease_terms.term_name', samples_disease_terms_term_name))
            
        if samples_modifications is not None:
            
            _query_params.append(('samples.modifications', samples_modifications))
            
        if samples_sample_terms_id is not None:
            
            _query_params.append(('samples.sample_terms.@id', samples_sample_terms_id))
            
        if samples_sample_terms_aliases is not None:
            
            _query_params.append(('samples.sample_terms.aliases', samples_sample_terms_aliases))
            
        if samples_sample_terms_status is not None:
            
            _query_params.append(('samples.sample_terms.status', samples_sample_terms_status))
            
        if samples_sample_terms_summary is not None:
            
            _query_params.append(('samples.sample_terms.summary', samples_sample_terms_summary))
            
        if samples_sample_terms_term_name is not None:
            
            _query_params.append(('samples.sample_terms.term_name', samples_sample_terms_term_name))
            
        if samples_status is not None:
            
            _query_params.append(('samples.status', samples_status))
            
        if samples_summary is not None:
            
            _query_params.append(('samples.summary', samples_summary))
            
        if samples_targeted_sample_term_id is not None:
            
            _query_params.append(('samples.targeted_sample_term.@id', samples_targeted_sample_term_id))
            
        if samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('samples.targeted_sample_term.term_name', samples_targeted_sample_term_term_name))
            
        if samples_taxa is not None:
            
            _query_params.append(('samples.taxa', samples_taxa))
            
        if samples_treatments is not None:
            
            _query_params.append(('samples.treatments', samples_treatments))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/auxiliary-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def awards(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        component: Annotated[Optional[List[StrictStr]], Field(description="Filter by component")] = None,
        contact_pi: Annotated[Optional[List[StrictStr]], Field(description="Filter by contact_pi")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        end_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by end_date")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        pis: Annotated[Optional[List[StrictStr]], Field(description="Filter by pis")] = None,
        project: Annotated[Optional[List[StrictStr]], Field(description="Filter by project")] = None,
        start_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by start_date")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        viewing_group: Annotated[Optional[List[StrictStr]], Field(description="Filter by viewing_group")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> AwardResults:
        """List items in the Award collection.

        Collection endpoint that accepts various query parameters to filter and sort Award items. Supports filtering on fields within Award items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param component: Filter by component
        :type component: List[str]
        :param contact_pi: Filter by contact_pi
        :type contact_pi: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param end_date: Filter by end_date
        :type end_date: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param pis: Filter by pis
        :type pis: List[str]
        :param project: Filter by project
        :type project: List[str]
        :param start_date: Filter by start_date
        :type start_date: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param viewing_group: Filter by viewing_group
        :type viewing_group: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._awards_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            component=component,
            contact_pi=contact_pi,
            creation_timestamp=creation_timestamp,
            description=description,
            end_date=end_date,
            name=name,
            notes=notes,
            pis=pis,
            project=project,
            start_date=start_date,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            viewing_group=viewing_group,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AwardResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def awards_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        component: Annotated[Optional[List[StrictStr]], Field(description="Filter by component")] = None,
        contact_pi: Annotated[Optional[List[StrictStr]], Field(description="Filter by contact_pi")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        end_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by end_date")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        pis: Annotated[Optional[List[StrictStr]], Field(description="Filter by pis")] = None,
        project: Annotated[Optional[List[StrictStr]], Field(description="Filter by project")] = None,
        start_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by start_date")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        viewing_group: Annotated[Optional[List[StrictStr]], Field(description="Filter by viewing_group")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[AwardResults]:
        """List items in the Award collection.

        Collection endpoint that accepts various query parameters to filter and sort Award items. Supports filtering on fields within Award items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param component: Filter by component
        :type component: List[str]
        :param contact_pi: Filter by contact_pi
        :type contact_pi: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param end_date: Filter by end_date
        :type end_date: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param pis: Filter by pis
        :type pis: List[str]
        :param project: Filter by project
        :type project: List[str]
        :param start_date: Filter by start_date
        :type start_date: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param viewing_group: Filter by viewing_group
        :type viewing_group: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._awards_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            component=component,
            contact_pi=contact_pi,
            creation_timestamp=creation_timestamp,
            description=description,
            end_date=end_date,
            name=name,
            notes=notes,
            pis=pis,
            project=project,
            start_date=start_date,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            viewing_group=viewing_group,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AwardResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def awards_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        component: Annotated[Optional[List[StrictStr]], Field(description="Filter by component")] = None,
        contact_pi: Annotated[Optional[List[StrictStr]], Field(description="Filter by contact_pi")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        end_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by end_date")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        pis: Annotated[Optional[List[StrictStr]], Field(description="Filter by pis")] = None,
        project: Annotated[Optional[List[StrictStr]], Field(description="Filter by project")] = None,
        start_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by start_date")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        viewing_group: Annotated[Optional[List[StrictStr]], Field(description="Filter by viewing_group")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Award collection.

        Collection endpoint that accepts various query parameters to filter and sort Award items. Supports filtering on fields within Award items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param component: Filter by component
        :type component: List[str]
        :param contact_pi: Filter by contact_pi
        :type contact_pi: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param end_date: Filter by end_date
        :type end_date: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param pis: Filter by pis
        :type pis: List[str]
        :param project: Filter by project
        :type project: List[str]
        :param start_date: Filter by start_date
        :type start_date: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param viewing_group: Filter by viewing_group
        :type viewing_group: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._awards_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            component=component,
            contact_pi=contact_pi,
            creation_timestamp=creation_timestamp,
            description=description,
            end_date=end_date,
            name=name,
            notes=notes,
            pis=pis,
            project=project,
            start_date=start_date,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            viewing_group=viewing_group,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "AwardResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _awards_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        component,
        contact_pi,
        creation_timestamp,
        description,
        end_date,
        name,
        notes,
        pis,
        project,
        start_date,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        url,
        uuid,
        viewing_group,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'component': 'multi',
            'contact_pi': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'end_date': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'pis': 'multi',
            'project': 'multi',
            'start_date': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'viewing_group': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if component is not None:
            
            _query_params.append(('component', component))
            
        if contact_pi is not None:
            
            _query_params.append(('contact_pi', contact_pi))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if end_date is not None:
            
            _query_params.append(('end_date', end_date))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if pis is not None:
            
            _query_params.append(('pis', pis))
            
        if project is not None:
            
            _query_params.append(('project', project))
            
        if start_date is not None:
            
            _query_params.append(('start_date', start_date))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if viewing_group is not None:
            
            _query_params.append(('viewing_group', viewing_group))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/awards/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def batch_download(
        self,
        type: Annotated[List[StrictStr], Field(description="Type of objects to return. Can be repeated for multiple types.")],
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> str:
        """List files to download based on search query. All results are returned.

        Generates TSV of files contained in FileSets in search results.

        :param type: Type of objects to return. Can be repeated for multiple types. (required)
        :type type: List[str]
        :param query: Query string for searching.
        :type query: str
        :param field_filters: Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.
        :type field_filters: object
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._batch_download_serialize(
            type=type,
            query=query,
            field_filters=field_filters,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "str",
            '400': "object",
            '404': "object",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def batch_download_with_http_info(
        self,
        type: Annotated[List[StrictStr], Field(description="Type of objects to return. Can be repeated for multiple types.")],
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[str]:
        """List files to download based on search query. All results are returned.

        Generates TSV of files contained in FileSets in search results.

        :param type: Type of objects to return. Can be repeated for multiple types. (required)
        :type type: List[str]
        :param query: Query string for searching.
        :type query: str
        :param field_filters: Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.
        :type field_filters: object
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._batch_download_serialize(
            type=type,
            query=query,
            field_filters=field_filters,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "str",
            '400': "object",
            '404': "object",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def batch_download_without_preload_content(
        self,
        type: Annotated[List[StrictStr], Field(description="Type of objects to return. Can be repeated for multiple types.")],
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List files to download based on search query. All results are returned.

        Generates TSV of files contained in FileSets in search results.

        :param type: Type of objects to return. Can be repeated for multiple types. (required)
        :type type: List[str]
        :param query: Query string for searching.
        :type query: str
        :param field_filters: Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.
        :type field_filters: object
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._batch_download_serialize(
            type=type,
            query=query,
            field_filters=field_filters,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "str",
            '400': "object",
            '404': "object",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _batch_download_serialize(
        self,
        type,
        query,
        field_filters,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'type': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if type is not None:
            
            _query_params.append(('type', type))
            
        if query is not None:
            
            _query_params.append(('query', query))
            
        if field_filters is not None:
             for k, v in field_filters.items():
                if isinstance(v, list):
                    for item in v:
                        _query_params.append((k, item))
                else:
                    _query_params.append((k, v))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter


        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'text/tab-separated-values', 
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/batch-download',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def biomarkers(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.name")] = None,
        biomarker_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarker_for")] = None,
        classification: Annotated[Optional[List[StrictStr]], Field(description="Filter by classification")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        gene: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        name_quantification: Annotated[Optional[List[StrictStr]], Field(description="Filter by name_quantification")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        quantification: Annotated[Optional[List[StrictStr]], Field(description="Filter by quantification")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> BiomarkerResults:
        """List items in the Biomarker collection.

        Collection endpoint that accepts various query parameters to filter and sort Biomarker items. Supports filtering on fields within Biomarker items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_name: Filter by award.name
        :type award_name: List[str]
        :param biomarker_for: Filter by biomarker_for
        :type biomarker_for: List[str]
        :param classification: Filter by classification
        :type classification: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param gene: Filter by gene
        :type gene: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param name_quantification: Filter by name_quantification
        :type name_quantification: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param quantification: Filter by quantification
        :type quantification: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._biomarkers_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            award_name=award_name,
            biomarker_for=biomarker_for,
            classification=classification,
            creation_timestamp=creation_timestamp,
            description=description,
            gene=gene,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            name_quantification=name_quantification,
            notes=notes,
            quantification=quantification,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "BiomarkerResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def biomarkers_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.name")] = None,
        biomarker_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarker_for")] = None,
        classification: Annotated[Optional[List[StrictStr]], Field(description="Filter by classification")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        gene: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        name_quantification: Annotated[Optional[List[StrictStr]], Field(description="Filter by name_quantification")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        quantification: Annotated[Optional[List[StrictStr]], Field(description="Filter by quantification")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[BiomarkerResults]:
        """List items in the Biomarker collection.

        Collection endpoint that accepts various query parameters to filter and sort Biomarker items. Supports filtering on fields within Biomarker items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_name: Filter by award.name
        :type award_name: List[str]
        :param biomarker_for: Filter by biomarker_for
        :type biomarker_for: List[str]
        :param classification: Filter by classification
        :type classification: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param gene: Filter by gene
        :type gene: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param name_quantification: Filter by name_quantification
        :type name_quantification: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param quantification: Filter by quantification
        :type quantification: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._biomarkers_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            award_name=award_name,
            biomarker_for=biomarker_for,
            classification=classification,
            creation_timestamp=creation_timestamp,
            description=description,
            gene=gene,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            name_quantification=name_quantification,
            notes=notes,
            quantification=quantification,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "BiomarkerResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def biomarkers_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.name")] = None,
        biomarker_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarker_for")] = None,
        classification: Annotated[Optional[List[StrictStr]], Field(description="Filter by classification")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        gene: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        name_quantification: Annotated[Optional[List[StrictStr]], Field(description="Filter by name_quantification")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        quantification: Annotated[Optional[List[StrictStr]], Field(description="Filter by quantification")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Biomarker collection.

        Collection endpoint that accepts various query parameters to filter and sort Biomarker items. Supports filtering on fields within Biomarker items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_name: Filter by award.name
        :type award_name: List[str]
        :param biomarker_for: Filter by biomarker_for
        :type biomarker_for: List[str]
        :param classification: Filter by classification
        :type classification: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param gene: Filter by gene
        :type gene: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param name_quantification: Filter by name_quantification
        :type name_quantification: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param quantification: Filter by quantification
        :type quantification: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._biomarkers_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            award_name=award_name,
            biomarker_for=biomarker_for,
            classification=classification,
            creation_timestamp=creation_timestamp,
            description=description,
            gene=gene,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            name_quantification=name_quantification,
            notes=notes,
            quantification=quantification,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "BiomarkerResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _biomarkers_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award_id,
        award_component,
        award_name,
        biomarker_for,
        classification,
        creation_timestamp,
        description,
        gene,
        lab_id,
        lab_title,
        name,
        name_quantification,
        notes,
        quantification,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        synonyms,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.name': 'multi',
            'biomarker_for': 'multi',
            'classification': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'gene': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'name': 'multi',
            'name_quantification': 'multi',
            'notes': 'multi',
            'quantification': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'synonyms': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_name is not None:
            
            _query_params.append(('award.name', award_name))
            
        if biomarker_for is not None:
            
            _query_params.append(('biomarker_for', biomarker_for))
            
        if classification is not None:
            
            _query_params.append(('classification', classification))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if gene is not None:
            
            _query_params.append(('gene', gene))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if name_quantification is not None:
            
            _query_params.append(('name_quantification', name_quantification))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if quantification is not None:
            
            _query_params.append(('quantification', quantification))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if synonyms is not None:
            
            _query_params.append(('synonyms', synonyms))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/biomarkers/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def configuration_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        seqspec_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by seqspec_of")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ConfigurationFileResults:
        """List items in the ConfigurationFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ConfigurationFile items. Supports filtering on fields within ConfigurationFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param seqspec_of: Filter by seqspec_of
        :type seqspec_of: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._configuration_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            seqspec_of=seqspec_of,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ConfigurationFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def configuration_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        seqspec_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by seqspec_of")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ConfigurationFileResults]:
        """List items in the ConfigurationFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ConfigurationFile items. Supports filtering on fields within ConfigurationFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param seqspec_of: Filter by seqspec_of
        :type seqspec_of: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._configuration_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            seqspec_of=seqspec_of,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ConfigurationFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def configuration_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        seqspec_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by seqspec_of")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the ConfigurationFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ConfigurationFile items. Supports filtering on fields within ConfigurationFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param seqspec_of: Filter by seqspec_of
        :type seqspec_of: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._configuration_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            seqspec_of=seqspec_of,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ConfigurationFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _configuration_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        assay_titles,
        award_id,
        award_component,
        collections,
        content_md5sum,
        content_type,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        release_timestamp,
        revoke_detail,
        s3_uri,
        seqspec_of,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'assay_titles': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'seqspec_of': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if seqspec_of is not None:
            
            _query_params.append(('seqspec_of', seqspec_of))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/configuration-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def construct_library_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        applied_to_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.@id")] = None,
        applied_to_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.accession")] = None,
        applied_to_samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.aliases")] = None,
        applied_to_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.classifications")] = None,
        applied_to_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.disease_terms.@id")] = None,
        applied_to_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.disease_terms.term_name")] = None,
        applied_to_samples_donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.donors.@id")] = None,
        applied_to_samples_donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.donors.taxa")] = None,
        applied_to_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.sample_terms.@id")] = None,
        applied_to_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.sample_terms.term_name")] = None,
        applied_to_samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.status")] = None,
        applied_to_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.summary")] = None,
        applied_to_samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.targeted_sample_term.@id")] = None,
        applied_to_samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.targeted_sample_term.term_name")] = None,
        applied_to_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.taxa")] = None,
        associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by associated_phenotypes.@id")] = None,
        associated_phenotypes_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by associated_phenotypes.term_id")] = None,
        associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by associated_phenotypes.term_name")] = None,
        average_guide_coverage: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by average_guide_coverage")] = None,
        average_insert_size: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by average_insert_size")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        exon: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by exon")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        guide_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by guide_type")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        integrated_content_files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.@id")] = None,
        integrated_content_files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.accession")] = None,
        integrated_content_files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_content_files.aliases")] = None,
        integrated_content_files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.content_type")] = None,
        integrated_content_files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.file_format")] = None,
        integrated_content_files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.upload_status")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        large_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.@id")] = None,
        large_scale_gene_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.accession")] = None,
        large_scale_gene_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_gene_list.aliases")] = None,
        large_scale_loci_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.@id")] = None,
        large_scale_loci_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.accession")] = None,
        large_scale_loci_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_loci_list.aliases")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_guide_coverage: Annotated[Optional[List[StrictInt]], Field(description="Filter by lower_bound_guide_coverage")] = None,
        lower_bound_insert_size: Annotated[Optional[List[StrictInt]], Field(description="Filter by lower_bound_insert_size")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        orf_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.@id")] = None,
        orf_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_list.aliases")] = None,
        orf_list_genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.genes.@id")] = None,
        orf_list_genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.genes.symbol")] = None,
        orf_list_orf_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_list.orf_id")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        scope: Annotated[Optional[List[StrictStr]], Field(description="Filter by scope")] = None,
        selection_criteria: Annotated[Optional[List[StrictStr]], Field(description="Filter by selection_criteria")] = None,
        small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.@id")] = None,
        small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by small_scale_gene_list.geneid")] = None,
        small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.name")] = None,
        small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.symbol")] = None,
        small_scale_gene_list_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.synonyms")] = None,
        small_scale_loci_list: Annotated[Optional[List[Locus]], Field(description="Filter by small_scale_loci_list")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeton: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by targeton")] = None,
        tiling_modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by tiling_modality")] = None,
        upper_bound_guide_coverage: Annotated[Optional[List[StrictInt]], Field(description="Filter by upper_bound_guide_coverage")] = None,
        upper_bound_insert_size: Annotated[Optional[List[StrictInt]], Field(description="Filter by upper_bound_insert_size")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ConstructLibrarySetResults:
        """List items in the ConstructLibrarySet collection.

        Collection endpoint that accepts various query parameters to filter and sort ConstructLibrarySet items. Supports filtering on fields within ConstructLibrarySet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param applied_to_samples_id: Filter by applied_to_samples.@id
        :type applied_to_samples_id: List[str]
        :param applied_to_samples_accession: Filter by applied_to_samples.accession
        :type applied_to_samples_accession: List[str]
        :param applied_to_samples_aliases: Filter by applied_to_samples.aliases
        :type applied_to_samples_aliases: List[str]
        :param applied_to_samples_classifications: Filter by applied_to_samples.classifications
        :type applied_to_samples_classifications: List[str]
        :param applied_to_samples_disease_terms_id: Filter by applied_to_samples.disease_terms.@id
        :type applied_to_samples_disease_terms_id: List[str]
        :param applied_to_samples_disease_terms_term_name: Filter by applied_to_samples.disease_terms.term_name
        :type applied_to_samples_disease_terms_term_name: List[str]
        :param applied_to_samples_donors_id: Filter by applied_to_samples.donors.@id
        :type applied_to_samples_donors_id: List[str]
        :param applied_to_samples_donors_taxa: Filter by applied_to_samples.donors.taxa
        :type applied_to_samples_donors_taxa: List[str]
        :param applied_to_samples_sample_terms_id: Filter by applied_to_samples.sample_terms.@id
        :type applied_to_samples_sample_terms_id: List[str]
        :param applied_to_samples_sample_terms_term_name: Filter by applied_to_samples.sample_terms.term_name
        :type applied_to_samples_sample_terms_term_name: List[str]
        :param applied_to_samples_status: Filter by applied_to_samples.status
        :type applied_to_samples_status: List[str]
        :param applied_to_samples_summary: Filter by applied_to_samples.summary
        :type applied_to_samples_summary: List[str]
        :param applied_to_samples_targeted_sample_term_id: Filter by applied_to_samples.targeted_sample_term.@id
        :type applied_to_samples_targeted_sample_term_id: List[str]
        :param applied_to_samples_targeted_sample_term_term_name: Filter by applied_to_samples.targeted_sample_term.term_name
        :type applied_to_samples_targeted_sample_term_term_name: List[str]
        :param applied_to_samples_taxa: Filter by applied_to_samples.taxa
        :type applied_to_samples_taxa: List[str]
        :param associated_phenotypes_id: Filter by associated_phenotypes.@id
        :type associated_phenotypes_id: List[str]
        :param associated_phenotypes_term_id: Filter by associated_phenotypes.term_id
        :type associated_phenotypes_term_id: List[str]
        :param associated_phenotypes_term_name: Filter by associated_phenotypes.term_name
        :type associated_phenotypes_term_name: List[str]
        :param average_guide_coverage: Filter by average_guide_coverage
        :type average_guide_coverage: List[float]
        :param average_insert_size: Filter by average_insert_size
        :type average_insert_size: List[float]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param exon: Filter by exon
        :type exon: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param guide_type: Filter by guide_type
        :type guide_type: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param integrated_content_files_id: Filter by integrated_content_files.@id
        :type integrated_content_files_id: List[str]
        :param integrated_content_files_accession: Filter by integrated_content_files.accession
        :type integrated_content_files_accession: List[str]
        :param integrated_content_files_aliases: Filter by integrated_content_files.aliases
        :type integrated_content_files_aliases: List[str]
        :param integrated_content_files_content_type: Filter by integrated_content_files.content_type
        :type integrated_content_files_content_type: List[str]
        :param integrated_content_files_file_format: Filter by integrated_content_files.file_format
        :type integrated_content_files_file_format: List[str]
        :param integrated_content_files_upload_status: Filter by integrated_content_files.upload_status
        :type integrated_content_files_upload_status: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param large_scale_gene_list_id: Filter by large_scale_gene_list.@id
        :type large_scale_gene_list_id: List[str]
        :param large_scale_gene_list_accession: Filter by large_scale_gene_list.accession
        :type large_scale_gene_list_accession: List[str]
        :param large_scale_gene_list_aliases: Filter by large_scale_gene_list.aliases
        :type large_scale_gene_list_aliases: List[str]
        :param large_scale_loci_list_id: Filter by large_scale_loci_list.@id
        :type large_scale_loci_list_id: List[str]
        :param large_scale_loci_list_accession: Filter by large_scale_loci_list.accession
        :type large_scale_loci_list_accession: List[str]
        :param large_scale_loci_list_aliases: Filter by large_scale_loci_list.aliases
        :type large_scale_loci_list_aliases: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_guide_coverage: Filter by lower_bound_guide_coverage
        :type lower_bound_guide_coverage: List[int]
        :param lower_bound_insert_size: Filter by lower_bound_insert_size
        :type lower_bound_insert_size: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param orf_list_id: Filter by orf_list.@id
        :type orf_list_id: List[str]
        :param orf_list_aliases: Filter by orf_list.aliases
        :type orf_list_aliases: List[str]
        :param orf_list_genes_id: Filter by orf_list.genes.@id
        :type orf_list_genes_id: List[str]
        :param orf_list_genes_symbol: Filter by orf_list.genes.symbol
        :type orf_list_genes_symbol: List[str]
        :param orf_list_orf_id: Filter by orf_list.orf_id
        :type orf_list_orf_id: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param scope: Filter by scope
        :type scope: List[str]
        :param selection_criteria: Filter by selection_criteria
        :type selection_criteria: List[str]
        :param small_scale_gene_list_id: Filter by small_scale_gene_list.@id
        :type small_scale_gene_list_id: List[str]
        :param small_scale_gene_list_geneid: Filter by small_scale_gene_list.geneid
        :type small_scale_gene_list_geneid: List[str]
        :param small_scale_gene_list_name: Filter by small_scale_gene_list.name
        :type small_scale_gene_list_name: List[str]
        :param small_scale_gene_list_symbol: Filter by small_scale_gene_list.symbol
        :type small_scale_gene_list_symbol: List[str]
        :param small_scale_gene_list_synonyms: Filter by small_scale_gene_list.synonyms
        :type small_scale_gene_list_synonyms: List[str]
        :param small_scale_loci_list: Filter by small_scale_loci_list
        :type small_scale_loci_list: List[Locus]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeton: Filter by targeton
        :type targeton: List[str]
        :param tiling_modality: Filter by tiling_modality
        :type tiling_modality: List[str]
        :param upper_bound_guide_coverage: Filter by upper_bound_guide_coverage
        :type upper_bound_guide_coverage: List[int]
        :param upper_bound_insert_size: Filter by upper_bound_insert_size
        :type upper_bound_insert_size: List[int]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._construct_library_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            applied_to_samples_id=applied_to_samples_id,
            applied_to_samples_accession=applied_to_samples_accession,
            applied_to_samples_aliases=applied_to_samples_aliases,
            applied_to_samples_classifications=applied_to_samples_classifications,
            applied_to_samples_disease_terms_id=applied_to_samples_disease_terms_id,
            applied_to_samples_disease_terms_term_name=applied_to_samples_disease_terms_term_name,
            applied_to_samples_donors_id=applied_to_samples_donors_id,
            applied_to_samples_donors_taxa=applied_to_samples_donors_taxa,
            applied_to_samples_sample_terms_id=applied_to_samples_sample_terms_id,
            applied_to_samples_sample_terms_term_name=applied_to_samples_sample_terms_term_name,
            applied_to_samples_status=applied_to_samples_status,
            applied_to_samples_summary=applied_to_samples_summary,
            applied_to_samples_targeted_sample_term_id=applied_to_samples_targeted_sample_term_id,
            applied_to_samples_targeted_sample_term_term_name=applied_to_samples_targeted_sample_term_term_name,
            applied_to_samples_taxa=applied_to_samples_taxa,
            associated_phenotypes_id=associated_phenotypes_id,
            associated_phenotypes_term_id=associated_phenotypes_term_id,
            associated_phenotypes_term_name=associated_phenotypes_term_name,
            average_guide_coverage=average_guide_coverage,
            average_insert_size=average_insert_size,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            exon=exon,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_file_format=files_file_format,
            files_upload_status=files_upload_status,
            guide_type=guide_type,
            input_file_set_for=input_file_set_for,
            integrated_content_files_id=integrated_content_files_id,
            integrated_content_files_accession=integrated_content_files_accession,
            integrated_content_files_aliases=integrated_content_files_aliases,
            integrated_content_files_content_type=integrated_content_files_content_type,
            integrated_content_files_file_format=integrated_content_files_file_format,
            integrated_content_files_upload_status=integrated_content_files_upload_status,
            lab_id=lab_id,
            lab_title=lab_title,
            large_scale_gene_list_id=large_scale_gene_list_id,
            large_scale_gene_list_accession=large_scale_gene_list_accession,
            large_scale_gene_list_aliases=large_scale_gene_list_aliases,
            large_scale_loci_list_id=large_scale_loci_list_id,
            large_scale_loci_list_accession=large_scale_loci_list_accession,
            large_scale_loci_list_aliases=large_scale_loci_list_aliases,
            lot_id=lot_id,
            lower_bound_guide_coverage=lower_bound_guide_coverage,
            lower_bound_insert_size=lower_bound_insert_size,
            notes=notes,
            orf_list_id=orf_list_id,
            orf_list_aliases=orf_list_aliases,
            orf_list_genes_id=orf_list_genes_id,
            orf_list_genes_symbol=orf_list_genes_symbol,
            orf_list_orf_id=orf_list_orf_id,
            product_id=product_id,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            scope=scope,
            selection_criteria=selection_criteria,
            small_scale_gene_list_id=small_scale_gene_list_id,
            small_scale_gene_list_geneid=small_scale_gene_list_geneid,
            small_scale_gene_list_name=small_scale_gene_list_name,
            small_scale_gene_list_symbol=small_scale_gene_list_symbol,
            small_scale_gene_list_synonyms=small_scale_gene_list_synonyms,
            small_scale_loci_list=small_scale_loci_list,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            targeton=targeton,
            tiling_modality=tiling_modality,
            upper_bound_guide_coverage=upper_bound_guide_coverage,
            upper_bound_insert_size=upper_bound_insert_size,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ConstructLibrarySetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def construct_library_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        applied_to_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.@id")] = None,
        applied_to_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.accession")] = None,
        applied_to_samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.aliases")] = None,
        applied_to_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.classifications")] = None,
        applied_to_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.disease_terms.@id")] = None,
        applied_to_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.disease_terms.term_name")] = None,
        applied_to_samples_donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.donors.@id")] = None,
        applied_to_samples_donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.donors.taxa")] = None,
        applied_to_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.sample_terms.@id")] = None,
        applied_to_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.sample_terms.term_name")] = None,
        applied_to_samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.status")] = None,
        applied_to_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.summary")] = None,
        applied_to_samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.targeted_sample_term.@id")] = None,
        applied_to_samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.targeted_sample_term.term_name")] = None,
        applied_to_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.taxa")] = None,
        associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by associated_phenotypes.@id")] = None,
        associated_phenotypes_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by associated_phenotypes.term_id")] = None,
        associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by associated_phenotypes.term_name")] = None,
        average_guide_coverage: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by average_guide_coverage")] = None,
        average_insert_size: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by average_insert_size")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        exon: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by exon")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        guide_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by guide_type")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        integrated_content_files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.@id")] = None,
        integrated_content_files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.accession")] = None,
        integrated_content_files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_content_files.aliases")] = None,
        integrated_content_files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.content_type")] = None,
        integrated_content_files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.file_format")] = None,
        integrated_content_files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.upload_status")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        large_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.@id")] = None,
        large_scale_gene_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.accession")] = None,
        large_scale_gene_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_gene_list.aliases")] = None,
        large_scale_loci_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.@id")] = None,
        large_scale_loci_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.accession")] = None,
        large_scale_loci_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_loci_list.aliases")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_guide_coverage: Annotated[Optional[List[StrictInt]], Field(description="Filter by lower_bound_guide_coverage")] = None,
        lower_bound_insert_size: Annotated[Optional[List[StrictInt]], Field(description="Filter by lower_bound_insert_size")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        orf_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.@id")] = None,
        orf_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_list.aliases")] = None,
        orf_list_genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.genes.@id")] = None,
        orf_list_genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.genes.symbol")] = None,
        orf_list_orf_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_list.orf_id")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        scope: Annotated[Optional[List[StrictStr]], Field(description="Filter by scope")] = None,
        selection_criteria: Annotated[Optional[List[StrictStr]], Field(description="Filter by selection_criteria")] = None,
        small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.@id")] = None,
        small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by small_scale_gene_list.geneid")] = None,
        small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.name")] = None,
        small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.symbol")] = None,
        small_scale_gene_list_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.synonyms")] = None,
        small_scale_loci_list: Annotated[Optional[List[Locus]], Field(description="Filter by small_scale_loci_list")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeton: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by targeton")] = None,
        tiling_modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by tiling_modality")] = None,
        upper_bound_guide_coverage: Annotated[Optional[List[StrictInt]], Field(description="Filter by upper_bound_guide_coverage")] = None,
        upper_bound_insert_size: Annotated[Optional[List[StrictInt]], Field(description="Filter by upper_bound_insert_size")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ConstructLibrarySetResults]:
        """List items in the ConstructLibrarySet collection.

        Collection endpoint that accepts various query parameters to filter and sort ConstructLibrarySet items. Supports filtering on fields within ConstructLibrarySet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param applied_to_samples_id: Filter by applied_to_samples.@id
        :type applied_to_samples_id: List[str]
        :param applied_to_samples_accession: Filter by applied_to_samples.accession
        :type applied_to_samples_accession: List[str]
        :param applied_to_samples_aliases: Filter by applied_to_samples.aliases
        :type applied_to_samples_aliases: List[str]
        :param applied_to_samples_classifications: Filter by applied_to_samples.classifications
        :type applied_to_samples_classifications: List[str]
        :param applied_to_samples_disease_terms_id: Filter by applied_to_samples.disease_terms.@id
        :type applied_to_samples_disease_terms_id: List[str]
        :param applied_to_samples_disease_terms_term_name: Filter by applied_to_samples.disease_terms.term_name
        :type applied_to_samples_disease_terms_term_name: List[str]
        :param applied_to_samples_donors_id: Filter by applied_to_samples.donors.@id
        :type applied_to_samples_donors_id: List[str]
        :param applied_to_samples_donors_taxa: Filter by applied_to_samples.donors.taxa
        :type applied_to_samples_donors_taxa: List[str]
        :param applied_to_samples_sample_terms_id: Filter by applied_to_samples.sample_terms.@id
        :type applied_to_samples_sample_terms_id: List[str]
        :param applied_to_samples_sample_terms_term_name: Filter by applied_to_samples.sample_terms.term_name
        :type applied_to_samples_sample_terms_term_name: List[str]
        :param applied_to_samples_status: Filter by applied_to_samples.status
        :type applied_to_samples_status: List[str]
        :param applied_to_samples_summary: Filter by applied_to_samples.summary
        :type applied_to_samples_summary: List[str]
        :param applied_to_samples_targeted_sample_term_id: Filter by applied_to_samples.targeted_sample_term.@id
        :type applied_to_samples_targeted_sample_term_id: List[str]
        :param applied_to_samples_targeted_sample_term_term_name: Filter by applied_to_samples.targeted_sample_term.term_name
        :type applied_to_samples_targeted_sample_term_term_name: List[str]
        :param applied_to_samples_taxa: Filter by applied_to_samples.taxa
        :type applied_to_samples_taxa: List[str]
        :param associated_phenotypes_id: Filter by associated_phenotypes.@id
        :type associated_phenotypes_id: List[str]
        :param associated_phenotypes_term_id: Filter by associated_phenotypes.term_id
        :type associated_phenotypes_term_id: List[str]
        :param associated_phenotypes_term_name: Filter by associated_phenotypes.term_name
        :type associated_phenotypes_term_name: List[str]
        :param average_guide_coverage: Filter by average_guide_coverage
        :type average_guide_coverage: List[float]
        :param average_insert_size: Filter by average_insert_size
        :type average_insert_size: List[float]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param exon: Filter by exon
        :type exon: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param guide_type: Filter by guide_type
        :type guide_type: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param integrated_content_files_id: Filter by integrated_content_files.@id
        :type integrated_content_files_id: List[str]
        :param integrated_content_files_accession: Filter by integrated_content_files.accession
        :type integrated_content_files_accession: List[str]
        :param integrated_content_files_aliases: Filter by integrated_content_files.aliases
        :type integrated_content_files_aliases: List[str]
        :param integrated_content_files_content_type: Filter by integrated_content_files.content_type
        :type integrated_content_files_content_type: List[str]
        :param integrated_content_files_file_format: Filter by integrated_content_files.file_format
        :type integrated_content_files_file_format: List[str]
        :param integrated_content_files_upload_status: Filter by integrated_content_files.upload_status
        :type integrated_content_files_upload_status: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param large_scale_gene_list_id: Filter by large_scale_gene_list.@id
        :type large_scale_gene_list_id: List[str]
        :param large_scale_gene_list_accession: Filter by large_scale_gene_list.accession
        :type large_scale_gene_list_accession: List[str]
        :param large_scale_gene_list_aliases: Filter by large_scale_gene_list.aliases
        :type large_scale_gene_list_aliases: List[str]
        :param large_scale_loci_list_id: Filter by large_scale_loci_list.@id
        :type large_scale_loci_list_id: List[str]
        :param large_scale_loci_list_accession: Filter by large_scale_loci_list.accession
        :type large_scale_loci_list_accession: List[str]
        :param large_scale_loci_list_aliases: Filter by large_scale_loci_list.aliases
        :type large_scale_loci_list_aliases: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_guide_coverage: Filter by lower_bound_guide_coverage
        :type lower_bound_guide_coverage: List[int]
        :param lower_bound_insert_size: Filter by lower_bound_insert_size
        :type lower_bound_insert_size: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param orf_list_id: Filter by orf_list.@id
        :type orf_list_id: List[str]
        :param orf_list_aliases: Filter by orf_list.aliases
        :type orf_list_aliases: List[str]
        :param orf_list_genes_id: Filter by orf_list.genes.@id
        :type orf_list_genes_id: List[str]
        :param orf_list_genes_symbol: Filter by orf_list.genes.symbol
        :type orf_list_genes_symbol: List[str]
        :param orf_list_orf_id: Filter by orf_list.orf_id
        :type orf_list_orf_id: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param scope: Filter by scope
        :type scope: List[str]
        :param selection_criteria: Filter by selection_criteria
        :type selection_criteria: List[str]
        :param small_scale_gene_list_id: Filter by small_scale_gene_list.@id
        :type small_scale_gene_list_id: List[str]
        :param small_scale_gene_list_geneid: Filter by small_scale_gene_list.geneid
        :type small_scale_gene_list_geneid: List[str]
        :param small_scale_gene_list_name: Filter by small_scale_gene_list.name
        :type small_scale_gene_list_name: List[str]
        :param small_scale_gene_list_symbol: Filter by small_scale_gene_list.symbol
        :type small_scale_gene_list_symbol: List[str]
        :param small_scale_gene_list_synonyms: Filter by small_scale_gene_list.synonyms
        :type small_scale_gene_list_synonyms: List[str]
        :param small_scale_loci_list: Filter by small_scale_loci_list
        :type small_scale_loci_list: List[Locus]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeton: Filter by targeton
        :type targeton: List[str]
        :param tiling_modality: Filter by tiling_modality
        :type tiling_modality: List[str]
        :param upper_bound_guide_coverage: Filter by upper_bound_guide_coverage
        :type upper_bound_guide_coverage: List[int]
        :param upper_bound_insert_size: Filter by upper_bound_insert_size
        :type upper_bound_insert_size: List[int]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._construct_library_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            applied_to_samples_id=applied_to_samples_id,
            applied_to_samples_accession=applied_to_samples_accession,
            applied_to_samples_aliases=applied_to_samples_aliases,
            applied_to_samples_classifications=applied_to_samples_classifications,
            applied_to_samples_disease_terms_id=applied_to_samples_disease_terms_id,
            applied_to_samples_disease_terms_term_name=applied_to_samples_disease_terms_term_name,
            applied_to_samples_donors_id=applied_to_samples_donors_id,
            applied_to_samples_donors_taxa=applied_to_samples_donors_taxa,
            applied_to_samples_sample_terms_id=applied_to_samples_sample_terms_id,
            applied_to_samples_sample_terms_term_name=applied_to_samples_sample_terms_term_name,
            applied_to_samples_status=applied_to_samples_status,
            applied_to_samples_summary=applied_to_samples_summary,
            applied_to_samples_targeted_sample_term_id=applied_to_samples_targeted_sample_term_id,
            applied_to_samples_targeted_sample_term_term_name=applied_to_samples_targeted_sample_term_term_name,
            applied_to_samples_taxa=applied_to_samples_taxa,
            associated_phenotypes_id=associated_phenotypes_id,
            associated_phenotypes_term_id=associated_phenotypes_term_id,
            associated_phenotypes_term_name=associated_phenotypes_term_name,
            average_guide_coverage=average_guide_coverage,
            average_insert_size=average_insert_size,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            exon=exon,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_file_format=files_file_format,
            files_upload_status=files_upload_status,
            guide_type=guide_type,
            input_file_set_for=input_file_set_for,
            integrated_content_files_id=integrated_content_files_id,
            integrated_content_files_accession=integrated_content_files_accession,
            integrated_content_files_aliases=integrated_content_files_aliases,
            integrated_content_files_content_type=integrated_content_files_content_type,
            integrated_content_files_file_format=integrated_content_files_file_format,
            integrated_content_files_upload_status=integrated_content_files_upload_status,
            lab_id=lab_id,
            lab_title=lab_title,
            large_scale_gene_list_id=large_scale_gene_list_id,
            large_scale_gene_list_accession=large_scale_gene_list_accession,
            large_scale_gene_list_aliases=large_scale_gene_list_aliases,
            large_scale_loci_list_id=large_scale_loci_list_id,
            large_scale_loci_list_accession=large_scale_loci_list_accession,
            large_scale_loci_list_aliases=large_scale_loci_list_aliases,
            lot_id=lot_id,
            lower_bound_guide_coverage=lower_bound_guide_coverage,
            lower_bound_insert_size=lower_bound_insert_size,
            notes=notes,
            orf_list_id=orf_list_id,
            orf_list_aliases=orf_list_aliases,
            orf_list_genes_id=orf_list_genes_id,
            orf_list_genes_symbol=orf_list_genes_symbol,
            orf_list_orf_id=orf_list_orf_id,
            product_id=product_id,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            scope=scope,
            selection_criteria=selection_criteria,
            small_scale_gene_list_id=small_scale_gene_list_id,
            small_scale_gene_list_geneid=small_scale_gene_list_geneid,
            small_scale_gene_list_name=small_scale_gene_list_name,
            small_scale_gene_list_symbol=small_scale_gene_list_symbol,
            small_scale_gene_list_synonyms=small_scale_gene_list_synonyms,
            small_scale_loci_list=small_scale_loci_list,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            targeton=targeton,
            tiling_modality=tiling_modality,
            upper_bound_guide_coverage=upper_bound_guide_coverage,
            upper_bound_insert_size=upper_bound_insert_size,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ConstructLibrarySetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def construct_library_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        applied_to_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.@id")] = None,
        applied_to_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.accession")] = None,
        applied_to_samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.aliases")] = None,
        applied_to_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.classifications")] = None,
        applied_to_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.disease_terms.@id")] = None,
        applied_to_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.disease_terms.term_name")] = None,
        applied_to_samples_donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.donors.@id")] = None,
        applied_to_samples_donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.donors.taxa")] = None,
        applied_to_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.sample_terms.@id")] = None,
        applied_to_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.sample_terms.term_name")] = None,
        applied_to_samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.status")] = None,
        applied_to_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.summary")] = None,
        applied_to_samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.targeted_sample_term.@id")] = None,
        applied_to_samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by applied_to_samples.targeted_sample_term.term_name")] = None,
        applied_to_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by applied_to_samples.taxa")] = None,
        associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by associated_phenotypes.@id")] = None,
        associated_phenotypes_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by associated_phenotypes.term_id")] = None,
        associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by associated_phenotypes.term_name")] = None,
        average_guide_coverage: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by average_guide_coverage")] = None,
        average_insert_size: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by average_insert_size")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        exon: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by exon")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        guide_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by guide_type")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        integrated_content_files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.@id")] = None,
        integrated_content_files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.accession")] = None,
        integrated_content_files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_content_files.aliases")] = None,
        integrated_content_files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.content_type")] = None,
        integrated_content_files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.file_format")] = None,
        integrated_content_files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_content_files.upload_status")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        large_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.@id")] = None,
        large_scale_gene_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.accession")] = None,
        large_scale_gene_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_gene_list.aliases")] = None,
        large_scale_loci_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.@id")] = None,
        large_scale_loci_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.accession")] = None,
        large_scale_loci_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_loci_list.aliases")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_guide_coverage: Annotated[Optional[List[StrictInt]], Field(description="Filter by lower_bound_guide_coverage")] = None,
        lower_bound_insert_size: Annotated[Optional[List[StrictInt]], Field(description="Filter by lower_bound_insert_size")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        orf_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.@id")] = None,
        orf_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_list.aliases")] = None,
        orf_list_genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.genes.@id")] = None,
        orf_list_genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by orf_list.genes.symbol")] = None,
        orf_list_orf_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_list.orf_id")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        scope: Annotated[Optional[List[StrictStr]], Field(description="Filter by scope")] = None,
        selection_criteria: Annotated[Optional[List[StrictStr]], Field(description="Filter by selection_criteria")] = None,
        small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.@id")] = None,
        small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by small_scale_gene_list.geneid")] = None,
        small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.name")] = None,
        small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.symbol")] = None,
        small_scale_gene_list_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.synonyms")] = None,
        small_scale_loci_list: Annotated[Optional[List[Locus]], Field(description="Filter by small_scale_loci_list")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeton: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by targeton")] = None,
        tiling_modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by tiling_modality")] = None,
        upper_bound_guide_coverage: Annotated[Optional[List[StrictInt]], Field(description="Filter by upper_bound_guide_coverage")] = None,
        upper_bound_insert_size: Annotated[Optional[List[StrictInt]], Field(description="Filter by upper_bound_insert_size")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the ConstructLibrarySet collection.

        Collection endpoint that accepts various query parameters to filter and sort ConstructLibrarySet items. Supports filtering on fields within ConstructLibrarySet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param applied_to_samples_id: Filter by applied_to_samples.@id
        :type applied_to_samples_id: List[str]
        :param applied_to_samples_accession: Filter by applied_to_samples.accession
        :type applied_to_samples_accession: List[str]
        :param applied_to_samples_aliases: Filter by applied_to_samples.aliases
        :type applied_to_samples_aliases: List[str]
        :param applied_to_samples_classifications: Filter by applied_to_samples.classifications
        :type applied_to_samples_classifications: List[str]
        :param applied_to_samples_disease_terms_id: Filter by applied_to_samples.disease_terms.@id
        :type applied_to_samples_disease_terms_id: List[str]
        :param applied_to_samples_disease_terms_term_name: Filter by applied_to_samples.disease_terms.term_name
        :type applied_to_samples_disease_terms_term_name: List[str]
        :param applied_to_samples_donors_id: Filter by applied_to_samples.donors.@id
        :type applied_to_samples_donors_id: List[str]
        :param applied_to_samples_donors_taxa: Filter by applied_to_samples.donors.taxa
        :type applied_to_samples_donors_taxa: List[str]
        :param applied_to_samples_sample_terms_id: Filter by applied_to_samples.sample_terms.@id
        :type applied_to_samples_sample_terms_id: List[str]
        :param applied_to_samples_sample_terms_term_name: Filter by applied_to_samples.sample_terms.term_name
        :type applied_to_samples_sample_terms_term_name: List[str]
        :param applied_to_samples_status: Filter by applied_to_samples.status
        :type applied_to_samples_status: List[str]
        :param applied_to_samples_summary: Filter by applied_to_samples.summary
        :type applied_to_samples_summary: List[str]
        :param applied_to_samples_targeted_sample_term_id: Filter by applied_to_samples.targeted_sample_term.@id
        :type applied_to_samples_targeted_sample_term_id: List[str]
        :param applied_to_samples_targeted_sample_term_term_name: Filter by applied_to_samples.targeted_sample_term.term_name
        :type applied_to_samples_targeted_sample_term_term_name: List[str]
        :param applied_to_samples_taxa: Filter by applied_to_samples.taxa
        :type applied_to_samples_taxa: List[str]
        :param associated_phenotypes_id: Filter by associated_phenotypes.@id
        :type associated_phenotypes_id: List[str]
        :param associated_phenotypes_term_id: Filter by associated_phenotypes.term_id
        :type associated_phenotypes_term_id: List[str]
        :param associated_phenotypes_term_name: Filter by associated_phenotypes.term_name
        :type associated_phenotypes_term_name: List[str]
        :param average_guide_coverage: Filter by average_guide_coverage
        :type average_guide_coverage: List[float]
        :param average_insert_size: Filter by average_insert_size
        :type average_insert_size: List[float]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param exon: Filter by exon
        :type exon: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param guide_type: Filter by guide_type
        :type guide_type: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param integrated_content_files_id: Filter by integrated_content_files.@id
        :type integrated_content_files_id: List[str]
        :param integrated_content_files_accession: Filter by integrated_content_files.accession
        :type integrated_content_files_accession: List[str]
        :param integrated_content_files_aliases: Filter by integrated_content_files.aliases
        :type integrated_content_files_aliases: List[str]
        :param integrated_content_files_content_type: Filter by integrated_content_files.content_type
        :type integrated_content_files_content_type: List[str]
        :param integrated_content_files_file_format: Filter by integrated_content_files.file_format
        :type integrated_content_files_file_format: List[str]
        :param integrated_content_files_upload_status: Filter by integrated_content_files.upload_status
        :type integrated_content_files_upload_status: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param large_scale_gene_list_id: Filter by large_scale_gene_list.@id
        :type large_scale_gene_list_id: List[str]
        :param large_scale_gene_list_accession: Filter by large_scale_gene_list.accession
        :type large_scale_gene_list_accession: List[str]
        :param large_scale_gene_list_aliases: Filter by large_scale_gene_list.aliases
        :type large_scale_gene_list_aliases: List[str]
        :param large_scale_loci_list_id: Filter by large_scale_loci_list.@id
        :type large_scale_loci_list_id: List[str]
        :param large_scale_loci_list_accession: Filter by large_scale_loci_list.accession
        :type large_scale_loci_list_accession: List[str]
        :param large_scale_loci_list_aliases: Filter by large_scale_loci_list.aliases
        :type large_scale_loci_list_aliases: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_guide_coverage: Filter by lower_bound_guide_coverage
        :type lower_bound_guide_coverage: List[int]
        :param lower_bound_insert_size: Filter by lower_bound_insert_size
        :type lower_bound_insert_size: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param orf_list_id: Filter by orf_list.@id
        :type orf_list_id: List[str]
        :param orf_list_aliases: Filter by orf_list.aliases
        :type orf_list_aliases: List[str]
        :param orf_list_genes_id: Filter by orf_list.genes.@id
        :type orf_list_genes_id: List[str]
        :param orf_list_genes_symbol: Filter by orf_list.genes.symbol
        :type orf_list_genes_symbol: List[str]
        :param orf_list_orf_id: Filter by orf_list.orf_id
        :type orf_list_orf_id: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param scope: Filter by scope
        :type scope: List[str]
        :param selection_criteria: Filter by selection_criteria
        :type selection_criteria: List[str]
        :param small_scale_gene_list_id: Filter by small_scale_gene_list.@id
        :type small_scale_gene_list_id: List[str]
        :param small_scale_gene_list_geneid: Filter by small_scale_gene_list.geneid
        :type small_scale_gene_list_geneid: List[str]
        :param small_scale_gene_list_name: Filter by small_scale_gene_list.name
        :type small_scale_gene_list_name: List[str]
        :param small_scale_gene_list_symbol: Filter by small_scale_gene_list.symbol
        :type small_scale_gene_list_symbol: List[str]
        :param small_scale_gene_list_synonyms: Filter by small_scale_gene_list.synonyms
        :type small_scale_gene_list_synonyms: List[str]
        :param small_scale_loci_list: Filter by small_scale_loci_list
        :type small_scale_loci_list: List[Locus]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeton: Filter by targeton
        :type targeton: List[str]
        :param tiling_modality: Filter by tiling_modality
        :type tiling_modality: List[str]
        :param upper_bound_guide_coverage: Filter by upper_bound_guide_coverage
        :type upper_bound_guide_coverage: List[int]
        :param upper_bound_insert_size: Filter by upper_bound_insert_size
        :type upper_bound_insert_size: List[int]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._construct_library_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            applied_to_samples_id=applied_to_samples_id,
            applied_to_samples_accession=applied_to_samples_accession,
            applied_to_samples_aliases=applied_to_samples_aliases,
            applied_to_samples_classifications=applied_to_samples_classifications,
            applied_to_samples_disease_terms_id=applied_to_samples_disease_terms_id,
            applied_to_samples_disease_terms_term_name=applied_to_samples_disease_terms_term_name,
            applied_to_samples_donors_id=applied_to_samples_donors_id,
            applied_to_samples_donors_taxa=applied_to_samples_donors_taxa,
            applied_to_samples_sample_terms_id=applied_to_samples_sample_terms_id,
            applied_to_samples_sample_terms_term_name=applied_to_samples_sample_terms_term_name,
            applied_to_samples_status=applied_to_samples_status,
            applied_to_samples_summary=applied_to_samples_summary,
            applied_to_samples_targeted_sample_term_id=applied_to_samples_targeted_sample_term_id,
            applied_to_samples_targeted_sample_term_term_name=applied_to_samples_targeted_sample_term_term_name,
            applied_to_samples_taxa=applied_to_samples_taxa,
            associated_phenotypes_id=associated_phenotypes_id,
            associated_phenotypes_term_id=associated_phenotypes_term_id,
            associated_phenotypes_term_name=associated_phenotypes_term_name,
            average_guide_coverage=average_guide_coverage,
            average_insert_size=average_insert_size,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            exon=exon,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_file_format=files_file_format,
            files_upload_status=files_upload_status,
            guide_type=guide_type,
            input_file_set_for=input_file_set_for,
            integrated_content_files_id=integrated_content_files_id,
            integrated_content_files_accession=integrated_content_files_accession,
            integrated_content_files_aliases=integrated_content_files_aliases,
            integrated_content_files_content_type=integrated_content_files_content_type,
            integrated_content_files_file_format=integrated_content_files_file_format,
            integrated_content_files_upload_status=integrated_content_files_upload_status,
            lab_id=lab_id,
            lab_title=lab_title,
            large_scale_gene_list_id=large_scale_gene_list_id,
            large_scale_gene_list_accession=large_scale_gene_list_accession,
            large_scale_gene_list_aliases=large_scale_gene_list_aliases,
            large_scale_loci_list_id=large_scale_loci_list_id,
            large_scale_loci_list_accession=large_scale_loci_list_accession,
            large_scale_loci_list_aliases=large_scale_loci_list_aliases,
            lot_id=lot_id,
            lower_bound_guide_coverage=lower_bound_guide_coverage,
            lower_bound_insert_size=lower_bound_insert_size,
            notes=notes,
            orf_list_id=orf_list_id,
            orf_list_aliases=orf_list_aliases,
            orf_list_genes_id=orf_list_genes_id,
            orf_list_genes_symbol=orf_list_genes_symbol,
            orf_list_orf_id=orf_list_orf_id,
            product_id=product_id,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            scope=scope,
            selection_criteria=selection_criteria,
            small_scale_gene_list_id=small_scale_gene_list_id,
            small_scale_gene_list_geneid=small_scale_gene_list_geneid,
            small_scale_gene_list_name=small_scale_gene_list_name,
            small_scale_gene_list_symbol=small_scale_gene_list_symbol,
            small_scale_gene_list_synonyms=small_scale_gene_list_synonyms,
            small_scale_loci_list=small_scale_loci_list,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            targeton=targeton,
            tiling_modality=tiling_modality,
            upper_bound_guide_coverage=upper_bound_guide_coverage,
            upper_bound_insert_size=upper_bound_insert_size,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ConstructLibrarySetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _construct_library_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        applied_to_samples_id,
        applied_to_samples_accession,
        applied_to_samples_aliases,
        applied_to_samples_classifications,
        applied_to_samples_disease_terms_id,
        applied_to_samples_disease_terms_term_name,
        applied_to_samples_donors_id,
        applied_to_samples_donors_taxa,
        applied_to_samples_sample_terms_id,
        applied_to_samples_sample_terms_term_name,
        applied_to_samples_status,
        applied_to_samples_summary,
        applied_to_samples_targeted_sample_term_id,
        applied_to_samples_targeted_sample_term_term_name,
        applied_to_samples_taxa,
        associated_phenotypes_id,
        associated_phenotypes_term_id,
        associated_phenotypes_term_name,
        average_guide_coverage,
        average_insert_size,
        award_id,
        award_component,
        collections,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        description,
        documents,
        exon,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_file_format,
        files_upload_status,
        guide_type,
        input_file_set_for,
        integrated_content_files_id,
        integrated_content_files_accession,
        integrated_content_files_aliases,
        integrated_content_files_content_type,
        integrated_content_files_file_format,
        integrated_content_files_upload_status,
        lab_id,
        lab_title,
        large_scale_gene_list_id,
        large_scale_gene_list_accession,
        large_scale_gene_list_aliases,
        large_scale_loci_list_id,
        large_scale_loci_list_accession,
        large_scale_loci_list_aliases,
        lot_id,
        lower_bound_guide_coverage,
        lower_bound_insert_size,
        notes,
        orf_list_id,
        orf_list_aliases,
        orf_list_genes_id,
        orf_list_genes_symbol,
        orf_list_orf_id,
        product_id,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        scope,
        selection_criteria,
        small_scale_gene_list_id,
        small_scale_gene_list_geneid,
        small_scale_gene_list_name,
        small_scale_gene_list_symbol,
        small_scale_gene_list_synonyms,
        small_scale_loci_list,
        sources,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        targeton,
        tiling_modality,
        upper_bound_guide_coverage,
        upper_bound_insert_size,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'applied_to_samples.@id': 'multi',
            'applied_to_samples.accession': 'multi',
            'applied_to_samples.aliases': 'multi',
            'applied_to_samples.classifications': 'multi',
            'applied_to_samples.disease_terms.@id': 'multi',
            'applied_to_samples.disease_terms.term_name': 'multi',
            'applied_to_samples.donors.@id': 'multi',
            'applied_to_samples.donors.taxa': 'multi',
            'applied_to_samples.sample_terms.@id': 'multi',
            'applied_to_samples.sample_terms.term_name': 'multi',
            'applied_to_samples.status': 'multi',
            'applied_to_samples.summary': 'multi',
            'applied_to_samples.targeted_sample_term.@id': 'multi',
            'applied_to_samples.targeted_sample_term.term_name': 'multi',
            'applied_to_samples.taxa': 'multi',
            'associated_phenotypes.@id': 'multi',
            'associated_phenotypes.term_id': 'multi',
            'associated_phenotypes.term_name': 'multi',
            'average_guide_coverage': 'multi',
            'average_insert_size': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'exon': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.file_format': 'multi',
            'files.upload_status': 'multi',
            'guide_type': 'multi',
            'input_file_set_for': 'multi',
            'integrated_content_files.@id': 'multi',
            'integrated_content_files.accession': 'multi',
            'integrated_content_files.aliases': 'multi',
            'integrated_content_files.content_type': 'multi',
            'integrated_content_files.file_format': 'multi',
            'integrated_content_files.upload_status': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'large_scale_gene_list.@id': 'multi',
            'large_scale_gene_list.accession': 'multi',
            'large_scale_gene_list.aliases': 'multi',
            'large_scale_loci_list.@id': 'multi',
            'large_scale_loci_list.accession': 'multi',
            'large_scale_loci_list.aliases': 'multi',
            'lot_id': 'multi',
            'lower_bound_guide_coverage': 'multi',
            'lower_bound_insert_size': 'multi',
            'notes': 'multi',
            'orf_list.@id': 'multi',
            'orf_list.aliases': 'multi',
            'orf_list.genes.@id': 'multi',
            'orf_list.genes.symbol': 'multi',
            'orf_list.orf_id': 'multi',
            'product_id': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'scope': 'multi',
            'selection_criteria': 'multi',
            'small_scale_gene_list.@id': 'multi',
            'small_scale_gene_list.geneid': 'multi',
            'small_scale_gene_list.name': 'multi',
            'small_scale_gene_list.symbol': 'multi',
            'small_scale_gene_list.synonyms': 'multi',
            'small_scale_loci_list': 'multi',
            'sources': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'targeton': 'multi',
            'tiling_modality': 'multi',
            'upper_bound_guide_coverage': 'multi',
            'upper_bound_insert_size': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if applied_to_samples_id is not None:
            
            _query_params.append(('applied_to_samples.@id', applied_to_samples_id))
            
        if applied_to_samples_accession is not None:
            
            _query_params.append(('applied_to_samples.accession', applied_to_samples_accession))
            
        if applied_to_samples_aliases is not None:
            
            _query_params.append(('applied_to_samples.aliases', applied_to_samples_aliases))
            
        if applied_to_samples_classifications is not None:
            
            _query_params.append(('applied_to_samples.classifications', applied_to_samples_classifications))
            
        if applied_to_samples_disease_terms_id is not None:
            
            _query_params.append(('applied_to_samples.disease_terms.@id', applied_to_samples_disease_terms_id))
            
        if applied_to_samples_disease_terms_term_name is not None:
            
            _query_params.append(('applied_to_samples.disease_terms.term_name', applied_to_samples_disease_terms_term_name))
            
        if applied_to_samples_donors_id is not None:
            
            _query_params.append(('applied_to_samples.donors.@id', applied_to_samples_donors_id))
            
        if applied_to_samples_donors_taxa is not None:
            
            _query_params.append(('applied_to_samples.donors.taxa', applied_to_samples_donors_taxa))
            
        if applied_to_samples_sample_terms_id is not None:
            
            _query_params.append(('applied_to_samples.sample_terms.@id', applied_to_samples_sample_terms_id))
            
        if applied_to_samples_sample_terms_term_name is not None:
            
            _query_params.append(('applied_to_samples.sample_terms.term_name', applied_to_samples_sample_terms_term_name))
            
        if applied_to_samples_status is not None:
            
            _query_params.append(('applied_to_samples.status', applied_to_samples_status))
            
        if applied_to_samples_summary is not None:
            
            _query_params.append(('applied_to_samples.summary', applied_to_samples_summary))
            
        if applied_to_samples_targeted_sample_term_id is not None:
            
            _query_params.append(('applied_to_samples.targeted_sample_term.@id', applied_to_samples_targeted_sample_term_id))
            
        if applied_to_samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('applied_to_samples.targeted_sample_term.term_name', applied_to_samples_targeted_sample_term_term_name))
            
        if applied_to_samples_taxa is not None:
            
            _query_params.append(('applied_to_samples.taxa', applied_to_samples_taxa))
            
        if associated_phenotypes_id is not None:
            
            _query_params.append(('associated_phenotypes.@id', associated_phenotypes_id))
            
        if associated_phenotypes_term_id is not None:
            
            _query_params.append(('associated_phenotypes.term_id', associated_phenotypes_term_id))
            
        if associated_phenotypes_term_name is not None:
            
            _query_params.append(('associated_phenotypes.term_name', associated_phenotypes_term_name))
            
        if average_guide_coverage is not None:
            
            _query_params.append(('average_guide_coverage', average_guide_coverage))
            
        if average_insert_size is not None:
            
            _query_params.append(('average_insert_size', average_insert_size))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if exon is not None:
            
            _query_params.append(('exon', exon))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if guide_type is not None:
            
            _query_params.append(('guide_type', guide_type))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if integrated_content_files_id is not None:
            
            _query_params.append(('integrated_content_files.@id', integrated_content_files_id))
            
        if integrated_content_files_accession is not None:
            
            _query_params.append(('integrated_content_files.accession', integrated_content_files_accession))
            
        if integrated_content_files_aliases is not None:
            
            _query_params.append(('integrated_content_files.aliases', integrated_content_files_aliases))
            
        if integrated_content_files_content_type is not None:
            
            _query_params.append(('integrated_content_files.content_type', integrated_content_files_content_type))
            
        if integrated_content_files_file_format is not None:
            
            _query_params.append(('integrated_content_files.file_format', integrated_content_files_file_format))
            
        if integrated_content_files_upload_status is not None:
            
            _query_params.append(('integrated_content_files.upload_status', integrated_content_files_upload_status))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if large_scale_gene_list_id is not None:
            
            _query_params.append(('large_scale_gene_list.@id', large_scale_gene_list_id))
            
        if large_scale_gene_list_accession is not None:
            
            _query_params.append(('large_scale_gene_list.accession', large_scale_gene_list_accession))
            
        if large_scale_gene_list_aliases is not None:
            
            _query_params.append(('large_scale_gene_list.aliases', large_scale_gene_list_aliases))
            
        if large_scale_loci_list_id is not None:
            
            _query_params.append(('large_scale_loci_list.@id', large_scale_loci_list_id))
            
        if large_scale_loci_list_accession is not None:
            
            _query_params.append(('large_scale_loci_list.accession', large_scale_loci_list_accession))
            
        if large_scale_loci_list_aliases is not None:
            
            _query_params.append(('large_scale_loci_list.aliases', large_scale_loci_list_aliases))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if lower_bound_guide_coverage is not None:
            
            _query_params.append(('lower_bound_guide_coverage', lower_bound_guide_coverage))
            
        if lower_bound_insert_size is not None:
            
            _query_params.append(('lower_bound_insert_size', lower_bound_insert_size))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if orf_list_id is not None:
            
            _query_params.append(('orf_list.@id', orf_list_id))
            
        if orf_list_aliases is not None:
            
            _query_params.append(('orf_list.aliases', orf_list_aliases))
            
        if orf_list_genes_id is not None:
            
            _query_params.append(('orf_list.genes.@id', orf_list_genes_id))
            
        if orf_list_genes_symbol is not None:
            
            _query_params.append(('orf_list.genes.symbol', orf_list_genes_symbol))
            
        if orf_list_orf_id is not None:
            
            _query_params.append(('orf_list.orf_id', orf_list_orf_id))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if scope is not None:
            
            _query_params.append(('scope', scope))
            
        if selection_criteria is not None:
            
            _query_params.append(('selection_criteria', selection_criteria))
            
        if small_scale_gene_list_id is not None:
            
            _query_params.append(('small_scale_gene_list.@id', small_scale_gene_list_id))
            
        if small_scale_gene_list_geneid is not None:
            
            _query_params.append(('small_scale_gene_list.geneid', small_scale_gene_list_geneid))
            
        if small_scale_gene_list_name is not None:
            
            _query_params.append(('small_scale_gene_list.name', small_scale_gene_list_name))
            
        if small_scale_gene_list_symbol is not None:
            
            _query_params.append(('small_scale_gene_list.symbol', small_scale_gene_list_symbol))
            
        if small_scale_gene_list_synonyms is not None:
            
            _query_params.append(('small_scale_gene_list.synonyms', small_scale_gene_list_synonyms))
            
        if small_scale_loci_list is not None:
            
            _query_params.append(('small_scale_loci_list', small_scale_loci_list))
            
        if sources is not None:
            
            _query_params.append(('sources', sources))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if targeton is not None:
            
            _query_params.append(('targeton', targeton))
            
        if tiling_modality is not None:
            
            _query_params.append(('tiling_modality', tiling_modality))
            
        if upper_bound_guide_coverage is not None:
            
            _query_params.append(('upper_bound_guide_coverage', upper_bound_guide_coverage))
            
        if upper_bound_insert_size is not None:
            
            _query_params.append(('upper_bound_insert_size', upper_bound_insert_size))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/construct-library-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def crispr_modifications(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        activated: Annotated[Optional[List[StrictBool]], Field(description="Filter by activated")] = None,
        activating_agent_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by activating_agent_term_id")] = None,
        activating_agent_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by activating_agent_term_name")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_modified: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_modified")] = None,
        cas: Annotated[Optional[List[StrictStr]], Field(description="Filter by cas")] = None,
        cas_species: Annotated[Optional[List[StrictStr]], Field(description="Filter by cas_species")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        fused_domain: Annotated[Optional[List[StrictStr]], Field(description="Filter by fused_domain")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by modality")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        tagged_protein: Annotated[Optional[List[StrictStr]], Field(description="Filter by tagged_protein")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> CrisprModificationResults:
        """List items in the CrisprModification collection.

        Collection endpoint that accepts various query parameters to filter and sort CrisprModification items. Supports filtering on fields within CrisprModification items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param activated: Filter by activated
        :type activated: List[bool]
        :param activating_agent_term_id: Filter by activating_agent_term_id
        :type activating_agent_term_id: List[str]
        :param activating_agent_term_name: Filter by activating_agent_term_name
        :type activating_agent_term_name: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_modified: Filter by biosamples_modified
        :type biosamples_modified: List[str]
        :param cas: Filter by cas
        :type cas: List[str]
        :param cas_species: Filter by cas_species
        :type cas_species: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param fused_domain: Filter by fused_domain
        :type fused_domain: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param modality: Filter by modality
        :type modality: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param tagged_protein: Filter by tagged_protein
        :type tagged_protein: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._crispr_modifications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            activated=activated,
            activating_agent_term_id=activating_agent_term_id,
            activating_agent_term_name=activating_agent_term_name,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            biosamples_modified=biosamples_modified,
            cas=cas,
            cas_species=cas_species,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            fused_domain=fused_domain,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            modality=modality,
            notes=notes,
            product_id=product_id,
            release_timestamp=release_timestamp,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            tagged_protein=tagged_protein,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "CrisprModificationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def crispr_modifications_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        activated: Annotated[Optional[List[StrictBool]], Field(description="Filter by activated")] = None,
        activating_agent_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by activating_agent_term_id")] = None,
        activating_agent_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by activating_agent_term_name")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_modified: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_modified")] = None,
        cas: Annotated[Optional[List[StrictStr]], Field(description="Filter by cas")] = None,
        cas_species: Annotated[Optional[List[StrictStr]], Field(description="Filter by cas_species")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        fused_domain: Annotated[Optional[List[StrictStr]], Field(description="Filter by fused_domain")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by modality")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        tagged_protein: Annotated[Optional[List[StrictStr]], Field(description="Filter by tagged_protein")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[CrisprModificationResults]:
        """List items in the CrisprModification collection.

        Collection endpoint that accepts various query parameters to filter and sort CrisprModification items. Supports filtering on fields within CrisprModification items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param activated: Filter by activated
        :type activated: List[bool]
        :param activating_agent_term_id: Filter by activating_agent_term_id
        :type activating_agent_term_id: List[str]
        :param activating_agent_term_name: Filter by activating_agent_term_name
        :type activating_agent_term_name: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_modified: Filter by biosamples_modified
        :type biosamples_modified: List[str]
        :param cas: Filter by cas
        :type cas: List[str]
        :param cas_species: Filter by cas_species
        :type cas_species: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param fused_domain: Filter by fused_domain
        :type fused_domain: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param modality: Filter by modality
        :type modality: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param tagged_protein: Filter by tagged_protein
        :type tagged_protein: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._crispr_modifications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            activated=activated,
            activating_agent_term_id=activating_agent_term_id,
            activating_agent_term_name=activating_agent_term_name,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            biosamples_modified=biosamples_modified,
            cas=cas,
            cas_species=cas_species,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            fused_domain=fused_domain,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            modality=modality,
            notes=notes,
            product_id=product_id,
            release_timestamp=release_timestamp,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            tagged_protein=tagged_protein,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "CrisprModificationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def crispr_modifications_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        activated: Annotated[Optional[List[StrictBool]], Field(description="Filter by activated")] = None,
        activating_agent_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by activating_agent_term_id")] = None,
        activating_agent_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by activating_agent_term_name")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_modified: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_modified")] = None,
        cas: Annotated[Optional[List[StrictStr]], Field(description="Filter by cas")] = None,
        cas_species: Annotated[Optional[List[StrictStr]], Field(description="Filter by cas_species")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        fused_domain: Annotated[Optional[List[StrictStr]], Field(description="Filter by fused_domain")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by modality")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        tagged_protein: Annotated[Optional[List[StrictStr]], Field(description="Filter by tagged_protein")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the CrisprModification collection.

        Collection endpoint that accepts various query parameters to filter and sort CrisprModification items. Supports filtering on fields within CrisprModification items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param activated: Filter by activated
        :type activated: List[bool]
        :param activating_agent_term_id: Filter by activating_agent_term_id
        :type activating_agent_term_id: List[str]
        :param activating_agent_term_name: Filter by activating_agent_term_name
        :type activating_agent_term_name: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_modified: Filter by biosamples_modified
        :type biosamples_modified: List[str]
        :param cas: Filter by cas
        :type cas: List[str]
        :param cas_species: Filter by cas_species
        :type cas_species: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param fused_domain: Filter by fused_domain
        :type fused_domain: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param modality: Filter by modality
        :type modality: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param tagged_protein: Filter by tagged_protein
        :type tagged_protein: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._crispr_modifications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            activated=activated,
            activating_agent_term_id=activating_agent_term_id,
            activating_agent_term_name=activating_agent_term_name,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            biosamples_modified=biosamples_modified,
            cas=cas,
            cas_species=cas_species,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            fused_domain=fused_domain,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            modality=modality,
            notes=notes,
            product_id=product_id,
            release_timestamp=release_timestamp,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            tagged_protein=tagged_protein,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "CrisprModificationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _crispr_modifications_serialize(
        self,
        query,
        limit,
        sort,
        id,
        activated,
        activating_agent_term_id,
        activating_agent_term_name,
        aliases,
        award_id,
        award_component,
        biosamples_modified,
        cas,
        cas_species,
        creation_timestamp,
        description,
        documents,
        fused_domain,
        lab_id,
        lab_title,
        lot_id,
        modality,
        notes,
        product_id,
        release_timestamp,
        sources,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        tagged_protein,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'activated': 'multi',
            'activating_agent_term_id': 'multi',
            'activating_agent_term_name': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biosamples_modified': 'multi',
            'cas': 'multi',
            'cas_species': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'fused_domain': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'modality': 'multi',
            'notes': 'multi',
            'product_id': 'multi',
            'release_timestamp': 'multi',
            'sources': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'tagged_protein': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if activated is not None:
            
            _query_params.append(('activated', activated))
            
        if activating_agent_term_id is not None:
            
            _query_params.append(('activating_agent_term_id', activating_agent_term_id))
            
        if activating_agent_term_name is not None:
            
            _query_params.append(('activating_agent_term_name', activating_agent_term_name))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biosamples_modified is not None:
            
            _query_params.append(('biosamples_modified', biosamples_modified))
            
        if cas is not None:
            
            _query_params.append(('cas', cas))
            
        if cas_species is not None:
            
            _query_params.append(('cas_species', cas_species))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if fused_domain is not None:
            
            _query_params.append(('fused_domain', fused_domain))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if modality is not None:
            
            _query_params.append(('modality', modality))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if sources is not None:
            
            _query_params.append(('sources', sources))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if tagged_protein is not None:
            
            _query_params.append(('tagged_protein', tagged_protein))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/crispr-modifications/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def curated_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assemblies: Annotated[Optional[List[StrictStr]], Field(description="Filter by assemblies")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        transcriptome_annotations: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotations")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> CuratedSetResults:
        """List items in the CuratedSet collection.

        Collection endpoint that accepts various query parameters to filter and sort CuratedSet items. Supports filtering on fields within CuratedSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assemblies: Filter by assemblies
        :type assemblies: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param transcriptome_annotations: Filter by transcriptome_annotations
        :type transcriptome_annotations: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._curated_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assemblies=assemblies,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            transcriptome_annotations=transcriptome_annotations,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "CuratedSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def curated_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assemblies: Annotated[Optional[List[StrictStr]], Field(description="Filter by assemblies")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        transcriptome_annotations: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotations")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[CuratedSetResults]:
        """List items in the CuratedSet collection.

        Collection endpoint that accepts various query parameters to filter and sort CuratedSet items. Supports filtering on fields within CuratedSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assemblies: Filter by assemblies
        :type assemblies: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param transcriptome_annotations: Filter by transcriptome_annotations
        :type transcriptome_annotations: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._curated_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assemblies=assemblies,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            transcriptome_annotations=transcriptome_annotations,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "CuratedSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def curated_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assemblies: Annotated[Optional[List[StrictStr]], Field(description="Filter by assemblies")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        transcriptome_annotations: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotations")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the CuratedSet collection.

        Collection endpoint that accepts various query parameters to filter and sort CuratedSet items. Supports filtering on fields within CuratedSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assemblies: Filter by assemblies
        :type assemblies: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param transcriptome_annotations: Filter by transcriptome_annotations
        :type transcriptome_annotations: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._curated_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assemblies=assemblies,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            transcriptome_annotations=transcriptome_annotations,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "CuratedSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _curated_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        assemblies,
        award_id,
        award_component,
        award_contact_pi_id,
        award_contact_pi_title,
        award_title,
        collections,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        donors_id,
        donors_accession,
        donors_aliases,
        donors_sex,
        donors_status,
        donors_taxa,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_creation_timestamp,
        files_file_format,
        files_file_size,
        files_href,
        files_s3_uri,
        files_sequencing_platform,
        files_submitted_file_name,
        files_upload_status,
        input_file_set_for,
        lab_id,
        lab_title,
        notes,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        samples_id,
        samples_accession,
        samples_aliases,
        samples_cell_fate_change_treatments,
        samples_classifications,
        samples_construct_library_sets,
        samples_disease_terms_id,
        samples_disease_terms_term_name,
        samples_modifications,
        samples_sample_terms_id,
        samples_sample_terms_aliases,
        samples_sample_terms_status,
        samples_sample_terms_summary,
        samples_sample_terms_term_name,
        samples_status,
        samples_summary,
        samples_targeted_sample_term_id,
        samples_targeted_sample_term_term_name,
        samples_taxa,
        samples_treatments,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        taxa,
        transcriptome_annotations,
        url,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'assemblies': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.contact_pi.@id': 'multi',
            'award.contact_pi.title': 'multi',
            'award.title': 'multi',
            'collections': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'donors.@id': 'multi',
            'donors.accession': 'multi',
            'donors.aliases': 'multi',
            'donors.sex': 'multi',
            'donors.status': 'multi',
            'donors.taxa': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.creation_timestamp': 'multi',
            'files.file_format': 'multi',
            'files.file_size': 'multi',
            'files.href': 'multi',
            'files.s3_uri': 'multi',
            'files.sequencing_platform': 'multi',
            'files.submitted_file_name': 'multi',
            'files.upload_status': 'multi',
            'input_file_set_for': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'samples.@id': 'multi',
            'samples.accession': 'multi',
            'samples.aliases': 'multi',
            'samples.cell_fate_change_treatments': 'multi',
            'samples.classifications': 'multi',
            'samples.construct_library_sets': 'multi',
            'samples.disease_terms.@id': 'multi',
            'samples.disease_terms.term_name': 'multi',
            'samples.modifications': 'multi',
            'samples.sample_terms.@id': 'multi',
            'samples.sample_terms.aliases': 'multi',
            'samples.sample_terms.status': 'multi',
            'samples.sample_terms.summary': 'multi',
            'samples.sample_terms.term_name': 'multi',
            'samples.status': 'multi',
            'samples.summary': 'multi',
            'samples.targeted_sample_term.@id': 'multi',
            'samples.targeted_sample_term.term_name': 'multi',
            'samples.taxa': 'multi',
            'samples.treatments': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'transcriptome_annotations': 'multi',
            'url': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if assemblies is not None:
            
            _query_params.append(('assemblies', assemblies))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_contact_pi_id is not None:
            
            _query_params.append(('award.contact_pi.@id', award_contact_pi_id))
            
        if award_contact_pi_title is not None:
            
            _query_params.append(('award.contact_pi.title', award_contact_pi_title))
            
        if award_title is not None:
            
            _query_params.append(('award.title', award_title))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors_id is not None:
            
            _query_params.append(('donors.@id', donors_id))
            
        if donors_accession is not None:
            
            _query_params.append(('donors.accession', donors_accession))
            
        if donors_aliases is not None:
            
            _query_params.append(('donors.aliases', donors_aliases))
            
        if donors_sex is not None:
            
            _query_params.append(('donors.sex', donors_sex))
            
        if donors_status is not None:
            
            _query_params.append(('donors.status', donors_status))
            
        if donors_taxa is not None:
            
            _query_params.append(('donors.taxa', donors_taxa))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_creation_timestamp is not None:
            
            _query_params.append(('files.creation_timestamp', files_creation_timestamp))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_file_size is not None:
            
            _query_params.append(('files.file_size', files_file_size))
            
        if files_href is not None:
            
            _query_params.append(('files.href', files_href))
            
        if files_s3_uri is not None:
            
            _query_params.append(('files.s3_uri', files_s3_uri))
            
        if files_sequencing_platform is not None:
            
            _query_params.append(('files.sequencing_platform', files_sequencing_platform))
            
        if files_submitted_file_name is not None:
            
            _query_params.append(('files.submitted_file_name', files_submitted_file_name))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if samples_id is not None:
            
            _query_params.append(('samples.@id', samples_id))
            
        if samples_accession is not None:
            
            _query_params.append(('samples.accession', samples_accession))
            
        if samples_aliases is not None:
            
            _query_params.append(('samples.aliases', samples_aliases))
            
        if samples_cell_fate_change_treatments is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments', samples_cell_fate_change_treatments))
            
        if samples_classifications is not None:
            
            _query_params.append(('samples.classifications', samples_classifications))
            
        if samples_construct_library_sets is not None:
            
            _query_params.append(('samples.construct_library_sets', samples_construct_library_sets))
            
        if samples_disease_terms_id is not None:
            
            _query_params.append(('samples.disease_terms.@id', samples_disease_terms_id))
            
        if samples_disease_terms_term_name is not None:
            
            _query_params.append(('samples.disease_terms.term_name', samples_disease_terms_term_name))
            
        if samples_modifications is not None:
            
            _query_params.append(('samples.modifications', samples_modifications))
            
        if samples_sample_terms_id is not None:
            
            _query_params.append(('samples.sample_terms.@id', samples_sample_terms_id))
            
        if samples_sample_terms_aliases is not None:
            
            _query_params.append(('samples.sample_terms.aliases', samples_sample_terms_aliases))
            
        if samples_sample_terms_status is not None:
            
            _query_params.append(('samples.sample_terms.status', samples_sample_terms_status))
            
        if samples_sample_terms_summary is not None:
            
            _query_params.append(('samples.sample_terms.summary', samples_sample_terms_summary))
            
        if samples_sample_terms_term_name is not None:
            
            _query_params.append(('samples.sample_terms.term_name', samples_sample_terms_term_name))
            
        if samples_status is not None:
            
            _query_params.append(('samples.status', samples_status))
            
        if samples_summary is not None:
            
            _query_params.append(('samples.summary', samples_summary))
            
        if samples_targeted_sample_term_id is not None:
            
            _query_params.append(('samples.targeted_sample_term.@id', samples_targeted_sample_term_id))
            
        if samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('samples.targeted_sample_term.term_name', samples_targeted_sample_term_term_name))
            
        if samples_taxa is not None:
            
            _query_params.append(('samples.taxa', samples_taxa))
            
        if samples_treatments is not None:
            
            _query_params.append(('samples.treatments', samples_treatments))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if transcriptome_annotations is not None:
            
            _query_params.append(('transcriptome_annotations', transcriptome_annotations))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/curated-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def degron_modifications(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        activated: Annotated[Optional[List[StrictBool]], Field(description="Filter by activated")] = None,
        activating_agent_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by activating_agent_term_id")] = None,
        activating_agent_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by activating_agent_term_name")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_modified: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_modified")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        degron_system: Annotated[Optional[List[StrictStr]], Field(description="Filter by degron_system")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by modality")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        tagged_proteins: Annotated[Optional[List[StrictStr]], Field(description="Filter by tagged_proteins")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> DegronModificationResults:
        """List items in the DegronModification collection.

        Collection endpoint that accepts various query parameters to filter and sort DegronModification items. Supports filtering on fields within DegronModification items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param activated: Filter by activated
        :type activated: List[bool]
        :param activating_agent_term_id: Filter by activating_agent_term_id
        :type activating_agent_term_id: List[str]
        :param activating_agent_term_name: Filter by activating_agent_term_name
        :type activating_agent_term_name: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_modified: Filter by biosamples_modified
        :type biosamples_modified: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param degron_system: Filter by degron_system
        :type degron_system: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param modality: Filter by modality
        :type modality: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param tagged_proteins: Filter by tagged_proteins
        :type tagged_proteins: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._degron_modifications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            activated=activated,
            activating_agent_term_id=activating_agent_term_id,
            activating_agent_term_name=activating_agent_term_name,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            biosamples_modified=biosamples_modified,
            creation_timestamp=creation_timestamp,
            degron_system=degron_system,
            description=description,
            documents=documents,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            modality=modality,
            notes=notes,
            product_id=product_id,
            release_timestamp=release_timestamp,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            tagged_proteins=tagged_proteins,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "DegronModificationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def degron_modifications_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        activated: Annotated[Optional[List[StrictBool]], Field(description="Filter by activated")] = None,
        activating_agent_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by activating_agent_term_id")] = None,
        activating_agent_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by activating_agent_term_name")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_modified: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_modified")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        degron_system: Annotated[Optional[List[StrictStr]], Field(description="Filter by degron_system")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by modality")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        tagged_proteins: Annotated[Optional[List[StrictStr]], Field(description="Filter by tagged_proteins")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[DegronModificationResults]:
        """List items in the DegronModification collection.

        Collection endpoint that accepts various query parameters to filter and sort DegronModification items. Supports filtering on fields within DegronModification items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param activated: Filter by activated
        :type activated: List[bool]
        :param activating_agent_term_id: Filter by activating_agent_term_id
        :type activating_agent_term_id: List[str]
        :param activating_agent_term_name: Filter by activating_agent_term_name
        :type activating_agent_term_name: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_modified: Filter by biosamples_modified
        :type biosamples_modified: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param degron_system: Filter by degron_system
        :type degron_system: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param modality: Filter by modality
        :type modality: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param tagged_proteins: Filter by tagged_proteins
        :type tagged_proteins: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._degron_modifications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            activated=activated,
            activating_agent_term_id=activating_agent_term_id,
            activating_agent_term_name=activating_agent_term_name,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            biosamples_modified=biosamples_modified,
            creation_timestamp=creation_timestamp,
            degron_system=degron_system,
            description=description,
            documents=documents,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            modality=modality,
            notes=notes,
            product_id=product_id,
            release_timestamp=release_timestamp,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            tagged_proteins=tagged_proteins,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "DegronModificationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def degron_modifications_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        activated: Annotated[Optional[List[StrictBool]], Field(description="Filter by activated")] = None,
        activating_agent_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by activating_agent_term_id")] = None,
        activating_agent_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by activating_agent_term_name")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_modified: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_modified")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        degron_system: Annotated[Optional[List[StrictStr]], Field(description="Filter by degron_system")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        modality: Annotated[Optional[List[StrictStr]], Field(description="Filter by modality")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        tagged_proteins: Annotated[Optional[List[StrictStr]], Field(description="Filter by tagged_proteins")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the DegronModification collection.

        Collection endpoint that accepts various query parameters to filter and sort DegronModification items. Supports filtering on fields within DegronModification items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param activated: Filter by activated
        :type activated: List[bool]
        :param activating_agent_term_id: Filter by activating_agent_term_id
        :type activating_agent_term_id: List[str]
        :param activating_agent_term_name: Filter by activating_agent_term_name
        :type activating_agent_term_name: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_modified: Filter by biosamples_modified
        :type biosamples_modified: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param degron_system: Filter by degron_system
        :type degron_system: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param modality: Filter by modality
        :type modality: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param tagged_proteins: Filter by tagged_proteins
        :type tagged_proteins: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._degron_modifications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            activated=activated,
            activating_agent_term_id=activating_agent_term_id,
            activating_agent_term_name=activating_agent_term_name,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            biosamples_modified=biosamples_modified,
            creation_timestamp=creation_timestamp,
            degron_system=degron_system,
            description=description,
            documents=documents,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            modality=modality,
            notes=notes,
            product_id=product_id,
            release_timestamp=release_timestamp,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            tagged_proteins=tagged_proteins,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "DegronModificationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _degron_modifications_serialize(
        self,
        query,
        limit,
        sort,
        id,
        activated,
        activating_agent_term_id,
        activating_agent_term_name,
        aliases,
        award_id,
        award_component,
        biosamples_modified,
        creation_timestamp,
        degron_system,
        description,
        documents,
        lab_id,
        lab_title,
        lot_id,
        modality,
        notes,
        product_id,
        release_timestamp,
        sources,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        tagged_proteins,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'activated': 'multi',
            'activating_agent_term_id': 'multi',
            'activating_agent_term_name': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biosamples_modified': 'multi',
            'creation_timestamp': 'multi',
            'degron_system': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'modality': 'multi',
            'notes': 'multi',
            'product_id': 'multi',
            'release_timestamp': 'multi',
            'sources': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'tagged_proteins': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if activated is not None:
            
            _query_params.append(('activated', activated))
            
        if activating_agent_term_id is not None:
            
            _query_params.append(('activating_agent_term_id', activating_agent_term_id))
            
        if activating_agent_term_name is not None:
            
            _query_params.append(('activating_agent_term_name', activating_agent_term_name))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biosamples_modified is not None:
            
            _query_params.append(('biosamples_modified', biosamples_modified))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if degron_system is not None:
            
            _query_params.append(('degron_system', degron_system))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if modality is not None:
            
            _query_params.append(('modality', modality))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if sources is not None:
            
            _query_params.append(('sources', sources))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if tagged_proteins is not None:
            
            _query_params.append(('tagged_proteins', tagged_proteins))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/degron-modifications/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def documents(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        characterization_method: Annotated[Optional[List[StrictStr]], Field(description="Filter by characterization_method")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        document_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by document_type")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        urls: Annotated[Optional[List[StrictStr]], Field(description="Filter by urls")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> DocumentResults:
        """List items in the Document collection.

        Collection endpoint that accepts various query parameters to filter and sort Document items. Supports filtering on fields within Document items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param characterization_method: Filter by characterization_method
        :type characterization_method: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param document_type: Filter by document_type
        :type document_type: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param urls: Filter by urls
        :type urls: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._documents_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            characterization_method=characterization_method,
            creation_timestamp=creation_timestamp,
            description=description,
            document_type=document_type,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            urls=urls,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "DocumentResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def documents_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        characterization_method: Annotated[Optional[List[StrictStr]], Field(description="Filter by characterization_method")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        document_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by document_type")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        urls: Annotated[Optional[List[StrictStr]], Field(description="Filter by urls")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[DocumentResults]:
        """List items in the Document collection.

        Collection endpoint that accepts various query parameters to filter and sort Document items. Supports filtering on fields within Document items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param characterization_method: Filter by characterization_method
        :type characterization_method: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param document_type: Filter by document_type
        :type document_type: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param urls: Filter by urls
        :type urls: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._documents_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            characterization_method=characterization_method,
            creation_timestamp=creation_timestamp,
            description=description,
            document_type=document_type,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            urls=urls,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "DocumentResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def documents_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        characterization_method: Annotated[Optional[List[StrictStr]], Field(description="Filter by characterization_method")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        document_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by document_type")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        urls: Annotated[Optional[List[StrictStr]], Field(description="Filter by urls")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Document collection.

        Collection endpoint that accepts various query parameters to filter and sort Document items. Supports filtering on fields within Document items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param characterization_method: Filter by characterization_method
        :type characterization_method: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param document_type: Filter by document_type
        :type document_type: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param urls: Filter by urls
        :type urls: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._documents_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            characterization_method=characterization_method,
            creation_timestamp=creation_timestamp,
            description=description,
            document_type=document_type,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            urls=urls,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "DocumentResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _documents_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award_id,
        award_component,
        characterization_method,
        creation_timestamp,
        description,
        document_type,
        lab_id,
        lab_title,
        notes,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        urls,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'characterization_method': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'document_type': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'urls': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if characterization_method is not None:
            
            _query_params.append(('characterization_method', characterization_method))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if document_type is not None:
            
            _query_params.append(('document_type', document_type))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if urls is not None:
            
            _query_params.append(('urls', urls))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/documents/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def download(
        self,
        file_id: Annotated[StrictStr, Field(description="The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03).")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> bytearray:
        """Download file.

        Returns underlying file associated with file metadata

        :param file_id: The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03). (required)
        :type file_id: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._download_serialize(
            file_id=file_id,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "bytearray",
            '404': None,
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def download_with_http_info(
        self,
        file_id: Annotated[StrictStr, Field(description="The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03).")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[bytearray]:
        """Download file.

        Returns underlying file associated with file metadata

        :param file_id: The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03). (required)
        :type file_id: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._download_serialize(
            file_id=file_id,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "bytearray",
            '404': None,
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def download_without_preload_content(
        self,
        file_id: Annotated[StrictStr, Field(description="The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03).")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """Download file.

        Returns underlying file associated with file metadata

        :param file_id: The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03). (required)
        :type file_id: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._download_serialize(
            file_id=file_id,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "bytearray",
            '404': None,
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _download_serialize(
        self,
        file_id,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        if file_id is not None:
            _path_params['file_id'] = file_id
        # process the query parameters
        # process the header parameters
        # process the form parameters
        # process the body parameter


        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/octet-stream'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/{file_id}/@@download',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def genes(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by geneid")] = None,
        geneid_with_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by geneid_with_version")] = None,
        locations: Annotated[Optional[List[GeneLocation]], Field(description="Filter by locations")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by symbol")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        version_number: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by version_number")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> GeneResults:
        """List items in the Gene collection.

        Collection endpoint that accepts various query parameters to filter and sort Gene items. Supports filtering on fields within Gene items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param geneid: Filter by geneid
        :type geneid: List[str]
        :param geneid_with_version: Filter by geneid_with_version
        :type geneid_with_version: List[str]
        :param locations: Filter by locations
        :type locations: List[GeneLocation]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param symbol: Filter by symbol
        :type symbol: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param version_number: Filter by version_number
        :type version_number: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._genes_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            geneid=geneid,
            geneid_with_version=geneid_with_version,
            locations=locations,
            name=name,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            symbol=symbol,
            synonyms=synonyms,
            taxa=taxa,
            title=title,
            transcriptome_annotation=transcriptome_annotation,
            uuid=uuid,
            version_number=version_number,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "GeneResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def genes_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by geneid")] = None,
        geneid_with_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by geneid_with_version")] = None,
        locations: Annotated[Optional[List[GeneLocation]], Field(description="Filter by locations")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by symbol")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        version_number: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by version_number")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[GeneResults]:
        """List items in the Gene collection.

        Collection endpoint that accepts various query parameters to filter and sort Gene items. Supports filtering on fields within Gene items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param geneid: Filter by geneid
        :type geneid: List[str]
        :param geneid_with_version: Filter by geneid_with_version
        :type geneid_with_version: List[str]
        :param locations: Filter by locations
        :type locations: List[GeneLocation]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param symbol: Filter by symbol
        :type symbol: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param version_number: Filter by version_number
        :type version_number: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._genes_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            geneid=geneid,
            geneid_with_version=geneid_with_version,
            locations=locations,
            name=name,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            symbol=symbol,
            synonyms=synonyms,
            taxa=taxa,
            title=title,
            transcriptome_annotation=transcriptome_annotation,
            uuid=uuid,
            version_number=version_number,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "GeneResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def genes_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by geneid")] = None,
        geneid_with_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by geneid_with_version")] = None,
        locations: Annotated[Optional[List[GeneLocation]], Field(description="Filter by locations")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by symbol")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        version_number: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by version_number")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Gene collection.

        Collection endpoint that accepts various query parameters to filter and sort Gene items. Supports filtering on fields within Gene items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param geneid: Filter by geneid
        :type geneid: List[str]
        :param geneid_with_version: Filter by geneid_with_version
        :type geneid_with_version: List[str]
        :param locations: Filter by locations
        :type locations: List[GeneLocation]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param symbol: Filter by symbol
        :type symbol: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param version_number: Filter by version_number
        :type version_number: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._genes_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            geneid=geneid,
            geneid_with_version=geneid_with_version,
            locations=locations,
            name=name,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            symbol=symbol,
            synonyms=synonyms,
            taxa=taxa,
            title=title,
            transcriptome_annotation=transcriptome_annotation,
            uuid=uuid,
            version_number=version_number,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "GeneResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _genes_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        creation_timestamp,
        dbxrefs,
        description,
        geneid,
        geneid_with_version,
        locations,
        name,
        notes,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        symbol,
        synonyms,
        taxa,
        title,
        transcriptome_annotation,
        uuid,
        version_number,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'geneid': 'multi',
            'geneid_with_version': 'multi',
            'locations': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'symbol': 'multi',
            'synonyms': 'multi',
            'taxa': 'multi',
            'title': 'multi',
            'transcriptome_annotation': 'multi',
            'uuid': 'multi',
            'version_number': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if geneid is not None:
            
            _query_params.append(('geneid', geneid))
            
        if geneid_with_version is not None:
            
            _query_params.append(('geneid_with_version', geneid_with_version))
            
        if locations is not None:
            
            _query_params.append(('locations', locations))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if symbol is not None:
            
            _query_params.append(('symbol', symbol))
            
        if synonyms is not None:
            
            _query_params.append(('synonyms', synonyms))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if transcriptome_annotation is not None:
            
            _query_params.append(('transcriptome_annotation', transcriptome_annotation))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if version_number is not None:
            
            _query_params.append(('version_number', version_number))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/genes/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def genome_browser_annotation_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> GenomeBrowserAnnotationFileResults:
        """List items in the GenomeBrowserAnnotationFile collection.

        Collection endpoint that accepts various query parameters to filter and sort GenomeBrowserAnnotationFile items. Supports filtering on fields within GenomeBrowserAnnotationFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._genome_browser_annotation_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "GenomeBrowserAnnotationFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def genome_browser_annotation_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[GenomeBrowserAnnotationFileResults]:
        """List items in the GenomeBrowserAnnotationFile collection.

        Collection endpoint that accepts various query parameters to filter and sort GenomeBrowserAnnotationFile items. Supports filtering on fields within GenomeBrowserAnnotationFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._genome_browser_annotation_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "GenomeBrowserAnnotationFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def genome_browser_annotation_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the GenomeBrowserAnnotationFile collection.

        Collection endpoint that accepts various query parameters to filter and sort GenomeBrowserAnnotationFile items. Supports filtering on fields within GenomeBrowserAnnotationFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._genome_browser_annotation_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "GenomeBrowserAnnotationFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _genome_browser_annotation_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        assay_titles,
        assembly,
        award_id,
        award_component,
        cell_type_annotation_id,
        cell_type_annotation_term_name,
        collections,
        content_md5sum,
        content_type,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_format_type,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        release_timestamp,
        revoke_detail,
        s3_uri,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        transcriptome_annotation,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'assay_titles': 'multi',
            'assembly': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'cell_type_annotation.@id': 'multi',
            'cell_type_annotation.term_name': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_format_type': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'transcriptome_annotation': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if assembly is not None:
            
            _query_params.append(('assembly', assembly))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if cell_type_annotation_id is not None:
            
            _query_params.append(('cell_type_annotation.@id', cell_type_annotation_id))
            
        if cell_type_annotation_term_name is not None:
            
            _query_params.append(('cell_type_annotation.term_name', cell_type_annotation_term_name))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_format_type is not None:
            
            _query_params.append(('file_format_type', file_format_type))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if transcriptome_annotation is not None:
            
            _query_params.append(('transcriptome_annotation', transcriptome_annotation))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/genome-browser-annotation-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def get_by_id(
        self,
        resource_id: Annotated[StrictStr, Field(description="The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`).")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> Item:
        """Get item information

        Retrieve detailed information about a specific item using its @id or uuid.

        :param resource_id: The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`). (required)
        :type resource_id: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._get_by_id_serialize(
            resource_id=resource_id,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "Item",
            '404': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def get_by_id_with_http_info(
        self,
        resource_id: Annotated[StrictStr, Field(description="The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`).")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[Item]:
        """Get item information

        Retrieve detailed information about a specific item using its @id or uuid.

        :param resource_id: The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`). (required)
        :type resource_id: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._get_by_id_serialize(
            resource_id=resource_id,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "Item",
            '404': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def get_by_id_without_preload_content(
        self,
        resource_id: Annotated[StrictStr, Field(description="The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`).")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """Get item information

        Retrieve detailed information about a specific item using its @id or uuid.

        :param resource_id: The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`). (required)
        :type resource_id: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._get_by_id_serialize(
            resource_id=resource_id,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "Item",
            '404': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _get_by_id_serialize(
        self,
        resource_id,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        if resource_id is not None:
            _path_params['resource_id'] = resource_id
        # process the query parameters
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/{resource_id}',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def human_donors(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        ethnicities: Annotated[Optional[List[StrictStr]], Field(description="Filter by ethnicities")] = None,
        human_donor_identifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by human_donor_identifiers")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        phenotypic_features_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.@id")] = None,
        phenotypic_features_feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.feature.@id")] = None,
        phenotypic_features_feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_id")] = None,
        phenotypic_features_feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_name")] = None,
        phenotypic_features_observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.observation_date")] = None,
        phenotypic_features_quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.quantity_units")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        related_donors_donor_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_donors.donor.@id")] = None,
        related_donors_donor_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_donors.donor.accession")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> HumanDonorResults:
        """List items in the HumanDonor collection.

        Collection endpoint that accepts various query parameters to filter and sort HumanDonor items. Supports filtering on fields within HumanDonor items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param ethnicities: Filter by ethnicities
        :type ethnicities: List[str]
        :param human_donor_identifiers: Filter by human_donor_identifiers
        :type human_donor_identifiers: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param phenotypic_features_id: Filter by phenotypic_features.@id
        :type phenotypic_features_id: List[str]
        :param phenotypic_features_feature_id: Filter by phenotypic_features.feature.@id
        :type phenotypic_features_feature_id: List[str]
        :param phenotypic_features_feature_term_id: Filter by phenotypic_features.feature.term_id
        :type phenotypic_features_feature_term_id: List[str]
        :param phenotypic_features_feature_term_name: Filter by phenotypic_features.feature.term_name
        :type phenotypic_features_feature_term_name: List[str]
        :param phenotypic_features_observation_date: Filter by phenotypic_features.observation_date
        :type phenotypic_features_observation_date: List[str]
        :param phenotypic_features_quantity_units: Filter by phenotypic_features.quantity_units
        :type phenotypic_features_quantity_units: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param related_donors_donor_id: Filter by related_donors.donor.@id
        :type related_donors_donor_id: List[str]
        :param related_donors_donor_accession: Filter by related_donors.donor.accession
        :type related_donors_donor_accession: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._human_donors_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            ethnicities=ethnicities,
            human_donor_identifiers=human_donor_identifiers,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            phenotypic_features_id=phenotypic_features_id,
            phenotypic_features_feature_id=phenotypic_features_feature_id,
            phenotypic_features_feature_term_id=phenotypic_features_feature_term_id,
            phenotypic_features_feature_term_name=phenotypic_features_feature_term_name,
            phenotypic_features_observation_date=phenotypic_features_observation_date,
            phenotypic_features_quantity_units=phenotypic_features_quantity_units,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            related_donors_donor_id=related_donors_donor_id,
            related_donors_donor_accession=related_donors_donor_accession,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sex=sex,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "HumanDonorResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def human_donors_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        ethnicities: Annotated[Optional[List[StrictStr]], Field(description="Filter by ethnicities")] = None,
        human_donor_identifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by human_donor_identifiers")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        phenotypic_features_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.@id")] = None,
        phenotypic_features_feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.feature.@id")] = None,
        phenotypic_features_feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_id")] = None,
        phenotypic_features_feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_name")] = None,
        phenotypic_features_observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.observation_date")] = None,
        phenotypic_features_quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.quantity_units")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        related_donors_donor_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_donors.donor.@id")] = None,
        related_donors_donor_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_donors.donor.accession")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[HumanDonorResults]:
        """List items in the HumanDonor collection.

        Collection endpoint that accepts various query parameters to filter and sort HumanDonor items. Supports filtering on fields within HumanDonor items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param ethnicities: Filter by ethnicities
        :type ethnicities: List[str]
        :param human_donor_identifiers: Filter by human_donor_identifiers
        :type human_donor_identifiers: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param phenotypic_features_id: Filter by phenotypic_features.@id
        :type phenotypic_features_id: List[str]
        :param phenotypic_features_feature_id: Filter by phenotypic_features.feature.@id
        :type phenotypic_features_feature_id: List[str]
        :param phenotypic_features_feature_term_id: Filter by phenotypic_features.feature.term_id
        :type phenotypic_features_feature_term_id: List[str]
        :param phenotypic_features_feature_term_name: Filter by phenotypic_features.feature.term_name
        :type phenotypic_features_feature_term_name: List[str]
        :param phenotypic_features_observation_date: Filter by phenotypic_features.observation_date
        :type phenotypic_features_observation_date: List[str]
        :param phenotypic_features_quantity_units: Filter by phenotypic_features.quantity_units
        :type phenotypic_features_quantity_units: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param related_donors_donor_id: Filter by related_donors.donor.@id
        :type related_donors_donor_id: List[str]
        :param related_donors_donor_accession: Filter by related_donors.donor.accession
        :type related_donors_donor_accession: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._human_donors_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            ethnicities=ethnicities,
            human_donor_identifiers=human_donor_identifiers,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            phenotypic_features_id=phenotypic_features_id,
            phenotypic_features_feature_id=phenotypic_features_feature_id,
            phenotypic_features_feature_term_id=phenotypic_features_feature_term_id,
            phenotypic_features_feature_term_name=phenotypic_features_feature_term_name,
            phenotypic_features_observation_date=phenotypic_features_observation_date,
            phenotypic_features_quantity_units=phenotypic_features_quantity_units,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            related_donors_donor_id=related_donors_donor_id,
            related_donors_donor_accession=related_donors_donor_accession,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sex=sex,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "HumanDonorResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def human_donors_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        ethnicities: Annotated[Optional[List[StrictStr]], Field(description="Filter by ethnicities")] = None,
        human_donor_identifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by human_donor_identifiers")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        phenotypic_features_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.@id")] = None,
        phenotypic_features_feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.feature.@id")] = None,
        phenotypic_features_feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_id")] = None,
        phenotypic_features_feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_name")] = None,
        phenotypic_features_observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.observation_date")] = None,
        phenotypic_features_quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.quantity_units")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        related_donors_donor_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_donors.donor.@id")] = None,
        related_donors_donor_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_donors.donor.accession")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the HumanDonor collection.

        Collection endpoint that accepts various query parameters to filter and sort HumanDonor items. Supports filtering on fields within HumanDonor items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param ethnicities: Filter by ethnicities
        :type ethnicities: List[str]
        :param human_donor_identifiers: Filter by human_donor_identifiers
        :type human_donor_identifiers: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param phenotypic_features_id: Filter by phenotypic_features.@id
        :type phenotypic_features_id: List[str]
        :param phenotypic_features_feature_id: Filter by phenotypic_features.feature.@id
        :type phenotypic_features_feature_id: List[str]
        :param phenotypic_features_feature_term_id: Filter by phenotypic_features.feature.term_id
        :type phenotypic_features_feature_term_id: List[str]
        :param phenotypic_features_feature_term_name: Filter by phenotypic_features.feature.term_name
        :type phenotypic_features_feature_term_name: List[str]
        :param phenotypic_features_observation_date: Filter by phenotypic_features.observation_date
        :type phenotypic_features_observation_date: List[str]
        :param phenotypic_features_quantity_units: Filter by phenotypic_features.quantity_units
        :type phenotypic_features_quantity_units: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param related_donors_donor_id: Filter by related_donors.donor.@id
        :type related_donors_donor_id: List[str]
        :param related_donors_donor_accession: Filter by related_donors.donor.accession
        :type related_donors_donor_accession: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._human_donors_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            ethnicities=ethnicities,
            human_donor_identifiers=human_donor_identifiers,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            phenotypic_features_id=phenotypic_features_id,
            phenotypic_features_feature_id=phenotypic_features_feature_id,
            phenotypic_features_feature_term_id=phenotypic_features_feature_term_id,
            phenotypic_features_feature_term_name=phenotypic_features_feature_term_name,
            phenotypic_features_observation_date=phenotypic_features_observation_date,
            phenotypic_features_quantity_units=phenotypic_features_quantity_units,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            related_donors_donor_id=related_donors_donor_id,
            related_donors_donor_accession=related_donors_donor_accession,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sex=sex,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "HumanDonorResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _human_donors_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        collections,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        ethnicities,
        human_donor_identifiers,
        lab_id,
        lab_title,
        notes,
        phenotypic_features_id,
        phenotypic_features_feature_id,
        phenotypic_features_feature_term_id,
        phenotypic_features_feature_term_name,
        phenotypic_features_observation_date,
        phenotypic_features_quantity_units,
        publications_id,
        publications_publication_identifiers,
        related_donors_donor_id,
        related_donors_donor_accession,
        release_timestamp,
        revoke_detail,
        sex,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        taxa,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'ethnicities': 'multi',
            'human_donor_identifiers': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'phenotypic_features.@id': 'multi',
            'phenotypic_features.feature.@id': 'multi',
            'phenotypic_features.feature.term_id': 'multi',
            'phenotypic_features.feature.term_name': 'multi',
            'phenotypic_features.observation_date': 'multi',
            'phenotypic_features.quantity_units': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'related_donors.donor.@id': 'multi',
            'related_donors.donor.accession': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sex': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if ethnicities is not None:
            
            _query_params.append(('ethnicities', ethnicities))
            
        if human_donor_identifiers is not None:
            
            _query_params.append(('human_donor_identifiers', human_donor_identifiers))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if phenotypic_features_id is not None:
            
            _query_params.append(('phenotypic_features.@id', phenotypic_features_id))
            
        if phenotypic_features_feature_id is not None:
            
            _query_params.append(('phenotypic_features.feature.@id', phenotypic_features_feature_id))
            
        if phenotypic_features_feature_term_id is not None:
            
            _query_params.append(('phenotypic_features.feature.term_id', phenotypic_features_feature_term_id))
            
        if phenotypic_features_feature_term_name is not None:
            
            _query_params.append(('phenotypic_features.feature.term_name', phenotypic_features_feature_term_name))
            
        if phenotypic_features_observation_date is not None:
            
            _query_params.append(('phenotypic_features.observation_date', phenotypic_features_observation_date))
            
        if phenotypic_features_quantity_units is not None:
            
            _query_params.append(('phenotypic_features.quantity_units', phenotypic_features_quantity_units))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if related_donors_donor_id is not None:
            
            _query_params.append(('related_donors.donor.@id', related_donors_donor_id))
            
        if related_donors_donor_accession is not None:
            
            _query_params.append(('related_donors.donor.accession', related_donors_donor_accession))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sex is not None:
            
            _query_params.append(('sex', sex))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/human-donors/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def image_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ImageFileResults:
        """List items in the ImageFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ImageFile items. Supports filtering on fields within ImageFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._image_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ImageFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def image_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ImageFileResults]:
        """List items in the ImageFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ImageFile items. Supports filtering on fields within ImageFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._image_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ImageFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def image_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the ImageFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ImageFile items. Supports filtering on fields within ImageFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._image_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ImageFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _image_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        assay_titles,
        award_id,
        award_component,
        collections,
        content_md5sum,
        content_type,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        release_timestamp,
        revoke_detail,
        s3_uri,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'assay_titles': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/image-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def images(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        caption: Annotated[Optional[List[StrictStr]], Field(description="Filter by caption")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        download_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by download_url")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        thumb_nail: Annotated[Optional[List[StrictStr]], Field(description="Filter by thumb_nail")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ImageResults:
        """List items in the Image collection.

        Collection endpoint that accepts various query parameters to filter and sort Image items. Supports filtering on fields within Image items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param caption: Filter by caption
        :type caption: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param download_url: Filter by download_url
        :type download_url: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param thumb_nail: Filter by thumb_nail
        :type thumb_nail: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._images_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            caption=caption,
            creation_timestamp=creation_timestamp,
            description=description,
            download_url=download_url,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            thumb_nail=thumb_nail,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ImageResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def images_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        caption: Annotated[Optional[List[StrictStr]], Field(description="Filter by caption")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        download_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by download_url")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        thumb_nail: Annotated[Optional[List[StrictStr]], Field(description="Filter by thumb_nail")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ImageResults]:
        """List items in the Image collection.

        Collection endpoint that accepts various query parameters to filter and sort Image items. Supports filtering on fields within Image items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param caption: Filter by caption
        :type caption: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param download_url: Filter by download_url
        :type download_url: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param thumb_nail: Filter by thumb_nail
        :type thumb_nail: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._images_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            caption=caption,
            creation_timestamp=creation_timestamp,
            description=description,
            download_url=download_url,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            thumb_nail=thumb_nail,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ImageResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def images_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        caption: Annotated[Optional[List[StrictStr]], Field(description="Filter by caption")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        download_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by download_url")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        thumb_nail: Annotated[Optional[List[StrictStr]], Field(description="Filter by thumb_nail")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Image collection.

        Collection endpoint that accepts various query parameters to filter and sort Image items. Supports filtering on fields within Image items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param caption: Filter by caption
        :type caption: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param download_url: Filter by download_url
        :type download_url: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param thumb_nail: Filter by thumb_nail
        :type thumb_nail: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._images_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            caption=caption,
            creation_timestamp=creation_timestamp,
            description=description,
            download_url=download_url,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            thumb_nail=thumb_nail,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ImageResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _images_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        caption,
        creation_timestamp,
        description,
        download_url,
        notes,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        thumb_nail,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'caption': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'download_url': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'thumb_nail': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if caption is not None:
            
            _query_params.append(('caption', caption))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if download_url is not None:
            
            _query_params.append(('download_url', download_url))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if thumb_nail is not None:
            
            _query_params.append(('thumb_nail', thumb_nail))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/images/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def in_vitro_systems(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        biosample_qualifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosample_qualifiers")] = None,
        cell_fate_change_protocol: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_protocol")] = None,
        cell_fate_change_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.@id")] = None,
        cell_fate_change_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.purpose")] = None,
        cell_fate_change_treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.status")] = None,
        cell_fate_change_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.summary")] = None,
        cell_fate_change_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.treatment_type")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        demultiplexed_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by demultiplexed_from")] = None,
        demultiplexed_to: Annotated[Optional[List[StrictStr]], Field(description="Filter by demultiplexed_to")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        growth_medium: Annotated[Optional[List[StrictStr]], Field(description="Filter by growth_medium")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from.@id")] = None,
        originated_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from.accession")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        passage_number: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by passage_number")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeted_sample_term: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_sample_term")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_change: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_change")] = None,
        time_post_change_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_change_units")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> InVitroSystemResults:
        """List items in the InVitroSystem collection.

        Collection endpoint that accepts various query parameters to filter and sort InVitroSystem items. Supports filtering on fields within InVitroSystem items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param biosample_qualifiers: Filter by biosample_qualifiers
        :type biosample_qualifiers: List[str]
        :param cell_fate_change_protocol: Filter by cell_fate_change_protocol
        :type cell_fate_change_protocol: List[str]
        :param cell_fate_change_treatments_id: Filter by cell_fate_change_treatments.@id
        :type cell_fate_change_treatments_id: List[str]
        :param cell_fate_change_treatments_purpose: Filter by cell_fate_change_treatments.purpose
        :type cell_fate_change_treatments_purpose: List[str]
        :param cell_fate_change_treatments_status: Filter by cell_fate_change_treatments.status
        :type cell_fate_change_treatments_status: List[str]
        :param cell_fate_change_treatments_summary: Filter by cell_fate_change_treatments.summary
        :type cell_fate_change_treatments_summary: List[str]
        :param cell_fate_change_treatments_treatment_type: Filter by cell_fate_change_treatments.treatment_type
        :type cell_fate_change_treatments_treatment_type: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param demultiplexed_from: Filter by demultiplexed_from
        :type demultiplexed_from: List[str]
        :param demultiplexed_to: Filter by demultiplexed_to
        :type demultiplexed_to: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param growth_medium: Filter by growth_medium
        :type growth_medium: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from_id: Filter by originated_from.@id
        :type originated_from_id: List[str]
        :param originated_from_accession: Filter by originated_from.accession
        :type originated_from_accession: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param passage_number: Filter by passage_number
        :type passage_number: List[int]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeted_sample_term: Filter by targeted_sample_term
        :type targeted_sample_term: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_change: Filter by time_post_change
        :type time_post_change: List[float]
        :param time_post_change_units: Filter by time_post_change_units
        :type time_post_change_units: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._in_vitro_systems_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            biosample_qualifiers=biosample_qualifiers,
            cell_fate_change_protocol=cell_fate_change_protocol,
            cell_fate_change_treatments_id=cell_fate_change_treatments_id,
            cell_fate_change_treatments_purpose=cell_fate_change_treatments_purpose,
            cell_fate_change_treatments_status=cell_fate_change_treatments_status,
            cell_fate_change_treatments_summary=cell_fate_change_treatments_summary,
            cell_fate_change_treatments_treatment_type=cell_fate_change_treatments_treatment_type,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            demultiplexed_from=demultiplexed_from,
            demultiplexed_to=demultiplexed_to,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            growth_medium=growth_medium,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from_id=originated_from_id,
            originated_from_accession=originated_from_accession,
            part_of=part_of,
            parts=parts,
            passage_number=passage_number,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            targeted_sample_term=targeted_sample_term,
            taxa=taxa,
            time_post_change=time_post_change,
            time_post_change_units=time_post_change_units,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "InVitroSystemResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def in_vitro_systems_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        biosample_qualifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosample_qualifiers")] = None,
        cell_fate_change_protocol: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_protocol")] = None,
        cell_fate_change_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.@id")] = None,
        cell_fate_change_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.purpose")] = None,
        cell_fate_change_treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.status")] = None,
        cell_fate_change_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.summary")] = None,
        cell_fate_change_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.treatment_type")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        demultiplexed_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by demultiplexed_from")] = None,
        demultiplexed_to: Annotated[Optional[List[StrictStr]], Field(description="Filter by demultiplexed_to")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        growth_medium: Annotated[Optional[List[StrictStr]], Field(description="Filter by growth_medium")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from.@id")] = None,
        originated_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from.accession")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        passage_number: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by passage_number")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeted_sample_term: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_sample_term")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_change: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_change")] = None,
        time_post_change_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_change_units")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[InVitroSystemResults]:
        """List items in the InVitroSystem collection.

        Collection endpoint that accepts various query parameters to filter and sort InVitroSystem items. Supports filtering on fields within InVitroSystem items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param biosample_qualifiers: Filter by biosample_qualifiers
        :type biosample_qualifiers: List[str]
        :param cell_fate_change_protocol: Filter by cell_fate_change_protocol
        :type cell_fate_change_protocol: List[str]
        :param cell_fate_change_treatments_id: Filter by cell_fate_change_treatments.@id
        :type cell_fate_change_treatments_id: List[str]
        :param cell_fate_change_treatments_purpose: Filter by cell_fate_change_treatments.purpose
        :type cell_fate_change_treatments_purpose: List[str]
        :param cell_fate_change_treatments_status: Filter by cell_fate_change_treatments.status
        :type cell_fate_change_treatments_status: List[str]
        :param cell_fate_change_treatments_summary: Filter by cell_fate_change_treatments.summary
        :type cell_fate_change_treatments_summary: List[str]
        :param cell_fate_change_treatments_treatment_type: Filter by cell_fate_change_treatments.treatment_type
        :type cell_fate_change_treatments_treatment_type: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param demultiplexed_from: Filter by demultiplexed_from
        :type demultiplexed_from: List[str]
        :param demultiplexed_to: Filter by demultiplexed_to
        :type demultiplexed_to: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param growth_medium: Filter by growth_medium
        :type growth_medium: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from_id: Filter by originated_from.@id
        :type originated_from_id: List[str]
        :param originated_from_accession: Filter by originated_from.accession
        :type originated_from_accession: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param passage_number: Filter by passage_number
        :type passage_number: List[int]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeted_sample_term: Filter by targeted_sample_term
        :type targeted_sample_term: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_change: Filter by time_post_change
        :type time_post_change: List[float]
        :param time_post_change_units: Filter by time_post_change_units
        :type time_post_change_units: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._in_vitro_systems_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            biosample_qualifiers=biosample_qualifiers,
            cell_fate_change_protocol=cell_fate_change_protocol,
            cell_fate_change_treatments_id=cell_fate_change_treatments_id,
            cell_fate_change_treatments_purpose=cell_fate_change_treatments_purpose,
            cell_fate_change_treatments_status=cell_fate_change_treatments_status,
            cell_fate_change_treatments_summary=cell_fate_change_treatments_summary,
            cell_fate_change_treatments_treatment_type=cell_fate_change_treatments_treatment_type,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            demultiplexed_from=demultiplexed_from,
            demultiplexed_to=demultiplexed_to,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            growth_medium=growth_medium,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from_id=originated_from_id,
            originated_from_accession=originated_from_accession,
            part_of=part_of,
            parts=parts,
            passage_number=passage_number,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            targeted_sample_term=targeted_sample_term,
            taxa=taxa,
            time_post_change=time_post_change,
            time_post_change_units=time_post_change_units,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "InVitroSystemResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def in_vitro_systems_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        biosample_qualifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosample_qualifiers")] = None,
        cell_fate_change_protocol: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_protocol")] = None,
        cell_fate_change_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.@id")] = None,
        cell_fate_change_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.purpose")] = None,
        cell_fate_change_treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.status")] = None,
        cell_fate_change_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.summary")] = None,
        cell_fate_change_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_fate_change_treatments.treatment_type")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        demultiplexed_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by demultiplexed_from")] = None,
        demultiplexed_to: Annotated[Optional[List[StrictStr]], Field(description="Filter by demultiplexed_to")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        growth_medium: Annotated[Optional[List[StrictStr]], Field(description="Filter by growth_medium")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from.@id")] = None,
        originated_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from.accession")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        passage_number: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by passage_number")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeted_sample_term: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_sample_term")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_change: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_change")] = None,
        time_post_change_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_change_units")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the InVitroSystem collection.

        Collection endpoint that accepts various query parameters to filter and sort InVitroSystem items. Supports filtering on fields within InVitroSystem items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param biosample_qualifiers: Filter by biosample_qualifiers
        :type biosample_qualifiers: List[str]
        :param cell_fate_change_protocol: Filter by cell_fate_change_protocol
        :type cell_fate_change_protocol: List[str]
        :param cell_fate_change_treatments_id: Filter by cell_fate_change_treatments.@id
        :type cell_fate_change_treatments_id: List[str]
        :param cell_fate_change_treatments_purpose: Filter by cell_fate_change_treatments.purpose
        :type cell_fate_change_treatments_purpose: List[str]
        :param cell_fate_change_treatments_status: Filter by cell_fate_change_treatments.status
        :type cell_fate_change_treatments_status: List[str]
        :param cell_fate_change_treatments_summary: Filter by cell_fate_change_treatments.summary
        :type cell_fate_change_treatments_summary: List[str]
        :param cell_fate_change_treatments_treatment_type: Filter by cell_fate_change_treatments.treatment_type
        :type cell_fate_change_treatments_treatment_type: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param demultiplexed_from: Filter by demultiplexed_from
        :type demultiplexed_from: List[str]
        :param demultiplexed_to: Filter by demultiplexed_to
        :type demultiplexed_to: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param growth_medium: Filter by growth_medium
        :type growth_medium: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from_id: Filter by originated_from.@id
        :type originated_from_id: List[str]
        :param originated_from_accession: Filter by originated_from.accession
        :type originated_from_accession: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param passage_number: Filter by passage_number
        :type passage_number: List[int]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeted_sample_term: Filter by targeted_sample_term
        :type targeted_sample_term: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_change: Filter by time_post_change
        :type time_post_change: List[float]
        :param time_post_change_units: Filter by time_post_change_units
        :type time_post_change_units: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._in_vitro_systems_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            biosample_qualifiers=biosample_qualifiers,
            cell_fate_change_protocol=cell_fate_change_protocol,
            cell_fate_change_treatments_id=cell_fate_change_treatments_id,
            cell_fate_change_treatments_purpose=cell_fate_change_treatments_purpose,
            cell_fate_change_treatments_status=cell_fate_change_treatments_status,
            cell_fate_change_treatments_summary=cell_fate_change_treatments_summary,
            cell_fate_change_treatments_treatment_type=cell_fate_change_treatments_treatment_type,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            demultiplexed_from=demultiplexed_from,
            demultiplexed_to=demultiplexed_to,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            growth_medium=growth_medium,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from_id=originated_from_id,
            originated_from_accession=originated_from_accession,
            part_of=part_of,
            parts=parts,
            passage_number=passage_number,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            targeted_sample_term=targeted_sample_term,
            taxa=taxa,
            time_post_change=time_post_change,
            time_post_change_units=time_post_change_units,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "InVitroSystemResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _in_vitro_systems_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        age,
        age_units,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        biomarkers,
        biosample_qualifiers,
        cell_fate_change_protocol,
        cell_fate_change_treatments_id,
        cell_fate_change_treatments_purpose,
        cell_fate_change_treatments_status,
        cell_fate_change_treatments_summary,
        cell_fate_change_treatments_treatment_type,
        cellular_sub_pool,
        classifications,
        collections,
        construct_library_sets,
        creation_timestamp,
        date_obtained,
        dbxrefs,
        demultiplexed_from,
        demultiplexed_to,
        description,
        disease_terms_id,
        disease_terms_term_name,
        documents,
        donors,
        embryonic,
        file_sets_id,
        file_sets_accession,
        file_sets_aliases,
        file_sets_assay_term_term_name,
        file_sets_lab_title,
        file_sets_status,
        file_sets_summary,
        growth_medium,
        institutional_certificates_id,
        institutional_certificates_certificate_identifier,
        lab_id,
        lab_title,
        lot_id,
        lower_bound_age,
        lower_bound_age_in_hours,
        modifications_id,
        modifications_status,
        modifications_summary,
        moi,
        multiplexed_in_id,
        multiplexed_in_accession,
        notes,
        nucleic_acid_delivery,
        origin_of,
        originated_from_id,
        originated_from_accession,
        part_of,
        parts,
        passage_number,
        pooled_from,
        pooled_in,
        product_id,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_terms_id,
        sample_terms_term_name,
        sex,
        sorted_fractions,
        sorted_from_id,
        sorted_from_accession,
        sorted_from_detail,
        sources_id,
        starting_amount,
        starting_amount_units,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        targeted_sample_term,
        taxa,
        time_post_change,
        time_post_change_units,
        time_post_library_delivery,
        time_post_library_delivery_units,
        treatments_id,
        treatments_purpose,
        treatments_status,
        treatments_summary,
        treatments_treatment_type,
        upper_bound_age,
        upper_bound_age_in_hours,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'age': 'multi',
            'age_units': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biomarkers': 'multi',
            'biosample_qualifiers': 'multi',
            'cell_fate_change_protocol': 'multi',
            'cell_fate_change_treatments.@id': 'multi',
            'cell_fate_change_treatments.purpose': 'multi',
            'cell_fate_change_treatments.status': 'multi',
            'cell_fate_change_treatments.summary': 'multi',
            'cell_fate_change_treatments.treatment_type': 'multi',
            'cellular_sub_pool': 'multi',
            'classifications': 'multi',
            'collections': 'multi',
            'construct_library_sets': 'multi',
            'creation_timestamp': 'multi',
            'date_obtained': 'multi',
            'dbxrefs': 'multi',
            'demultiplexed_from': 'multi',
            'demultiplexed_to': 'multi',
            'description': 'multi',
            'disease_terms.@id': 'multi',
            'disease_terms.term_name': 'multi',
            'documents': 'multi',
            'donors': 'multi',
            'embryonic': 'multi',
            'file_sets.@id': 'multi',
            'file_sets.accession': 'multi',
            'file_sets.aliases': 'multi',
            'file_sets.assay_term.term_name': 'multi',
            'file_sets.lab.title': 'multi',
            'file_sets.status': 'multi',
            'file_sets.summary': 'multi',
            'growth_medium': 'multi',
            'institutional_certificates.@id': 'multi',
            'institutional_certificates.certificate_identifier': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'lower_bound_age': 'multi',
            'lower_bound_age_in_hours': 'multi',
            'modifications.@id': 'multi',
            'modifications.status': 'multi',
            'modifications.summary': 'multi',
            'moi': 'multi',
            'multiplexed_in.@id': 'multi',
            'multiplexed_in.accession': 'multi',
            'notes': 'multi',
            'nucleic_acid_delivery': 'multi',
            'origin_of': 'multi',
            'originated_from.@id': 'multi',
            'originated_from.accession': 'multi',
            'part_of': 'multi',
            'parts': 'multi',
            'passage_number': 'multi',
            'pooled_from': 'multi',
            'pooled_in': 'multi',
            'product_id': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_terms.@id': 'multi',
            'sample_terms.term_name': 'multi',
            'sex': 'multi',
            'sorted_fractions': 'multi',
            'sorted_from.@id': 'multi',
            'sorted_from.accession': 'multi',
            'sorted_from_detail': 'multi',
            'sources.@id': 'multi',
            'starting_amount': 'multi',
            'starting_amount_units': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'targeted_sample_term': 'multi',
            'taxa': 'multi',
            'time_post_change': 'multi',
            'time_post_change_units': 'multi',
            'time_post_library_delivery': 'multi',
            'time_post_library_delivery_units': 'multi',
            'treatments.@id': 'multi',
            'treatments.purpose': 'multi',
            'treatments.status': 'multi',
            'treatments.summary': 'multi',
            'treatments.treatment_type': 'multi',
            'upper_bound_age': 'multi',
            'upper_bound_age_in_hours': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if age is not None:
            
            _query_params.append(('age', age))
            
        if age_units is not None:
            
            _query_params.append(('age_units', age_units))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biomarkers is not None:
            
            _query_params.append(('biomarkers', biomarkers))
            
        if biosample_qualifiers is not None:
            
            _query_params.append(('biosample_qualifiers', biosample_qualifiers))
            
        if cell_fate_change_protocol is not None:
            
            _query_params.append(('cell_fate_change_protocol', cell_fate_change_protocol))
            
        if cell_fate_change_treatments_id is not None:
            
            _query_params.append(('cell_fate_change_treatments.@id', cell_fate_change_treatments_id))
            
        if cell_fate_change_treatments_purpose is not None:
            
            _query_params.append(('cell_fate_change_treatments.purpose', cell_fate_change_treatments_purpose))
            
        if cell_fate_change_treatments_status is not None:
            
            _query_params.append(('cell_fate_change_treatments.status', cell_fate_change_treatments_status))
            
        if cell_fate_change_treatments_summary is not None:
            
            _query_params.append(('cell_fate_change_treatments.summary', cell_fate_change_treatments_summary))
            
        if cell_fate_change_treatments_treatment_type is not None:
            
            _query_params.append(('cell_fate_change_treatments.treatment_type', cell_fate_change_treatments_treatment_type))
            
        if cellular_sub_pool is not None:
            
            _query_params.append(('cellular_sub_pool', cellular_sub_pool))
            
        if classifications is not None:
            
            _query_params.append(('classifications', classifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if construct_library_sets is not None:
            
            _query_params.append(('construct_library_sets', construct_library_sets))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_obtained is not None:
            
            _query_params.append(('date_obtained', date_obtained))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if demultiplexed_from is not None:
            
            _query_params.append(('demultiplexed_from', demultiplexed_from))
            
        if demultiplexed_to is not None:
            
            _query_params.append(('demultiplexed_to', demultiplexed_to))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if disease_terms_id is not None:
            
            _query_params.append(('disease_terms.@id', disease_terms_id))
            
        if disease_terms_term_name is not None:
            
            _query_params.append(('disease_terms.term_name', disease_terms_term_name))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors is not None:
            
            _query_params.append(('donors', donors))
            
        if embryonic is not None:
            
            _query_params.append(('embryonic', embryonic))
            
        if file_sets_id is not None:
            
            _query_params.append(('file_sets.@id', file_sets_id))
            
        if file_sets_accession is not None:
            
            _query_params.append(('file_sets.accession', file_sets_accession))
            
        if file_sets_aliases is not None:
            
            _query_params.append(('file_sets.aliases', file_sets_aliases))
            
        if file_sets_assay_term_term_name is not None:
            
            _query_params.append(('file_sets.assay_term.term_name', file_sets_assay_term_term_name))
            
        if file_sets_lab_title is not None:
            
            _query_params.append(('file_sets.lab.title', file_sets_lab_title))
            
        if file_sets_status is not None:
            
            _query_params.append(('file_sets.status', file_sets_status))
            
        if file_sets_summary is not None:
            
            _query_params.append(('file_sets.summary', file_sets_summary))
            
        if growth_medium is not None:
            
            _query_params.append(('growth_medium', growth_medium))
            
        if institutional_certificates_id is not None:
            
            _query_params.append(('institutional_certificates.@id', institutional_certificates_id))
            
        if institutional_certificates_certificate_identifier is not None:
            
            _query_params.append(('institutional_certificates.certificate_identifier', institutional_certificates_certificate_identifier))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if lower_bound_age is not None:
            
            _query_params.append(('lower_bound_age', lower_bound_age))
            
        if lower_bound_age_in_hours is not None:
            
            _query_params.append(('lower_bound_age_in_hours', lower_bound_age_in_hours))
            
        if modifications_id is not None:
            
            _query_params.append(('modifications.@id', modifications_id))
            
        if modifications_status is not None:
            
            _query_params.append(('modifications.status', modifications_status))
            
        if modifications_summary is not None:
            
            _query_params.append(('modifications.summary', modifications_summary))
            
        if moi is not None:
            
            _query_params.append(('moi', moi))
            
        if multiplexed_in_id is not None:
            
            _query_params.append(('multiplexed_in.@id', multiplexed_in_id))
            
        if multiplexed_in_accession is not None:
            
            _query_params.append(('multiplexed_in.accession', multiplexed_in_accession))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if nucleic_acid_delivery is not None:
            
            _query_params.append(('nucleic_acid_delivery', nucleic_acid_delivery))
            
        if origin_of is not None:
            
            _query_params.append(('origin_of', origin_of))
            
        if originated_from_id is not None:
            
            _query_params.append(('originated_from.@id', originated_from_id))
            
        if originated_from_accession is not None:
            
            _query_params.append(('originated_from.accession', originated_from_accession))
            
        if part_of is not None:
            
            _query_params.append(('part_of', part_of))
            
        if parts is not None:
            
            _query_params.append(('parts', parts))
            
        if passage_number is not None:
            
            _query_params.append(('passage_number', passage_number))
            
        if pooled_from is not None:
            
            _query_params.append(('pooled_from', pooled_from))
            
        if pooled_in is not None:
            
            _query_params.append(('pooled_in', pooled_in))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_terms_id is not None:
            
            _query_params.append(('sample_terms.@id', sample_terms_id))
            
        if sample_terms_term_name is not None:
            
            _query_params.append(('sample_terms.term_name', sample_terms_term_name))
            
        if sex is not None:
            
            _query_params.append(('sex', sex))
            
        if sorted_fractions is not None:
            
            _query_params.append(('sorted_fractions', sorted_fractions))
            
        if sorted_from_id is not None:
            
            _query_params.append(('sorted_from.@id', sorted_from_id))
            
        if sorted_from_accession is not None:
            
            _query_params.append(('sorted_from.accession', sorted_from_accession))
            
        if sorted_from_detail is not None:
            
            _query_params.append(('sorted_from_detail', sorted_from_detail))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if starting_amount is not None:
            
            _query_params.append(('starting_amount', starting_amount))
            
        if starting_amount_units is not None:
            
            _query_params.append(('starting_amount_units', starting_amount_units))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if targeted_sample_term is not None:
            
            _query_params.append(('targeted_sample_term', targeted_sample_term))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if time_post_change is not None:
            
            _query_params.append(('time_post_change', time_post_change))
            
        if time_post_change_units is not None:
            
            _query_params.append(('time_post_change_units', time_post_change_units))
            
        if time_post_library_delivery is not None:
            
            _query_params.append(('time_post_library_delivery', time_post_library_delivery))
            
        if time_post_library_delivery_units is not None:
            
            _query_params.append(('time_post_library_delivery_units', time_post_library_delivery_units))
            
        if treatments_id is not None:
            
            _query_params.append(('treatments.@id', treatments_id))
            
        if treatments_purpose is not None:
            
            _query_params.append(('treatments.purpose', treatments_purpose))
            
        if treatments_status is not None:
            
            _query_params.append(('treatments.status', treatments_status))
            
        if treatments_summary is not None:
            
            _query_params.append(('treatments.summary', treatments_summary))
            
        if treatments_treatment_type is not None:
            
            _query_params.append(('treatments.treatment_type', treatments_treatment_type))
            
        if upper_bound_age is not None:
            
            _query_params.append(('upper_bound_age', upper_bound_age))
            
        if upper_bound_age_in_hours is not None:
            
            _query_params.append(('upper_bound_age_in_hours', upper_bound_age_in_hours))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/in-vitro-systems/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def institutional_certificates(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by certificate_identifier")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        data_use_limitation: Annotated[Optional[List[StrictStr]], Field(description="Filter by data_use_limitation")] = None,
        data_use_limitation_modifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by data_use_limitation_modifiers")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        samples: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        urls: Annotated[Optional[List[StrictStr]], Field(description="Filter by urls")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> InstitutionalCertificateResults:
        """List items in the InstitutionalCertificate collection.

        Collection endpoint that accepts various query parameters to filter and sort InstitutionalCertificate items. Supports filtering on fields within InstitutionalCertificate items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param certificate_identifier: Filter by certificate_identifier
        :type certificate_identifier: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param data_use_limitation: Filter by data_use_limitation
        :type data_use_limitation: List[str]
        :param data_use_limitation_modifiers: Filter by data_use_limitation_modifiers
        :type data_use_limitation_modifiers: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param samples: Filter by samples
        :type samples: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param urls: Filter by urls
        :type urls: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._institutional_certificates_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            certificate_identifier=certificate_identifier,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            data_use_limitation=data_use_limitation,
            data_use_limitation_modifiers=data_use_limitation_modifiers,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            samples=samples,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            urls=urls,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "InstitutionalCertificateResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def institutional_certificates_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by certificate_identifier")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        data_use_limitation: Annotated[Optional[List[StrictStr]], Field(description="Filter by data_use_limitation")] = None,
        data_use_limitation_modifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by data_use_limitation_modifiers")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        samples: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        urls: Annotated[Optional[List[StrictStr]], Field(description="Filter by urls")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[InstitutionalCertificateResults]:
        """List items in the InstitutionalCertificate collection.

        Collection endpoint that accepts various query parameters to filter and sort InstitutionalCertificate items. Supports filtering on fields within InstitutionalCertificate items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param certificate_identifier: Filter by certificate_identifier
        :type certificate_identifier: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param data_use_limitation: Filter by data_use_limitation
        :type data_use_limitation: List[str]
        :param data_use_limitation_modifiers: Filter by data_use_limitation_modifiers
        :type data_use_limitation_modifiers: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param samples: Filter by samples
        :type samples: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param urls: Filter by urls
        :type urls: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._institutional_certificates_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            certificate_identifier=certificate_identifier,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            data_use_limitation=data_use_limitation,
            data_use_limitation_modifiers=data_use_limitation_modifiers,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            samples=samples,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            urls=urls,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "InstitutionalCertificateResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def institutional_certificates_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by certificate_identifier")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        data_use_limitation: Annotated[Optional[List[StrictStr]], Field(description="Filter by data_use_limitation")] = None,
        data_use_limitation_modifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by data_use_limitation_modifiers")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        samples: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        urls: Annotated[Optional[List[StrictStr]], Field(description="Filter by urls")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the InstitutionalCertificate collection.

        Collection endpoint that accepts various query parameters to filter and sort InstitutionalCertificate items. Supports filtering on fields within InstitutionalCertificate items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param certificate_identifier: Filter by certificate_identifier
        :type certificate_identifier: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param data_use_limitation: Filter by data_use_limitation
        :type data_use_limitation: List[str]
        :param data_use_limitation_modifiers: Filter by data_use_limitation_modifiers
        :type data_use_limitation_modifiers: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param samples: Filter by samples
        :type samples: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param urls: Filter by urls
        :type urls: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._institutional_certificates_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            certificate_identifier=certificate_identifier,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            data_use_limitation=data_use_limitation,
            data_use_limitation_modifiers=data_use_limitation_modifiers,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            release_timestamp=release_timestamp,
            samples=samples,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            urls=urls,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "InstitutionalCertificateResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _institutional_certificates_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award_id,
        award_component,
        certificate_identifier,
        controlled_access,
        creation_timestamp,
        data_use_limitation,
        data_use_limitation_modifiers,
        description,
        lab_id,
        lab_title,
        notes,
        release_timestamp,
        samples,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        urls,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'certificate_identifier': 'multi',
            'controlled_access': 'multi',
            'creation_timestamp': 'multi',
            'data_use_limitation': 'multi',
            'data_use_limitation_modifiers': 'multi',
            'description': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'samples': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'urls': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if certificate_identifier is not None:
            
            _query_params.append(('certificate_identifier', certificate_identifier))
            
        if controlled_access is not None:
            
            _query_params.append(('controlled_access', controlled_access))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if data_use_limitation is not None:
            
            _query_params.append(('data_use_limitation', data_use_limitation))
            
        if data_use_limitation_modifiers is not None:
            
            _query_params.append(('data_use_limitation_modifiers', data_use_limitation_modifiers))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if samples is not None:
            
            _query_params.append(('samples', samples))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if urls is not None:
            
            _query_params.append(('urls', urls))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/institutional-certificates/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def labs(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        awards_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by awards.@id")] = None,
        awards_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by awards.component")] = None,
        awards_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by awards.name")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        institute_label: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institute_label")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        pi: Annotated[Optional[List[StrictStr]], Field(description="Filter by pi")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> LabResults:
        """List items in the Lab collection.

        Collection endpoint that accepts various query parameters to filter and sort Lab items. Supports filtering on fields within Lab items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param awards_id: Filter by awards.@id
        :type awards_id: List[str]
        :param awards_component: Filter by awards.component
        :type awards_component: List[str]
        :param awards_name: Filter by awards.name
        :type awards_name: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param institute_label: Filter by institute_label
        :type institute_label: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param pi: Filter by pi
        :type pi: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._labs_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            awards_id=awards_id,
            awards_component=awards_component,
            awards_name=awards_name,
            creation_timestamp=creation_timestamp,
            description=description,
            institute_label=institute_label,
            name=name,
            notes=notes,
            pi=pi,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "LabResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def labs_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        awards_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by awards.@id")] = None,
        awards_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by awards.component")] = None,
        awards_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by awards.name")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        institute_label: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institute_label")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        pi: Annotated[Optional[List[StrictStr]], Field(description="Filter by pi")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[LabResults]:
        """List items in the Lab collection.

        Collection endpoint that accepts various query parameters to filter and sort Lab items. Supports filtering on fields within Lab items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param awards_id: Filter by awards.@id
        :type awards_id: List[str]
        :param awards_component: Filter by awards.component
        :type awards_component: List[str]
        :param awards_name: Filter by awards.name
        :type awards_name: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param institute_label: Filter by institute_label
        :type institute_label: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param pi: Filter by pi
        :type pi: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._labs_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            awards_id=awards_id,
            awards_component=awards_component,
            awards_name=awards_name,
            creation_timestamp=creation_timestamp,
            description=description,
            institute_label=institute_label,
            name=name,
            notes=notes,
            pi=pi,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "LabResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def labs_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        awards_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by awards.@id")] = None,
        awards_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by awards.component")] = None,
        awards_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by awards.name")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        institute_label: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institute_label")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        pi: Annotated[Optional[List[StrictStr]], Field(description="Filter by pi")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Lab collection.

        Collection endpoint that accepts various query parameters to filter and sort Lab items. Supports filtering on fields within Lab items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param awards_id: Filter by awards.@id
        :type awards_id: List[str]
        :param awards_component: Filter by awards.component
        :type awards_component: List[str]
        :param awards_name: Filter by awards.name
        :type awards_name: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param institute_label: Filter by institute_label
        :type institute_label: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param pi: Filter by pi
        :type pi: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._labs_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            awards_id=awards_id,
            awards_component=awards_component,
            awards_name=awards_name,
            creation_timestamp=creation_timestamp,
            description=description,
            institute_label=institute_label,
            name=name,
            notes=notes,
            pi=pi,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "LabResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _labs_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        awards_id,
        awards_component,
        awards_name,
        creation_timestamp,
        description,
        institute_label,
        name,
        notes,
        pi,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        url,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'awards.@id': 'multi',
            'awards.component': 'multi',
            'awards.name': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'institute_label': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'pi': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'url': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if awards_id is not None:
            
            _query_params.append(('awards.@id', awards_id))
            
        if awards_component is not None:
            
            _query_params.append(('awards.component', awards_component))
            
        if awards_name is not None:
            
            _query_params.append(('awards.name', awards_name))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if institute_label is not None:
            
            _query_params.append(('institute_label', institute_label))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if pi is not None:
            
            _query_params.append(('pi', pi))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/labs/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def matrix_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        principal_dimension: Annotated[Optional[List[StrictStr]], Field(description="Filter by principal_dimension")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        secondary_dimensions: Annotated[Optional[List[StrictStr]], Field(description="Filter by secondary_dimensions")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> MatrixFileResults:
        """List items in the MatrixFile collection.

        Collection endpoint that accepts various query parameters to filter and sort MatrixFile items. Supports filtering on fields within MatrixFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param principal_dimension: Filter by principal_dimension
        :type principal_dimension: List[str]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param secondary_dimensions: Filter by secondary_dimensions
        :type secondary_dimensions: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._matrix_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            principal_dimension=principal_dimension,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            secondary_dimensions=secondary_dimensions,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MatrixFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def matrix_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        principal_dimension: Annotated[Optional[List[StrictStr]], Field(description="Filter by principal_dimension")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        secondary_dimensions: Annotated[Optional[List[StrictStr]], Field(description="Filter by secondary_dimensions")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[MatrixFileResults]:
        """List items in the MatrixFile collection.

        Collection endpoint that accepts various query parameters to filter and sort MatrixFile items. Supports filtering on fields within MatrixFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param principal_dimension: Filter by principal_dimension
        :type principal_dimension: List[str]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param secondary_dimensions: Filter by secondary_dimensions
        :type secondary_dimensions: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._matrix_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            principal_dimension=principal_dimension,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            secondary_dimensions=secondary_dimensions,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MatrixFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def matrix_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        principal_dimension: Annotated[Optional[List[StrictStr]], Field(description="Filter by principal_dimension")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        secondary_dimensions: Annotated[Optional[List[StrictStr]], Field(description="Filter by secondary_dimensions")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the MatrixFile collection.

        Collection endpoint that accepts various query parameters to filter and sort MatrixFile items. Supports filtering on fields within MatrixFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param principal_dimension: Filter by principal_dimension
        :type principal_dimension: List[str]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param secondary_dimensions: Filter by secondary_dimensions
        :type secondary_dimensions: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._matrix_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            principal_dimension=principal_dimension,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            secondary_dimensions=secondary_dimensions,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MatrixFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _matrix_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        assay_titles,
        award_id,
        award_component,
        collections,
        content_md5sum,
        content_summary,
        content_type,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        principal_dimension,
        reference_files,
        release_timestamp,
        revoke_detail,
        s3_uri,
        secondary_dimensions,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'assay_titles': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_summary': 'multi',
            'content_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'principal_dimension': 'multi',
            'reference_files': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'secondary_dimensions': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_summary is not None:
            
            _query_params.append(('content_summary', content_summary))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if principal_dimension is not None:
            
            _query_params.append(('principal_dimension', principal_dimension))
            
        if reference_files is not None:
            
            _query_params.append(('reference_files', reference_files))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if secondary_dimensions is not None:
            
            _query_params.append(('secondary_dimensions', secondary_dimensions))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/matrix-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def measurement_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assay_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_term.@id")] = None,
        assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by assay_term.term_name")] = None,
        auxiliary_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.@id")] = None,
        auxiliary_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.accession")] = None,
        auxiliary_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by auxiliary_sets.aliases")] = None,
        auxiliary_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.file_set_type")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_file_sets.@id")] = None,
        control_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_file_sets.accession")] = None,
        control_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_file_sets.aliases")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_image_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by external_image_url")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform.@id")] = None,
        files_sequencing_platform_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.sequencing_platform.term_name")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        functional_assay_mechanisms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by functional_assay_mechanisms.@id")] = None,
        functional_assay_mechanisms_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_id")] = None,
        functional_assay_mechanisms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_name")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        multiome_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=2)]]], Field(description="Filter by multiome_size")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        preferred_assay_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by preferred_assay_title")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        related_multiome_datasets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_multiome_datasets.@id")] = None,
        related_multiome_datasets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_multiome_datasets.accession")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.@id")] = None,
        samples_cell_fate_change_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.purpose")] = None,
        samples_cell_fate_change_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.summary")] = None,
        samples_cell_fate_change_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.treatment_type")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.@id")] = None,
        samples_construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.accession")] = None,
        samples_construct_library_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.file_set_type")] = None,
        samples_construct_library_sets_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.@id")] = None,
        samples_construct_library_sets_small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.geneid")] = None,
        samples_construct_library_sets_small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.name")] = None,
        samples_construct_library_sets_small_scale_gene_list_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.summary")] = None,
        samples_construct_library_sets_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.symbol")] = None,
        samples_construct_library_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.summary")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications.@id")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.@id")] = None,
        samples_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.purpose")] = None,
        samples_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.summary")] = None,
        samples_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.treatment_type")] = None,
        sequencing_library_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_library_types")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeted_genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.@id")] = None,
        targeted_genes_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by targeted_genes.geneid")] = None,
        targeted_genes_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.name")] = None,
        targeted_genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.symbol")] = None,
        targeted_genes_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.synonyms")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> MeasurementSetResults:
        """List items in the MeasurementSet collection.

        Collection endpoint that accepts various query parameters to filter and sort MeasurementSet items. Supports filtering on fields within MeasurementSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assay_term_id: Filter by assay_term.@id
        :type assay_term_id: List[str]
        :param assay_term_term_name: Filter by assay_term.term_name
        :type assay_term_term_name: List[str]
        :param auxiliary_sets_id: Filter by auxiliary_sets.@id
        :type auxiliary_sets_id: List[str]
        :param auxiliary_sets_accession: Filter by auxiliary_sets.accession
        :type auxiliary_sets_accession: List[str]
        :param auxiliary_sets_aliases: Filter by auxiliary_sets.aliases
        :type auxiliary_sets_aliases: List[str]
        :param auxiliary_sets_file_set_type: Filter by auxiliary_sets.file_set_type
        :type auxiliary_sets_file_set_type: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_file_sets_id: Filter by control_file_sets.@id
        :type control_file_sets_id: List[str]
        :param control_file_sets_accession: Filter by control_file_sets.accession
        :type control_file_sets_accession: List[str]
        :param control_file_sets_aliases: Filter by control_file_sets.aliases
        :type control_file_sets_aliases: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_image_url: Filter by external_image_url
        :type external_image_url: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform_id: Filter by files.sequencing_platform.@id
        :type files_sequencing_platform_id: List[str]
        :param files_sequencing_platform_term_name: Filter by files.sequencing_platform.term_name
        :type files_sequencing_platform_term_name: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param functional_assay_mechanisms_id: Filter by functional_assay_mechanisms.@id
        :type functional_assay_mechanisms_id: List[str]
        :param functional_assay_mechanisms_term_id: Filter by functional_assay_mechanisms.term_id
        :type functional_assay_mechanisms_term_id: List[str]
        :param functional_assay_mechanisms_term_name: Filter by functional_assay_mechanisms.term_name
        :type functional_assay_mechanisms_term_name: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param multiome_size: Filter by multiome_size
        :type multiome_size: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param preferred_assay_title: Filter by preferred_assay_title
        :type preferred_assay_title: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param related_multiome_datasets_id: Filter by related_multiome_datasets.@id
        :type related_multiome_datasets_id: List[str]
        :param related_multiome_datasets_accession: Filter by related_multiome_datasets.accession
        :type related_multiome_datasets_accession: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments_id: Filter by samples.cell_fate_change_treatments.@id
        :type samples_cell_fate_change_treatments_id: List[str]
        :param samples_cell_fate_change_treatments_purpose: Filter by samples.cell_fate_change_treatments.purpose
        :type samples_cell_fate_change_treatments_purpose: List[str]
        :param samples_cell_fate_change_treatments_summary: Filter by samples.cell_fate_change_treatments.summary
        :type samples_cell_fate_change_treatments_summary: List[str]
        :param samples_cell_fate_change_treatments_treatment_type: Filter by samples.cell_fate_change_treatments.treatment_type
        :type samples_cell_fate_change_treatments_treatment_type: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets_id: Filter by samples.construct_library_sets.@id
        :type samples_construct_library_sets_id: List[str]
        :param samples_construct_library_sets_accession: Filter by samples.construct_library_sets.accession
        :type samples_construct_library_sets_accession: List[str]
        :param samples_construct_library_sets_file_set_type: Filter by samples.construct_library_sets.file_set_type
        :type samples_construct_library_sets_file_set_type: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_id: Filter by samples.construct_library_sets.small_scale_gene_list.@id
        :type samples_construct_library_sets_small_scale_gene_list_id: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_geneid: Filter by samples.construct_library_sets.small_scale_gene_list.geneid
        :type samples_construct_library_sets_small_scale_gene_list_geneid: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_name: Filter by samples.construct_library_sets.small_scale_gene_list.name
        :type samples_construct_library_sets_small_scale_gene_list_name: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_summary: Filter by samples.construct_library_sets.small_scale_gene_list.summary
        :type samples_construct_library_sets_small_scale_gene_list_summary: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_symbol: Filter by samples.construct_library_sets.small_scale_gene_list.symbol
        :type samples_construct_library_sets_small_scale_gene_list_symbol: List[str]
        :param samples_construct_library_sets_summary: Filter by samples.construct_library_sets.summary
        :type samples_construct_library_sets_summary: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications_id: Filter by samples.modifications.@id
        :type samples_modifications_id: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments_id: Filter by samples.treatments.@id
        :type samples_treatments_id: List[str]
        :param samples_treatments_purpose: Filter by samples.treatments.purpose
        :type samples_treatments_purpose: List[str]
        :param samples_treatments_summary: Filter by samples.treatments.summary
        :type samples_treatments_summary: List[str]
        :param samples_treatments_treatment_type: Filter by samples.treatments.treatment_type
        :type samples_treatments_treatment_type: List[str]
        :param sequencing_library_types: Filter by sequencing_library_types
        :type sequencing_library_types: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeted_genes_id: Filter by targeted_genes.@id
        :type targeted_genes_id: List[str]
        :param targeted_genes_geneid: Filter by targeted_genes.geneid
        :type targeted_genes_geneid: List[str]
        :param targeted_genes_name: Filter by targeted_genes.name
        :type targeted_genes_name: List[str]
        :param targeted_genes_symbol: Filter by targeted_genes.symbol
        :type targeted_genes_symbol: List[str]
        :param targeted_genes_synonyms: Filter by targeted_genes.synonyms
        :type targeted_genes_synonyms: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._measurement_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assay_term_id=assay_term_id,
            assay_term_term_name=assay_term_term_name,
            auxiliary_sets_id=auxiliary_sets_id,
            auxiliary_sets_accession=auxiliary_sets_accession,
            auxiliary_sets_aliases=auxiliary_sets_aliases,
            auxiliary_sets_file_set_type=auxiliary_sets_file_set_type,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_file_sets_id=control_file_sets_id,
            control_file_sets_accession=control_file_sets_accession,
            control_file_sets_aliases=control_file_sets_aliases,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_image_url=external_image_url,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform_id=files_sequencing_platform_id,
            files_sequencing_platform_term_name=files_sequencing_platform_term_name,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            functional_assay_mechanisms_id=functional_assay_mechanisms_id,
            functional_assay_mechanisms_term_id=functional_assay_mechanisms_term_id,
            functional_assay_mechanisms_term_name=functional_assay_mechanisms_term_name,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            multiome_size=multiome_size,
            notes=notes,
            preferred_assay_title=preferred_assay_title,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            related_multiome_datasets_id=related_multiome_datasets_id,
            related_multiome_datasets_accession=related_multiome_datasets_accession,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments_id=samples_cell_fate_change_treatments_id,
            samples_cell_fate_change_treatments_purpose=samples_cell_fate_change_treatments_purpose,
            samples_cell_fate_change_treatments_summary=samples_cell_fate_change_treatments_summary,
            samples_cell_fate_change_treatments_treatment_type=samples_cell_fate_change_treatments_treatment_type,
            samples_classifications=samples_classifications,
            samples_construct_library_sets_id=samples_construct_library_sets_id,
            samples_construct_library_sets_accession=samples_construct_library_sets_accession,
            samples_construct_library_sets_file_set_type=samples_construct_library_sets_file_set_type,
            samples_construct_library_sets_small_scale_gene_list_id=samples_construct_library_sets_small_scale_gene_list_id,
            samples_construct_library_sets_small_scale_gene_list_geneid=samples_construct_library_sets_small_scale_gene_list_geneid,
            samples_construct_library_sets_small_scale_gene_list_name=samples_construct_library_sets_small_scale_gene_list_name,
            samples_construct_library_sets_small_scale_gene_list_summary=samples_construct_library_sets_small_scale_gene_list_summary,
            samples_construct_library_sets_small_scale_gene_list_symbol=samples_construct_library_sets_small_scale_gene_list_symbol,
            samples_construct_library_sets_summary=samples_construct_library_sets_summary,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications_id=samples_modifications_id,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments_id=samples_treatments_id,
            samples_treatments_purpose=samples_treatments_purpose,
            samples_treatments_summary=samples_treatments_summary,
            samples_treatments_treatment_type=samples_treatments_treatment_type,
            sequencing_library_types=sequencing_library_types,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            targeted_genes_id=targeted_genes_id,
            targeted_genes_geneid=targeted_genes_geneid,
            targeted_genes_name=targeted_genes_name,
            targeted_genes_symbol=targeted_genes_symbol,
            targeted_genes_synonyms=targeted_genes_synonyms,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MeasurementSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def measurement_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assay_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_term.@id")] = None,
        assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by assay_term.term_name")] = None,
        auxiliary_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.@id")] = None,
        auxiliary_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.accession")] = None,
        auxiliary_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by auxiliary_sets.aliases")] = None,
        auxiliary_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.file_set_type")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_file_sets.@id")] = None,
        control_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_file_sets.accession")] = None,
        control_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_file_sets.aliases")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_image_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by external_image_url")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform.@id")] = None,
        files_sequencing_platform_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.sequencing_platform.term_name")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        functional_assay_mechanisms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by functional_assay_mechanisms.@id")] = None,
        functional_assay_mechanisms_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_id")] = None,
        functional_assay_mechanisms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_name")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        multiome_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=2)]]], Field(description="Filter by multiome_size")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        preferred_assay_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by preferred_assay_title")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        related_multiome_datasets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_multiome_datasets.@id")] = None,
        related_multiome_datasets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_multiome_datasets.accession")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.@id")] = None,
        samples_cell_fate_change_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.purpose")] = None,
        samples_cell_fate_change_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.summary")] = None,
        samples_cell_fate_change_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.treatment_type")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.@id")] = None,
        samples_construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.accession")] = None,
        samples_construct_library_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.file_set_type")] = None,
        samples_construct_library_sets_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.@id")] = None,
        samples_construct_library_sets_small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.geneid")] = None,
        samples_construct_library_sets_small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.name")] = None,
        samples_construct_library_sets_small_scale_gene_list_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.summary")] = None,
        samples_construct_library_sets_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.symbol")] = None,
        samples_construct_library_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.summary")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications.@id")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.@id")] = None,
        samples_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.purpose")] = None,
        samples_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.summary")] = None,
        samples_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.treatment_type")] = None,
        sequencing_library_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_library_types")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeted_genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.@id")] = None,
        targeted_genes_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by targeted_genes.geneid")] = None,
        targeted_genes_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.name")] = None,
        targeted_genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.symbol")] = None,
        targeted_genes_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.synonyms")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[MeasurementSetResults]:
        """List items in the MeasurementSet collection.

        Collection endpoint that accepts various query parameters to filter and sort MeasurementSet items. Supports filtering on fields within MeasurementSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assay_term_id: Filter by assay_term.@id
        :type assay_term_id: List[str]
        :param assay_term_term_name: Filter by assay_term.term_name
        :type assay_term_term_name: List[str]
        :param auxiliary_sets_id: Filter by auxiliary_sets.@id
        :type auxiliary_sets_id: List[str]
        :param auxiliary_sets_accession: Filter by auxiliary_sets.accession
        :type auxiliary_sets_accession: List[str]
        :param auxiliary_sets_aliases: Filter by auxiliary_sets.aliases
        :type auxiliary_sets_aliases: List[str]
        :param auxiliary_sets_file_set_type: Filter by auxiliary_sets.file_set_type
        :type auxiliary_sets_file_set_type: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_file_sets_id: Filter by control_file_sets.@id
        :type control_file_sets_id: List[str]
        :param control_file_sets_accession: Filter by control_file_sets.accession
        :type control_file_sets_accession: List[str]
        :param control_file_sets_aliases: Filter by control_file_sets.aliases
        :type control_file_sets_aliases: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_image_url: Filter by external_image_url
        :type external_image_url: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform_id: Filter by files.sequencing_platform.@id
        :type files_sequencing_platform_id: List[str]
        :param files_sequencing_platform_term_name: Filter by files.sequencing_platform.term_name
        :type files_sequencing_platform_term_name: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param functional_assay_mechanisms_id: Filter by functional_assay_mechanisms.@id
        :type functional_assay_mechanisms_id: List[str]
        :param functional_assay_mechanisms_term_id: Filter by functional_assay_mechanisms.term_id
        :type functional_assay_mechanisms_term_id: List[str]
        :param functional_assay_mechanisms_term_name: Filter by functional_assay_mechanisms.term_name
        :type functional_assay_mechanisms_term_name: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param multiome_size: Filter by multiome_size
        :type multiome_size: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param preferred_assay_title: Filter by preferred_assay_title
        :type preferred_assay_title: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param related_multiome_datasets_id: Filter by related_multiome_datasets.@id
        :type related_multiome_datasets_id: List[str]
        :param related_multiome_datasets_accession: Filter by related_multiome_datasets.accession
        :type related_multiome_datasets_accession: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments_id: Filter by samples.cell_fate_change_treatments.@id
        :type samples_cell_fate_change_treatments_id: List[str]
        :param samples_cell_fate_change_treatments_purpose: Filter by samples.cell_fate_change_treatments.purpose
        :type samples_cell_fate_change_treatments_purpose: List[str]
        :param samples_cell_fate_change_treatments_summary: Filter by samples.cell_fate_change_treatments.summary
        :type samples_cell_fate_change_treatments_summary: List[str]
        :param samples_cell_fate_change_treatments_treatment_type: Filter by samples.cell_fate_change_treatments.treatment_type
        :type samples_cell_fate_change_treatments_treatment_type: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets_id: Filter by samples.construct_library_sets.@id
        :type samples_construct_library_sets_id: List[str]
        :param samples_construct_library_sets_accession: Filter by samples.construct_library_sets.accession
        :type samples_construct_library_sets_accession: List[str]
        :param samples_construct_library_sets_file_set_type: Filter by samples.construct_library_sets.file_set_type
        :type samples_construct_library_sets_file_set_type: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_id: Filter by samples.construct_library_sets.small_scale_gene_list.@id
        :type samples_construct_library_sets_small_scale_gene_list_id: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_geneid: Filter by samples.construct_library_sets.small_scale_gene_list.geneid
        :type samples_construct_library_sets_small_scale_gene_list_geneid: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_name: Filter by samples.construct_library_sets.small_scale_gene_list.name
        :type samples_construct_library_sets_small_scale_gene_list_name: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_summary: Filter by samples.construct_library_sets.small_scale_gene_list.summary
        :type samples_construct_library_sets_small_scale_gene_list_summary: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_symbol: Filter by samples.construct_library_sets.small_scale_gene_list.symbol
        :type samples_construct_library_sets_small_scale_gene_list_symbol: List[str]
        :param samples_construct_library_sets_summary: Filter by samples.construct_library_sets.summary
        :type samples_construct_library_sets_summary: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications_id: Filter by samples.modifications.@id
        :type samples_modifications_id: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments_id: Filter by samples.treatments.@id
        :type samples_treatments_id: List[str]
        :param samples_treatments_purpose: Filter by samples.treatments.purpose
        :type samples_treatments_purpose: List[str]
        :param samples_treatments_summary: Filter by samples.treatments.summary
        :type samples_treatments_summary: List[str]
        :param samples_treatments_treatment_type: Filter by samples.treatments.treatment_type
        :type samples_treatments_treatment_type: List[str]
        :param sequencing_library_types: Filter by sequencing_library_types
        :type sequencing_library_types: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeted_genes_id: Filter by targeted_genes.@id
        :type targeted_genes_id: List[str]
        :param targeted_genes_geneid: Filter by targeted_genes.geneid
        :type targeted_genes_geneid: List[str]
        :param targeted_genes_name: Filter by targeted_genes.name
        :type targeted_genes_name: List[str]
        :param targeted_genes_symbol: Filter by targeted_genes.symbol
        :type targeted_genes_symbol: List[str]
        :param targeted_genes_synonyms: Filter by targeted_genes.synonyms
        :type targeted_genes_synonyms: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._measurement_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assay_term_id=assay_term_id,
            assay_term_term_name=assay_term_term_name,
            auxiliary_sets_id=auxiliary_sets_id,
            auxiliary_sets_accession=auxiliary_sets_accession,
            auxiliary_sets_aliases=auxiliary_sets_aliases,
            auxiliary_sets_file_set_type=auxiliary_sets_file_set_type,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_file_sets_id=control_file_sets_id,
            control_file_sets_accession=control_file_sets_accession,
            control_file_sets_aliases=control_file_sets_aliases,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_image_url=external_image_url,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform_id=files_sequencing_platform_id,
            files_sequencing_platform_term_name=files_sequencing_platform_term_name,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            functional_assay_mechanisms_id=functional_assay_mechanisms_id,
            functional_assay_mechanisms_term_id=functional_assay_mechanisms_term_id,
            functional_assay_mechanisms_term_name=functional_assay_mechanisms_term_name,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            multiome_size=multiome_size,
            notes=notes,
            preferred_assay_title=preferred_assay_title,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            related_multiome_datasets_id=related_multiome_datasets_id,
            related_multiome_datasets_accession=related_multiome_datasets_accession,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments_id=samples_cell_fate_change_treatments_id,
            samples_cell_fate_change_treatments_purpose=samples_cell_fate_change_treatments_purpose,
            samples_cell_fate_change_treatments_summary=samples_cell_fate_change_treatments_summary,
            samples_cell_fate_change_treatments_treatment_type=samples_cell_fate_change_treatments_treatment_type,
            samples_classifications=samples_classifications,
            samples_construct_library_sets_id=samples_construct_library_sets_id,
            samples_construct_library_sets_accession=samples_construct_library_sets_accession,
            samples_construct_library_sets_file_set_type=samples_construct_library_sets_file_set_type,
            samples_construct_library_sets_small_scale_gene_list_id=samples_construct_library_sets_small_scale_gene_list_id,
            samples_construct_library_sets_small_scale_gene_list_geneid=samples_construct_library_sets_small_scale_gene_list_geneid,
            samples_construct_library_sets_small_scale_gene_list_name=samples_construct_library_sets_small_scale_gene_list_name,
            samples_construct_library_sets_small_scale_gene_list_summary=samples_construct_library_sets_small_scale_gene_list_summary,
            samples_construct_library_sets_small_scale_gene_list_symbol=samples_construct_library_sets_small_scale_gene_list_symbol,
            samples_construct_library_sets_summary=samples_construct_library_sets_summary,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications_id=samples_modifications_id,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments_id=samples_treatments_id,
            samples_treatments_purpose=samples_treatments_purpose,
            samples_treatments_summary=samples_treatments_summary,
            samples_treatments_treatment_type=samples_treatments_treatment_type,
            sequencing_library_types=sequencing_library_types,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            targeted_genes_id=targeted_genes_id,
            targeted_genes_geneid=targeted_genes_geneid,
            targeted_genes_name=targeted_genes_name,
            targeted_genes_symbol=targeted_genes_symbol,
            targeted_genes_synonyms=targeted_genes_synonyms,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MeasurementSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def measurement_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assay_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_term.@id")] = None,
        assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by assay_term.term_name")] = None,
        auxiliary_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.@id")] = None,
        auxiliary_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.accession")] = None,
        auxiliary_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by auxiliary_sets.aliases")] = None,
        auxiliary_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by auxiliary_sets.file_set_type")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_file_sets.@id")] = None,
        control_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_file_sets.accession")] = None,
        control_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_file_sets.aliases")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_image_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by external_image_url")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform.@id")] = None,
        files_sequencing_platform_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.sequencing_platform.term_name")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        functional_assay_mechanisms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by functional_assay_mechanisms.@id")] = None,
        functional_assay_mechanisms_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_id")] = None,
        functional_assay_mechanisms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by functional_assay_mechanisms.term_name")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        multiome_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=2)]]], Field(description="Filter by multiome_size")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        preferred_assay_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by preferred_assay_title")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        related_multiome_datasets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_multiome_datasets.@id")] = None,
        related_multiome_datasets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by related_multiome_datasets.accession")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.@id")] = None,
        samples_cell_fate_change_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.purpose")] = None,
        samples_cell_fate_change_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.summary")] = None,
        samples_cell_fate_change_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments.treatment_type")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.@id")] = None,
        samples_construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.accession")] = None,
        samples_construct_library_sets_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.file_set_type")] = None,
        samples_construct_library_sets_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.@id")] = None,
        samples_construct_library_sets_small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.geneid")] = None,
        samples_construct_library_sets_small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.name")] = None,
        samples_construct_library_sets_small_scale_gene_list_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.summary")] = None,
        samples_construct_library_sets_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.small_scale_gene_list.symbol")] = None,
        samples_construct_library_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.summary")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications.@id")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.@id")] = None,
        samples_treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.purpose")] = None,
        samples_treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.summary")] = None,
        samples_treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments.treatment_type")] = None,
        sequencing_library_types: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_library_types")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        targeted_genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.@id")] = None,
        targeted_genes_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by targeted_genes.geneid")] = None,
        targeted_genes_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.name")] = None,
        targeted_genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.symbol")] = None,
        targeted_genes_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by targeted_genes.synonyms")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the MeasurementSet collection.

        Collection endpoint that accepts various query parameters to filter and sort MeasurementSet items. Supports filtering on fields within MeasurementSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assay_term_id: Filter by assay_term.@id
        :type assay_term_id: List[str]
        :param assay_term_term_name: Filter by assay_term.term_name
        :type assay_term_term_name: List[str]
        :param auxiliary_sets_id: Filter by auxiliary_sets.@id
        :type auxiliary_sets_id: List[str]
        :param auxiliary_sets_accession: Filter by auxiliary_sets.accession
        :type auxiliary_sets_accession: List[str]
        :param auxiliary_sets_aliases: Filter by auxiliary_sets.aliases
        :type auxiliary_sets_aliases: List[str]
        :param auxiliary_sets_file_set_type: Filter by auxiliary_sets.file_set_type
        :type auxiliary_sets_file_set_type: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_file_sets_id: Filter by control_file_sets.@id
        :type control_file_sets_id: List[str]
        :param control_file_sets_accession: Filter by control_file_sets.accession
        :type control_file_sets_accession: List[str]
        :param control_file_sets_aliases: Filter by control_file_sets.aliases
        :type control_file_sets_aliases: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_image_url: Filter by external_image_url
        :type external_image_url: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform_id: Filter by files.sequencing_platform.@id
        :type files_sequencing_platform_id: List[str]
        :param files_sequencing_platform_term_name: Filter by files.sequencing_platform.term_name
        :type files_sequencing_platform_term_name: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param functional_assay_mechanisms_id: Filter by functional_assay_mechanisms.@id
        :type functional_assay_mechanisms_id: List[str]
        :param functional_assay_mechanisms_term_id: Filter by functional_assay_mechanisms.term_id
        :type functional_assay_mechanisms_term_id: List[str]
        :param functional_assay_mechanisms_term_name: Filter by functional_assay_mechanisms.term_name
        :type functional_assay_mechanisms_term_name: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param multiome_size: Filter by multiome_size
        :type multiome_size: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param preferred_assay_title: Filter by preferred_assay_title
        :type preferred_assay_title: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param related_multiome_datasets_id: Filter by related_multiome_datasets.@id
        :type related_multiome_datasets_id: List[str]
        :param related_multiome_datasets_accession: Filter by related_multiome_datasets.accession
        :type related_multiome_datasets_accession: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments_id: Filter by samples.cell_fate_change_treatments.@id
        :type samples_cell_fate_change_treatments_id: List[str]
        :param samples_cell_fate_change_treatments_purpose: Filter by samples.cell_fate_change_treatments.purpose
        :type samples_cell_fate_change_treatments_purpose: List[str]
        :param samples_cell_fate_change_treatments_summary: Filter by samples.cell_fate_change_treatments.summary
        :type samples_cell_fate_change_treatments_summary: List[str]
        :param samples_cell_fate_change_treatments_treatment_type: Filter by samples.cell_fate_change_treatments.treatment_type
        :type samples_cell_fate_change_treatments_treatment_type: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets_id: Filter by samples.construct_library_sets.@id
        :type samples_construct_library_sets_id: List[str]
        :param samples_construct_library_sets_accession: Filter by samples.construct_library_sets.accession
        :type samples_construct_library_sets_accession: List[str]
        :param samples_construct_library_sets_file_set_type: Filter by samples.construct_library_sets.file_set_type
        :type samples_construct_library_sets_file_set_type: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_id: Filter by samples.construct_library_sets.small_scale_gene_list.@id
        :type samples_construct_library_sets_small_scale_gene_list_id: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_geneid: Filter by samples.construct_library_sets.small_scale_gene_list.geneid
        :type samples_construct_library_sets_small_scale_gene_list_geneid: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_name: Filter by samples.construct_library_sets.small_scale_gene_list.name
        :type samples_construct_library_sets_small_scale_gene_list_name: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_summary: Filter by samples.construct_library_sets.small_scale_gene_list.summary
        :type samples_construct_library_sets_small_scale_gene_list_summary: List[str]
        :param samples_construct_library_sets_small_scale_gene_list_symbol: Filter by samples.construct_library_sets.small_scale_gene_list.symbol
        :type samples_construct_library_sets_small_scale_gene_list_symbol: List[str]
        :param samples_construct_library_sets_summary: Filter by samples.construct_library_sets.summary
        :type samples_construct_library_sets_summary: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications_id: Filter by samples.modifications.@id
        :type samples_modifications_id: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments_id: Filter by samples.treatments.@id
        :type samples_treatments_id: List[str]
        :param samples_treatments_purpose: Filter by samples.treatments.purpose
        :type samples_treatments_purpose: List[str]
        :param samples_treatments_summary: Filter by samples.treatments.summary
        :type samples_treatments_summary: List[str]
        :param samples_treatments_treatment_type: Filter by samples.treatments.treatment_type
        :type samples_treatments_treatment_type: List[str]
        :param sequencing_library_types: Filter by sequencing_library_types
        :type sequencing_library_types: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param targeted_genes_id: Filter by targeted_genes.@id
        :type targeted_genes_id: List[str]
        :param targeted_genes_geneid: Filter by targeted_genes.geneid
        :type targeted_genes_geneid: List[str]
        :param targeted_genes_name: Filter by targeted_genes.name
        :type targeted_genes_name: List[str]
        :param targeted_genes_symbol: Filter by targeted_genes.symbol
        :type targeted_genes_symbol: List[str]
        :param targeted_genes_synonyms: Filter by targeted_genes.synonyms
        :type targeted_genes_synonyms: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._measurement_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assay_term_id=assay_term_id,
            assay_term_term_name=assay_term_term_name,
            auxiliary_sets_id=auxiliary_sets_id,
            auxiliary_sets_accession=auxiliary_sets_accession,
            auxiliary_sets_aliases=auxiliary_sets_aliases,
            auxiliary_sets_file_set_type=auxiliary_sets_file_set_type,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_file_sets_id=control_file_sets_id,
            control_file_sets_accession=control_file_sets_accession,
            control_file_sets_aliases=control_file_sets_aliases,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_image_url=external_image_url,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform_id=files_sequencing_platform_id,
            files_sequencing_platform_term_name=files_sequencing_platform_term_name,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            functional_assay_mechanisms_id=functional_assay_mechanisms_id,
            functional_assay_mechanisms_term_id=functional_assay_mechanisms_term_id,
            functional_assay_mechanisms_term_name=functional_assay_mechanisms_term_name,
            input_file_set_for=input_file_set_for,
            lab_id=lab_id,
            lab_title=lab_title,
            multiome_size=multiome_size,
            notes=notes,
            preferred_assay_title=preferred_assay_title,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            related_multiome_datasets_id=related_multiome_datasets_id,
            related_multiome_datasets_accession=related_multiome_datasets_accession,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments_id=samples_cell_fate_change_treatments_id,
            samples_cell_fate_change_treatments_purpose=samples_cell_fate_change_treatments_purpose,
            samples_cell_fate_change_treatments_summary=samples_cell_fate_change_treatments_summary,
            samples_cell_fate_change_treatments_treatment_type=samples_cell_fate_change_treatments_treatment_type,
            samples_classifications=samples_classifications,
            samples_construct_library_sets_id=samples_construct_library_sets_id,
            samples_construct_library_sets_accession=samples_construct_library_sets_accession,
            samples_construct_library_sets_file_set_type=samples_construct_library_sets_file_set_type,
            samples_construct_library_sets_small_scale_gene_list_id=samples_construct_library_sets_small_scale_gene_list_id,
            samples_construct_library_sets_small_scale_gene_list_geneid=samples_construct_library_sets_small_scale_gene_list_geneid,
            samples_construct_library_sets_small_scale_gene_list_name=samples_construct_library_sets_small_scale_gene_list_name,
            samples_construct_library_sets_small_scale_gene_list_summary=samples_construct_library_sets_small_scale_gene_list_summary,
            samples_construct_library_sets_small_scale_gene_list_symbol=samples_construct_library_sets_small_scale_gene_list_symbol,
            samples_construct_library_sets_summary=samples_construct_library_sets_summary,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications_id=samples_modifications_id,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments_id=samples_treatments_id,
            samples_treatments_purpose=samples_treatments_purpose,
            samples_treatments_summary=samples_treatments_summary,
            samples_treatments_treatment_type=samples_treatments_treatment_type,
            sequencing_library_types=sequencing_library_types,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            targeted_genes_id=targeted_genes_id,
            targeted_genes_geneid=targeted_genes_geneid,
            targeted_genes_name=targeted_genes_name,
            targeted_genes_symbol=targeted_genes_symbol,
            targeted_genes_synonyms=targeted_genes_synonyms,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MeasurementSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _measurement_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        assay_term_id,
        assay_term_term_name,
        auxiliary_sets_id,
        auxiliary_sets_accession,
        auxiliary_sets_aliases,
        auxiliary_sets_file_set_type,
        award_id,
        award_component,
        award_contact_pi_id,
        award_contact_pi_title,
        award_title,
        collections,
        control_file_sets_id,
        control_file_sets_accession,
        control_file_sets_aliases,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        donors_id,
        donors_accession,
        donors_aliases,
        donors_sex,
        donors_status,
        donors_taxa,
        external_image_url,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_creation_timestamp,
        files_file_format,
        files_file_size,
        files_href,
        files_s3_uri,
        files_sequencing_platform_id,
        files_sequencing_platform_term_name,
        files_submitted_file_name,
        files_upload_status,
        functional_assay_mechanisms_id,
        functional_assay_mechanisms_term_id,
        functional_assay_mechanisms_term_name,
        input_file_set_for,
        lab_id,
        lab_title,
        multiome_size,
        notes,
        preferred_assay_title,
        protocols,
        publications_id,
        publications_publication_identifiers,
        related_multiome_datasets_id,
        related_multiome_datasets_accession,
        release_timestamp,
        revoke_detail,
        samples_id,
        samples_accession,
        samples_aliases,
        samples_cell_fate_change_treatments_id,
        samples_cell_fate_change_treatments_purpose,
        samples_cell_fate_change_treatments_summary,
        samples_cell_fate_change_treatments_treatment_type,
        samples_classifications,
        samples_construct_library_sets_id,
        samples_construct_library_sets_accession,
        samples_construct_library_sets_file_set_type,
        samples_construct_library_sets_small_scale_gene_list_id,
        samples_construct_library_sets_small_scale_gene_list_geneid,
        samples_construct_library_sets_small_scale_gene_list_name,
        samples_construct_library_sets_small_scale_gene_list_summary,
        samples_construct_library_sets_small_scale_gene_list_symbol,
        samples_construct_library_sets_summary,
        samples_disease_terms_id,
        samples_disease_terms_term_name,
        samples_modifications_id,
        samples_sample_terms_id,
        samples_sample_terms_aliases,
        samples_sample_terms_status,
        samples_sample_terms_summary,
        samples_sample_terms_term_name,
        samples_status,
        samples_summary,
        samples_targeted_sample_term_id,
        samples_targeted_sample_term_term_name,
        samples_taxa,
        samples_treatments_id,
        samples_treatments_purpose,
        samples_treatments_summary,
        samples_treatments_treatment_type,
        sequencing_library_types,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        targeted_genes_id,
        targeted_genes_geneid,
        targeted_genes_name,
        targeted_genes_symbol,
        targeted_genes_synonyms,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'assay_term.@id': 'multi',
            'assay_term.term_name': 'multi',
            'auxiliary_sets.@id': 'multi',
            'auxiliary_sets.accession': 'multi',
            'auxiliary_sets.aliases': 'multi',
            'auxiliary_sets.file_set_type': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.contact_pi.@id': 'multi',
            'award.contact_pi.title': 'multi',
            'award.title': 'multi',
            'collections': 'multi',
            'control_file_sets.@id': 'multi',
            'control_file_sets.accession': 'multi',
            'control_file_sets.aliases': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'donors.@id': 'multi',
            'donors.accession': 'multi',
            'donors.aliases': 'multi',
            'donors.sex': 'multi',
            'donors.status': 'multi',
            'donors.taxa': 'multi',
            'external_image_url': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.creation_timestamp': 'multi',
            'files.file_format': 'multi',
            'files.file_size': 'multi',
            'files.href': 'multi',
            'files.s3_uri': 'multi',
            'files.sequencing_platform.@id': 'multi',
            'files.sequencing_platform.term_name': 'multi',
            'files.submitted_file_name': 'multi',
            'files.upload_status': 'multi',
            'functional_assay_mechanisms.@id': 'multi',
            'functional_assay_mechanisms.term_id': 'multi',
            'functional_assay_mechanisms.term_name': 'multi',
            'input_file_set_for': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'multiome_size': 'multi',
            'notes': 'multi',
            'preferred_assay_title': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'related_multiome_datasets.@id': 'multi',
            'related_multiome_datasets.accession': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'samples.@id': 'multi',
            'samples.accession': 'multi',
            'samples.aliases': 'multi',
            'samples.cell_fate_change_treatments.@id': 'multi',
            'samples.cell_fate_change_treatments.purpose': 'multi',
            'samples.cell_fate_change_treatments.summary': 'multi',
            'samples.cell_fate_change_treatments.treatment_type': 'multi',
            'samples.classifications': 'multi',
            'samples.construct_library_sets.@id': 'multi',
            'samples.construct_library_sets.accession': 'multi',
            'samples.construct_library_sets.file_set_type': 'multi',
            'samples.construct_library_sets.small_scale_gene_list.@id': 'multi',
            'samples.construct_library_sets.small_scale_gene_list.geneid': 'multi',
            'samples.construct_library_sets.small_scale_gene_list.name': 'multi',
            'samples.construct_library_sets.small_scale_gene_list.summary': 'multi',
            'samples.construct_library_sets.small_scale_gene_list.symbol': 'multi',
            'samples.construct_library_sets.summary': 'multi',
            'samples.disease_terms.@id': 'multi',
            'samples.disease_terms.term_name': 'multi',
            'samples.modifications.@id': 'multi',
            'samples.sample_terms.@id': 'multi',
            'samples.sample_terms.aliases': 'multi',
            'samples.sample_terms.status': 'multi',
            'samples.sample_terms.summary': 'multi',
            'samples.sample_terms.term_name': 'multi',
            'samples.status': 'multi',
            'samples.summary': 'multi',
            'samples.targeted_sample_term.@id': 'multi',
            'samples.targeted_sample_term.term_name': 'multi',
            'samples.taxa': 'multi',
            'samples.treatments.@id': 'multi',
            'samples.treatments.purpose': 'multi',
            'samples.treatments.summary': 'multi',
            'samples.treatments.treatment_type': 'multi',
            'sequencing_library_types': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'targeted_genes.@id': 'multi',
            'targeted_genes.geneid': 'multi',
            'targeted_genes.name': 'multi',
            'targeted_genes.symbol': 'multi',
            'targeted_genes.synonyms': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if assay_term_id is not None:
            
            _query_params.append(('assay_term.@id', assay_term_id))
            
        if assay_term_term_name is not None:
            
            _query_params.append(('assay_term.term_name', assay_term_term_name))
            
        if auxiliary_sets_id is not None:
            
            _query_params.append(('auxiliary_sets.@id', auxiliary_sets_id))
            
        if auxiliary_sets_accession is not None:
            
            _query_params.append(('auxiliary_sets.accession', auxiliary_sets_accession))
            
        if auxiliary_sets_aliases is not None:
            
            _query_params.append(('auxiliary_sets.aliases', auxiliary_sets_aliases))
            
        if auxiliary_sets_file_set_type is not None:
            
            _query_params.append(('auxiliary_sets.file_set_type', auxiliary_sets_file_set_type))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_contact_pi_id is not None:
            
            _query_params.append(('award.contact_pi.@id', award_contact_pi_id))
            
        if award_contact_pi_title is not None:
            
            _query_params.append(('award.contact_pi.title', award_contact_pi_title))
            
        if award_title is not None:
            
            _query_params.append(('award.title', award_title))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_file_sets_id is not None:
            
            _query_params.append(('control_file_sets.@id', control_file_sets_id))
            
        if control_file_sets_accession is not None:
            
            _query_params.append(('control_file_sets.accession', control_file_sets_accession))
            
        if control_file_sets_aliases is not None:
            
            _query_params.append(('control_file_sets.aliases', control_file_sets_aliases))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors_id is not None:
            
            _query_params.append(('donors.@id', donors_id))
            
        if donors_accession is not None:
            
            _query_params.append(('donors.accession', donors_accession))
            
        if donors_aliases is not None:
            
            _query_params.append(('donors.aliases', donors_aliases))
            
        if donors_sex is not None:
            
            _query_params.append(('donors.sex', donors_sex))
            
        if donors_status is not None:
            
            _query_params.append(('donors.status', donors_status))
            
        if donors_taxa is not None:
            
            _query_params.append(('donors.taxa', donors_taxa))
            
        if external_image_url is not None:
            
            _query_params.append(('external_image_url', external_image_url))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_creation_timestamp is not None:
            
            _query_params.append(('files.creation_timestamp', files_creation_timestamp))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_file_size is not None:
            
            _query_params.append(('files.file_size', files_file_size))
            
        if files_href is not None:
            
            _query_params.append(('files.href', files_href))
            
        if files_s3_uri is not None:
            
            _query_params.append(('files.s3_uri', files_s3_uri))
            
        if files_sequencing_platform_id is not None:
            
            _query_params.append(('files.sequencing_platform.@id', files_sequencing_platform_id))
            
        if files_sequencing_platform_term_name is not None:
            
            _query_params.append(('files.sequencing_platform.term_name', files_sequencing_platform_term_name))
            
        if files_submitted_file_name is not None:
            
            _query_params.append(('files.submitted_file_name', files_submitted_file_name))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if functional_assay_mechanisms_id is not None:
            
            _query_params.append(('functional_assay_mechanisms.@id', functional_assay_mechanisms_id))
            
        if functional_assay_mechanisms_term_id is not None:
            
            _query_params.append(('functional_assay_mechanisms.term_id', functional_assay_mechanisms_term_id))
            
        if functional_assay_mechanisms_term_name is not None:
            
            _query_params.append(('functional_assay_mechanisms.term_name', functional_assay_mechanisms_term_name))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if multiome_size is not None:
            
            _query_params.append(('multiome_size', multiome_size))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if preferred_assay_title is not None:
            
            _query_params.append(('preferred_assay_title', preferred_assay_title))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if related_multiome_datasets_id is not None:
            
            _query_params.append(('related_multiome_datasets.@id', related_multiome_datasets_id))
            
        if related_multiome_datasets_accession is not None:
            
            _query_params.append(('related_multiome_datasets.accession', related_multiome_datasets_accession))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if samples_id is not None:
            
            _query_params.append(('samples.@id', samples_id))
            
        if samples_accession is not None:
            
            _query_params.append(('samples.accession', samples_accession))
            
        if samples_aliases is not None:
            
            _query_params.append(('samples.aliases', samples_aliases))
            
        if samples_cell_fate_change_treatments_id is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments.@id', samples_cell_fate_change_treatments_id))
            
        if samples_cell_fate_change_treatments_purpose is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments.purpose', samples_cell_fate_change_treatments_purpose))
            
        if samples_cell_fate_change_treatments_summary is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments.summary', samples_cell_fate_change_treatments_summary))
            
        if samples_cell_fate_change_treatments_treatment_type is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments.treatment_type', samples_cell_fate_change_treatments_treatment_type))
            
        if samples_classifications is not None:
            
            _query_params.append(('samples.classifications', samples_classifications))
            
        if samples_construct_library_sets_id is not None:
            
            _query_params.append(('samples.construct_library_sets.@id', samples_construct_library_sets_id))
            
        if samples_construct_library_sets_accession is not None:
            
            _query_params.append(('samples.construct_library_sets.accession', samples_construct_library_sets_accession))
            
        if samples_construct_library_sets_file_set_type is not None:
            
            _query_params.append(('samples.construct_library_sets.file_set_type', samples_construct_library_sets_file_set_type))
            
        if samples_construct_library_sets_small_scale_gene_list_id is not None:
            
            _query_params.append(('samples.construct_library_sets.small_scale_gene_list.@id', samples_construct_library_sets_small_scale_gene_list_id))
            
        if samples_construct_library_sets_small_scale_gene_list_geneid is not None:
            
            _query_params.append(('samples.construct_library_sets.small_scale_gene_list.geneid', samples_construct_library_sets_small_scale_gene_list_geneid))
            
        if samples_construct_library_sets_small_scale_gene_list_name is not None:
            
            _query_params.append(('samples.construct_library_sets.small_scale_gene_list.name', samples_construct_library_sets_small_scale_gene_list_name))
            
        if samples_construct_library_sets_small_scale_gene_list_summary is not None:
            
            _query_params.append(('samples.construct_library_sets.small_scale_gene_list.summary', samples_construct_library_sets_small_scale_gene_list_summary))
            
        if samples_construct_library_sets_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('samples.construct_library_sets.small_scale_gene_list.symbol', samples_construct_library_sets_small_scale_gene_list_symbol))
            
        if samples_construct_library_sets_summary is not None:
            
            _query_params.append(('samples.construct_library_sets.summary', samples_construct_library_sets_summary))
            
        if samples_disease_terms_id is not None:
            
            _query_params.append(('samples.disease_terms.@id', samples_disease_terms_id))
            
        if samples_disease_terms_term_name is not None:
            
            _query_params.append(('samples.disease_terms.term_name', samples_disease_terms_term_name))
            
        if samples_modifications_id is not None:
            
            _query_params.append(('samples.modifications.@id', samples_modifications_id))
            
        if samples_sample_terms_id is not None:
            
            _query_params.append(('samples.sample_terms.@id', samples_sample_terms_id))
            
        if samples_sample_terms_aliases is not None:
            
            _query_params.append(('samples.sample_terms.aliases', samples_sample_terms_aliases))
            
        if samples_sample_terms_status is not None:
            
            _query_params.append(('samples.sample_terms.status', samples_sample_terms_status))
            
        if samples_sample_terms_summary is not None:
            
            _query_params.append(('samples.sample_terms.summary', samples_sample_terms_summary))
            
        if samples_sample_terms_term_name is not None:
            
            _query_params.append(('samples.sample_terms.term_name', samples_sample_terms_term_name))
            
        if samples_status is not None:
            
            _query_params.append(('samples.status', samples_status))
            
        if samples_summary is not None:
            
            _query_params.append(('samples.summary', samples_summary))
            
        if samples_targeted_sample_term_id is not None:
            
            _query_params.append(('samples.targeted_sample_term.@id', samples_targeted_sample_term_id))
            
        if samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('samples.targeted_sample_term.term_name', samples_targeted_sample_term_term_name))
            
        if samples_taxa is not None:
            
            _query_params.append(('samples.taxa', samples_taxa))
            
        if samples_treatments_id is not None:
            
            _query_params.append(('samples.treatments.@id', samples_treatments_id))
            
        if samples_treatments_purpose is not None:
            
            _query_params.append(('samples.treatments.purpose', samples_treatments_purpose))
            
        if samples_treatments_summary is not None:
            
            _query_params.append(('samples.treatments.summary', samples_treatments_summary))
            
        if samples_treatments_treatment_type is not None:
            
            _query_params.append(('samples.treatments.treatment_type', samples_treatments_treatment_type))
            
        if sequencing_library_types is not None:
            
            _query_params.append(('sequencing_library_types', sequencing_library_types))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if targeted_genes_id is not None:
            
            _query_params.append(('targeted_genes.@id', targeted_genes_id))
            
        if targeted_genes_geneid is not None:
            
            _query_params.append(('targeted_genes.geneid', targeted_genes_geneid))
            
        if targeted_genes_name is not None:
            
            _query_params.append(('targeted_genes.name', targeted_genes_name))
            
        if targeted_genes_symbol is not None:
            
            _query_params.append(('targeted_genes.symbol', targeted_genes_symbol))
            
        if targeted_genes_synonyms is not None:
            
            _query_params.append(('targeted_genes.synonyms', targeted_genes_synonyms))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/measurement-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def model_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ModelFileResults:
        """List items in the ModelFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ModelFile items. Supports filtering on fields within ModelFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._model_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ModelFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def model_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ModelFileResults]:
        """List items in the ModelFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ModelFile items. Supports filtering on fields within ModelFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._model_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ModelFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def model_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the ModelFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ModelFile items. Supports filtering on fields within ModelFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._model_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ModelFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _model_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        anvil_url,
        assay_titles,
        award_id,
        award_component,
        collections,
        content_md5sum,
        content_type,
        controlled_access,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        release_timestamp,
        revoke_detail,
        s3_uri,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'anvil_url': 'multi',
            'assay_titles': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'controlled_access': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if anvil_url is not None:
            
            _query_params.append(('anvil_url', anvil_url))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if controlled_access is not None:
            
            _query_params.append(('controlled_access', controlled_access))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/model-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def model_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assessed_genes: Annotated[Optional[List[StrictStr]], Field(description="Filter by assessed_genes")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_input_data: Annotated[Optional[List[StrictStr]], Field(description="Filter by external_input_data")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.@id")] = None,
        input_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.accession")] = None,
        input_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by input_file_sets.aliases")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        model_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by model_name")] = None,
        model_version: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by model_version")] = None,
        model_zoo_location: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by model_zoo_location")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        prediction_objects: Annotated[Optional[List[StrictStr]], Field(description="Filter by prediction_objects")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        software_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_version")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ModelSetResults:
        """List items in the ModelSet collection.

        Collection endpoint that accepts various query parameters to filter and sort ModelSet items. Supports filtering on fields within ModelSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assessed_genes: Filter by assessed_genes
        :type assessed_genes: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_input_data: Filter by external_input_data
        :type external_input_data: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets_id: Filter by input_file_sets.@id
        :type input_file_sets_id: List[str]
        :param input_file_sets_accession: Filter by input_file_sets.accession
        :type input_file_sets_accession: List[str]
        :param input_file_sets_aliases: Filter by input_file_sets.aliases
        :type input_file_sets_aliases: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param model_name: Filter by model_name
        :type model_name: List[str]
        :param model_version: Filter by model_version
        :type model_version: List[str]
        :param model_zoo_location: Filter by model_zoo_location
        :type model_zoo_location: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param prediction_objects: Filter by prediction_objects
        :type prediction_objects: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param software_version: Filter by software_version
        :type software_version: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._model_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assessed_genes=assessed_genes,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_input_data=external_input_data,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            input_file_sets_id=input_file_sets_id,
            input_file_sets_accession=input_file_sets_accession,
            input_file_sets_aliases=input_file_sets_aliases,
            lab_id=lab_id,
            lab_title=lab_title,
            model_name=model_name,
            model_version=model_version,
            model_zoo_location=model_zoo_location,
            notes=notes,
            prediction_objects=prediction_objects,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            software_version=software_version,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ModelSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def model_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assessed_genes: Annotated[Optional[List[StrictStr]], Field(description="Filter by assessed_genes")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_input_data: Annotated[Optional[List[StrictStr]], Field(description="Filter by external_input_data")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.@id")] = None,
        input_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.accession")] = None,
        input_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by input_file_sets.aliases")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        model_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by model_name")] = None,
        model_version: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by model_version")] = None,
        model_zoo_location: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by model_zoo_location")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        prediction_objects: Annotated[Optional[List[StrictStr]], Field(description="Filter by prediction_objects")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        software_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_version")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ModelSetResults]:
        """List items in the ModelSet collection.

        Collection endpoint that accepts various query parameters to filter and sort ModelSet items. Supports filtering on fields within ModelSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assessed_genes: Filter by assessed_genes
        :type assessed_genes: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_input_data: Filter by external_input_data
        :type external_input_data: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets_id: Filter by input_file_sets.@id
        :type input_file_sets_id: List[str]
        :param input_file_sets_accession: Filter by input_file_sets.accession
        :type input_file_sets_accession: List[str]
        :param input_file_sets_aliases: Filter by input_file_sets.aliases
        :type input_file_sets_aliases: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param model_name: Filter by model_name
        :type model_name: List[str]
        :param model_version: Filter by model_version
        :type model_version: List[str]
        :param model_zoo_location: Filter by model_zoo_location
        :type model_zoo_location: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param prediction_objects: Filter by prediction_objects
        :type prediction_objects: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param software_version: Filter by software_version
        :type software_version: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._model_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assessed_genes=assessed_genes,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_input_data=external_input_data,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            input_file_sets_id=input_file_sets_id,
            input_file_sets_accession=input_file_sets_accession,
            input_file_sets_aliases=input_file_sets_aliases,
            lab_id=lab_id,
            lab_title=lab_title,
            model_name=model_name,
            model_version=model_version,
            model_zoo_location=model_zoo_location,
            notes=notes,
            prediction_objects=prediction_objects,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            software_version=software_version,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ModelSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def model_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assessed_genes: Annotated[Optional[List[StrictStr]], Field(description="Filter by assessed_genes")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        external_input_data: Annotated[Optional[List[StrictStr]], Field(description="Filter by external_input_data")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.@id")] = None,
        input_file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets.accession")] = None,
        input_file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by input_file_sets.aliases")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        model_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by model_name")] = None,
        model_version: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by model_version")] = None,
        model_zoo_location: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by model_zoo_location")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        prediction_objects: Annotated[Optional[List[StrictStr]], Field(description="Filter by prediction_objects")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        software_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_version")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the ModelSet collection.

        Collection endpoint that accepts various query parameters to filter and sort ModelSet items. Supports filtering on fields within ModelSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assessed_genes: Filter by assessed_genes
        :type assessed_genes: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param external_input_data: Filter by external_input_data
        :type external_input_data: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets_id: Filter by input_file_sets.@id
        :type input_file_sets_id: List[str]
        :param input_file_sets_accession: Filter by input_file_sets.accession
        :type input_file_sets_accession: List[str]
        :param input_file_sets_aliases: Filter by input_file_sets.aliases
        :type input_file_sets_aliases: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param model_name: Filter by model_name
        :type model_name: List[str]
        :param model_version: Filter by model_version
        :type model_version: List[str]
        :param model_zoo_location: Filter by model_zoo_location
        :type model_zoo_location: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param prediction_objects: Filter by prediction_objects
        :type prediction_objects: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets: Filter by samples.construct_library_sets
        :type samples_construct_library_sets: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param software_version: Filter by software_version
        :type software_version: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._model_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assessed_genes=assessed_genes,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            external_input_data=external_input_data,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            input_file_sets_id=input_file_sets_id,
            input_file_sets_accession=input_file_sets_accession,
            input_file_sets_aliases=input_file_sets_aliases,
            lab_id=lab_id,
            lab_title=lab_title,
            model_name=model_name,
            model_version=model_version,
            model_zoo_location=model_zoo_location,
            notes=notes,
            prediction_objects=prediction_objects,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets=samples_construct_library_sets,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            software_version=software_version,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ModelSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _model_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        assessed_genes,
        award_id,
        award_component,
        award_contact_pi_id,
        award_contact_pi_title,
        award_title,
        collections,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        donors_id,
        donors_accession,
        donors_aliases,
        donors_sex,
        donors_status,
        donors_taxa,
        external_input_data,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_creation_timestamp,
        files_file_format,
        files_file_size,
        files_href,
        files_s3_uri,
        files_sequencing_platform,
        files_submitted_file_name,
        files_upload_status,
        input_file_set_for,
        input_file_sets_id,
        input_file_sets_accession,
        input_file_sets_aliases,
        lab_id,
        lab_title,
        model_name,
        model_version,
        model_zoo_location,
        notes,
        prediction_objects,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        samples_id,
        samples_accession,
        samples_aliases,
        samples_cell_fate_change_treatments,
        samples_classifications,
        samples_construct_library_sets,
        samples_disease_terms_id,
        samples_disease_terms_term_name,
        samples_modifications,
        samples_sample_terms_id,
        samples_sample_terms_aliases,
        samples_sample_terms_status,
        samples_sample_terms_summary,
        samples_sample_terms_term_name,
        samples_status,
        samples_summary,
        samples_targeted_sample_term_id,
        samples_targeted_sample_term_term_name,
        samples_taxa,
        samples_treatments,
        software_version,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        url,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'assessed_genes': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.contact_pi.@id': 'multi',
            'award.contact_pi.title': 'multi',
            'award.title': 'multi',
            'collections': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'donors.@id': 'multi',
            'donors.accession': 'multi',
            'donors.aliases': 'multi',
            'donors.sex': 'multi',
            'donors.status': 'multi',
            'donors.taxa': 'multi',
            'external_input_data': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.creation_timestamp': 'multi',
            'files.file_format': 'multi',
            'files.file_size': 'multi',
            'files.href': 'multi',
            'files.s3_uri': 'multi',
            'files.sequencing_platform': 'multi',
            'files.submitted_file_name': 'multi',
            'files.upload_status': 'multi',
            'input_file_set_for': 'multi',
            'input_file_sets.@id': 'multi',
            'input_file_sets.accession': 'multi',
            'input_file_sets.aliases': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'model_name': 'multi',
            'model_version': 'multi',
            'model_zoo_location': 'multi',
            'notes': 'multi',
            'prediction_objects': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'samples.@id': 'multi',
            'samples.accession': 'multi',
            'samples.aliases': 'multi',
            'samples.cell_fate_change_treatments': 'multi',
            'samples.classifications': 'multi',
            'samples.construct_library_sets': 'multi',
            'samples.disease_terms.@id': 'multi',
            'samples.disease_terms.term_name': 'multi',
            'samples.modifications': 'multi',
            'samples.sample_terms.@id': 'multi',
            'samples.sample_terms.aliases': 'multi',
            'samples.sample_terms.status': 'multi',
            'samples.sample_terms.summary': 'multi',
            'samples.sample_terms.term_name': 'multi',
            'samples.status': 'multi',
            'samples.summary': 'multi',
            'samples.targeted_sample_term.@id': 'multi',
            'samples.targeted_sample_term.term_name': 'multi',
            'samples.taxa': 'multi',
            'samples.treatments': 'multi',
            'software_version': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'url': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if assessed_genes is not None:
            
            _query_params.append(('assessed_genes', assessed_genes))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_contact_pi_id is not None:
            
            _query_params.append(('award.contact_pi.@id', award_contact_pi_id))
            
        if award_contact_pi_title is not None:
            
            _query_params.append(('award.contact_pi.title', award_contact_pi_title))
            
        if award_title is not None:
            
            _query_params.append(('award.title', award_title))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors_id is not None:
            
            _query_params.append(('donors.@id', donors_id))
            
        if donors_accession is not None:
            
            _query_params.append(('donors.accession', donors_accession))
            
        if donors_aliases is not None:
            
            _query_params.append(('donors.aliases', donors_aliases))
            
        if donors_sex is not None:
            
            _query_params.append(('donors.sex', donors_sex))
            
        if donors_status is not None:
            
            _query_params.append(('donors.status', donors_status))
            
        if donors_taxa is not None:
            
            _query_params.append(('donors.taxa', donors_taxa))
            
        if external_input_data is not None:
            
            _query_params.append(('external_input_data', external_input_data))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_creation_timestamp is not None:
            
            _query_params.append(('files.creation_timestamp', files_creation_timestamp))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_file_size is not None:
            
            _query_params.append(('files.file_size', files_file_size))
            
        if files_href is not None:
            
            _query_params.append(('files.href', files_href))
            
        if files_s3_uri is not None:
            
            _query_params.append(('files.s3_uri', files_s3_uri))
            
        if files_sequencing_platform is not None:
            
            _query_params.append(('files.sequencing_platform', files_sequencing_platform))
            
        if files_submitted_file_name is not None:
            
            _query_params.append(('files.submitted_file_name', files_submitted_file_name))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if input_file_sets_id is not None:
            
            _query_params.append(('input_file_sets.@id', input_file_sets_id))
            
        if input_file_sets_accession is not None:
            
            _query_params.append(('input_file_sets.accession', input_file_sets_accession))
            
        if input_file_sets_aliases is not None:
            
            _query_params.append(('input_file_sets.aliases', input_file_sets_aliases))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if model_name is not None:
            
            _query_params.append(('model_name', model_name))
            
        if model_version is not None:
            
            _query_params.append(('model_version', model_version))
            
        if model_zoo_location is not None:
            
            _query_params.append(('model_zoo_location', model_zoo_location))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if prediction_objects is not None:
            
            _query_params.append(('prediction_objects', prediction_objects))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if samples_id is not None:
            
            _query_params.append(('samples.@id', samples_id))
            
        if samples_accession is not None:
            
            _query_params.append(('samples.accession', samples_accession))
            
        if samples_aliases is not None:
            
            _query_params.append(('samples.aliases', samples_aliases))
            
        if samples_cell_fate_change_treatments is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments', samples_cell_fate_change_treatments))
            
        if samples_classifications is not None:
            
            _query_params.append(('samples.classifications', samples_classifications))
            
        if samples_construct_library_sets is not None:
            
            _query_params.append(('samples.construct_library_sets', samples_construct_library_sets))
            
        if samples_disease_terms_id is not None:
            
            _query_params.append(('samples.disease_terms.@id', samples_disease_terms_id))
            
        if samples_disease_terms_term_name is not None:
            
            _query_params.append(('samples.disease_terms.term_name', samples_disease_terms_term_name))
            
        if samples_modifications is not None:
            
            _query_params.append(('samples.modifications', samples_modifications))
            
        if samples_sample_terms_id is not None:
            
            _query_params.append(('samples.sample_terms.@id', samples_sample_terms_id))
            
        if samples_sample_terms_aliases is not None:
            
            _query_params.append(('samples.sample_terms.aliases', samples_sample_terms_aliases))
            
        if samples_sample_terms_status is not None:
            
            _query_params.append(('samples.sample_terms.status', samples_sample_terms_status))
            
        if samples_sample_terms_summary is not None:
            
            _query_params.append(('samples.sample_terms.summary', samples_sample_terms_summary))
            
        if samples_sample_terms_term_name is not None:
            
            _query_params.append(('samples.sample_terms.term_name', samples_sample_terms_term_name))
            
        if samples_status is not None:
            
            _query_params.append(('samples.status', samples_status))
            
        if samples_summary is not None:
            
            _query_params.append(('samples.summary', samples_summary))
            
        if samples_targeted_sample_term_id is not None:
            
            _query_params.append(('samples.targeted_sample_term.@id', samples_targeted_sample_term_id))
            
        if samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('samples.targeted_sample_term.term_name', samples_targeted_sample_term_term_name))
            
        if samples_taxa is not None:
            
            _query_params.append(('samples.taxa', samples_taxa))
            
        if samples_treatments is not None:
            
            _query_params.append(('samples.treatments', samples_treatments))
            
        if software_version is not None:
            
            _query_params.append(('software_version', software_version))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/model-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def multiplexed_samples(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        barcode_map: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets.@id")] = None,
        construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets.accession")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        multiplexed_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.@id")] = None,
        multiplexed_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.accession")] = None,
        multiplexed_samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.construct_library_sets")] = None,
        multiplexed_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.disease_terms.@id")] = None,
        multiplexed_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by multiplexed_samples.disease_terms.term_name")] = None,
        multiplexed_samples_donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.donors.@id")] = None,
        multiplexed_samples_donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.donors.accession")] = None,
        multiplexed_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.sample_terms.@id")] = None,
        multiplexed_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by multiplexed_samples.sample_terms.term_name")] = None,
        multiplexed_samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.status")] = None,
        multiplexed_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.summary")] = None,
        multiplexing_methods: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexing_methods")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> MultiplexedSampleResults:
        """List items in the MultiplexedSample collection.

        Collection endpoint that accepts various query parameters to filter and sort MultiplexedSample items. Supports filtering on fields within MultiplexedSample items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param barcode_map: Filter by barcode_map
        :type barcode_map: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets_id: Filter by construct_library_sets.@id
        :type construct_library_sets_id: List[str]
        :param construct_library_sets_accession: Filter by construct_library_sets.accession
        :type construct_library_sets_accession: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates: Filter by institutional_certificates
        :type institutional_certificates: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param multiplexed_samples_id: Filter by multiplexed_samples.@id
        :type multiplexed_samples_id: List[str]
        :param multiplexed_samples_accession: Filter by multiplexed_samples.accession
        :type multiplexed_samples_accession: List[str]
        :param multiplexed_samples_construct_library_sets: Filter by multiplexed_samples.construct_library_sets
        :type multiplexed_samples_construct_library_sets: List[str]
        :param multiplexed_samples_disease_terms_id: Filter by multiplexed_samples.disease_terms.@id
        :type multiplexed_samples_disease_terms_id: List[str]
        :param multiplexed_samples_disease_terms_term_name: Filter by multiplexed_samples.disease_terms.term_name
        :type multiplexed_samples_disease_terms_term_name: List[str]
        :param multiplexed_samples_donors_id: Filter by multiplexed_samples.donors.@id
        :type multiplexed_samples_donors_id: List[str]
        :param multiplexed_samples_donors_accession: Filter by multiplexed_samples.donors.accession
        :type multiplexed_samples_donors_accession: List[str]
        :param multiplexed_samples_sample_terms_id: Filter by multiplexed_samples.sample_terms.@id
        :type multiplexed_samples_sample_terms_id: List[str]
        :param multiplexed_samples_sample_terms_term_name: Filter by multiplexed_samples.sample_terms.term_name
        :type multiplexed_samples_sample_terms_term_name: List[str]
        :param multiplexed_samples_status: Filter by multiplexed_samples.status
        :type multiplexed_samples_status: List[str]
        :param multiplexed_samples_summary: Filter by multiplexed_samples.summary
        :type multiplexed_samples_summary: List[str]
        :param multiplexing_methods: Filter by multiplexing_methods
        :type multiplexing_methods: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._multiplexed_samples_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            barcode_map=barcode_map,
            biomarkers=biomarkers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets_id=construct_library_sets_id,
            construct_library_sets_accession=construct_library_sets_accession,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates=institutional_certificates,
            lab_id=lab_id,
            lab_title=lab_title,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            multiplexed_samples_id=multiplexed_samples_id,
            multiplexed_samples_accession=multiplexed_samples_accession,
            multiplexed_samples_construct_library_sets=multiplexed_samples_construct_library_sets,
            multiplexed_samples_disease_terms_id=multiplexed_samples_disease_terms_id,
            multiplexed_samples_disease_terms_term_name=multiplexed_samples_disease_terms_term_name,
            multiplexed_samples_donors_id=multiplexed_samples_donors_id,
            multiplexed_samples_donors_accession=multiplexed_samples_donors_accession,
            multiplexed_samples_sample_terms_id=multiplexed_samples_sample_terms_id,
            multiplexed_samples_sample_terms_term_name=multiplexed_samples_sample_terms_term_name,
            multiplexed_samples_status=multiplexed_samples_status,
            multiplexed_samples_summary=multiplexed_samples_summary,
            multiplexing_methods=multiplexing_methods,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MultiplexedSampleResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def multiplexed_samples_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        barcode_map: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets.@id")] = None,
        construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets.accession")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        multiplexed_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.@id")] = None,
        multiplexed_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.accession")] = None,
        multiplexed_samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.construct_library_sets")] = None,
        multiplexed_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.disease_terms.@id")] = None,
        multiplexed_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by multiplexed_samples.disease_terms.term_name")] = None,
        multiplexed_samples_donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.donors.@id")] = None,
        multiplexed_samples_donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.donors.accession")] = None,
        multiplexed_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.sample_terms.@id")] = None,
        multiplexed_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by multiplexed_samples.sample_terms.term_name")] = None,
        multiplexed_samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.status")] = None,
        multiplexed_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.summary")] = None,
        multiplexing_methods: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexing_methods")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[MultiplexedSampleResults]:
        """List items in the MultiplexedSample collection.

        Collection endpoint that accepts various query parameters to filter and sort MultiplexedSample items. Supports filtering on fields within MultiplexedSample items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param barcode_map: Filter by barcode_map
        :type barcode_map: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets_id: Filter by construct_library_sets.@id
        :type construct_library_sets_id: List[str]
        :param construct_library_sets_accession: Filter by construct_library_sets.accession
        :type construct_library_sets_accession: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates: Filter by institutional_certificates
        :type institutional_certificates: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param multiplexed_samples_id: Filter by multiplexed_samples.@id
        :type multiplexed_samples_id: List[str]
        :param multiplexed_samples_accession: Filter by multiplexed_samples.accession
        :type multiplexed_samples_accession: List[str]
        :param multiplexed_samples_construct_library_sets: Filter by multiplexed_samples.construct_library_sets
        :type multiplexed_samples_construct_library_sets: List[str]
        :param multiplexed_samples_disease_terms_id: Filter by multiplexed_samples.disease_terms.@id
        :type multiplexed_samples_disease_terms_id: List[str]
        :param multiplexed_samples_disease_terms_term_name: Filter by multiplexed_samples.disease_terms.term_name
        :type multiplexed_samples_disease_terms_term_name: List[str]
        :param multiplexed_samples_donors_id: Filter by multiplexed_samples.donors.@id
        :type multiplexed_samples_donors_id: List[str]
        :param multiplexed_samples_donors_accession: Filter by multiplexed_samples.donors.accession
        :type multiplexed_samples_donors_accession: List[str]
        :param multiplexed_samples_sample_terms_id: Filter by multiplexed_samples.sample_terms.@id
        :type multiplexed_samples_sample_terms_id: List[str]
        :param multiplexed_samples_sample_terms_term_name: Filter by multiplexed_samples.sample_terms.term_name
        :type multiplexed_samples_sample_terms_term_name: List[str]
        :param multiplexed_samples_status: Filter by multiplexed_samples.status
        :type multiplexed_samples_status: List[str]
        :param multiplexed_samples_summary: Filter by multiplexed_samples.summary
        :type multiplexed_samples_summary: List[str]
        :param multiplexing_methods: Filter by multiplexing_methods
        :type multiplexing_methods: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._multiplexed_samples_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            barcode_map=barcode_map,
            biomarkers=biomarkers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets_id=construct_library_sets_id,
            construct_library_sets_accession=construct_library_sets_accession,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates=institutional_certificates,
            lab_id=lab_id,
            lab_title=lab_title,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            multiplexed_samples_id=multiplexed_samples_id,
            multiplexed_samples_accession=multiplexed_samples_accession,
            multiplexed_samples_construct_library_sets=multiplexed_samples_construct_library_sets,
            multiplexed_samples_disease_terms_id=multiplexed_samples_disease_terms_id,
            multiplexed_samples_disease_terms_term_name=multiplexed_samples_disease_terms_term_name,
            multiplexed_samples_donors_id=multiplexed_samples_donors_id,
            multiplexed_samples_donors_accession=multiplexed_samples_donors_accession,
            multiplexed_samples_sample_terms_id=multiplexed_samples_sample_terms_id,
            multiplexed_samples_sample_terms_term_name=multiplexed_samples_sample_terms_term_name,
            multiplexed_samples_status=multiplexed_samples_status,
            multiplexed_samples_summary=multiplexed_samples_summary,
            multiplexing_methods=multiplexing_methods,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MultiplexedSampleResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def multiplexed_samples_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        barcode_map: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets.@id")] = None,
        construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets.accession")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        multiplexed_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.@id")] = None,
        multiplexed_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.accession")] = None,
        multiplexed_samples_construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.construct_library_sets")] = None,
        multiplexed_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.disease_terms.@id")] = None,
        multiplexed_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by multiplexed_samples.disease_terms.term_name")] = None,
        multiplexed_samples_donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.donors.@id")] = None,
        multiplexed_samples_donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.donors.accession")] = None,
        multiplexed_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.sample_terms.@id")] = None,
        multiplexed_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by multiplexed_samples.sample_terms.term_name")] = None,
        multiplexed_samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.status")] = None,
        multiplexed_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_samples.summary")] = None,
        multiplexing_methods: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexing_methods")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the MultiplexedSample collection.

        Collection endpoint that accepts various query parameters to filter and sort MultiplexedSample items. Supports filtering on fields within MultiplexedSample items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param barcode_map: Filter by barcode_map
        :type barcode_map: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets_id: Filter by construct_library_sets.@id
        :type construct_library_sets_id: List[str]
        :param construct_library_sets_accession: Filter by construct_library_sets.accession
        :type construct_library_sets_accession: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates: Filter by institutional_certificates
        :type institutional_certificates: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param multiplexed_samples_id: Filter by multiplexed_samples.@id
        :type multiplexed_samples_id: List[str]
        :param multiplexed_samples_accession: Filter by multiplexed_samples.accession
        :type multiplexed_samples_accession: List[str]
        :param multiplexed_samples_construct_library_sets: Filter by multiplexed_samples.construct_library_sets
        :type multiplexed_samples_construct_library_sets: List[str]
        :param multiplexed_samples_disease_terms_id: Filter by multiplexed_samples.disease_terms.@id
        :type multiplexed_samples_disease_terms_id: List[str]
        :param multiplexed_samples_disease_terms_term_name: Filter by multiplexed_samples.disease_terms.term_name
        :type multiplexed_samples_disease_terms_term_name: List[str]
        :param multiplexed_samples_donors_id: Filter by multiplexed_samples.donors.@id
        :type multiplexed_samples_donors_id: List[str]
        :param multiplexed_samples_donors_accession: Filter by multiplexed_samples.donors.accession
        :type multiplexed_samples_donors_accession: List[str]
        :param multiplexed_samples_sample_terms_id: Filter by multiplexed_samples.sample_terms.@id
        :type multiplexed_samples_sample_terms_id: List[str]
        :param multiplexed_samples_sample_terms_term_name: Filter by multiplexed_samples.sample_terms.term_name
        :type multiplexed_samples_sample_terms_term_name: List[str]
        :param multiplexed_samples_status: Filter by multiplexed_samples.status
        :type multiplexed_samples_status: List[str]
        :param multiplexed_samples_summary: Filter by multiplexed_samples.summary
        :type multiplexed_samples_summary: List[str]
        :param multiplexing_methods: Filter by multiplexing_methods
        :type multiplexing_methods: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._multiplexed_samples_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            barcode_map=barcode_map,
            biomarkers=biomarkers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets_id=construct_library_sets_id,
            construct_library_sets_accession=construct_library_sets_accession,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates=institutional_certificates,
            lab_id=lab_id,
            lab_title=lab_title,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            multiplexed_samples_id=multiplexed_samples_id,
            multiplexed_samples_accession=multiplexed_samples_accession,
            multiplexed_samples_construct_library_sets=multiplexed_samples_construct_library_sets,
            multiplexed_samples_disease_terms_id=multiplexed_samples_disease_terms_id,
            multiplexed_samples_disease_terms_term_name=multiplexed_samples_disease_terms_term_name,
            multiplexed_samples_donors_id=multiplexed_samples_donors_id,
            multiplexed_samples_donors_accession=multiplexed_samples_donors_accession,
            multiplexed_samples_sample_terms_id=multiplexed_samples_sample_terms_id,
            multiplexed_samples_sample_terms_term_name=multiplexed_samples_sample_terms_term_name,
            multiplexed_samples_status=multiplexed_samples_status,
            multiplexed_samples_summary=multiplexed_samples_summary,
            multiplexing_methods=multiplexing_methods,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "MultiplexedSampleResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _multiplexed_samples_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        barcode_map,
        biomarkers,
        cellular_sub_pool,
        classifications,
        collections,
        construct_library_sets_id,
        construct_library_sets_accession,
        creation_timestamp,
        date_obtained,
        dbxrefs,
        description,
        disease_terms_id,
        disease_terms_term_name,
        documents,
        donors,
        file_sets_id,
        file_sets_accession,
        file_sets_aliases,
        file_sets_assay_term_term_name,
        file_sets_lab_title,
        file_sets_status,
        file_sets_summary,
        institutional_certificates,
        lab_id,
        lab_title,
        modifications_id,
        modifications_status,
        modifications_summary,
        moi,
        multiplexed_in_id,
        multiplexed_in_accession,
        multiplexed_samples_id,
        multiplexed_samples_accession,
        multiplexed_samples_construct_library_sets,
        multiplexed_samples_disease_terms_id,
        multiplexed_samples_disease_terms_term_name,
        multiplexed_samples_donors_id,
        multiplexed_samples_donors_accession,
        multiplexed_samples_sample_terms_id,
        multiplexed_samples_sample_terms_term_name,
        multiplexed_samples_status,
        multiplexed_samples_summary,
        multiplexing_methods,
        notes,
        nucleic_acid_delivery,
        origin_of,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_terms_id,
        sample_terms_term_name,
        sorted_fractions,
        sorted_from_id,
        sorted_from_accession,
        sorted_from_detail,
        sources_id,
        starting_amount,
        starting_amount_units,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        time_post_library_delivery,
        time_post_library_delivery_units,
        treatments_id,
        treatments_purpose,
        treatments_status,
        treatments_summary,
        treatments_treatment_type,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'barcode_map': 'multi',
            'biomarkers': 'multi',
            'cellular_sub_pool': 'multi',
            'classifications': 'multi',
            'collections': 'multi',
            'construct_library_sets.@id': 'multi',
            'construct_library_sets.accession': 'multi',
            'creation_timestamp': 'multi',
            'date_obtained': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'disease_terms.@id': 'multi',
            'disease_terms.term_name': 'multi',
            'documents': 'multi',
            'donors': 'multi',
            'file_sets.@id': 'multi',
            'file_sets.accession': 'multi',
            'file_sets.aliases': 'multi',
            'file_sets.assay_term.term_name': 'multi',
            'file_sets.lab.title': 'multi',
            'file_sets.status': 'multi',
            'file_sets.summary': 'multi',
            'institutional_certificates': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'modifications.@id': 'multi',
            'modifications.status': 'multi',
            'modifications.summary': 'multi',
            'moi': 'multi',
            'multiplexed_in.@id': 'multi',
            'multiplexed_in.accession': 'multi',
            'multiplexed_samples.@id': 'multi',
            'multiplexed_samples.accession': 'multi',
            'multiplexed_samples.construct_library_sets': 'multi',
            'multiplexed_samples.disease_terms.@id': 'multi',
            'multiplexed_samples.disease_terms.term_name': 'multi',
            'multiplexed_samples.donors.@id': 'multi',
            'multiplexed_samples.donors.accession': 'multi',
            'multiplexed_samples.sample_terms.@id': 'multi',
            'multiplexed_samples.sample_terms.term_name': 'multi',
            'multiplexed_samples.status': 'multi',
            'multiplexed_samples.summary': 'multi',
            'multiplexing_methods': 'multi',
            'notes': 'multi',
            'nucleic_acid_delivery': 'multi',
            'origin_of': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_terms.@id': 'multi',
            'sample_terms.term_name': 'multi',
            'sorted_fractions': 'multi',
            'sorted_from.@id': 'multi',
            'sorted_from.accession': 'multi',
            'sorted_from_detail': 'multi',
            'sources.@id': 'multi',
            'starting_amount': 'multi',
            'starting_amount_units': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'time_post_library_delivery': 'multi',
            'time_post_library_delivery_units': 'multi',
            'treatments.@id': 'multi',
            'treatments.purpose': 'multi',
            'treatments.status': 'multi',
            'treatments.summary': 'multi',
            'treatments.treatment_type': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if barcode_map is not None:
            
            _query_params.append(('barcode_map', barcode_map))
            
        if biomarkers is not None:
            
            _query_params.append(('biomarkers', biomarkers))
            
        if cellular_sub_pool is not None:
            
            _query_params.append(('cellular_sub_pool', cellular_sub_pool))
            
        if classifications is not None:
            
            _query_params.append(('classifications', classifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if construct_library_sets_id is not None:
            
            _query_params.append(('construct_library_sets.@id', construct_library_sets_id))
            
        if construct_library_sets_accession is not None:
            
            _query_params.append(('construct_library_sets.accession', construct_library_sets_accession))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_obtained is not None:
            
            _query_params.append(('date_obtained', date_obtained))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if disease_terms_id is not None:
            
            _query_params.append(('disease_terms.@id', disease_terms_id))
            
        if disease_terms_term_name is not None:
            
            _query_params.append(('disease_terms.term_name', disease_terms_term_name))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors is not None:
            
            _query_params.append(('donors', donors))
            
        if file_sets_id is not None:
            
            _query_params.append(('file_sets.@id', file_sets_id))
            
        if file_sets_accession is not None:
            
            _query_params.append(('file_sets.accession', file_sets_accession))
            
        if file_sets_aliases is not None:
            
            _query_params.append(('file_sets.aliases', file_sets_aliases))
            
        if file_sets_assay_term_term_name is not None:
            
            _query_params.append(('file_sets.assay_term.term_name', file_sets_assay_term_term_name))
            
        if file_sets_lab_title is not None:
            
            _query_params.append(('file_sets.lab.title', file_sets_lab_title))
            
        if file_sets_status is not None:
            
            _query_params.append(('file_sets.status', file_sets_status))
            
        if file_sets_summary is not None:
            
            _query_params.append(('file_sets.summary', file_sets_summary))
            
        if institutional_certificates is not None:
            
            _query_params.append(('institutional_certificates', institutional_certificates))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if modifications_id is not None:
            
            _query_params.append(('modifications.@id', modifications_id))
            
        if modifications_status is not None:
            
            _query_params.append(('modifications.status', modifications_status))
            
        if modifications_summary is not None:
            
            _query_params.append(('modifications.summary', modifications_summary))
            
        if moi is not None:
            
            _query_params.append(('moi', moi))
            
        if multiplexed_in_id is not None:
            
            _query_params.append(('multiplexed_in.@id', multiplexed_in_id))
            
        if multiplexed_in_accession is not None:
            
            _query_params.append(('multiplexed_in.accession', multiplexed_in_accession))
            
        if multiplexed_samples_id is not None:
            
            _query_params.append(('multiplexed_samples.@id', multiplexed_samples_id))
            
        if multiplexed_samples_accession is not None:
            
            _query_params.append(('multiplexed_samples.accession', multiplexed_samples_accession))
            
        if multiplexed_samples_construct_library_sets is not None:
            
            _query_params.append(('multiplexed_samples.construct_library_sets', multiplexed_samples_construct_library_sets))
            
        if multiplexed_samples_disease_terms_id is not None:
            
            _query_params.append(('multiplexed_samples.disease_terms.@id', multiplexed_samples_disease_terms_id))
            
        if multiplexed_samples_disease_terms_term_name is not None:
            
            _query_params.append(('multiplexed_samples.disease_terms.term_name', multiplexed_samples_disease_terms_term_name))
            
        if multiplexed_samples_donors_id is not None:
            
            _query_params.append(('multiplexed_samples.donors.@id', multiplexed_samples_donors_id))
            
        if multiplexed_samples_donors_accession is not None:
            
            _query_params.append(('multiplexed_samples.donors.accession', multiplexed_samples_donors_accession))
            
        if multiplexed_samples_sample_terms_id is not None:
            
            _query_params.append(('multiplexed_samples.sample_terms.@id', multiplexed_samples_sample_terms_id))
            
        if multiplexed_samples_sample_terms_term_name is not None:
            
            _query_params.append(('multiplexed_samples.sample_terms.term_name', multiplexed_samples_sample_terms_term_name))
            
        if multiplexed_samples_status is not None:
            
            _query_params.append(('multiplexed_samples.status', multiplexed_samples_status))
            
        if multiplexed_samples_summary is not None:
            
            _query_params.append(('multiplexed_samples.summary', multiplexed_samples_summary))
            
        if multiplexing_methods is not None:
            
            _query_params.append(('multiplexing_methods', multiplexing_methods))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if nucleic_acid_delivery is not None:
            
            _query_params.append(('nucleic_acid_delivery', nucleic_acid_delivery))
            
        if origin_of is not None:
            
            _query_params.append(('origin_of', origin_of))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_terms_id is not None:
            
            _query_params.append(('sample_terms.@id', sample_terms_id))
            
        if sample_terms_term_name is not None:
            
            _query_params.append(('sample_terms.term_name', sample_terms_term_name))
            
        if sorted_fractions is not None:
            
            _query_params.append(('sorted_fractions', sorted_fractions))
            
        if sorted_from_id is not None:
            
            _query_params.append(('sorted_from.@id', sorted_from_id))
            
        if sorted_from_accession is not None:
            
            _query_params.append(('sorted_from.accession', sorted_from_accession))
            
        if sorted_from_detail is not None:
            
            _query_params.append(('sorted_from_detail', sorted_from_detail))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if starting_amount is not None:
            
            _query_params.append(('starting_amount', starting_amount))
            
        if starting_amount_units is not None:
            
            _query_params.append(('starting_amount_units', starting_amount_units))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if time_post_library_delivery is not None:
            
            _query_params.append(('time_post_library_delivery', time_post_library_delivery))
            
        if time_post_library_delivery_units is not None:
            
            _query_params.append(('time_post_library_delivery_units', time_post_library_delivery_units))
            
        if treatments_id is not None:
            
            _query_params.append(('treatments.@id', treatments_id))
            
        if treatments_purpose is not None:
            
            _query_params.append(('treatments.purpose', treatments_purpose))
            
        if treatments_status is not None:
            
            _query_params.append(('treatments.status', treatments_status))
            
        if treatments_summary is not None:
            
            _query_params.append(('treatments.summary', treatments_summary))
            
        if treatments_treatment_type is not None:
            
            _query_params.append(('treatments.treatment_type', treatments_treatment_type))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/multiplexed-samples/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def open_reading_frames(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award: Annotated[Optional[List[StrictStr]], Field(description="Filter by award")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by genes.@id")] = None,
        genes_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by genes.geneid")] = None,
        genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by genes.symbol")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        orf_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_id")] = None,
        pct_coverage_orf: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_coverage_orf")] = None,
        pct_coverage_protein: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_coverage_protein")] = None,
        pct_identical_protein: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_identical_protein")] = None,
        protein_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protein_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> OpenReadingFrameResults:
        """List items in the OpenReadingFrame collection.

        Collection endpoint that accepts various query parameters to filter and sort OpenReadingFrame items. Supports filtering on fields within OpenReadingFrame items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award: Filter by award
        :type award: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param genes_id: Filter by genes.@id
        :type genes_id: List[str]
        :param genes_geneid: Filter by genes.geneid
        :type genes_geneid: List[str]
        :param genes_symbol: Filter by genes.symbol
        :type genes_symbol: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param orf_id: Filter by orf_id
        :type orf_id: List[str]
        :param pct_coverage_orf: Filter by pct_coverage_orf
        :type pct_coverage_orf: List[float]
        :param pct_coverage_protein: Filter by pct_coverage_protein
        :type pct_coverage_protein: List[float]
        :param pct_identical_protein: Filter by pct_identical_protein
        :type pct_identical_protein: List[float]
        :param protein_id: Filter by protein_id
        :type protein_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by: Filter by submitted_by
        :type submitted_by: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._open_reading_frames_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award=award,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            genes_id=genes_id,
            genes_geneid=genes_geneid,
            genes_symbol=genes_symbol,
            lab=lab,
            notes=notes,
            orf_id=orf_id,
            pct_coverage_orf=pct_coverage_orf,
            pct_coverage_protein=pct_coverage_protein,
            pct_identical_protein=pct_identical_protein,
            protein_id=protein_id,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by=submitted_by,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "OpenReadingFrameResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def open_reading_frames_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award: Annotated[Optional[List[StrictStr]], Field(description="Filter by award")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by genes.@id")] = None,
        genes_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by genes.geneid")] = None,
        genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by genes.symbol")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        orf_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_id")] = None,
        pct_coverage_orf: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_coverage_orf")] = None,
        pct_coverage_protein: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_coverage_protein")] = None,
        pct_identical_protein: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_identical_protein")] = None,
        protein_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protein_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[OpenReadingFrameResults]:
        """List items in the OpenReadingFrame collection.

        Collection endpoint that accepts various query parameters to filter and sort OpenReadingFrame items. Supports filtering on fields within OpenReadingFrame items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award: Filter by award
        :type award: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param genes_id: Filter by genes.@id
        :type genes_id: List[str]
        :param genes_geneid: Filter by genes.geneid
        :type genes_geneid: List[str]
        :param genes_symbol: Filter by genes.symbol
        :type genes_symbol: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param orf_id: Filter by orf_id
        :type orf_id: List[str]
        :param pct_coverage_orf: Filter by pct_coverage_orf
        :type pct_coverage_orf: List[float]
        :param pct_coverage_protein: Filter by pct_coverage_protein
        :type pct_coverage_protein: List[float]
        :param pct_identical_protein: Filter by pct_identical_protein
        :type pct_identical_protein: List[float]
        :param protein_id: Filter by protein_id
        :type protein_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by: Filter by submitted_by
        :type submitted_by: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._open_reading_frames_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award=award,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            genes_id=genes_id,
            genes_geneid=genes_geneid,
            genes_symbol=genes_symbol,
            lab=lab,
            notes=notes,
            orf_id=orf_id,
            pct_coverage_orf=pct_coverage_orf,
            pct_coverage_protein=pct_coverage_protein,
            pct_identical_protein=pct_identical_protein,
            protein_id=protein_id,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by=submitted_by,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "OpenReadingFrameResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def open_reading_frames_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award: Annotated[Optional[List[StrictStr]], Field(description="Filter by award")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        genes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by genes.@id")] = None,
        genes_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by genes.geneid")] = None,
        genes_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by genes.symbol")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        orf_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by orf_id")] = None,
        pct_coverage_orf: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_coverage_orf")] = None,
        pct_coverage_protein: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_coverage_protein")] = None,
        pct_identical_protein: Annotated[Optional[List[Union[Annotated[float, Field(le=100, strict=True, ge=0)], Annotated[int, Field(le=100, strict=True, ge=0)]]]], Field(description="Filter by pct_identical_protein")] = None,
        protein_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protein_id")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the OpenReadingFrame collection.

        Collection endpoint that accepts various query parameters to filter and sort OpenReadingFrame items. Supports filtering on fields within OpenReadingFrame items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award: Filter by award
        :type award: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param genes_id: Filter by genes.@id
        :type genes_id: List[str]
        :param genes_geneid: Filter by genes.geneid
        :type genes_geneid: List[str]
        :param genes_symbol: Filter by genes.symbol
        :type genes_symbol: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param orf_id: Filter by orf_id
        :type orf_id: List[str]
        :param pct_coverage_orf: Filter by pct_coverage_orf
        :type pct_coverage_orf: List[float]
        :param pct_coverage_protein: Filter by pct_coverage_protein
        :type pct_coverage_protein: List[float]
        :param pct_identical_protein: Filter by pct_identical_protein
        :type pct_identical_protein: List[float]
        :param protein_id: Filter by protein_id
        :type protein_id: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by: Filter by submitted_by
        :type submitted_by: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._open_reading_frames_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award=award,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            genes_id=genes_id,
            genes_geneid=genes_geneid,
            genes_symbol=genes_symbol,
            lab=lab,
            notes=notes,
            orf_id=orf_id,
            pct_coverage_orf=pct_coverage_orf,
            pct_coverage_protein=pct_coverage_protein,
            pct_identical_protein=pct_identical_protein,
            protein_id=protein_id,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by=submitted_by,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "OpenReadingFrameResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _open_reading_frames_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award,
        creation_timestamp,
        dbxrefs,
        description,
        genes_id,
        genes_geneid,
        genes_symbol,
        lab,
        notes,
        orf_id,
        pct_coverage_orf,
        pct_coverage_protein,
        pct_identical_protein,
        protein_id,
        release_timestamp,
        status,
        submitted_by,
        submitter_comment,
        summary,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'genes.@id': 'multi',
            'genes.geneid': 'multi',
            'genes.symbol': 'multi',
            'lab': 'multi',
            'notes': 'multi',
            'orf_id': 'multi',
            'pct_coverage_orf': 'multi',
            'pct_coverage_protein': 'multi',
            'pct_identical_protein': 'multi',
            'protein_id': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award is not None:
            
            _query_params.append(('award', award))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if genes_id is not None:
            
            _query_params.append(('genes.@id', genes_id))
            
        if genes_geneid is not None:
            
            _query_params.append(('genes.geneid', genes_geneid))
            
        if genes_symbol is not None:
            
            _query_params.append(('genes.symbol', genes_symbol))
            
        if lab is not None:
            
            _query_params.append(('lab', lab))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if orf_id is not None:
            
            _query_params.append(('orf_id', orf_id))
            
        if pct_coverage_orf is not None:
            
            _query_params.append(('pct_coverage_orf', pct_coverage_orf))
            
        if pct_coverage_protein is not None:
            
            _query_params.append(('pct_coverage_protein', pct_coverage_protein))
            
        if pct_identical_protein is not None:
            
            _query_params.append(('pct_identical_protein', pct_identical_protein))
            
        if protein_id is not None:
            
            _query_params.append(('protein_id', protein_id))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by is not None:
            
            _query_params.append(('submitted_by', submitted_by))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/open-reading-frames/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def pages(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award: Annotated[Optional[List[StrictStr]], Field(description="Filter by award")] = None,
        canonical_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by canonical_uri")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        parent: Annotated[Optional[List[StrictStr]], Field(description="Filter by parent")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PageResults:
        """List items in the Page collection.

        Collection endpoint that accepts various query parameters to filter and sort Page items. Supports filtering on fields within Page items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award: Filter by award
        :type award: List[str]
        :param canonical_uri: Filter by canonical_uri
        :type canonical_uri: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param parent: Filter by parent
        :type parent: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._pages_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award=award,
            canonical_uri=canonical_uri,
            creation_timestamp=creation_timestamp,
            description=description,
            lab=lab,
            name=name,
            notes=notes,
            parent=parent,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PageResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def pages_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award: Annotated[Optional[List[StrictStr]], Field(description="Filter by award")] = None,
        canonical_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by canonical_uri")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        parent: Annotated[Optional[List[StrictStr]], Field(description="Filter by parent")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PageResults]:
        """List items in the Page collection.

        Collection endpoint that accepts various query parameters to filter and sort Page items. Supports filtering on fields within Page items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award: Filter by award
        :type award: List[str]
        :param canonical_uri: Filter by canonical_uri
        :type canonical_uri: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param parent: Filter by parent
        :type parent: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._pages_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award=award,
            canonical_uri=canonical_uri,
            creation_timestamp=creation_timestamp,
            description=description,
            lab=lab,
            name=name,
            notes=notes,
            parent=parent,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PageResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def pages_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award: Annotated[Optional[List[StrictStr]], Field(description="Filter by award")] = None,
        canonical_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by canonical_uri")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        parent: Annotated[Optional[List[StrictStr]], Field(description="Filter by parent")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Page collection.

        Collection endpoint that accepts various query parameters to filter and sort Page items. Supports filtering on fields within Page items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award: Filter by award
        :type award: List[str]
        :param canonical_uri: Filter by canonical_uri
        :type canonical_uri: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param parent: Filter by parent
        :type parent: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._pages_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award=award,
            canonical_uri=canonical_uri,
            creation_timestamp=creation_timestamp,
            description=description,
            lab=lab,
            name=name,
            notes=notes,
            parent=parent,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PageResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _pages_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award,
        canonical_uri,
        creation_timestamp,
        description,
        lab,
        name,
        notes,
        parent,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award': 'multi',
            'canonical_uri': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'lab': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'parent': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award is not None:
            
            _query_params.append(('award', award))
            
        if canonical_uri is not None:
            
            _query_params.append(('canonical_uri', canonical_uri))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if lab is not None:
            
            _query_params.append(('lab', lab))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if parent is not None:
            
            _query_params.append(('parent', parent))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/pages/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def phenotype_terms(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PhenotypeTermResults:
        """List items in the PhenotypeTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort PhenotypeTerm items. Supports filtering on fields within PhenotypeTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._phenotype_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PhenotypeTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def phenotype_terms_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PhenotypeTermResults]:
        """List items in the PhenotypeTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort PhenotypeTerm items. Supports filtering on fields within PhenotypeTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._phenotype_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PhenotypeTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def phenotype_terms_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the PhenotypeTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort PhenotypeTerm items. Supports filtering on fields within PhenotypeTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._phenotype_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PhenotypeTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _phenotype_terms_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        ancestors,
        creation_timestamp,
        deprecated_ntr_terms,
        description,
        is_a,
        name,
        notes,
        ontology,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        synonyms,
        term_id,
        term_name,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'ancestors': 'multi',
            'creation_timestamp': 'multi',
            'deprecated_ntr_terms': 'multi',
            'description': 'multi',
            'is_a': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'ontology': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'synonyms': 'multi',
            'term_id': 'multi',
            'term_name': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if ancestors is not None:
            
            _query_params.append(('ancestors', ancestors))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if deprecated_ntr_terms is not None:
            
            _query_params.append(('deprecated_ntr_terms', deprecated_ntr_terms))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if is_a is not None:
            
            _query_params.append(('is_a', is_a))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if ontology is not None:
            
            _query_params.append(('ontology', ontology))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if synonyms is not None:
            
            _query_params.append(('synonyms', synonyms))
            
        if term_id is not None:
            
            _query_params.append(('term_id', term_id))
            
        if term_name is not None:
            
            _query_params.append(('term_name', term_name))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/phenotype-terms/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def phenotypic_features(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by feature.@id")] = None,
        feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by feature.term_id")] = None,
        feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by feature.term_name")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by observation_date")] = None,
        quality: Annotated[Optional[List[StrictStr]], Field(description="Filter by quality")] = None,
        quantity: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by quantity")] = None,
        quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by quantity_units")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PhenotypicFeatureResults:
        """List items in the PhenotypicFeature collection.

        Collection endpoint that accepts various query parameters to filter and sort PhenotypicFeature items. Supports filtering on fields within PhenotypicFeature items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param feature_id: Filter by feature.@id
        :type feature_id: List[str]
        :param feature_term_id: Filter by feature.term_id
        :type feature_term_id: List[str]
        :param feature_term_name: Filter by feature.term_name
        :type feature_term_name: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param observation_date: Filter by observation_date
        :type observation_date: List[str]
        :param quality: Filter by quality
        :type quality: List[str]
        :param quantity: Filter by quantity
        :type quantity: List[float]
        :param quantity_units: Filter by quantity_units
        :type quantity_units: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._phenotypic_features_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            feature_id=feature_id,
            feature_term_id=feature_term_id,
            feature_term_name=feature_term_name,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            observation_date=observation_date,
            quality=quality,
            quantity=quantity,
            quantity_units=quantity_units,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PhenotypicFeatureResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def phenotypic_features_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by feature.@id")] = None,
        feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by feature.term_id")] = None,
        feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by feature.term_name")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by observation_date")] = None,
        quality: Annotated[Optional[List[StrictStr]], Field(description="Filter by quality")] = None,
        quantity: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by quantity")] = None,
        quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by quantity_units")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PhenotypicFeatureResults]:
        """List items in the PhenotypicFeature collection.

        Collection endpoint that accepts various query parameters to filter and sort PhenotypicFeature items. Supports filtering on fields within PhenotypicFeature items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param feature_id: Filter by feature.@id
        :type feature_id: List[str]
        :param feature_term_id: Filter by feature.term_id
        :type feature_term_id: List[str]
        :param feature_term_name: Filter by feature.term_name
        :type feature_term_name: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param observation_date: Filter by observation_date
        :type observation_date: List[str]
        :param quality: Filter by quality
        :type quality: List[str]
        :param quantity: Filter by quantity
        :type quantity: List[float]
        :param quantity_units: Filter by quantity_units
        :type quantity_units: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._phenotypic_features_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            feature_id=feature_id,
            feature_term_id=feature_term_id,
            feature_term_name=feature_term_name,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            observation_date=observation_date,
            quality=quality,
            quantity=quantity,
            quantity_units=quantity_units,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PhenotypicFeatureResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def phenotypic_features_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by feature.@id")] = None,
        feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by feature.term_id")] = None,
        feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by feature.term_name")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by observation_date")] = None,
        quality: Annotated[Optional[List[StrictStr]], Field(description="Filter by quality")] = None,
        quantity: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by quantity")] = None,
        quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by quantity_units")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the PhenotypicFeature collection.

        Collection endpoint that accepts various query parameters to filter and sort PhenotypicFeature items. Supports filtering on fields within PhenotypicFeature items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param feature_id: Filter by feature.@id
        :type feature_id: List[str]
        :param feature_term_id: Filter by feature.term_id
        :type feature_term_id: List[str]
        :param feature_term_name: Filter by feature.term_name
        :type feature_term_name: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param observation_date: Filter by observation_date
        :type observation_date: List[str]
        :param quality: Filter by quality
        :type quality: List[str]
        :param quantity: Filter by quantity
        :type quantity: List[float]
        :param quantity_units: Filter by quantity_units
        :type quantity_units: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._phenotypic_features_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            feature_id=feature_id,
            feature_term_id=feature_term_id,
            feature_term_name=feature_term_name,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            observation_date=observation_date,
            quality=quality,
            quantity=quantity,
            quantity_units=quantity_units,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PhenotypicFeatureResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _phenotypic_features_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award_id,
        award_component,
        creation_timestamp,
        description,
        feature_id,
        feature_term_id,
        feature_term_name,
        lab_id,
        lab_title,
        notes,
        observation_date,
        quality,
        quantity,
        quantity_units,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'feature.@id': 'multi',
            'feature.term_id': 'multi',
            'feature.term_name': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'observation_date': 'multi',
            'quality': 'multi',
            'quantity': 'multi',
            'quantity_units': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if feature_id is not None:
            
            _query_params.append(('feature.@id', feature_id))
            
        if feature_term_id is not None:
            
            _query_params.append(('feature.term_id', feature_term_id))
            
        if feature_term_name is not None:
            
            _query_params.append(('feature.term_name', feature_term_name))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if observation_date is not None:
            
            _query_params.append(('observation_date', observation_date))
            
        if quality is not None:
            
            _query_params.append(('quality', quality))
            
        if quantity is not None:
            
            _query_params.append(('quantity', quantity))
            
        if quantity_units is not None:
            
            _query_params.append(('quantity_units', quantity_units))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/phenotypic-features/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def platform_terms(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        company: Annotated[Optional[List[StrictStr]], Field(description="Filter by company")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sequencing_kits: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_kits")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PlatformTermResults:
        """List items in the PlatformTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort PlatformTerm items. Supports filtering on fields within PlatformTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param company: Filter by company
        :type company: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sequencing_kits: Filter by sequencing_kits
        :type sequencing_kits: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._platform_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            company=company,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            release_timestamp=release_timestamp,
            sequencing_kits=sequencing_kits,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PlatformTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def platform_terms_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        company: Annotated[Optional[List[StrictStr]], Field(description="Filter by company")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sequencing_kits: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_kits")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PlatformTermResults]:
        """List items in the PlatformTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort PlatformTerm items. Supports filtering on fields within PlatformTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param company: Filter by company
        :type company: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sequencing_kits: Filter by sequencing_kits
        :type sequencing_kits: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._platform_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            company=company,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            release_timestamp=release_timestamp,
            sequencing_kits=sequencing_kits,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PlatformTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def platform_terms_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        company: Annotated[Optional[List[StrictStr]], Field(description="Filter by company")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sequencing_kits: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_kits")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the PlatformTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort PlatformTerm items. Supports filtering on fields within PlatformTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param company: Filter by company
        :type company: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sequencing_kits: Filter by sequencing_kits
        :type sequencing_kits: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._platform_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            company=company,
            creation_timestamp=creation_timestamp,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            release_timestamp=release_timestamp,
            sequencing_kits=sequencing_kits,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PlatformTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _platform_terms_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        ancestors,
        company,
        creation_timestamp,
        deprecated_ntr_terms,
        description,
        is_a,
        name,
        notes,
        ontology,
        release_timestamp,
        sequencing_kits,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        synonyms,
        term_id,
        term_name,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'ancestors': 'multi',
            'company': 'multi',
            'creation_timestamp': 'multi',
            'deprecated_ntr_terms': 'multi',
            'description': 'multi',
            'is_a': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'ontology': 'multi',
            'release_timestamp': 'multi',
            'sequencing_kits': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'synonyms': 'multi',
            'term_id': 'multi',
            'term_name': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if ancestors is not None:
            
            _query_params.append(('ancestors', ancestors))
            
        if company is not None:
            
            _query_params.append(('company', company))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if deprecated_ntr_terms is not None:
            
            _query_params.append(('deprecated_ntr_terms', deprecated_ntr_terms))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if is_a is not None:
            
            _query_params.append(('is_a', is_a))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if ontology is not None:
            
            _query_params.append(('ontology', ontology))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if sequencing_kits is not None:
            
            _query_params.append(('sequencing_kits', sequencing_kits))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if synonyms is not None:
            
            _query_params.append(('synonyms', synonyms))
            
        if term_id is not None:
            
            _query_params.append(('term_id', term_id))
            
        if term_name is not None:
            
            _query_params.append(('term_name', term_name))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/platform-terms/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def prediction_sets(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assessed_genes: Annotated[Optional[List[StrictStr]], Field(description="Filter by assessed_genes")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        large_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.@id")] = None,
        large_scale_gene_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.accession")] = None,
        large_scale_gene_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_gene_list.aliases")] = None,
        large_scale_loci_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.@id")] = None,
        large_scale_loci_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.accession")] = None,
        large_scale_loci_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_loci_list.aliases")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.@id")] = None,
        samples_construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.accession")] = None,
        samples_construct_library_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.summary")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        scope: Annotated[Optional[List[StrictStr]], Field(description="Filter by scope")] = None,
        small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.@id")] = None,
        small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by small_scale_gene_list.geneid")] = None,
        small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.name")] = None,
        small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.symbol")] = None,
        small_scale_gene_list_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.synonyms")] = None,
        small_scale_loci_list: Annotated[Optional[List[Locus]], Field(description="Filter by small_scale_loci_list")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PredictionSetResults:
        """List items in the PredictionSet collection.

        Collection endpoint that accepts various query parameters to filter and sort PredictionSet items. Supports filtering on fields within PredictionSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assessed_genes: Filter by assessed_genes
        :type assessed_genes: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets: Filter by input_file_sets
        :type input_file_sets: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param large_scale_gene_list_id: Filter by large_scale_gene_list.@id
        :type large_scale_gene_list_id: List[str]
        :param large_scale_gene_list_accession: Filter by large_scale_gene_list.accession
        :type large_scale_gene_list_accession: List[str]
        :param large_scale_gene_list_aliases: Filter by large_scale_gene_list.aliases
        :type large_scale_gene_list_aliases: List[str]
        :param large_scale_loci_list_id: Filter by large_scale_loci_list.@id
        :type large_scale_loci_list_id: List[str]
        :param large_scale_loci_list_accession: Filter by large_scale_loci_list.accession
        :type large_scale_loci_list_accession: List[str]
        :param large_scale_loci_list_aliases: Filter by large_scale_loci_list.aliases
        :type large_scale_loci_list_aliases: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets_id: Filter by samples.construct_library_sets.@id
        :type samples_construct_library_sets_id: List[str]
        :param samples_construct_library_sets_accession: Filter by samples.construct_library_sets.accession
        :type samples_construct_library_sets_accession: List[str]
        :param samples_construct_library_sets_summary: Filter by samples.construct_library_sets.summary
        :type samples_construct_library_sets_summary: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param scope: Filter by scope
        :type scope: List[str]
        :param small_scale_gene_list_id: Filter by small_scale_gene_list.@id
        :type small_scale_gene_list_id: List[str]
        :param small_scale_gene_list_geneid: Filter by small_scale_gene_list.geneid
        :type small_scale_gene_list_geneid: List[str]
        :param small_scale_gene_list_name: Filter by small_scale_gene_list.name
        :type small_scale_gene_list_name: List[str]
        :param small_scale_gene_list_symbol: Filter by small_scale_gene_list.symbol
        :type small_scale_gene_list_symbol: List[str]
        :param small_scale_gene_list_synonyms: Filter by small_scale_gene_list.synonyms
        :type small_scale_gene_list_synonyms: List[str]
        :param small_scale_loci_list: Filter by small_scale_loci_list
        :type small_scale_loci_list: List[Locus]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._prediction_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assessed_genes=assessed_genes,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            input_file_sets=input_file_sets,
            lab_id=lab_id,
            lab_title=lab_title,
            large_scale_gene_list_id=large_scale_gene_list_id,
            large_scale_gene_list_accession=large_scale_gene_list_accession,
            large_scale_gene_list_aliases=large_scale_gene_list_aliases,
            large_scale_loci_list_id=large_scale_loci_list_id,
            large_scale_loci_list_accession=large_scale_loci_list_accession,
            large_scale_loci_list_aliases=large_scale_loci_list_aliases,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets_id=samples_construct_library_sets_id,
            samples_construct_library_sets_accession=samples_construct_library_sets_accession,
            samples_construct_library_sets_summary=samples_construct_library_sets_summary,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            scope=scope,
            small_scale_gene_list_id=small_scale_gene_list_id,
            small_scale_gene_list_geneid=small_scale_gene_list_geneid,
            small_scale_gene_list_name=small_scale_gene_list_name,
            small_scale_gene_list_symbol=small_scale_gene_list_symbol,
            small_scale_gene_list_synonyms=small_scale_gene_list_synonyms,
            small_scale_loci_list=small_scale_loci_list,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PredictionSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def prediction_sets_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assessed_genes: Annotated[Optional[List[StrictStr]], Field(description="Filter by assessed_genes")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        large_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.@id")] = None,
        large_scale_gene_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.accession")] = None,
        large_scale_gene_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_gene_list.aliases")] = None,
        large_scale_loci_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.@id")] = None,
        large_scale_loci_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.accession")] = None,
        large_scale_loci_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_loci_list.aliases")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.@id")] = None,
        samples_construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.accession")] = None,
        samples_construct_library_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.summary")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        scope: Annotated[Optional[List[StrictStr]], Field(description="Filter by scope")] = None,
        small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.@id")] = None,
        small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by small_scale_gene_list.geneid")] = None,
        small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.name")] = None,
        small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.symbol")] = None,
        small_scale_gene_list_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.synonyms")] = None,
        small_scale_loci_list: Annotated[Optional[List[Locus]], Field(description="Filter by small_scale_loci_list")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PredictionSetResults]:
        """List items in the PredictionSet collection.

        Collection endpoint that accepts various query parameters to filter and sort PredictionSet items. Supports filtering on fields within PredictionSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assessed_genes: Filter by assessed_genes
        :type assessed_genes: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets: Filter by input_file_sets
        :type input_file_sets: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param large_scale_gene_list_id: Filter by large_scale_gene_list.@id
        :type large_scale_gene_list_id: List[str]
        :param large_scale_gene_list_accession: Filter by large_scale_gene_list.accession
        :type large_scale_gene_list_accession: List[str]
        :param large_scale_gene_list_aliases: Filter by large_scale_gene_list.aliases
        :type large_scale_gene_list_aliases: List[str]
        :param large_scale_loci_list_id: Filter by large_scale_loci_list.@id
        :type large_scale_loci_list_id: List[str]
        :param large_scale_loci_list_accession: Filter by large_scale_loci_list.accession
        :type large_scale_loci_list_accession: List[str]
        :param large_scale_loci_list_aliases: Filter by large_scale_loci_list.aliases
        :type large_scale_loci_list_aliases: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets_id: Filter by samples.construct_library_sets.@id
        :type samples_construct_library_sets_id: List[str]
        :param samples_construct_library_sets_accession: Filter by samples.construct_library_sets.accession
        :type samples_construct_library_sets_accession: List[str]
        :param samples_construct_library_sets_summary: Filter by samples.construct_library_sets.summary
        :type samples_construct_library_sets_summary: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param scope: Filter by scope
        :type scope: List[str]
        :param small_scale_gene_list_id: Filter by small_scale_gene_list.@id
        :type small_scale_gene_list_id: List[str]
        :param small_scale_gene_list_geneid: Filter by small_scale_gene_list.geneid
        :type small_scale_gene_list_geneid: List[str]
        :param small_scale_gene_list_name: Filter by small_scale_gene_list.name
        :type small_scale_gene_list_name: List[str]
        :param small_scale_gene_list_symbol: Filter by small_scale_gene_list.symbol
        :type small_scale_gene_list_symbol: List[str]
        :param small_scale_gene_list_synonyms: Filter by small_scale_gene_list.synonyms
        :type small_scale_gene_list_synonyms: List[str]
        :param small_scale_loci_list: Filter by small_scale_loci_list
        :type small_scale_loci_list: List[Locus]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._prediction_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assessed_genes=assessed_genes,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            input_file_sets=input_file_sets,
            lab_id=lab_id,
            lab_title=lab_title,
            large_scale_gene_list_id=large_scale_gene_list_id,
            large_scale_gene_list_accession=large_scale_gene_list_accession,
            large_scale_gene_list_aliases=large_scale_gene_list_aliases,
            large_scale_loci_list_id=large_scale_loci_list_id,
            large_scale_loci_list_accession=large_scale_loci_list_accession,
            large_scale_loci_list_aliases=large_scale_loci_list_aliases,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets_id=samples_construct_library_sets_id,
            samples_construct_library_sets_accession=samples_construct_library_sets_accession,
            samples_construct_library_sets_summary=samples_construct_library_sets_summary,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            scope=scope,
            small_scale_gene_list_id=small_scale_gene_list_id,
            small_scale_gene_list_geneid=small_scale_gene_list_geneid,
            small_scale_gene_list_name=small_scale_gene_list_name,
            small_scale_gene_list_symbol=small_scale_gene_list_symbol,
            small_scale_gene_list_synonyms=small_scale_gene_list_synonyms,
            small_scale_loci_list=small_scale_loci_list,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PredictionSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def prediction_sets_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        assessed_genes: Annotated[Optional[List[StrictStr]], Field(description="Filter by assessed_genes")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        award_contact_pi_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.@id")] = None,
        award_contact_pi_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.contact_pi.title")] = None,
        award_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by award.title")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        control_for_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.@id")] = None,
        control_for_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_for.accession")] = None,
        control_for_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by control_for.aliases")] = None,
        control_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by control_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.@id")] = None,
        donors_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.accession")] = None,
        donors_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by donors.aliases")] = None,
        donors_sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.sex")] = None,
        donors_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.status")] = None,
        donors_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors.taxa")] = None,
        file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set_type")] = None,
        files_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.@id")] = None,
        files_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.accession")] = None,
        files_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by files.aliases")] = None,
        files_content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.content_type")] = None,
        files_creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.creation_timestamp")] = None,
        files_file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.file_format")] = None,
        files_file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by files.file_size")] = None,
        files_href: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.href")] = None,
        files_s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.s3_uri")] = None,
        files_sequencing_platform: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.sequencing_platform")] = None,
        files_submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.submitted_file_name")] = None,
        files_upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by files.upload_status")] = None,
        input_file_set_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_set_for")] = None,
        input_file_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_sets")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        large_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.@id")] = None,
        large_scale_gene_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_gene_list.accession")] = None,
        large_scale_gene_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_gene_list.aliases")] = None,
        large_scale_loci_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.@id")] = None,
        large_scale_loci_list_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by large_scale_loci_list.accession")] = None,
        large_scale_loci_list_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by large_scale_loci_list.aliases")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.@id")] = None,
        samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.accession")] = None,
        samples_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.aliases")] = None,
        samples_cell_fate_change_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.cell_fate_change_treatments")] = None,
        samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.classifications")] = None,
        samples_construct_library_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.@id")] = None,
        samples_construct_library_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.accession")] = None,
        samples_construct_library_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.construct_library_sets.summary")] = None,
        samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.disease_terms.@id")] = None,
        samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.disease_terms.term_name")] = None,
        samples_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.modifications")] = None,
        samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.@id")] = None,
        samples_sample_terms_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.aliases")] = None,
        samples_sample_terms_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.status")] = None,
        samples_sample_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.sample_terms.summary")] = None,
        samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.sample_terms.term_name")] = None,
        samples_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.status")] = None,
        samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.summary")] = None,
        samples_targeted_sample_term_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.targeted_sample_term.@id")] = None,
        samples_targeted_sample_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by samples.targeted_sample_term.term_name")] = None,
        samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.taxa")] = None,
        samples_treatments: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples.treatments")] = None,
        scope: Annotated[Optional[List[StrictStr]], Field(description="Filter by scope")] = None,
        small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.@id")] = None,
        small_scale_gene_list_geneid: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by small_scale_gene_list.geneid")] = None,
        small_scale_gene_list_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.name")] = None,
        small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.symbol")] = None,
        small_scale_gene_list_synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by small_scale_gene_list.synonyms")] = None,
        small_scale_loci_list: Annotated[Optional[List[Locus]], Field(description="Filter by small_scale_loci_list")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_files_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_files_timestamp")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the PredictionSet collection.

        Collection endpoint that accepts various query parameters to filter and sort PredictionSet items. Supports filtering on fields within PredictionSet items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param assessed_genes: Filter by assessed_genes
        :type assessed_genes: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param award_contact_pi_id: Filter by award.contact_pi.@id
        :type award_contact_pi_id: List[str]
        :param award_contact_pi_title: Filter by award.contact_pi.title
        :type award_contact_pi_title: List[str]
        :param award_title: Filter by award.title
        :type award_title: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param control_for_id: Filter by control_for.@id
        :type control_for_id: List[str]
        :param control_for_accession: Filter by control_for.accession
        :type control_for_accession: List[str]
        :param control_for_aliases: Filter by control_for.aliases
        :type control_for_aliases: List[str]
        :param control_type: Filter by control_type
        :type control_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors_id: Filter by donors.@id
        :type donors_id: List[str]
        :param donors_accession: Filter by donors.accession
        :type donors_accession: List[str]
        :param donors_aliases: Filter by donors.aliases
        :type donors_aliases: List[str]
        :param donors_sex: Filter by donors.sex
        :type donors_sex: List[str]
        :param donors_status: Filter by donors.status
        :type donors_status: List[str]
        :param donors_taxa: Filter by donors.taxa
        :type donors_taxa: List[str]
        :param file_set_type: Filter by file_set_type
        :type file_set_type: List[str]
        :param files_id: Filter by files.@id
        :type files_id: List[str]
        :param files_accession: Filter by files.accession
        :type files_accession: List[str]
        :param files_aliases: Filter by files.aliases
        :type files_aliases: List[str]
        :param files_content_type: Filter by files.content_type
        :type files_content_type: List[str]
        :param files_creation_timestamp: Filter by files.creation_timestamp
        :type files_creation_timestamp: List[str]
        :param files_file_format: Filter by files.file_format
        :type files_file_format: List[str]
        :param files_file_size: Filter by files.file_size
        :type files_file_size: List[int]
        :param files_href: Filter by files.href
        :type files_href: List[str]
        :param files_s3_uri: Filter by files.s3_uri
        :type files_s3_uri: List[str]
        :param files_sequencing_platform: Filter by files.sequencing_platform
        :type files_sequencing_platform: List[str]
        :param files_submitted_file_name: Filter by files.submitted_file_name
        :type files_submitted_file_name: List[str]
        :param files_upload_status: Filter by files.upload_status
        :type files_upload_status: List[str]
        :param input_file_set_for: Filter by input_file_set_for
        :type input_file_set_for: List[str]
        :param input_file_sets: Filter by input_file_sets
        :type input_file_sets: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param large_scale_gene_list_id: Filter by large_scale_gene_list.@id
        :type large_scale_gene_list_id: List[str]
        :param large_scale_gene_list_accession: Filter by large_scale_gene_list.accession
        :type large_scale_gene_list_accession: List[str]
        :param large_scale_gene_list_aliases: Filter by large_scale_gene_list.aliases
        :type large_scale_gene_list_aliases: List[str]
        :param large_scale_loci_list_id: Filter by large_scale_loci_list.@id
        :type large_scale_loci_list_id: List[str]
        :param large_scale_loci_list_accession: Filter by large_scale_loci_list.accession
        :type large_scale_loci_list_accession: List[str]
        :param large_scale_loci_list_aliases: Filter by large_scale_loci_list.aliases
        :type large_scale_loci_list_aliases: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param samples_id: Filter by samples.@id
        :type samples_id: List[str]
        :param samples_accession: Filter by samples.accession
        :type samples_accession: List[str]
        :param samples_aliases: Filter by samples.aliases
        :type samples_aliases: List[str]
        :param samples_cell_fate_change_treatments: Filter by samples.cell_fate_change_treatments
        :type samples_cell_fate_change_treatments: List[str]
        :param samples_classifications: Filter by samples.classifications
        :type samples_classifications: List[str]
        :param samples_construct_library_sets_id: Filter by samples.construct_library_sets.@id
        :type samples_construct_library_sets_id: List[str]
        :param samples_construct_library_sets_accession: Filter by samples.construct_library_sets.accession
        :type samples_construct_library_sets_accession: List[str]
        :param samples_construct_library_sets_summary: Filter by samples.construct_library_sets.summary
        :type samples_construct_library_sets_summary: List[str]
        :param samples_disease_terms_id: Filter by samples.disease_terms.@id
        :type samples_disease_terms_id: List[str]
        :param samples_disease_terms_term_name: Filter by samples.disease_terms.term_name
        :type samples_disease_terms_term_name: List[str]
        :param samples_modifications: Filter by samples.modifications
        :type samples_modifications: List[str]
        :param samples_sample_terms_id: Filter by samples.sample_terms.@id
        :type samples_sample_terms_id: List[str]
        :param samples_sample_terms_aliases: Filter by samples.sample_terms.aliases
        :type samples_sample_terms_aliases: List[str]
        :param samples_sample_terms_status: Filter by samples.sample_terms.status
        :type samples_sample_terms_status: List[str]
        :param samples_sample_terms_summary: Filter by samples.sample_terms.summary
        :type samples_sample_terms_summary: List[str]
        :param samples_sample_terms_term_name: Filter by samples.sample_terms.term_name
        :type samples_sample_terms_term_name: List[str]
        :param samples_status: Filter by samples.status
        :type samples_status: List[str]
        :param samples_summary: Filter by samples.summary
        :type samples_summary: List[str]
        :param samples_targeted_sample_term_id: Filter by samples.targeted_sample_term.@id
        :type samples_targeted_sample_term_id: List[str]
        :param samples_targeted_sample_term_term_name: Filter by samples.targeted_sample_term.term_name
        :type samples_targeted_sample_term_term_name: List[str]
        :param samples_taxa: Filter by samples.taxa
        :type samples_taxa: List[str]
        :param samples_treatments: Filter by samples.treatments
        :type samples_treatments: List[str]
        :param scope: Filter by scope
        :type scope: List[str]
        :param small_scale_gene_list_id: Filter by small_scale_gene_list.@id
        :type small_scale_gene_list_id: List[str]
        :param small_scale_gene_list_geneid: Filter by small_scale_gene_list.geneid
        :type small_scale_gene_list_geneid: List[str]
        :param small_scale_gene_list_name: Filter by small_scale_gene_list.name
        :type small_scale_gene_list_name: List[str]
        :param small_scale_gene_list_symbol: Filter by small_scale_gene_list.symbol
        :type small_scale_gene_list_symbol: List[str]
        :param small_scale_gene_list_synonyms: Filter by small_scale_gene_list.synonyms
        :type small_scale_gene_list_synonyms: List[str]
        :param small_scale_loci_list: Filter by small_scale_loci_list
        :type small_scale_loci_list: List[Locus]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_files_timestamp: Filter by submitted_files_timestamp
        :type submitted_files_timestamp: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._prediction_sets_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            assessed_genes=assessed_genes,
            award_id=award_id,
            award_component=award_component,
            award_contact_pi_id=award_contact_pi_id,
            award_contact_pi_title=award_contact_pi_title,
            award_title=award_title,
            collections=collections,
            control_for_id=control_for_id,
            control_for_accession=control_for_accession,
            control_for_aliases=control_for_aliases,
            control_type=control_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            donors_id=donors_id,
            donors_accession=donors_accession,
            donors_aliases=donors_aliases,
            donors_sex=donors_sex,
            donors_status=donors_status,
            donors_taxa=donors_taxa,
            file_set_type=file_set_type,
            files_id=files_id,
            files_accession=files_accession,
            files_aliases=files_aliases,
            files_content_type=files_content_type,
            files_creation_timestamp=files_creation_timestamp,
            files_file_format=files_file_format,
            files_file_size=files_file_size,
            files_href=files_href,
            files_s3_uri=files_s3_uri,
            files_sequencing_platform=files_sequencing_platform,
            files_submitted_file_name=files_submitted_file_name,
            files_upload_status=files_upload_status,
            input_file_set_for=input_file_set_for,
            input_file_sets=input_file_sets,
            lab_id=lab_id,
            lab_title=lab_title,
            large_scale_gene_list_id=large_scale_gene_list_id,
            large_scale_gene_list_accession=large_scale_gene_list_accession,
            large_scale_gene_list_aliases=large_scale_gene_list_aliases,
            large_scale_loci_list_id=large_scale_loci_list_id,
            large_scale_loci_list_accession=large_scale_loci_list_accession,
            large_scale_loci_list_aliases=large_scale_loci_list_aliases,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            samples_id=samples_id,
            samples_accession=samples_accession,
            samples_aliases=samples_aliases,
            samples_cell_fate_change_treatments=samples_cell_fate_change_treatments,
            samples_classifications=samples_classifications,
            samples_construct_library_sets_id=samples_construct_library_sets_id,
            samples_construct_library_sets_accession=samples_construct_library_sets_accession,
            samples_construct_library_sets_summary=samples_construct_library_sets_summary,
            samples_disease_terms_id=samples_disease_terms_id,
            samples_disease_terms_term_name=samples_disease_terms_term_name,
            samples_modifications=samples_modifications,
            samples_sample_terms_id=samples_sample_terms_id,
            samples_sample_terms_aliases=samples_sample_terms_aliases,
            samples_sample_terms_status=samples_sample_terms_status,
            samples_sample_terms_summary=samples_sample_terms_summary,
            samples_sample_terms_term_name=samples_sample_terms_term_name,
            samples_status=samples_status,
            samples_summary=samples_summary,
            samples_targeted_sample_term_id=samples_targeted_sample_term_id,
            samples_targeted_sample_term_term_name=samples_targeted_sample_term_term_name,
            samples_taxa=samples_taxa,
            samples_treatments=samples_treatments,
            scope=scope,
            small_scale_gene_list_id=small_scale_gene_list_id,
            small_scale_gene_list_geneid=small_scale_gene_list_geneid,
            small_scale_gene_list_name=small_scale_gene_list_name,
            small_scale_gene_list_symbol=small_scale_gene_list_symbol,
            small_scale_gene_list_synonyms=small_scale_gene_list_synonyms,
            small_scale_loci_list=small_scale_loci_list,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_files_timestamp=submitted_files_timestamp,
            submitter_comment=submitter_comment,
            summary=summary,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PredictionSetResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _prediction_sets_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        assessed_genes,
        award_id,
        award_component,
        award_contact_pi_id,
        award_contact_pi_title,
        award_title,
        collections,
        control_for_id,
        control_for_accession,
        control_for_aliases,
        control_type,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        donors_id,
        donors_accession,
        donors_aliases,
        donors_sex,
        donors_status,
        donors_taxa,
        file_set_type,
        files_id,
        files_accession,
        files_aliases,
        files_content_type,
        files_creation_timestamp,
        files_file_format,
        files_file_size,
        files_href,
        files_s3_uri,
        files_sequencing_platform,
        files_submitted_file_name,
        files_upload_status,
        input_file_set_for,
        input_file_sets,
        lab_id,
        lab_title,
        large_scale_gene_list_id,
        large_scale_gene_list_accession,
        large_scale_gene_list_aliases,
        large_scale_loci_list_id,
        large_scale_loci_list_accession,
        large_scale_loci_list_aliases,
        notes,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        samples_id,
        samples_accession,
        samples_aliases,
        samples_cell_fate_change_treatments,
        samples_classifications,
        samples_construct_library_sets_id,
        samples_construct_library_sets_accession,
        samples_construct_library_sets_summary,
        samples_disease_terms_id,
        samples_disease_terms_term_name,
        samples_modifications,
        samples_sample_terms_id,
        samples_sample_terms_aliases,
        samples_sample_terms_status,
        samples_sample_terms_summary,
        samples_sample_terms_term_name,
        samples_status,
        samples_summary,
        samples_targeted_sample_term_id,
        samples_targeted_sample_term_term_name,
        samples_taxa,
        samples_treatments,
        scope,
        small_scale_gene_list_id,
        small_scale_gene_list_geneid,
        small_scale_gene_list_name,
        small_scale_gene_list_symbol,
        small_scale_gene_list_synonyms,
        small_scale_loci_list,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_files_timestamp,
        submitter_comment,
        summary,
        url,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'assessed_genes': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'award.contact_pi.@id': 'multi',
            'award.contact_pi.title': 'multi',
            'award.title': 'multi',
            'collections': 'multi',
            'control_for.@id': 'multi',
            'control_for.accession': 'multi',
            'control_for.aliases': 'multi',
            'control_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'donors.@id': 'multi',
            'donors.accession': 'multi',
            'donors.aliases': 'multi',
            'donors.sex': 'multi',
            'donors.status': 'multi',
            'donors.taxa': 'multi',
            'file_set_type': 'multi',
            'files.@id': 'multi',
            'files.accession': 'multi',
            'files.aliases': 'multi',
            'files.content_type': 'multi',
            'files.creation_timestamp': 'multi',
            'files.file_format': 'multi',
            'files.file_size': 'multi',
            'files.href': 'multi',
            'files.s3_uri': 'multi',
            'files.sequencing_platform': 'multi',
            'files.submitted_file_name': 'multi',
            'files.upload_status': 'multi',
            'input_file_set_for': 'multi',
            'input_file_sets': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'large_scale_gene_list.@id': 'multi',
            'large_scale_gene_list.accession': 'multi',
            'large_scale_gene_list.aliases': 'multi',
            'large_scale_loci_list.@id': 'multi',
            'large_scale_loci_list.accession': 'multi',
            'large_scale_loci_list.aliases': 'multi',
            'notes': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'samples.@id': 'multi',
            'samples.accession': 'multi',
            'samples.aliases': 'multi',
            'samples.cell_fate_change_treatments': 'multi',
            'samples.classifications': 'multi',
            'samples.construct_library_sets.@id': 'multi',
            'samples.construct_library_sets.accession': 'multi',
            'samples.construct_library_sets.summary': 'multi',
            'samples.disease_terms.@id': 'multi',
            'samples.disease_terms.term_name': 'multi',
            'samples.modifications': 'multi',
            'samples.sample_terms.@id': 'multi',
            'samples.sample_terms.aliases': 'multi',
            'samples.sample_terms.status': 'multi',
            'samples.sample_terms.summary': 'multi',
            'samples.sample_terms.term_name': 'multi',
            'samples.status': 'multi',
            'samples.summary': 'multi',
            'samples.targeted_sample_term.@id': 'multi',
            'samples.targeted_sample_term.term_name': 'multi',
            'samples.taxa': 'multi',
            'samples.treatments': 'multi',
            'scope': 'multi',
            'small_scale_gene_list.@id': 'multi',
            'small_scale_gene_list.geneid': 'multi',
            'small_scale_gene_list.name': 'multi',
            'small_scale_gene_list.symbol': 'multi',
            'small_scale_gene_list.synonyms': 'multi',
            'small_scale_loci_list': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_files_timestamp': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'url': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if assessed_genes is not None:
            
            _query_params.append(('assessed_genes', assessed_genes))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if award_contact_pi_id is not None:
            
            _query_params.append(('award.contact_pi.@id', award_contact_pi_id))
            
        if award_contact_pi_title is not None:
            
            _query_params.append(('award.contact_pi.title', award_contact_pi_title))
            
        if award_title is not None:
            
            _query_params.append(('award.title', award_title))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if control_for_id is not None:
            
            _query_params.append(('control_for.@id', control_for_id))
            
        if control_for_accession is not None:
            
            _query_params.append(('control_for.accession', control_for_accession))
            
        if control_for_aliases is not None:
            
            _query_params.append(('control_for.aliases', control_for_aliases))
            
        if control_type is not None:
            
            _query_params.append(('control_type', control_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors_id is not None:
            
            _query_params.append(('donors.@id', donors_id))
            
        if donors_accession is not None:
            
            _query_params.append(('donors.accession', donors_accession))
            
        if donors_aliases is not None:
            
            _query_params.append(('donors.aliases', donors_aliases))
            
        if donors_sex is not None:
            
            _query_params.append(('donors.sex', donors_sex))
            
        if donors_status is not None:
            
            _query_params.append(('donors.status', donors_status))
            
        if donors_taxa is not None:
            
            _query_params.append(('donors.taxa', donors_taxa))
            
        if file_set_type is not None:
            
            _query_params.append(('file_set_type', file_set_type))
            
        if files_id is not None:
            
            _query_params.append(('files.@id', files_id))
            
        if files_accession is not None:
            
            _query_params.append(('files.accession', files_accession))
            
        if files_aliases is not None:
            
            _query_params.append(('files.aliases', files_aliases))
            
        if files_content_type is not None:
            
            _query_params.append(('files.content_type', files_content_type))
            
        if files_creation_timestamp is not None:
            
            _query_params.append(('files.creation_timestamp', files_creation_timestamp))
            
        if files_file_format is not None:
            
            _query_params.append(('files.file_format', files_file_format))
            
        if files_file_size is not None:
            
            _query_params.append(('files.file_size', files_file_size))
            
        if files_href is not None:
            
            _query_params.append(('files.href', files_href))
            
        if files_s3_uri is not None:
            
            _query_params.append(('files.s3_uri', files_s3_uri))
            
        if files_sequencing_platform is not None:
            
            _query_params.append(('files.sequencing_platform', files_sequencing_platform))
            
        if files_submitted_file_name is not None:
            
            _query_params.append(('files.submitted_file_name', files_submitted_file_name))
            
        if files_upload_status is not None:
            
            _query_params.append(('files.upload_status', files_upload_status))
            
        if input_file_set_for is not None:
            
            _query_params.append(('input_file_set_for', input_file_set_for))
            
        if input_file_sets is not None:
            
            _query_params.append(('input_file_sets', input_file_sets))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if large_scale_gene_list_id is not None:
            
            _query_params.append(('large_scale_gene_list.@id', large_scale_gene_list_id))
            
        if large_scale_gene_list_accession is not None:
            
            _query_params.append(('large_scale_gene_list.accession', large_scale_gene_list_accession))
            
        if large_scale_gene_list_aliases is not None:
            
            _query_params.append(('large_scale_gene_list.aliases', large_scale_gene_list_aliases))
            
        if large_scale_loci_list_id is not None:
            
            _query_params.append(('large_scale_loci_list.@id', large_scale_loci_list_id))
            
        if large_scale_loci_list_accession is not None:
            
            _query_params.append(('large_scale_loci_list.accession', large_scale_loci_list_accession))
            
        if large_scale_loci_list_aliases is not None:
            
            _query_params.append(('large_scale_loci_list.aliases', large_scale_loci_list_aliases))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if samples_id is not None:
            
            _query_params.append(('samples.@id', samples_id))
            
        if samples_accession is not None:
            
            _query_params.append(('samples.accession', samples_accession))
            
        if samples_aliases is not None:
            
            _query_params.append(('samples.aliases', samples_aliases))
            
        if samples_cell_fate_change_treatments is not None:
            
            _query_params.append(('samples.cell_fate_change_treatments', samples_cell_fate_change_treatments))
            
        if samples_classifications is not None:
            
            _query_params.append(('samples.classifications', samples_classifications))
            
        if samples_construct_library_sets_id is not None:
            
            _query_params.append(('samples.construct_library_sets.@id', samples_construct_library_sets_id))
            
        if samples_construct_library_sets_accession is not None:
            
            _query_params.append(('samples.construct_library_sets.accession', samples_construct_library_sets_accession))
            
        if samples_construct_library_sets_summary is not None:
            
            _query_params.append(('samples.construct_library_sets.summary', samples_construct_library_sets_summary))
            
        if samples_disease_terms_id is not None:
            
            _query_params.append(('samples.disease_terms.@id', samples_disease_terms_id))
            
        if samples_disease_terms_term_name is not None:
            
            _query_params.append(('samples.disease_terms.term_name', samples_disease_terms_term_name))
            
        if samples_modifications is not None:
            
            _query_params.append(('samples.modifications', samples_modifications))
            
        if samples_sample_terms_id is not None:
            
            _query_params.append(('samples.sample_terms.@id', samples_sample_terms_id))
            
        if samples_sample_terms_aliases is not None:
            
            _query_params.append(('samples.sample_terms.aliases', samples_sample_terms_aliases))
            
        if samples_sample_terms_status is not None:
            
            _query_params.append(('samples.sample_terms.status', samples_sample_terms_status))
            
        if samples_sample_terms_summary is not None:
            
            _query_params.append(('samples.sample_terms.summary', samples_sample_terms_summary))
            
        if samples_sample_terms_term_name is not None:
            
            _query_params.append(('samples.sample_terms.term_name', samples_sample_terms_term_name))
            
        if samples_status is not None:
            
            _query_params.append(('samples.status', samples_status))
            
        if samples_summary is not None:
            
            _query_params.append(('samples.summary', samples_summary))
            
        if samples_targeted_sample_term_id is not None:
            
            _query_params.append(('samples.targeted_sample_term.@id', samples_targeted_sample_term_id))
            
        if samples_targeted_sample_term_term_name is not None:
            
            _query_params.append(('samples.targeted_sample_term.term_name', samples_targeted_sample_term_term_name))
            
        if samples_taxa is not None:
            
            _query_params.append(('samples.taxa', samples_taxa))
            
        if samples_treatments is not None:
            
            _query_params.append(('samples.treatments', samples_treatments))
            
        if scope is not None:
            
            _query_params.append(('scope', scope))
            
        if small_scale_gene_list_id is not None:
            
            _query_params.append(('small_scale_gene_list.@id', small_scale_gene_list_id))
            
        if small_scale_gene_list_geneid is not None:
            
            _query_params.append(('small_scale_gene_list.geneid', small_scale_gene_list_geneid))
            
        if small_scale_gene_list_name is not None:
            
            _query_params.append(('small_scale_gene_list.name', small_scale_gene_list_name))
            
        if small_scale_gene_list_symbol is not None:
            
            _query_params.append(('small_scale_gene_list.symbol', small_scale_gene_list_symbol))
            
        if small_scale_gene_list_synonyms is not None:
            
            _query_params.append(('small_scale_gene_list.synonyms', small_scale_gene_list_synonyms))
            
        if small_scale_loci_list is not None:
            
            _query_params.append(('small_scale_loci_list', small_scale_loci_list))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_files_timestamp is not None:
            
            _query_params.append(('submitted_files_timestamp', submitted_files_timestamp))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/prediction-sets/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def primary_cells(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        biosample_qualifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosample_qualifiers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        passage_number: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by passage_number")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PrimaryCellResults:
        """List items in the PrimaryCell collection.

        Collection endpoint that accepts various query parameters to filter and sort PrimaryCell items. Supports filtering on fields within PrimaryCell items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param biosample_qualifiers: Filter by biosample_qualifiers
        :type biosample_qualifiers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param passage_number: Filter by passage_number
        :type passage_number: List[int]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._primary_cells_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            biosample_qualifiers=biosample_qualifiers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            passage_number=passage_number,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PrimaryCellResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def primary_cells_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        biosample_qualifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosample_qualifiers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        passage_number: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by passage_number")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PrimaryCellResults]:
        """List items in the PrimaryCell collection.

        Collection endpoint that accepts various query parameters to filter and sort PrimaryCell items. Supports filtering on fields within PrimaryCell items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param biosample_qualifiers: Filter by biosample_qualifiers
        :type biosample_qualifiers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param passage_number: Filter by passage_number
        :type passage_number: List[int]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._primary_cells_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            biosample_qualifiers=biosample_qualifiers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            passage_number=passage_number,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PrimaryCellResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def primary_cells_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        biosample_qualifiers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosample_qualifiers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        passage_number: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by passage_number")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the PrimaryCell collection.

        Collection endpoint that accepts various query parameters to filter and sort PrimaryCell items. Supports filtering on fields within PrimaryCell items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param biosample_qualifiers: Filter by biosample_qualifiers
        :type biosample_qualifiers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param passage_number: Filter by passage_number
        :type passage_number: List[int]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._primary_cells_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            biosample_qualifiers=biosample_qualifiers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            passage_number=passage_number,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PrimaryCellResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _primary_cells_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        age,
        age_units,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        biomarkers,
        biosample_qualifiers,
        cellular_sub_pool,
        classifications,
        collections,
        construct_library_sets,
        creation_timestamp,
        date_obtained,
        dbxrefs,
        description,
        disease_terms_id,
        disease_terms_term_name,
        documents,
        donors,
        embryonic,
        file_sets_id,
        file_sets_accession,
        file_sets_aliases,
        file_sets_assay_term_term_name,
        file_sets_lab_title,
        file_sets_status,
        file_sets_summary,
        institutional_certificates_id,
        institutional_certificates_certificate_identifier,
        lab_id,
        lab_title,
        lot_id,
        lower_bound_age,
        lower_bound_age_in_hours,
        modifications_id,
        modifications_status,
        modifications_summary,
        moi,
        multiplexed_in_id,
        multiplexed_in_accession,
        notes,
        nucleic_acid_delivery,
        origin_of,
        originated_from,
        part_of,
        parts,
        passage_number,
        pooled_from,
        pooled_in,
        product_id,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_terms_id,
        sample_terms_term_name,
        sex,
        sorted_fractions,
        sorted_from_id,
        sorted_from_accession,
        sorted_from_detail,
        sources_id,
        starting_amount,
        starting_amount_units,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        taxa,
        time_post_library_delivery,
        time_post_library_delivery_units,
        treatments_id,
        treatments_purpose,
        treatments_status,
        treatments_summary,
        treatments_treatment_type,
        upper_bound_age,
        upper_bound_age_in_hours,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'age': 'multi',
            'age_units': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biomarkers': 'multi',
            'biosample_qualifiers': 'multi',
            'cellular_sub_pool': 'multi',
            'classifications': 'multi',
            'collections': 'multi',
            'construct_library_sets': 'multi',
            'creation_timestamp': 'multi',
            'date_obtained': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'disease_terms.@id': 'multi',
            'disease_terms.term_name': 'multi',
            'documents': 'multi',
            'donors': 'multi',
            'embryonic': 'multi',
            'file_sets.@id': 'multi',
            'file_sets.accession': 'multi',
            'file_sets.aliases': 'multi',
            'file_sets.assay_term.term_name': 'multi',
            'file_sets.lab.title': 'multi',
            'file_sets.status': 'multi',
            'file_sets.summary': 'multi',
            'institutional_certificates.@id': 'multi',
            'institutional_certificates.certificate_identifier': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'lower_bound_age': 'multi',
            'lower_bound_age_in_hours': 'multi',
            'modifications.@id': 'multi',
            'modifications.status': 'multi',
            'modifications.summary': 'multi',
            'moi': 'multi',
            'multiplexed_in.@id': 'multi',
            'multiplexed_in.accession': 'multi',
            'notes': 'multi',
            'nucleic_acid_delivery': 'multi',
            'origin_of': 'multi',
            'originated_from': 'multi',
            'part_of': 'multi',
            'parts': 'multi',
            'passage_number': 'multi',
            'pooled_from': 'multi',
            'pooled_in': 'multi',
            'product_id': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_terms.@id': 'multi',
            'sample_terms.term_name': 'multi',
            'sex': 'multi',
            'sorted_fractions': 'multi',
            'sorted_from.@id': 'multi',
            'sorted_from.accession': 'multi',
            'sorted_from_detail': 'multi',
            'sources.@id': 'multi',
            'starting_amount': 'multi',
            'starting_amount_units': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'time_post_library_delivery': 'multi',
            'time_post_library_delivery_units': 'multi',
            'treatments.@id': 'multi',
            'treatments.purpose': 'multi',
            'treatments.status': 'multi',
            'treatments.summary': 'multi',
            'treatments.treatment_type': 'multi',
            'upper_bound_age': 'multi',
            'upper_bound_age_in_hours': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if age is not None:
            
            _query_params.append(('age', age))
            
        if age_units is not None:
            
            _query_params.append(('age_units', age_units))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biomarkers is not None:
            
            _query_params.append(('biomarkers', biomarkers))
            
        if biosample_qualifiers is not None:
            
            _query_params.append(('biosample_qualifiers', biosample_qualifiers))
            
        if cellular_sub_pool is not None:
            
            _query_params.append(('cellular_sub_pool', cellular_sub_pool))
            
        if classifications is not None:
            
            _query_params.append(('classifications', classifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if construct_library_sets is not None:
            
            _query_params.append(('construct_library_sets', construct_library_sets))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_obtained is not None:
            
            _query_params.append(('date_obtained', date_obtained))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if disease_terms_id is not None:
            
            _query_params.append(('disease_terms.@id', disease_terms_id))
            
        if disease_terms_term_name is not None:
            
            _query_params.append(('disease_terms.term_name', disease_terms_term_name))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors is not None:
            
            _query_params.append(('donors', donors))
            
        if embryonic is not None:
            
            _query_params.append(('embryonic', embryonic))
            
        if file_sets_id is not None:
            
            _query_params.append(('file_sets.@id', file_sets_id))
            
        if file_sets_accession is not None:
            
            _query_params.append(('file_sets.accession', file_sets_accession))
            
        if file_sets_aliases is not None:
            
            _query_params.append(('file_sets.aliases', file_sets_aliases))
            
        if file_sets_assay_term_term_name is not None:
            
            _query_params.append(('file_sets.assay_term.term_name', file_sets_assay_term_term_name))
            
        if file_sets_lab_title is not None:
            
            _query_params.append(('file_sets.lab.title', file_sets_lab_title))
            
        if file_sets_status is not None:
            
            _query_params.append(('file_sets.status', file_sets_status))
            
        if file_sets_summary is not None:
            
            _query_params.append(('file_sets.summary', file_sets_summary))
            
        if institutional_certificates_id is not None:
            
            _query_params.append(('institutional_certificates.@id', institutional_certificates_id))
            
        if institutional_certificates_certificate_identifier is not None:
            
            _query_params.append(('institutional_certificates.certificate_identifier', institutional_certificates_certificate_identifier))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if lower_bound_age is not None:
            
            _query_params.append(('lower_bound_age', lower_bound_age))
            
        if lower_bound_age_in_hours is not None:
            
            _query_params.append(('lower_bound_age_in_hours', lower_bound_age_in_hours))
            
        if modifications_id is not None:
            
            _query_params.append(('modifications.@id', modifications_id))
            
        if modifications_status is not None:
            
            _query_params.append(('modifications.status', modifications_status))
            
        if modifications_summary is not None:
            
            _query_params.append(('modifications.summary', modifications_summary))
            
        if moi is not None:
            
            _query_params.append(('moi', moi))
            
        if multiplexed_in_id is not None:
            
            _query_params.append(('multiplexed_in.@id', multiplexed_in_id))
            
        if multiplexed_in_accession is not None:
            
            _query_params.append(('multiplexed_in.accession', multiplexed_in_accession))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if nucleic_acid_delivery is not None:
            
            _query_params.append(('nucleic_acid_delivery', nucleic_acid_delivery))
            
        if origin_of is not None:
            
            _query_params.append(('origin_of', origin_of))
            
        if originated_from is not None:
            
            _query_params.append(('originated_from', originated_from))
            
        if part_of is not None:
            
            _query_params.append(('part_of', part_of))
            
        if parts is not None:
            
            _query_params.append(('parts', parts))
            
        if passage_number is not None:
            
            _query_params.append(('passage_number', passage_number))
            
        if pooled_from is not None:
            
            _query_params.append(('pooled_from', pooled_from))
            
        if pooled_in is not None:
            
            _query_params.append(('pooled_in', pooled_in))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_terms_id is not None:
            
            _query_params.append(('sample_terms.@id', sample_terms_id))
            
        if sample_terms_term_name is not None:
            
            _query_params.append(('sample_terms.term_name', sample_terms_term_name))
            
        if sex is not None:
            
            _query_params.append(('sex', sex))
            
        if sorted_fractions is not None:
            
            _query_params.append(('sorted_fractions', sorted_fractions))
            
        if sorted_from_id is not None:
            
            _query_params.append(('sorted_from.@id', sorted_from_id))
            
        if sorted_from_accession is not None:
            
            _query_params.append(('sorted_from.accession', sorted_from_accession))
            
        if sorted_from_detail is not None:
            
            _query_params.append(('sorted_from_detail', sorted_from_detail))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if starting_amount is not None:
            
            _query_params.append(('starting_amount', starting_amount))
            
        if starting_amount_units is not None:
            
            _query_params.append(('starting_amount_units', starting_amount_units))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if time_post_library_delivery is not None:
            
            _query_params.append(('time_post_library_delivery', time_post_library_delivery))
            
        if time_post_library_delivery_units is not None:
            
            _query_params.append(('time_post_library_delivery_units', time_post_library_delivery_units))
            
        if treatments_id is not None:
            
            _query_params.append(('treatments.@id', treatments_id))
            
        if treatments_purpose is not None:
            
            _query_params.append(('treatments.purpose', treatments_purpose))
            
        if treatments_status is not None:
            
            _query_params.append(('treatments.status', treatments_status))
            
        if treatments_summary is not None:
            
            _query_params.append(('treatments.summary', treatments_summary))
            
        if treatments_treatment_type is not None:
            
            _query_params.append(('treatments.treatment_type', treatments_treatment_type))
            
        if upper_bound_age is not None:
            
            _query_params.append(('upper_bound_age', upper_bound_age))
            
        if upper_bound_age_in_hours is not None:
            
            _query_params.append(('upper_bound_age_in_hours', upper_bound_age_in_hours))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/primary-cells/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def publications(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        abstract: Annotated[Optional[List[StrictStr]], Field(description="Filter by abstract")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        authors: Annotated[Optional[List[StrictStr]], Field(description="Filter by authors")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_published: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_published")] = None,
        date_revised: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_revised")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        file_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets")] = None,
        issue: Annotated[Optional[List[StrictStr]], Field(description="Filter by issue")] = None,
        journal: Annotated[Optional[List[StrictStr]], Field(description="Filter by journal")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        page: Annotated[Optional[List[StrictStr]], Field(description="Filter by page")] = None,
        publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publication_identifiers")] = None,
        publication_year: Annotated[Optional[List[StrictInt]], Field(description="Filter by publication_year")] = None,
        published_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by published_by")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        samples: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples")] = None,
        software: Annotated[Optional[List[StrictStr]], Field(description="Filter by software")] = None,
        software_versions: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        volume: Annotated[Optional[List[StrictStr]], Field(description="Filter by volume")] = None,
        workflows: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflows")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> PublicationResults:
        """List items in the Publication collection.

        Collection endpoint that accepts various query parameters to filter and sort Publication items. Supports filtering on fields within Publication items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param abstract: Filter by abstract
        :type abstract: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param authors: Filter by authors
        :type authors: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_published: Filter by date_published
        :type date_published: List[str]
        :param date_revised: Filter by date_revised
        :type date_revised: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param file_sets: Filter by file_sets
        :type file_sets: List[str]
        :param issue: Filter by issue
        :type issue: List[str]
        :param journal: Filter by journal
        :type journal: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param page: Filter by page
        :type page: List[str]
        :param publication_identifiers: Filter by publication_identifiers
        :type publication_identifiers: List[str]
        :param publication_year: Filter by publication_year
        :type publication_year: List[int]
        :param published_by: Filter by published_by
        :type published_by: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param samples: Filter by samples
        :type samples: List[str]
        :param software: Filter by software
        :type software: List[str]
        :param software_versions: Filter by software_versions
        :type software_versions: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param volume: Filter by volume
        :type volume: List[str]
        :param workflows: Filter by workflows
        :type workflows: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._publications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            abstract=abstract,
            aliases=aliases,
            authors=authors,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            date_published=date_published,
            date_revised=date_revised,
            description=description,
            donors=donors,
            file_sets=file_sets,
            issue=issue,
            journal=journal,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            page=page,
            publication_identifiers=publication_identifiers,
            publication_year=publication_year,
            published_by=published_by,
            release_timestamp=release_timestamp,
            samples=samples,
            software=software,
            software_versions=software_versions,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            volume=volume,
            workflows=workflows,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PublicationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def publications_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        abstract: Annotated[Optional[List[StrictStr]], Field(description="Filter by abstract")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        authors: Annotated[Optional[List[StrictStr]], Field(description="Filter by authors")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_published: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_published")] = None,
        date_revised: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_revised")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        file_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets")] = None,
        issue: Annotated[Optional[List[StrictStr]], Field(description="Filter by issue")] = None,
        journal: Annotated[Optional[List[StrictStr]], Field(description="Filter by journal")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        page: Annotated[Optional[List[StrictStr]], Field(description="Filter by page")] = None,
        publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publication_identifiers")] = None,
        publication_year: Annotated[Optional[List[StrictInt]], Field(description="Filter by publication_year")] = None,
        published_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by published_by")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        samples: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples")] = None,
        software: Annotated[Optional[List[StrictStr]], Field(description="Filter by software")] = None,
        software_versions: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        volume: Annotated[Optional[List[StrictStr]], Field(description="Filter by volume")] = None,
        workflows: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflows")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[PublicationResults]:
        """List items in the Publication collection.

        Collection endpoint that accepts various query parameters to filter and sort Publication items. Supports filtering on fields within Publication items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param abstract: Filter by abstract
        :type abstract: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param authors: Filter by authors
        :type authors: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_published: Filter by date_published
        :type date_published: List[str]
        :param date_revised: Filter by date_revised
        :type date_revised: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param file_sets: Filter by file_sets
        :type file_sets: List[str]
        :param issue: Filter by issue
        :type issue: List[str]
        :param journal: Filter by journal
        :type journal: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param page: Filter by page
        :type page: List[str]
        :param publication_identifiers: Filter by publication_identifiers
        :type publication_identifiers: List[str]
        :param publication_year: Filter by publication_year
        :type publication_year: List[int]
        :param published_by: Filter by published_by
        :type published_by: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param samples: Filter by samples
        :type samples: List[str]
        :param software: Filter by software
        :type software: List[str]
        :param software_versions: Filter by software_versions
        :type software_versions: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param volume: Filter by volume
        :type volume: List[str]
        :param workflows: Filter by workflows
        :type workflows: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._publications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            abstract=abstract,
            aliases=aliases,
            authors=authors,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            date_published=date_published,
            date_revised=date_revised,
            description=description,
            donors=donors,
            file_sets=file_sets,
            issue=issue,
            journal=journal,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            page=page,
            publication_identifiers=publication_identifiers,
            publication_year=publication_year,
            published_by=published_by,
            release_timestamp=release_timestamp,
            samples=samples,
            software=software,
            software_versions=software_versions,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            volume=volume,
            workflows=workflows,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PublicationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def publications_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        abstract: Annotated[Optional[List[StrictStr]], Field(description="Filter by abstract")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        authors: Annotated[Optional[List[StrictStr]], Field(description="Filter by authors")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_published: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_published")] = None,
        date_revised: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_revised")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        file_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets")] = None,
        issue: Annotated[Optional[List[StrictStr]], Field(description="Filter by issue")] = None,
        journal: Annotated[Optional[List[StrictStr]], Field(description="Filter by journal")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        page: Annotated[Optional[List[StrictStr]], Field(description="Filter by page")] = None,
        publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publication_identifiers")] = None,
        publication_year: Annotated[Optional[List[StrictInt]], Field(description="Filter by publication_year")] = None,
        published_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by published_by")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        samples: Annotated[Optional[List[StrictStr]], Field(description="Filter by samples")] = None,
        software: Annotated[Optional[List[StrictStr]], Field(description="Filter by software")] = None,
        software_versions: Annotated[Optional[List[StrictStr]], Field(description="Filter by software_versions")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        volume: Annotated[Optional[List[StrictStr]], Field(description="Filter by volume")] = None,
        workflows: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflows")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Publication collection.

        Collection endpoint that accepts various query parameters to filter and sort Publication items. Supports filtering on fields within Publication items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param abstract: Filter by abstract
        :type abstract: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param authors: Filter by authors
        :type authors: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_published: Filter by date_published
        :type date_published: List[str]
        :param date_revised: Filter by date_revised
        :type date_revised: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param file_sets: Filter by file_sets
        :type file_sets: List[str]
        :param issue: Filter by issue
        :type issue: List[str]
        :param journal: Filter by journal
        :type journal: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param page: Filter by page
        :type page: List[str]
        :param publication_identifiers: Filter by publication_identifiers
        :type publication_identifiers: List[str]
        :param publication_year: Filter by publication_year
        :type publication_year: List[int]
        :param published_by: Filter by published_by
        :type published_by: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param samples: Filter by samples
        :type samples: List[str]
        :param software: Filter by software
        :type software: List[str]
        :param software_versions: Filter by software_versions
        :type software_versions: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param volume: Filter by volume
        :type volume: List[str]
        :param workflows: Filter by workflows
        :type workflows: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._publications_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            abstract=abstract,
            aliases=aliases,
            authors=authors,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            date_published=date_published,
            date_revised=date_revised,
            description=description,
            donors=donors,
            file_sets=file_sets,
            issue=issue,
            journal=journal,
            lab_id=lab_id,
            lab_title=lab_title,
            notes=notes,
            page=page,
            publication_identifiers=publication_identifiers,
            publication_year=publication_year,
            published_by=published_by,
            release_timestamp=release_timestamp,
            samples=samples,
            software=software,
            software_versions=software_versions,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            volume=volume,
            workflows=workflows,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "PublicationResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _publications_serialize(
        self,
        query,
        limit,
        sort,
        id,
        abstract,
        aliases,
        authors,
        award_id,
        award_component,
        creation_timestamp,
        date_published,
        date_revised,
        description,
        donors,
        file_sets,
        issue,
        journal,
        lab_id,
        lab_title,
        notes,
        page,
        publication_identifiers,
        publication_year,
        published_by,
        release_timestamp,
        samples,
        software,
        software_versions,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        uuid,
        volume,
        workflows,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'abstract': 'multi',
            'aliases': 'multi',
            'authors': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'creation_timestamp': 'multi',
            'date_published': 'multi',
            'date_revised': 'multi',
            'description': 'multi',
            'donors': 'multi',
            'file_sets': 'multi',
            'issue': 'multi',
            'journal': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'notes': 'multi',
            'page': 'multi',
            'publication_identifiers': 'multi',
            'publication_year': 'multi',
            'published_by': 'multi',
            'release_timestamp': 'multi',
            'samples': 'multi',
            'software': 'multi',
            'software_versions': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'uuid': 'multi',
            'volume': 'multi',
            'workflows': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if abstract is not None:
            
            _query_params.append(('abstract', abstract))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if authors is not None:
            
            _query_params.append(('authors', authors))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_published is not None:
            
            _query_params.append(('date_published', date_published))
            
        if date_revised is not None:
            
            _query_params.append(('date_revised', date_revised))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if donors is not None:
            
            _query_params.append(('donors', donors))
            
        if file_sets is not None:
            
            _query_params.append(('file_sets', file_sets))
            
        if issue is not None:
            
            _query_params.append(('issue', issue))
            
        if journal is not None:
            
            _query_params.append(('journal', journal))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if page is not None:
            
            _query_params.append(('page', page))
            
        if publication_identifiers is not None:
            
            _query_params.append(('publication_identifiers', publication_identifiers))
            
        if publication_year is not None:
            
            _query_params.append(('publication_year', publication_year))
            
        if published_by is not None:
            
            _query_params.append(('published_by', published_by))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if samples is not None:
            
            _query_params.append(('samples', samples))
            
        if software is not None:
            
            _query_params.append(('software', software))
            
        if software_versions is not None:
            
            _query_params.append(('software_versions', software_versions))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if volume is not None:
            
            _query_params.append(('volume', volume))
            
        if workflows is not None:
            
            _query_params.append(('workflows', workflows))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/publications/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def reference_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        external: Annotated[Optional[List[StrictBool]], Field(description="Filter by external")] = None,
        external_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by external_id")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        source_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by source_url")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ReferenceFileResults:
        """List items in the ReferenceFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ReferenceFile items. Supports filtering on fields within ReferenceFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param external: Filter by external
        :type external: List[bool]
        :param external_id: Filter by external_id
        :type external_id: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._reference_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            external=external,
            external_id=external_id,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            source_url=source_url,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ReferenceFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def reference_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        external: Annotated[Optional[List[StrictBool]], Field(description="Filter by external")] = None,
        external_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by external_id")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        source_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by source_url")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[ReferenceFileResults]:
        """List items in the ReferenceFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ReferenceFile items. Supports filtering on fields within ReferenceFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param external: Filter by external
        :type external: List[bool]
        :param external_id: Filter by external_id
        :type external_id: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._reference_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            external=external,
            external_id=external_id,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            source_url=source_url,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ReferenceFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def reference_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        external: Annotated[Optional[List[StrictBool]], Field(description="Filter by external")] = None,
        external_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by external_id")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        source_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by source_url")] = None,
        sources: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the ReferenceFile collection.

        Collection endpoint that accepts various query parameters to filter and sort ReferenceFile items. Supports filtering on fields within ReferenceFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param external: Filter by external
        :type external: List[bool]
        :param external_id: Filter by external_id
        :type external_id: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param sources: Filter by sources
        :type sources: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._reference_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            external=external,
            external_id=external_id,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            source_url=source_url,
            sources=sources,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "ReferenceFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _reference_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        anvil_url,
        assay_titles,
        assembly,
        award_id,
        award_component,
        collections,
        content_md5sum,
        content_type,
        controlled_access,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        external,
        external_id,
        file_format,
        file_format_specifications,
        file_format_type,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        release_timestamp,
        revoke_detail,
        s3_uri,
        source_url,
        sources,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        transcriptome_annotation,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'anvil_url': 'multi',
            'assay_titles': 'multi',
            'assembly': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'controlled_access': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'external': 'multi',
            'external_id': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_format_type': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'source_url': 'multi',
            'sources': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'transcriptome_annotation': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if anvil_url is not None:
            
            _query_params.append(('anvil_url', anvil_url))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if assembly is not None:
            
            _query_params.append(('assembly', assembly))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if controlled_access is not None:
            
            _query_params.append(('controlled_access', controlled_access))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if external is not None:
            
            _query_params.append(('external', external))
            
        if external_id is not None:
            
            _query_params.append(('external_id', external_id))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_format_type is not None:
            
            _query_params.append(('file_format_type', file_format_type))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if source_url is not None:
            
            _query_params.append(('source_url', source_url))
            
        if sources is not None:
            
            _query_params.append(('sources', sources))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if transcriptome_annotation is not None:
            
            _query_params.append(('transcriptome_annotation', transcriptome_annotation))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/reference-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def report(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        type: Annotated[Optional[List[StrictStr]], Field(description="Filter by item type.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}")] = None,
        include_fields: Annotated[Optional[List[StrictStr]], Field(description="Fields to include in the response. Can be repeated for multiple fields.")] = None,
        frame: Annotated[Optional[StrictStr], Field(description="Object with links, or object with some links embedded.")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> str:
        """Generate a TSV file report based on search query. All results are returned.

        Like /search endpoint but returns a TSV file instead of JSON. Must specify item type(s).

        :param query: Query string for searching.
        :type query: str
        :param type: Filter by item type.
        :type type: List[str]
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
        :type sort: List[str]
        :param field_filters: Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
        :type field_filters: object
        :param include_fields: Fields to include in the response. Can be repeated for multiple fields.
        :type include_fields: List[str]
        :param frame: Object with links, or object with some links embedded.
        :type frame: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._report_serialize(
            query=query,
            type=type,
            limit=limit,
            sort=sort,
            field_filters=field_filters,
            include_fields=include_fields,
            frame=frame,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "str",
            '400': "object",
            '404': "object",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def report_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        type: Annotated[Optional[List[StrictStr]], Field(description="Filter by item type.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}")] = None,
        include_fields: Annotated[Optional[List[StrictStr]], Field(description="Fields to include in the response. Can be repeated for multiple fields.")] = None,
        frame: Annotated[Optional[StrictStr], Field(description="Object with links, or object with some links embedded.")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[str]:
        """Generate a TSV file report based on search query. All results are returned.

        Like /search endpoint but returns a TSV file instead of JSON. Must specify item type(s).

        :param query: Query string for searching.
        :type query: str
        :param type: Filter by item type.
        :type type: List[str]
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
        :type sort: List[str]
        :param field_filters: Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
        :type field_filters: object
        :param include_fields: Fields to include in the response. Can be repeated for multiple fields.
        :type include_fields: List[str]
        :param frame: Object with links, or object with some links embedded.
        :type frame: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._report_serialize(
            query=query,
            type=type,
            limit=limit,
            sort=sort,
            field_filters=field_filters,
            include_fields=include_fields,
            frame=frame,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "str",
            '400': "object",
            '404': "object",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def report_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        type: Annotated[Optional[List[StrictStr]], Field(description="Filter by item type.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}")] = None,
        include_fields: Annotated[Optional[List[StrictStr]], Field(description="Fields to include in the response. Can be repeated for multiple fields.")] = None,
        frame: Annotated[Optional[StrictStr], Field(description="Object with links, or object with some links embedded.")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """Generate a TSV file report based on search query. All results are returned.

        Like /search endpoint but returns a TSV file instead of JSON. Must specify item type(s).

        :param query: Query string for searching.
        :type query: str
        :param type: Filter by item type.
        :type type: List[str]
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
        :type sort: List[str]
        :param field_filters: Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
        :type field_filters: object
        :param include_fields: Fields to include in the response. Can be repeated for multiple fields.
        :type include_fields: List[str]
        :param frame: Object with links, or object with some links embedded.
        :type frame: str
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._report_serialize(
            query=query,
            type=type,
            limit=limit,
            sort=sort,
            field_filters=field_filters,
            include_fields=include_fields,
            frame=frame,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "str",
            '400': "object",
            '404': "object",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _report_serialize(
        self,
        query,
        type,
        limit,
        sort,
        field_filters,
        include_fields,
        frame,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'type': 'multi',
            'sort': 'multi',
            'field': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if type is not None:
            
            _query_params.append(('type', type))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if field_filters is not None:
             for k, v in field_filters.items():
                if isinstance(v, list):
                    for item in v:
                        _query_params.append((k, item))
                else:
                    _query_params.append((k, v))
            
        if include_fields is not None:
            
            _query_params.append(('field', include_fields))
            
        if frame is not None:
            
            _query_params.append(('frame', frame))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter


        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'text/tab-separated-values', 
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/multireport.tsv',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def rodent_donors(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        genotype: Annotated[Optional[List[StrictStr]], Field(description="Filter by genotype")] = None,
        individual_rodent: Annotated[Optional[List[StrictBool]], Field(description="Filter by individual_rodent")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        phenotypic_features_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.@id")] = None,
        phenotypic_features_feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.feature.@id")] = None,
        phenotypic_features_feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_id")] = None,
        phenotypic_features_feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_name")] = None,
        phenotypic_features_observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.observation_date")] = None,
        phenotypic_features_quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.quantity_units")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        rodent_identifier: Annotated[Optional[List[StrictStr]], Field(description="Filter by rodent_identifier")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        strain: Annotated[Optional[List[StrictStr]], Field(description="Filter by strain")] = None,
        strain_background: Annotated[Optional[List[StrictStr]], Field(description="Filter by strain_background")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RodentDonorResults:
        """List items in the RodentDonor collection.

        Collection endpoint that accepts various query parameters to filter and sort RodentDonor items. Supports filtering on fields within RodentDonor items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param genotype: Filter by genotype
        :type genotype: List[str]
        :param individual_rodent: Filter by individual_rodent
        :type individual_rodent: List[bool]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param phenotypic_features_id: Filter by phenotypic_features.@id
        :type phenotypic_features_id: List[str]
        :param phenotypic_features_feature_id: Filter by phenotypic_features.feature.@id
        :type phenotypic_features_feature_id: List[str]
        :param phenotypic_features_feature_term_id: Filter by phenotypic_features.feature.term_id
        :type phenotypic_features_feature_term_id: List[str]
        :param phenotypic_features_feature_term_name: Filter by phenotypic_features.feature.term_name
        :type phenotypic_features_feature_term_name: List[str]
        :param phenotypic_features_observation_date: Filter by phenotypic_features.observation_date
        :type phenotypic_features_observation_date: List[str]
        :param phenotypic_features_quantity_units: Filter by phenotypic_features.quantity_units
        :type phenotypic_features_quantity_units: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param rodent_identifier: Filter by rodent_identifier
        :type rodent_identifier: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param strain: Filter by strain
        :type strain: List[str]
        :param strain_background: Filter by strain_background
        :type strain_background: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._rodent_donors_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            genotype=genotype,
            individual_rodent=individual_rodent,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            notes=notes,
            phenotypic_features_id=phenotypic_features_id,
            phenotypic_features_feature_id=phenotypic_features_feature_id,
            phenotypic_features_feature_term_id=phenotypic_features_feature_term_id,
            phenotypic_features_feature_term_name=phenotypic_features_feature_term_name,
            phenotypic_features_observation_date=phenotypic_features_observation_date,
            phenotypic_features_quantity_units=phenotypic_features_quantity_units,
            product_id=product_id,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            rodent_identifier=rodent_identifier,
            sex=sex,
            sources_id=sources_id,
            status=status,
            strain=strain,
            strain_background=strain_background,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "RodentDonorResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def rodent_donors_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        genotype: Annotated[Optional[List[StrictStr]], Field(description="Filter by genotype")] = None,
        individual_rodent: Annotated[Optional[List[StrictBool]], Field(description="Filter by individual_rodent")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        phenotypic_features_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.@id")] = None,
        phenotypic_features_feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.feature.@id")] = None,
        phenotypic_features_feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_id")] = None,
        phenotypic_features_feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_name")] = None,
        phenotypic_features_observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.observation_date")] = None,
        phenotypic_features_quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.quantity_units")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        rodent_identifier: Annotated[Optional[List[StrictStr]], Field(description="Filter by rodent_identifier")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        strain: Annotated[Optional[List[StrictStr]], Field(description="Filter by strain")] = None,
        strain_background: Annotated[Optional[List[StrictStr]], Field(description="Filter by strain_background")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[RodentDonorResults]:
        """List items in the RodentDonor collection.

        Collection endpoint that accepts various query parameters to filter and sort RodentDonor items. Supports filtering on fields within RodentDonor items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param genotype: Filter by genotype
        :type genotype: List[str]
        :param individual_rodent: Filter by individual_rodent
        :type individual_rodent: List[bool]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param phenotypic_features_id: Filter by phenotypic_features.@id
        :type phenotypic_features_id: List[str]
        :param phenotypic_features_feature_id: Filter by phenotypic_features.feature.@id
        :type phenotypic_features_feature_id: List[str]
        :param phenotypic_features_feature_term_id: Filter by phenotypic_features.feature.term_id
        :type phenotypic_features_feature_term_id: List[str]
        :param phenotypic_features_feature_term_name: Filter by phenotypic_features.feature.term_name
        :type phenotypic_features_feature_term_name: List[str]
        :param phenotypic_features_observation_date: Filter by phenotypic_features.observation_date
        :type phenotypic_features_observation_date: List[str]
        :param phenotypic_features_quantity_units: Filter by phenotypic_features.quantity_units
        :type phenotypic_features_quantity_units: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param rodent_identifier: Filter by rodent_identifier
        :type rodent_identifier: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param strain: Filter by strain
        :type strain: List[str]
        :param strain_background: Filter by strain_background
        :type strain_background: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._rodent_donors_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            genotype=genotype,
            individual_rodent=individual_rodent,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            notes=notes,
            phenotypic_features_id=phenotypic_features_id,
            phenotypic_features_feature_id=phenotypic_features_feature_id,
            phenotypic_features_feature_term_id=phenotypic_features_feature_term_id,
            phenotypic_features_feature_term_name=phenotypic_features_feature_term_name,
            phenotypic_features_observation_date=phenotypic_features_observation_date,
            phenotypic_features_quantity_units=phenotypic_features_quantity_units,
            product_id=product_id,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            rodent_identifier=rodent_identifier,
            sex=sex,
            sources_id=sources_id,
            status=status,
            strain=strain,
            strain_background=strain_background,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "RodentDonorResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def rodent_donors_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        genotype: Annotated[Optional[List[StrictStr]], Field(description="Filter by genotype")] = None,
        individual_rodent: Annotated[Optional[List[StrictBool]], Field(description="Filter by individual_rodent")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        phenotypic_features_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.@id")] = None,
        phenotypic_features_feature_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.feature.@id")] = None,
        phenotypic_features_feature_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_id")] = None,
        phenotypic_features_feature_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by phenotypic_features.feature.term_name")] = None,
        phenotypic_features_observation_date: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.observation_date")] = None,
        phenotypic_features_quantity_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by phenotypic_features.quantity_units")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        rodent_identifier: Annotated[Optional[List[StrictStr]], Field(description="Filter by rodent_identifier")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        strain: Annotated[Optional[List[StrictStr]], Field(description="Filter by strain")] = None,
        strain_background: Annotated[Optional[List[StrictStr]], Field(description="Filter by strain_background")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the RodentDonor collection.

        Collection endpoint that accepts various query parameters to filter and sort RodentDonor items. Supports filtering on fields within RodentDonor items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param genotype: Filter by genotype
        :type genotype: List[str]
        :param individual_rodent: Filter by individual_rodent
        :type individual_rodent: List[bool]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param phenotypic_features_id: Filter by phenotypic_features.@id
        :type phenotypic_features_id: List[str]
        :param phenotypic_features_feature_id: Filter by phenotypic_features.feature.@id
        :type phenotypic_features_feature_id: List[str]
        :param phenotypic_features_feature_term_id: Filter by phenotypic_features.feature.term_id
        :type phenotypic_features_feature_term_id: List[str]
        :param phenotypic_features_feature_term_name: Filter by phenotypic_features.feature.term_name
        :type phenotypic_features_feature_term_name: List[str]
        :param phenotypic_features_observation_date: Filter by phenotypic_features.observation_date
        :type phenotypic_features_observation_date: List[str]
        :param phenotypic_features_quantity_units: Filter by phenotypic_features.quantity_units
        :type phenotypic_features_quantity_units: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param rodent_identifier: Filter by rodent_identifier
        :type rodent_identifier: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param strain: Filter by strain
        :type strain: List[str]
        :param strain_background: Filter by strain_background
        :type strain_background: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._rodent_donors_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            genotype=genotype,
            individual_rodent=individual_rodent,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            notes=notes,
            phenotypic_features_id=phenotypic_features_id,
            phenotypic_features_feature_id=phenotypic_features_feature_id,
            phenotypic_features_feature_term_id=phenotypic_features_feature_term_id,
            phenotypic_features_feature_term_name=phenotypic_features_feature_term_name,
            phenotypic_features_observation_date=phenotypic_features_observation_date,
            phenotypic_features_quantity_units=phenotypic_features_quantity_units,
            product_id=product_id,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            rodent_identifier=rodent_identifier,
            sex=sex,
            sources_id=sources_id,
            status=status,
            strain=strain,
            strain_background=strain_background,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "RodentDonorResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _rodent_donors_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        collections,
        creation_timestamp,
        dbxrefs,
        description,
        documents,
        genotype,
        individual_rodent,
        lab_id,
        lab_title,
        lot_id,
        notes,
        phenotypic_features_id,
        phenotypic_features_feature_id,
        phenotypic_features_feature_term_id,
        phenotypic_features_feature_term_name,
        phenotypic_features_observation_date,
        phenotypic_features_quantity_units,
        product_id,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        rodent_identifier,
        sex,
        sources_id,
        status,
        strain,
        strain_background,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        taxa,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'genotype': 'multi',
            'individual_rodent': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'notes': 'multi',
            'phenotypic_features.@id': 'multi',
            'phenotypic_features.feature.@id': 'multi',
            'phenotypic_features.feature.term_id': 'multi',
            'phenotypic_features.feature.term_name': 'multi',
            'phenotypic_features.observation_date': 'multi',
            'phenotypic_features.quantity_units': 'multi',
            'product_id': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'rodent_identifier': 'multi',
            'sex': 'multi',
            'sources.@id': 'multi',
            'status': 'multi',
            'strain': 'multi',
            'strain_background': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if genotype is not None:
            
            _query_params.append(('genotype', genotype))
            
        if individual_rodent is not None:
            
            _query_params.append(('individual_rodent', individual_rodent))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if phenotypic_features_id is not None:
            
            _query_params.append(('phenotypic_features.@id', phenotypic_features_id))
            
        if phenotypic_features_feature_id is not None:
            
            _query_params.append(('phenotypic_features.feature.@id', phenotypic_features_feature_id))
            
        if phenotypic_features_feature_term_id is not None:
            
            _query_params.append(('phenotypic_features.feature.term_id', phenotypic_features_feature_term_id))
            
        if phenotypic_features_feature_term_name is not None:
            
            _query_params.append(('phenotypic_features.feature.term_name', phenotypic_features_feature_term_name))
            
        if phenotypic_features_observation_date is not None:
            
            _query_params.append(('phenotypic_features.observation_date', phenotypic_features_observation_date))
            
        if phenotypic_features_quantity_units is not None:
            
            _query_params.append(('phenotypic_features.quantity_units', phenotypic_features_quantity_units))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if rodent_identifier is not None:
            
            _query_params.append(('rodent_identifier', rodent_identifier))
            
        if sex is not None:
            
            _query_params.append(('sex', sex))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if strain is not None:
            
            _query_params.append(('strain', strain))
            
        if strain_background is not None:
            
            _query_params.append(('strain_background', strain_background))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/rodent-donors/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def sample_terms(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        cell_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_slims")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        developmental_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by developmental_slims")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        organ_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by organ_slims")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        system_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by system_slims")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SampleTermResults:
        """List items in the SampleTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort SampleTerm items. Supports filtering on fields within SampleTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param cell_slims: Filter by cell_slims
        :type cell_slims: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param developmental_slims: Filter by developmental_slims
        :type developmental_slims: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param organ_slims: Filter by organ_slims
        :type organ_slims: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param system_slims: Filter by system_slims
        :type system_slims: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sample_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            cell_slims=cell_slims,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            developmental_slims=developmental_slims,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            organ_slims=organ_slims,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            system_slims=system_slims,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SampleTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def sample_terms_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        cell_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_slims")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        developmental_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by developmental_slims")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        organ_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by organ_slims")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        system_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by system_slims")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SampleTermResults]:
        """List items in the SampleTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort SampleTerm items. Supports filtering on fields within SampleTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param cell_slims: Filter by cell_slims
        :type cell_slims: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param developmental_slims: Filter by developmental_slims
        :type developmental_slims: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param organ_slims: Filter by organ_slims
        :type organ_slims: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param system_slims: Filter by system_slims
        :type system_slims: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sample_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            cell_slims=cell_slims,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            developmental_slims=developmental_slims,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            organ_slims=organ_slims,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            system_slims=system_slims,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SampleTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def sample_terms_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        ancestors: Annotated[Optional[List[StrictStr]], Field(description="Filter by ancestors")] = None,
        cell_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_slims")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        deprecated_ntr_terms: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by deprecated_ntr_terms")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        developmental_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by developmental_slims")] = None,
        is_a: Annotated[Optional[List[StrictStr]], Field(description="Filter by is_a")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        ontology: Annotated[Optional[List[StrictStr]], Field(description="Filter by ontology")] = None,
        organ_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by organ_slims")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        synonyms: Annotated[Optional[List[StrictStr]], Field(description="Filter by synonyms")] = None,
        system_slims: Annotated[Optional[List[StrictStr]], Field(description="Filter by system_slims")] = None,
        term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_id")] = None,
        term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by term_name")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the SampleTerm collection.

        Collection endpoint that accepts various query parameters to filter and sort SampleTerm items. Supports filtering on fields within SampleTerm items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param ancestors: Filter by ancestors
        :type ancestors: List[str]
        :param cell_slims: Filter by cell_slims
        :type cell_slims: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param deprecated_ntr_terms: Filter by deprecated_ntr_terms
        :type deprecated_ntr_terms: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param developmental_slims: Filter by developmental_slims
        :type developmental_slims: List[str]
        :param is_a: Filter by is_a
        :type is_a: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param ontology: Filter by ontology
        :type ontology: List[str]
        :param organ_slims: Filter by organ_slims
        :type organ_slims: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param synonyms: Filter by synonyms
        :type synonyms: List[str]
        :param system_slims: Filter by system_slims
        :type system_slims: List[str]
        :param term_id: Filter by term_id
        :type term_id: List[str]
        :param term_name: Filter by term_name
        :type term_name: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sample_terms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            ancestors=ancestors,
            cell_slims=cell_slims,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            deprecated_ntr_terms=deprecated_ntr_terms,
            description=description,
            developmental_slims=developmental_slims,
            is_a=is_a,
            name=name,
            notes=notes,
            ontology=ontology,
            organ_slims=organ_slims,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            synonyms=synonyms,
            system_slims=system_slims,
            term_id=term_id,
            term_name=term_name,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SampleTermResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _sample_terms_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        ancestors,
        cell_slims,
        creation_timestamp,
        dbxrefs,
        deprecated_ntr_terms,
        description,
        developmental_slims,
        is_a,
        name,
        notes,
        ontology,
        organ_slims,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        synonyms,
        system_slims,
        term_id,
        term_name,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'ancestors': 'multi',
            'cell_slims': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'deprecated_ntr_terms': 'multi',
            'description': 'multi',
            'developmental_slims': 'multi',
            'is_a': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'ontology': 'multi',
            'organ_slims': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'synonyms': 'multi',
            'system_slims': 'multi',
            'term_id': 'multi',
            'term_name': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if ancestors is not None:
            
            _query_params.append(('ancestors', ancestors))
            
        if cell_slims is not None:
            
            _query_params.append(('cell_slims', cell_slims))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if deprecated_ntr_terms is not None:
            
            _query_params.append(('deprecated_ntr_terms', deprecated_ntr_terms))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if developmental_slims is not None:
            
            _query_params.append(('developmental_slims', developmental_slims))
            
        if is_a is not None:
            
            _query_params.append(('is_a', is_a))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if ontology is not None:
            
            _query_params.append(('ontology', ontology))
            
        if organ_slims is not None:
            
            _query_params.append(('organ_slims', organ_slims))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if synonyms is not None:
            
            _query_params.append(('synonyms', synonyms))
            
        if system_slims is not None:
            
            _query_params.append(('system_slims', system_slims))
            
        if term_id is not None:
            
            _query_params.append(('term_id', term_id))
            
        if term_name is not None:
            
            _query_params.append(('term_name', term_name))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/sample-terms/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def schema_for_item_type(
        self,
        item_type: Annotated[ItemType, Field(description="The name of the item type")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> object:
        """Retrieve JSON schema for item type

        Returns JSON schemas of all the item types defined in IGVF

        :param item_type: The name of the item type (required)
        :type item_type: ItemType
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._schema_for_item_type_serialize(
            item_type=item_type,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def schema_for_item_type_with_http_info(
        self,
        item_type: Annotated[ItemType, Field(description="The name of the item type")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[object]:
        """Retrieve JSON schema for item type

        Returns JSON schemas of all the item types defined in IGVF

        :param item_type: The name of the item type (required)
        :type item_type: ItemType
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._schema_for_item_type_serialize(
            item_type=item_type,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def schema_for_item_type_without_preload_content(
        self,
        item_type: Annotated[ItemType, Field(description="The name of the item type")],
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """Retrieve JSON schema for item type

        Returns JSON schemas of all the item types defined in IGVF

        :param item_type: The name of the item type (required)
        :type item_type: ItemType
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._schema_for_item_type_serialize(
            item_type=item_type,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _schema_for_item_type_serialize(
        self,
        item_type,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        if item_type is not None:
            _path_params['item_type'] = item_type.value
        # process the query parameters
        # process the header parameters
        # process the form parameters
        # process the body parameter


        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/profiles/{item_type}',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def schemas(
        self,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> object:
        """Retrieve JSON schemas for all item types

        Returns JSON schemas of all the item types defined in IGVF

        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._schemas_serialize(
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def schemas_with_http_info(
        self,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[object]:
        """Retrieve JSON schemas for all item types

        Returns JSON schemas of all the item types defined in IGVF

        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._schemas_serialize(
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def schemas_without_preload_content(
        self,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """Retrieve JSON schemas for all item types

        Returns JSON schemas of all the item types defined in IGVF

        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._schemas_serialize(
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "object",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _schemas_serialize(
        self,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        # process the header parameters
        # process the form parameters
        # process the body parameter


        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/profiles',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def search(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        type: Annotated[Optional[List[StrictStr]], Field(description="Filter by item type.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SearchResults:
        """Search for items on the IGVF Data Portal.

        Search endpoint that accepts various query parameters to filter and sort results. Supports complex filtering on item types and fields within items.

        :param query: Query string for searching.
        :type query: str
        :param type: Filter by item type.
        :type type: List[str]
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
        :type sort: List[str]
        :param field_filters: Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
        :type field_filters: object
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._search_serialize(
            query=query,
            type=type,
            limit=limit,
            sort=sort,
            field_filters=field_filters,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SearchResults",
            '404': "NoResultsResponse",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def search_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        type: Annotated[Optional[List[StrictStr]], Field(description="Filter by item type.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SearchResults]:
        """Search for items on the IGVF Data Portal.

        Search endpoint that accepts various query parameters to filter and sort results. Supports complex filtering on item types and fields within items.

        :param query: Query string for searching.
        :type query: str
        :param type: Filter by item type.
        :type type: List[str]
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
        :type sort: List[str]
        :param field_filters: Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
        :type field_filters: object
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._search_serialize(
            query=query,
            type=type,
            limit=limit,
            sort=sort,
            field_filters=field_filters,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SearchResults",
            '404': "NoResultsResponse",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def search_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        type: Annotated[Optional[List[StrictStr]], Field(description="Filter by item type.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.")] = None,
        field_filters: Annotated[Optional[Dict[str, Any]], Field(description="Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """Search for items on the IGVF Data Portal.

        Search endpoint that accepts various query parameters to filter and sort results. Supports complex filtering on item types and fields within items.

        :param query: Query string for searching.
        :type query: str
        :param type: Filter by item type.
        :type type: List[str]
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
        :type sort: List[str]
        :param field_filters: Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
        :type field_filters: object
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._search_serialize(
            query=query,
            type=type,
            limit=limit,
            sort=sort,
            field_filters=field_filters,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SearchResults",
            '404': "NoResultsResponse",
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _search_serialize(
        self,
        query,
        type,
        limit,
        sort,
        field_filters,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'type': 'multi',
            'sort': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if type is not None:
            
            _query_params.append(('type', type))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if field_filters is not None:
             for k, v in field_filters.items():
                if isinstance(v, list):
                    for item in v:
                        _query_params.append((k, item))
                else:
                    _query_params.append((k, v))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/search',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def sequence_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        base_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by base_modifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        flowcell_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by flowcell_id")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        illumina_read_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by illumina_read_type")] = None,
        index: Annotated[Optional[List[StrictStr]], Field(description="Filter by index")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lane: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by lane")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        maximum_read_length: Annotated[Optional[List[Annotated[int, Field(le=300000000, strict=True, ge=0)]]], Field(description="Filter by maximum_read_length")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        mean_read_length: Annotated[Optional[List[Union[Annotated[float, Field(le=300000000, strict=True, ge=0)], Annotated[int, Field(le=300000000, strict=True, ge=0)]]]], Field(description="Filter by mean_read_length")] = None,
        minimum_read_length: Annotated[Optional[List[Annotated[int, Field(le=300000000, strict=True, ge=0)]]], Field(description="Filter by minimum_read_length")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        read_count: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by read_count")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        seqspecs: Annotated[Optional[List[StrictStr]], Field(description="Filter by seqspecs")] = None,
        sequencing_kit: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_kit")] = None,
        sequencing_platform_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_platform.@id")] = None,
        sequencing_platform_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sequencing_platform.term_name")] = None,
        sequencing_run: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by sequencing_run")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SequenceFileResults:
        """List items in the SequenceFile collection.

        Collection endpoint that accepts various query parameters to filter and sort SequenceFile items. Supports filtering on fields within SequenceFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param base_modifications: Filter by base_modifications
        :type base_modifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param flowcell_id: Filter by flowcell_id
        :type flowcell_id: List[str]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param illumina_read_type: Filter by illumina_read_type
        :type illumina_read_type: List[str]
        :param index: Filter by index
        :type index: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lane: Filter by lane
        :type lane: List[int]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param maximum_read_length: Filter by maximum_read_length
        :type maximum_read_length: List[int]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param mean_read_length: Filter by mean_read_length
        :type mean_read_length: List[float]
        :param minimum_read_length: Filter by minimum_read_length
        :type minimum_read_length: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param read_count: Filter by read_count
        :type read_count: List[int]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param seqspecs: Filter by seqspecs
        :type seqspecs: List[str]
        :param sequencing_kit: Filter by sequencing_kit
        :type sequencing_kit: List[str]
        :param sequencing_platform_id: Filter by sequencing_platform.@id
        :type sequencing_platform_id: List[str]
        :param sequencing_platform_term_name: Filter by sequencing_platform.term_name
        :type sequencing_platform_term_name: List[str]
        :param sequencing_run: Filter by sequencing_run
        :type sequencing_run: List[int]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sequence_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            base_modifications=base_modifications,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            flowcell_id=flowcell_id,
            gene_list_for=gene_list_for,
            href=href,
            illumina_read_type=illumina_read_type,
            index=index,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            lane=lane,
            loci_list_for=loci_list_for,
            maximum_read_length=maximum_read_length,
            md5sum=md5sum,
            mean_read_length=mean_read_length,
            minimum_read_length=minimum_read_length,
            notes=notes,
            read_count=read_count,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            seqspecs=seqspecs,
            sequencing_kit=sequencing_kit,
            sequencing_platform_id=sequencing_platform_id,
            sequencing_platform_term_name=sequencing_platform_term_name,
            sequencing_run=sequencing_run,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SequenceFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def sequence_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        base_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by base_modifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        flowcell_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by flowcell_id")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        illumina_read_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by illumina_read_type")] = None,
        index: Annotated[Optional[List[StrictStr]], Field(description="Filter by index")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lane: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by lane")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        maximum_read_length: Annotated[Optional[List[Annotated[int, Field(le=300000000, strict=True, ge=0)]]], Field(description="Filter by maximum_read_length")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        mean_read_length: Annotated[Optional[List[Union[Annotated[float, Field(le=300000000, strict=True, ge=0)], Annotated[int, Field(le=300000000, strict=True, ge=0)]]]], Field(description="Filter by mean_read_length")] = None,
        minimum_read_length: Annotated[Optional[List[Annotated[int, Field(le=300000000, strict=True, ge=0)]]], Field(description="Filter by minimum_read_length")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        read_count: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by read_count")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        seqspecs: Annotated[Optional[List[StrictStr]], Field(description="Filter by seqspecs")] = None,
        sequencing_kit: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_kit")] = None,
        sequencing_platform_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_platform.@id")] = None,
        sequencing_platform_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sequencing_platform.term_name")] = None,
        sequencing_run: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by sequencing_run")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SequenceFileResults]:
        """List items in the SequenceFile collection.

        Collection endpoint that accepts various query parameters to filter and sort SequenceFile items. Supports filtering on fields within SequenceFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param base_modifications: Filter by base_modifications
        :type base_modifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param flowcell_id: Filter by flowcell_id
        :type flowcell_id: List[str]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param illumina_read_type: Filter by illumina_read_type
        :type illumina_read_type: List[str]
        :param index: Filter by index
        :type index: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lane: Filter by lane
        :type lane: List[int]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param maximum_read_length: Filter by maximum_read_length
        :type maximum_read_length: List[int]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param mean_read_length: Filter by mean_read_length
        :type mean_read_length: List[float]
        :param minimum_read_length: Filter by minimum_read_length
        :type minimum_read_length: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param read_count: Filter by read_count
        :type read_count: List[int]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param seqspecs: Filter by seqspecs
        :type seqspecs: List[str]
        :param sequencing_kit: Filter by sequencing_kit
        :type sequencing_kit: List[str]
        :param sequencing_platform_id: Filter by sequencing_platform.@id
        :type sequencing_platform_id: List[str]
        :param sequencing_platform_term_name: Filter by sequencing_platform.term_name
        :type sequencing_platform_term_name: List[str]
        :param sequencing_run: Filter by sequencing_run
        :type sequencing_run: List[int]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sequence_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            base_modifications=base_modifications,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            flowcell_id=flowcell_id,
            gene_list_for=gene_list_for,
            href=href,
            illumina_read_type=illumina_read_type,
            index=index,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            lane=lane,
            loci_list_for=loci_list_for,
            maximum_read_length=maximum_read_length,
            md5sum=md5sum,
            mean_read_length=mean_read_length,
            minimum_read_length=minimum_read_length,
            notes=notes,
            read_count=read_count,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            seqspecs=seqspecs,
            sequencing_kit=sequencing_kit,
            sequencing_platform_id=sequencing_platform_id,
            sequencing_platform_term_name=sequencing_platform_term_name,
            sequencing_run=sequencing_run,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SequenceFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def sequence_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        base_modifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by base_modifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        flowcell_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by flowcell_id")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        illumina_read_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by illumina_read_type")] = None,
        index: Annotated[Optional[List[StrictStr]], Field(description="Filter by index")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lane: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by lane")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        maximum_read_length: Annotated[Optional[List[Annotated[int, Field(le=300000000, strict=True, ge=0)]]], Field(description="Filter by maximum_read_length")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        mean_read_length: Annotated[Optional[List[Union[Annotated[float, Field(le=300000000, strict=True, ge=0)], Annotated[int, Field(le=300000000, strict=True, ge=0)]]]], Field(description="Filter by mean_read_length")] = None,
        minimum_read_length: Annotated[Optional[List[Annotated[int, Field(le=300000000, strict=True, ge=0)]]], Field(description="Filter by minimum_read_length")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        read_count: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by read_count")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        seqspecs: Annotated[Optional[List[StrictStr]], Field(description="Filter by seqspecs")] = None,
        sequencing_kit: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_kit")] = None,
        sequencing_platform_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sequencing_platform.@id")] = None,
        sequencing_platform_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sequencing_platform.term_name")] = None,
        sequencing_run: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by sequencing_run")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the SequenceFile collection.

        Collection endpoint that accepts various query parameters to filter and sort SequenceFile items. Supports filtering on fields within SequenceFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param base_modifications: Filter by base_modifications
        :type base_modifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param flowcell_id: Filter by flowcell_id
        :type flowcell_id: List[str]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param illumina_read_type: Filter by illumina_read_type
        :type illumina_read_type: List[str]
        :param index: Filter by index
        :type index: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lane: Filter by lane
        :type lane: List[int]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param maximum_read_length: Filter by maximum_read_length
        :type maximum_read_length: List[int]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param mean_read_length: Filter by mean_read_length
        :type mean_read_length: List[float]
        :param minimum_read_length: Filter by minimum_read_length
        :type minimum_read_length: List[int]
        :param notes: Filter by notes
        :type notes: List[str]
        :param read_count: Filter by read_count
        :type read_count: List[int]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param seqspecs: Filter by seqspecs
        :type seqspecs: List[str]
        :param sequencing_kit: Filter by sequencing_kit
        :type sequencing_kit: List[str]
        :param sequencing_platform_id: Filter by sequencing_platform.@id
        :type sequencing_platform_id: List[str]
        :param sequencing_platform_term_name: Filter by sequencing_platform.term_name
        :type sequencing_platform_term_name: List[str]
        :param sequencing_run: Filter by sequencing_run
        :type sequencing_run: List[int]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sequence_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            award_id=award_id,
            award_component=award_component,
            base_modifications=base_modifications,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            flowcell_id=flowcell_id,
            gene_list_for=gene_list_for,
            href=href,
            illumina_read_type=illumina_read_type,
            index=index,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            lane=lane,
            loci_list_for=loci_list_for,
            maximum_read_length=maximum_read_length,
            md5sum=md5sum,
            mean_read_length=mean_read_length,
            minimum_read_length=minimum_read_length,
            notes=notes,
            read_count=read_count,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            seqspecs=seqspecs,
            sequencing_kit=sequencing_kit,
            sequencing_platform_id=sequencing_platform_id,
            sequencing_platform_term_name=sequencing_platform_term_name,
            sequencing_run=sequencing_run,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SequenceFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _sequence_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        anvil_url,
        assay_titles,
        award_id,
        award_component,
        base_modifications,
        collections,
        content_md5sum,
        content_type,
        controlled_access,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        flowcell_id,
        gene_list_for,
        href,
        illumina_read_type,
        index,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        lane,
        loci_list_for,
        maximum_read_length,
        md5sum,
        mean_read_length,
        minimum_read_length,
        notes,
        read_count,
        release_timestamp,
        revoke_detail,
        s3_uri,
        seqspecs,
        sequencing_kit,
        sequencing_platform_id,
        sequencing_platform_term_name,
        sequencing_run,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'anvil_url': 'multi',
            'assay_titles': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'base_modifications': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'controlled_access': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'flowcell_id': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'illumina_read_type': 'multi',
            'index': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lane': 'multi',
            'loci_list_for': 'multi',
            'maximum_read_length': 'multi',
            'md5sum': 'multi',
            'mean_read_length': 'multi',
            'minimum_read_length': 'multi',
            'notes': 'multi',
            'read_count': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'seqspecs': 'multi',
            'sequencing_kit': 'multi',
            'sequencing_platform.@id': 'multi',
            'sequencing_platform.term_name': 'multi',
            'sequencing_run': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if anvil_url is not None:
            
            _query_params.append(('anvil_url', anvil_url))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if base_modifications is not None:
            
            _query_params.append(('base_modifications', base_modifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if controlled_access is not None:
            
            _query_params.append(('controlled_access', controlled_access))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if flowcell_id is not None:
            
            _query_params.append(('flowcell_id', flowcell_id))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if illumina_read_type is not None:
            
            _query_params.append(('illumina_read_type', illumina_read_type))
            
        if index is not None:
            
            _query_params.append(('index', index))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lane is not None:
            
            _query_params.append(('lane', lane))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if maximum_read_length is not None:
            
            _query_params.append(('maximum_read_length', maximum_read_length))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if mean_read_length is not None:
            
            _query_params.append(('mean_read_length', mean_read_length))
            
        if minimum_read_length is not None:
            
            _query_params.append(('minimum_read_length', minimum_read_length))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if read_count is not None:
            
            _query_params.append(('read_count', read_count))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if seqspecs is not None:
            
            _query_params.append(('seqspecs', seqspecs))
            
        if sequencing_kit is not None:
            
            _query_params.append(('sequencing_kit', sequencing_kit))
            
        if sequencing_platform_id is not None:
            
            _query_params.append(('sequencing_platform.@id', sequencing_platform_id))
            
        if sequencing_platform_term_name is not None:
            
            _query_params.append(('sequencing_platform.term_name', sequencing_platform_term_name))
            
        if sequencing_run is not None:
            
            _query_params.append(('sequencing_run', sequencing_run))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/sequence-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def signal_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        filtered: Annotated[Optional[List[StrictBool]], Field(description="Filter by filtered")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        normalized: Annotated[Optional[List[StrictBool]], Field(description="Filter by normalized")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        start_view_position: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by start_view_position")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        strand_specificity: Annotated[Optional[List[StrictStr]], Field(description="Filter by strand_specificity")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SignalFileResults:
        """List items in the SignalFile collection.

        Collection endpoint that accepts various query parameters to filter and sort SignalFile items. Supports filtering on fields within SignalFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param filtered: Filter by filtered
        :type filtered: List[bool]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param normalized: Filter by normalized
        :type normalized: List[bool]
        :param notes: Filter by notes
        :type notes: List[str]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param start_view_position: Filter by start_view_position
        :type start_view_position: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param strand_specificity: Filter by strand_specificity
        :type strand_specificity: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._signal_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            filtered=filtered,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            normalized=normalized,
            notes=notes,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            start_view_position=start_view_position,
            status=status,
            strand_specificity=strand_specificity,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SignalFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def signal_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        filtered: Annotated[Optional[List[StrictBool]], Field(description="Filter by filtered")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        normalized: Annotated[Optional[List[StrictBool]], Field(description="Filter by normalized")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        start_view_position: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by start_view_position")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        strand_specificity: Annotated[Optional[List[StrictStr]], Field(description="Filter by strand_specificity")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SignalFileResults]:
        """List items in the SignalFile collection.

        Collection endpoint that accepts various query parameters to filter and sort SignalFile items. Supports filtering on fields within SignalFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param filtered: Filter by filtered
        :type filtered: List[bool]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param normalized: Filter by normalized
        :type normalized: List[bool]
        :param notes: Filter by notes
        :type notes: List[str]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param start_view_position: Filter by start_view_position
        :type start_view_position: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param strand_specificity: Filter by strand_specificity
        :type strand_specificity: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._signal_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            filtered=filtered,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            normalized=normalized,
            notes=notes,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            start_view_position=start_view_position,
            status=status,
            strand_specificity=strand_specificity,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SignalFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def signal_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_summary")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        filtered: Annotated[Optional[List[StrictBool]], Field(description="Filter by filtered")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        normalized: Annotated[Optional[List[StrictBool]], Field(description="Filter by normalized")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        reference_files: Annotated[Optional[List[StrictStr]], Field(description="Filter by reference_files")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        start_view_position: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by start_view_position")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        strand_specificity: Annotated[Optional[List[StrictStr]], Field(description="Filter by strand_specificity")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the SignalFile collection.

        Collection endpoint that accepts various query parameters to filter and sort SignalFile items. Supports filtering on fields within SignalFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_summary: Filter by content_summary
        :type content_summary: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param filtered: Filter by filtered
        :type filtered: List[bool]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param normalized: Filter by normalized
        :type normalized: List[bool]
        :param notes: Filter by notes
        :type notes: List[str]
        :param reference_files: Filter by reference_files
        :type reference_files: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param start_view_position: Filter by start_view_position
        :type start_view_position: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param strand_specificity: Filter by strand_specificity
        :type strand_specificity: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._signal_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_summary=content_summary,
            content_type=content_type,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            filtered=filtered,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            normalized=normalized,
            notes=notes,
            reference_files=reference_files,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            start_view_position=start_view_position,
            status=status,
            strand_specificity=strand_specificity,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SignalFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _signal_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        assay_titles,
        assembly,
        award_id,
        award_component,
        cell_type_annotation_id,
        cell_type_annotation_term_name,
        collections,
        content_md5sum,
        content_summary,
        content_type,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        filtered,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        normalized,
        notes,
        reference_files,
        release_timestamp,
        revoke_detail,
        s3_uri,
        start_view_position,
        status,
        strand_specificity,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        transcriptome_annotation,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'assay_titles': 'multi',
            'assembly': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'cell_type_annotation.@id': 'multi',
            'cell_type_annotation.term_name': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_summary': 'multi',
            'content_type': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'filtered': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'normalized': 'multi',
            'notes': 'multi',
            'reference_files': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'start_view_position': 'multi',
            'status': 'multi',
            'strand_specificity': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'transcriptome_annotation': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if assembly is not None:
            
            _query_params.append(('assembly', assembly))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if cell_type_annotation_id is not None:
            
            _query_params.append(('cell_type_annotation.@id', cell_type_annotation_id))
            
        if cell_type_annotation_term_name is not None:
            
            _query_params.append(('cell_type_annotation.term_name', cell_type_annotation_term_name))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_summary is not None:
            
            _query_params.append(('content_summary', content_summary))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if filtered is not None:
            
            _query_params.append(('filtered', filtered))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if normalized is not None:
            
            _query_params.append(('normalized', normalized))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if reference_files is not None:
            
            _query_params.append(('reference_files', reference_files))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if start_view_position is not None:
            
            _query_params.append(('start_view_position', start_view_position))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if strand_specificity is not None:
            
            _query_params.append(('strand_specificity', strand_specificity))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if transcriptome_annotation is not None:
            
            _query_params.append(('transcriptome_annotation', transcriptome_annotation))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/signal-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def software(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        source_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by source_url")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        used_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by used_by")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        versions: Annotated[Optional[List[StrictStr]], Field(description="Filter by versions")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SoftwareResults:
        """List items in the Software collection.

        Collection endpoint that accepts various query parameters to filter and sort Software items. Supports filtering on fields within Software items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param used_by: Filter by used_by
        :type used_by: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param versions: Filter by versions
        :type versions: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._software_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            source_url=source_url,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            used_by=used_by,
            uuid=uuid,
            versions=versions,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SoftwareResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def software_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        source_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by source_url")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        used_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by used_by")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        versions: Annotated[Optional[List[StrictStr]], Field(description="Filter by versions")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SoftwareResults]:
        """List items in the Software collection.

        Collection endpoint that accepts various query parameters to filter and sort Software items. Supports filtering on fields within Software items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param used_by: Filter by used_by
        :type used_by: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param versions: Filter by versions
        :type versions: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._software_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            source_url=source_url,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            used_by=used_by,
            uuid=uuid,
            versions=versions,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SoftwareResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def software_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        source_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by source_url")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        used_by: Annotated[Optional[List[StrictStr]], Field(description="Filter by used_by")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        versions: Annotated[Optional[List[StrictStr]], Field(description="Filter by versions")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Software collection.

        Collection endpoint that accepts various query parameters to filter and sort Software items. Supports filtering on fields within Software items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param used_by: Filter by used_by
        :type used_by: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param versions: Filter by versions
        :type versions: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._software_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            source_url=source_url,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            used_by=used_by,
            uuid=uuid,
            versions=versions,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SoftwareResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _software_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award_id,
        award_component,
        creation_timestamp,
        description,
        lab_id,
        lab_title,
        name,
        notes,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        source_url,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        used_by,
        uuid,
        versions,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'source_url': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'used_by': 'multi',
            'uuid': 'multi',
            'versions': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if source_url is not None:
            
            _query_params.append(('source_url', source_url))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if used_by is not None:
            
            _query_params.append(('used_by', used_by))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if versions is not None:
            
            _query_params.append(('versions', versions))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/software/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def software_versions(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        download_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by download_id")] = None,
        downloaded_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by downloaded_url")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        software_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by software.@id")] = None,
        software_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by software.title")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        version: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by version")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SoftwareVersionResults:
        """List items in the SoftwareVersion collection.

        Collection endpoint that accepts various query parameters to filter and sort SoftwareVersion items. Supports filtering on fields within SoftwareVersion items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param download_id: Filter by download_id
        :type download_id: List[str]
        :param downloaded_url: Filter by downloaded_url
        :type downloaded_url: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param software_id: Filter by software.@id
        :type software_id: List[str]
        :param software_title: Filter by software.title
        :type software_title: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param version: Filter by version
        :type version: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._software_versions_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            download_id=download_id,
            downloaded_url=downloaded_url,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            software_id=software_id,
            software_title=software_title,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            version=version,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SoftwareVersionResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def software_versions_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        download_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by download_id")] = None,
        downloaded_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by downloaded_url")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        software_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by software.@id")] = None,
        software_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by software.title")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        version: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by version")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SoftwareVersionResults]:
        """List items in the SoftwareVersion collection.

        Collection endpoint that accepts various query parameters to filter and sort SoftwareVersion items. Supports filtering on fields within SoftwareVersion items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param download_id: Filter by download_id
        :type download_id: List[str]
        :param downloaded_url: Filter by downloaded_url
        :type downloaded_url: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param software_id: Filter by software.@id
        :type software_id: List[str]
        :param software_title: Filter by software.title
        :type software_title: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param version: Filter by version
        :type version: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._software_versions_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            download_id=download_id,
            downloaded_url=downloaded_url,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            software_id=software_id,
            software_title=software_title,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            version=version,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SoftwareVersionResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def software_versions_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        download_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by download_id")] = None,
        downloaded_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by downloaded_url")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[StrictStr]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        software_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by software.@id")] = None,
        software_title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by software.title")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        version: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by version")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the SoftwareVersion collection.

        Collection endpoint that accepts various query parameters to filter and sort SoftwareVersion items. Supports filtering on fields within SoftwareVersion items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param download_id: Filter by download_id
        :type download_id: List[str]
        :param downloaded_url: Filter by downloaded_url
        :type downloaded_url: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param software_id: Filter by software.@id
        :type software_id: List[str]
        :param software_title: Filter by software.title
        :type software_title: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param version: Filter by version
        :type version: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._software_versions_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            award_id=award_id,
            award_component=award_component,
            creation_timestamp=creation_timestamp,
            description=description,
            download_id=download_id,
            downloaded_url=downloaded_url,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            software_id=software_id,
            software_title=software_title,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            version=version,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SoftwareVersionResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _software_versions_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        award_id,
        award_component,
        creation_timestamp,
        description,
        download_id,
        downloaded_url,
        lab_id,
        lab_title,
        name,
        notes,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        software_id,
        software_title,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        uuid,
        version,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'download_id': 'multi',
            'downloaded_url': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'software.@id': 'multi',
            'software.title': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'uuid': 'multi',
            'version': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if download_id is not None:
            
            _query_params.append(('download_id', download_id))
            
        if downloaded_url is not None:
            
            _query_params.append(('downloaded_url', downloaded_url))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if software_id is not None:
            
            _query_params.append(('software.@id', software_id))
            
        if software_title is not None:
            
            _query_params.append(('software.title', software_title))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if version is not None:
            
            _query_params.append(('version', version))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/software-versions/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def sources(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> SourceResults:
        """List items in the Source collection.

        Collection endpoint that accepts various query parameters to filter and sort Source items. Supports filtering on fields within Source items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sources_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            name=name,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SourceResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def sources_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[SourceResults]:
        """List items in the Source collection.

        Collection endpoint that accepts various query parameters to filter and sort Source items. Supports filtering on fields within Source items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sources_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            name=name,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SourceResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def sources_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by title")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Source collection.

        Collection endpoint that accepts various query parameters to filter and sort Source items. Supports filtering on fields within Source items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._sources_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            name=name,
            notes=notes,
            release_timestamp=release_timestamp,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            url=url,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "SourceResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _sources_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        creation_timestamp,
        description,
        name,
        notes,
        release_timestamp,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        url,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'url': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/sources/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def tabular_files(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        barcode_map_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map_for")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> TabularFileResults:
        """List items in the TabularFile collection.

        Collection endpoint that accepts various query parameters to filter and sort TabularFile items. Supports filtering on fields within TabularFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param barcode_map_for: Filter by barcode_map_for
        :type barcode_map_for: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._tabular_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            barcode_map_for=barcode_map_for,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TabularFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def tabular_files_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        barcode_map_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map_for")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[TabularFileResults]:
        """List items in the TabularFile collection.

        Collection endpoint that accepts various query parameters to filter and sort TabularFile items. Supports filtering on fields within TabularFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param barcode_map_for: Filter by barcode_map_for
        :type barcode_map_for: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._tabular_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            barcode_map_for=barcode_map_for,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TabularFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def tabular_files_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_step_version: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_step_version")] = None,
        anvil_url: Annotated[Optional[List[StrictStr]], Field(description="Filter by anvil_url")] = None,
        assay_titles: Annotated[Optional[List[StrictStr]], Field(description="Filter by assay_titles")] = None,
        assembly: Annotated[Optional[List[StrictStr]], Field(description="Filter by assembly")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        barcode_map_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by barcode_map_for")] = None,
        cell_type_annotation_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by cell_type_annotation.@id")] = None,
        cell_type_annotation_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cell_type_annotation.term_name")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        content_md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by content_md5sum")] = None,
        content_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by content_type")] = None,
        controlled_access: Annotated[Optional[List[StrictBool]], Field(description="Filter by controlled_access")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        derived_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by derived_from")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_format: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format")] = None,
        file_format_specifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_specifications")] = None,
        file_format_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_format_type")] = None,
        file_set_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.@id")] = None,
        file_set_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.accession")] = None,
        file_set_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.file_set_type")] = None,
        file_set_samples_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.@id")] = None,
        file_set_samples_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.accession")] = None,
        file_set_samples_classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.classifications")] = None,
        file_set_samples_disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.@id")] = None,
        file_set_samples_disease_terms_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.disease_terms.summary")] = None,
        file_set_samples_disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.disease_terms.term_name")] = None,
        file_set_samples_sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.sample_terms.@id")] = None,
        file_set_samples_sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_set.samples.sample_terms.term_name")] = None,
        file_set_samples_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.summary")] = None,
        file_set_samples_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.samples.taxa")] = None,
        file_set_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.summary")] = None,
        file_set_taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_set.taxa")] = None,
        file_size: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=0)]]], Field(description="Filter by file_size")] = None,
        gene_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by gene_list_for")] = None,
        href: Annotated[Optional[List[StrictStr]], Field(description="Filter by href")] = None,
        input_file_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by input_file_for")] = None,
        integrated_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.@id")] = None,
        integrated_in_associated_phenotypes_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.@id")] = None,
        integrated_in_associated_phenotypes_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.associated_phenotypes.summary")] = None,
        integrated_in_associated_phenotypes_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by integrated_in.associated_phenotypes.term_name")] = None,
        integrated_in_file_set_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.file_set_type")] = None,
        integrated_in_small_scale_gene_list_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.@id")] = None,
        integrated_in_small_scale_gene_list_symbol: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.small_scale_gene_list.symbol")] = None,
        integrated_in_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by integrated_in.summary")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        loci_list_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by loci_list_for")] = None,
        md5sum: Annotated[Optional[List[Annotated[str, Field(strict=True, max_length=32)]]], Field(description="Filter by md5sum")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        s3_uri: Annotated[Optional[List[StrictStr]], Field(description="Filter by s3_uri")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitted_file_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_file_name")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        transcriptome_annotation: Annotated[Optional[List[StrictStr]], Field(description="Filter by transcriptome_annotation")] = None,
        upload_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by upload_status")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        validation_error_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by validation_error_detail")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the TabularFile collection.

        Collection endpoint that accepts various query parameters to filter and sort TabularFile items. Supports filtering on fields within TabularFile items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_step_version: Filter by analysis_step_version
        :type analysis_step_version: List[str]
        :param anvil_url: Filter by anvil_url
        :type anvil_url: List[str]
        :param assay_titles: Filter by assay_titles
        :type assay_titles: List[str]
        :param assembly: Filter by assembly
        :type assembly: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param barcode_map_for: Filter by barcode_map_for
        :type barcode_map_for: List[str]
        :param cell_type_annotation_id: Filter by cell_type_annotation.@id
        :type cell_type_annotation_id: List[str]
        :param cell_type_annotation_term_name: Filter by cell_type_annotation.term_name
        :type cell_type_annotation_term_name: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param content_md5sum: Filter by content_md5sum
        :type content_md5sum: List[str]
        :param content_type: Filter by content_type
        :type content_type: List[str]
        :param controlled_access: Filter by controlled_access
        :type controlled_access: List[bool]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param derived_from: Filter by derived_from
        :type derived_from: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_format: Filter by file_format
        :type file_format: List[str]
        :param file_format_specifications: Filter by file_format_specifications
        :type file_format_specifications: List[str]
        :param file_format_type: Filter by file_format_type
        :type file_format_type: List[str]
        :param file_set_id: Filter by file_set.@id
        :type file_set_id: List[str]
        :param file_set_accession: Filter by file_set.accession
        :type file_set_accession: List[str]
        :param file_set_file_set_type: Filter by file_set.file_set_type
        :type file_set_file_set_type: List[str]
        :param file_set_samples_id: Filter by file_set.samples.@id
        :type file_set_samples_id: List[str]
        :param file_set_samples_accession: Filter by file_set.samples.accession
        :type file_set_samples_accession: List[str]
        :param file_set_samples_classifications: Filter by file_set.samples.classifications
        :type file_set_samples_classifications: List[str]
        :param file_set_samples_disease_terms_id: Filter by file_set.samples.disease_terms.@id
        :type file_set_samples_disease_terms_id: List[str]
        :param file_set_samples_disease_terms_summary: Filter by file_set.samples.disease_terms.summary
        :type file_set_samples_disease_terms_summary: List[str]
        :param file_set_samples_disease_terms_term_name: Filter by file_set.samples.disease_terms.term_name
        :type file_set_samples_disease_terms_term_name: List[str]
        :param file_set_samples_sample_terms_id: Filter by file_set.samples.sample_terms.@id
        :type file_set_samples_sample_terms_id: List[str]
        :param file_set_samples_sample_terms_term_name: Filter by file_set.samples.sample_terms.term_name
        :type file_set_samples_sample_terms_term_name: List[str]
        :param file_set_samples_summary: Filter by file_set.samples.summary
        :type file_set_samples_summary: List[str]
        :param file_set_samples_taxa: Filter by file_set.samples.taxa
        :type file_set_samples_taxa: List[str]
        :param file_set_summary: Filter by file_set.summary
        :type file_set_summary: List[str]
        :param file_set_taxa: Filter by file_set.taxa
        :type file_set_taxa: List[str]
        :param file_size: Filter by file_size
        :type file_size: List[int]
        :param gene_list_for: Filter by gene_list_for
        :type gene_list_for: List[str]
        :param href: Filter by href
        :type href: List[str]
        :param input_file_for: Filter by input_file_for
        :type input_file_for: List[str]
        :param integrated_in_id: Filter by integrated_in.@id
        :type integrated_in_id: List[str]
        :param integrated_in_associated_phenotypes_id: Filter by integrated_in.associated_phenotypes.@id
        :type integrated_in_associated_phenotypes_id: List[str]
        :param integrated_in_associated_phenotypes_summary: Filter by integrated_in.associated_phenotypes.summary
        :type integrated_in_associated_phenotypes_summary: List[str]
        :param integrated_in_associated_phenotypes_term_name: Filter by integrated_in.associated_phenotypes.term_name
        :type integrated_in_associated_phenotypes_term_name: List[str]
        :param integrated_in_file_set_type: Filter by integrated_in.file_set_type
        :type integrated_in_file_set_type: List[str]
        :param integrated_in_small_scale_gene_list_id: Filter by integrated_in.small_scale_gene_list.@id
        :type integrated_in_small_scale_gene_list_id: List[str]
        :param integrated_in_small_scale_gene_list_symbol: Filter by integrated_in.small_scale_gene_list.symbol
        :type integrated_in_small_scale_gene_list_symbol: List[str]
        :param integrated_in_summary: Filter by integrated_in.summary
        :type integrated_in_summary: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param loci_list_for: Filter by loci_list_for
        :type loci_list_for: List[str]
        :param md5sum: Filter by md5sum
        :type md5sum: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param s3_uri: Filter by s3_uri
        :type s3_uri: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitted_file_name: Filter by submitted_file_name
        :type submitted_file_name: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param transcriptome_annotation: Filter by transcriptome_annotation
        :type transcriptome_annotation: List[str]
        :param upload_status: Filter by upload_status
        :type upload_status: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param validation_error_detail: Filter by validation_error_detail
        :type validation_error_detail: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._tabular_files_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_step_version=analysis_step_version,
            anvil_url=anvil_url,
            assay_titles=assay_titles,
            assembly=assembly,
            award_id=award_id,
            award_component=award_component,
            barcode_map_for=barcode_map_for,
            cell_type_annotation_id=cell_type_annotation_id,
            cell_type_annotation_term_name=cell_type_annotation_term_name,
            collections=collections,
            content_md5sum=content_md5sum,
            content_type=content_type,
            controlled_access=controlled_access,
            creation_timestamp=creation_timestamp,
            dbxrefs=dbxrefs,
            derived_from=derived_from,
            description=description,
            documents=documents,
            file_format=file_format,
            file_format_specifications=file_format_specifications,
            file_format_type=file_format_type,
            file_set_id=file_set_id,
            file_set_accession=file_set_accession,
            file_set_file_set_type=file_set_file_set_type,
            file_set_samples_id=file_set_samples_id,
            file_set_samples_accession=file_set_samples_accession,
            file_set_samples_classifications=file_set_samples_classifications,
            file_set_samples_disease_terms_id=file_set_samples_disease_terms_id,
            file_set_samples_disease_terms_summary=file_set_samples_disease_terms_summary,
            file_set_samples_disease_terms_term_name=file_set_samples_disease_terms_term_name,
            file_set_samples_sample_terms_id=file_set_samples_sample_terms_id,
            file_set_samples_sample_terms_term_name=file_set_samples_sample_terms_term_name,
            file_set_samples_summary=file_set_samples_summary,
            file_set_samples_taxa=file_set_samples_taxa,
            file_set_summary=file_set_summary,
            file_set_taxa=file_set_taxa,
            file_size=file_size,
            gene_list_for=gene_list_for,
            href=href,
            input_file_for=input_file_for,
            integrated_in_id=integrated_in_id,
            integrated_in_associated_phenotypes_id=integrated_in_associated_phenotypes_id,
            integrated_in_associated_phenotypes_summary=integrated_in_associated_phenotypes_summary,
            integrated_in_associated_phenotypes_term_name=integrated_in_associated_phenotypes_term_name,
            integrated_in_file_set_type=integrated_in_file_set_type,
            integrated_in_small_scale_gene_list_id=integrated_in_small_scale_gene_list_id,
            integrated_in_small_scale_gene_list_symbol=integrated_in_small_scale_gene_list_symbol,
            integrated_in_summary=integrated_in_summary,
            lab_id=lab_id,
            lab_title=lab_title,
            loci_list_for=loci_list_for,
            md5sum=md5sum,
            notes=notes,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            s3_uri=s3_uri,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitted_file_name=submitted_file_name,
            submitter_comment=submitter_comment,
            summary=summary,
            transcriptome_annotation=transcriptome_annotation,
            upload_status=upload_status,
            uuid=uuid,
            validation_error_detail=validation_error_detail,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TabularFileResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _tabular_files_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_step_version,
        anvil_url,
        assay_titles,
        assembly,
        award_id,
        award_component,
        barcode_map_for,
        cell_type_annotation_id,
        cell_type_annotation_term_name,
        collections,
        content_md5sum,
        content_type,
        controlled_access,
        creation_timestamp,
        dbxrefs,
        derived_from,
        description,
        documents,
        file_format,
        file_format_specifications,
        file_format_type,
        file_set_id,
        file_set_accession,
        file_set_file_set_type,
        file_set_samples_id,
        file_set_samples_accession,
        file_set_samples_classifications,
        file_set_samples_disease_terms_id,
        file_set_samples_disease_terms_summary,
        file_set_samples_disease_terms_term_name,
        file_set_samples_sample_terms_id,
        file_set_samples_sample_terms_term_name,
        file_set_samples_summary,
        file_set_samples_taxa,
        file_set_summary,
        file_set_taxa,
        file_size,
        gene_list_for,
        href,
        input_file_for,
        integrated_in_id,
        integrated_in_associated_phenotypes_id,
        integrated_in_associated_phenotypes_summary,
        integrated_in_associated_phenotypes_term_name,
        integrated_in_file_set_type,
        integrated_in_small_scale_gene_list_id,
        integrated_in_small_scale_gene_list_symbol,
        integrated_in_summary,
        lab_id,
        lab_title,
        loci_list_for,
        md5sum,
        notes,
        release_timestamp,
        revoke_detail,
        s3_uri,
        status,
        submitted_by_id,
        submitted_by_title,
        submitted_file_name,
        submitter_comment,
        summary,
        transcriptome_annotation,
        upload_status,
        uuid,
        validation_error_detail,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_step_version': 'multi',
            'anvil_url': 'multi',
            'assay_titles': 'multi',
            'assembly': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'barcode_map_for': 'multi',
            'cell_type_annotation.@id': 'multi',
            'cell_type_annotation.term_name': 'multi',
            'collections': 'multi',
            'content_md5sum': 'multi',
            'content_type': 'multi',
            'controlled_access': 'multi',
            'creation_timestamp': 'multi',
            'dbxrefs': 'multi',
            'derived_from': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_format': 'multi',
            'file_format_specifications': 'multi',
            'file_format_type': 'multi',
            'file_set.@id': 'multi',
            'file_set.accession': 'multi',
            'file_set.file_set_type': 'multi',
            'file_set.samples.@id': 'multi',
            'file_set.samples.accession': 'multi',
            'file_set.samples.classifications': 'multi',
            'file_set.samples.disease_terms.@id': 'multi',
            'file_set.samples.disease_terms.summary': 'multi',
            'file_set.samples.disease_terms.term_name': 'multi',
            'file_set.samples.sample_terms.@id': 'multi',
            'file_set.samples.sample_terms.term_name': 'multi',
            'file_set.samples.summary': 'multi',
            'file_set.samples.taxa': 'multi',
            'file_set.summary': 'multi',
            'file_set.taxa': 'multi',
            'file_size': 'multi',
            'gene_list_for': 'multi',
            'href': 'multi',
            'input_file_for': 'multi',
            'integrated_in.@id': 'multi',
            'integrated_in.associated_phenotypes.@id': 'multi',
            'integrated_in.associated_phenotypes.summary': 'multi',
            'integrated_in.associated_phenotypes.term_name': 'multi',
            'integrated_in.file_set_type': 'multi',
            'integrated_in.small_scale_gene_list.@id': 'multi',
            'integrated_in.small_scale_gene_list.symbol': 'multi',
            'integrated_in.summary': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'loci_list_for': 'multi',
            'md5sum': 'multi',
            'notes': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            's3_uri': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitted_file_name': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'transcriptome_annotation': 'multi',
            'upload_status': 'multi',
            'uuid': 'multi',
            'validation_error_detail': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_step_version is not None:
            
            _query_params.append(('analysis_step_version', analysis_step_version))
            
        if anvil_url is not None:
            
            _query_params.append(('anvil_url', anvil_url))
            
        if assay_titles is not None:
            
            _query_params.append(('assay_titles', assay_titles))
            
        if assembly is not None:
            
            _query_params.append(('assembly', assembly))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if barcode_map_for is not None:
            
            _query_params.append(('barcode_map_for', barcode_map_for))
            
        if cell_type_annotation_id is not None:
            
            _query_params.append(('cell_type_annotation.@id', cell_type_annotation_id))
            
        if cell_type_annotation_term_name is not None:
            
            _query_params.append(('cell_type_annotation.term_name', cell_type_annotation_term_name))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if content_md5sum is not None:
            
            _query_params.append(('content_md5sum', content_md5sum))
            
        if content_type is not None:
            
            _query_params.append(('content_type', content_type))
            
        if controlled_access is not None:
            
            _query_params.append(('controlled_access', controlled_access))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if derived_from is not None:
            
            _query_params.append(('derived_from', derived_from))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_format is not None:
            
            _query_params.append(('file_format', file_format))
            
        if file_format_specifications is not None:
            
            _query_params.append(('file_format_specifications', file_format_specifications))
            
        if file_format_type is not None:
            
            _query_params.append(('file_format_type', file_format_type))
            
        if file_set_id is not None:
            
            _query_params.append(('file_set.@id', file_set_id))
            
        if file_set_accession is not None:
            
            _query_params.append(('file_set.accession', file_set_accession))
            
        if file_set_file_set_type is not None:
            
            _query_params.append(('file_set.file_set_type', file_set_file_set_type))
            
        if file_set_samples_id is not None:
            
            _query_params.append(('file_set.samples.@id', file_set_samples_id))
            
        if file_set_samples_accession is not None:
            
            _query_params.append(('file_set.samples.accession', file_set_samples_accession))
            
        if file_set_samples_classifications is not None:
            
            _query_params.append(('file_set.samples.classifications', file_set_samples_classifications))
            
        if file_set_samples_disease_terms_id is not None:
            
            _query_params.append(('file_set.samples.disease_terms.@id', file_set_samples_disease_terms_id))
            
        if file_set_samples_disease_terms_summary is not None:
            
            _query_params.append(('file_set.samples.disease_terms.summary', file_set_samples_disease_terms_summary))
            
        if file_set_samples_disease_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.disease_terms.term_name', file_set_samples_disease_terms_term_name))
            
        if file_set_samples_sample_terms_id is not None:
            
            _query_params.append(('file_set.samples.sample_terms.@id', file_set_samples_sample_terms_id))
            
        if file_set_samples_sample_terms_term_name is not None:
            
            _query_params.append(('file_set.samples.sample_terms.term_name', file_set_samples_sample_terms_term_name))
            
        if file_set_samples_summary is not None:
            
            _query_params.append(('file_set.samples.summary', file_set_samples_summary))
            
        if file_set_samples_taxa is not None:
            
            _query_params.append(('file_set.samples.taxa', file_set_samples_taxa))
            
        if file_set_summary is not None:
            
            _query_params.append(('file_set.summary', file_set_summary))
            
        if file_set_taxa is not None:
            
            _query_params.append(('file_set.taxa', file_set_taxa))
            
        if file_size is not None:
            
            _query_params.append(('file_size', file_size))
            
        if gene_list_for is not None:
            
            _query_params.append(('gene_list_for', gene_list_for))
            
        if href is not None:
            
            _query_params.append(('href', href))
            
        if input_file_for is not None:
            
            _query_params.append(('input_file_for', input_file_for))
            
        if integrated_in_id is not None:
            
            _query_params.append(('integrated_in.@id', integrated_in_id))
            
        if integrated_in_associated_phenotypes_id is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.@id', integrated_in_associated_phenotypes_id))
            
        if integrated_in_associated_phenotypes_summary is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.summary', integrated_in_associated_phenotypes_summary))
            
        if integrated_in_associated_phenotypes_term_name is not None:
            
            _query_params.append(('integrated_in.associated_phenotypes.term_name', integrated_in_associated_phenotypes_term_name))
            
        if integrated_in_file_set_type is not None:
            
            _query_params.append(('integrated_in.file_set_type', integrated_in_file_set_type))
            
        if integrated_in_small_scale_gene_list_id is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.@id', integrated_in_small_scale_gene_list_id))
            
        if integrated_in_small_scale_gene_list_symbol is not None:
            
            _query_params.append(('integrated_in.small_scale_gene_list.symbol', integrated_in_small_scale_gene_list_symbol))
            
        if integrated_in_summary is not None:
            
            _query_params.append(('integrated_in.summary', integrated_in_summary))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if loci_list_for is not None:
            
            _query_params.append(('loci_list_for', loci_list_for))
            
        if md5sum is not None:
            
            _query_params.append(('md5sum', md5sum))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if s3_uri is not None:
            
            _query_params.append(('s3_uri', s3_uri))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitted_file_name is not None:
            
            _query_params.append(('submitted_file_name', submitted_file_name))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if transcriptome_annotation is not None:
            
            _query_params.append(('transcriptome_annotation', transcriptome_annotation))
            
        if upload_status is not None:
            
            _query_params.append(('upload_status', upload_status))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if validation_error_detail is not None:
            
            _query_params.append(('validation_error_detail', validation_error_detail))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/tabular-files/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def technical_samples(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_material: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_material")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> TechnicalSampleResults:
        """List items in the TechnicalSample collection.

        Collection endpoint that accepts various query parameters to filter and sort TechnicalSample items. Supports filtering on fields within TechnicalSample items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates: Filter by institutional_certificates
        :type institutional_certificates: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_material: Filter by sample_material
        :type sample_material: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._technical_samples_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates=institutional_certificates,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_material=sample_material,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TechnicalSampleResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def technical_samples_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_material: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_material")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[TechnicalSampleResults]:
        """List items in the TechnicalSample collection.

        Collection endpoint that accepts various query parameters to filter and sort TechnicalSample items. Supports filtering on fields within TechnicalSample items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates: Filter by institutional_certificates
        :type institutional_certificates: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_material: Filter by sample_material
        :type sample_material: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._technical_samples_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates=institutional_certificates,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_material=sample_material,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TechnicalSampleResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def technical_samples_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_material: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_material")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the TechnicalSample collection.

        Collection endpoint that accepts various query parameters to filter and sort TechnicalSample items. Supports filtering on fields within TechnicalSample items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates: Filter by institutional_certificates
        :type institutional_certificates: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_material: Filter by sample_material
        :type sample_material: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._technical_samples_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            documents=documents,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates=institutional_certificates,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_material=sample_material,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TechnicalSampleResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _technical_samples_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        classifications,
        collections,
        construct_library_sets,
        creation_timestamp,
        date_obtained,
        dbxrefs,
        description,
        documents,
        file_sets_id,
        file_sets_accession,
        file_sets_aliases,
        file_sets_assay_term_term_name,
        file_sets_lab_title,
        file_sets_status,
        file_sets_summary,
        institutional_certificates,
        lab_id,
        lab_title,
        lot_id,
        moi,
        multiplexed_in_id,
        multiplexed_in_accession,
        notes,
        nucleic_acid_delivery,
        origin_of,
        product_id,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_material,
        sample_terms_id,
        sample_terms_term_name,
        sorted_fractions,
        sorted_from_id,
        sorted_from_accession,
        sorted_from_detail,
        sources_id,
        starting_amount,
        starting_amount_units,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        taxa,
        time_post_library_delivery,
        time_post_library_delivery_units,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'classifications': 'multi',
            'collections': 'multi',
            'construct_library_sets': 'multi',
            'creation_timestamp': 'multi',
            'date_obtained': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'file_sets.@id': 'multi',
            'file_sets.accession': 'multi',
            'file_sets.aliases': 'multi',
            'file_sets.assay_term.term_name': 'multi',
            'file_sets.lab.title': 'multi',
            'file_sets.status': 'multi',
            'file_sets.summary': 'multi',
            'institutional_certificates': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'moi': 'multi',
            'multiplexed_in.@id': 'multi',
            'multiplexed_in.accession': 'multi',
            'notes': 'multi',
            'nucleic_acid_delivery': 'multi',
            'origin_of': 'multi',
            'product_id': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_material': 'multi',
            'sample_terms.@id': 'multi',
            'sample_terms.term_name': 'multi',
            'sorted_fractions': 'multi',
            'sorted_from.@id': 'multi',
            'sorted_from.accession': 'multi',
            'sorted_from_detail': 'multi',
            'sources.@id': 'multi',
            'starting_amount': 'multi',
            'starting_amount_units': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'time_post_library_delivery': 'multi',
            'time_post_library_delivery_units': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if classifications is not None:
            
            _query_params.append(('classifications', classifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if construct_library_sets is not None:
            
            _query_params.append(('construct_library_sets', construct_library_sets))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_obtained is not None:
            
            _query_params.append(('date_obtained', date_obtained))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if file_sets_id is not None:
            
            _query_params.append(('file_sets.@id', file_sets_id))
            
        if file_sets_accession is not None:
            
            _query_params.append(('file_sets.accession', file_sets_accession))
            
        if file_sets_aliases is not None:
            
            _query_params.append(('file_sets.aliases', file_sets_aliases))
            
        if file_sets_assay_term_term_name is not None:
            
            _query_params.append(('file_sets.assay_term.term_name', file_sets_assay_term_term_name))
            
        if file_sets_lab_title is not None:
            
            _query_params.append(('file_sets.lab.title', file_sets_lab_title))
            
        if file_sets_status is not None:
            
            _query_params.append(('file_sets.status', file_sets_status))
            
        if file_sets_summary is not None:
            
            _query_params.append(('file_sets.summary', file_sets_summary))
            
        if institutional_certificates is not None:
            
            _query_params.append(('institutional_certificates', institutional_certificates))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if moi is not None:
            
            _query_params.append(('moi', moi))
            
        if multiplexed_in_id is not None:
            
            _query_params.append(('multiplexed_in.@id', multiplexed_in_id))
            
        if multiplexed_in_accession is not None:
            
            _query_params.append(('multiplexed_in.accession', multiplexed_in_accession))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if nucleic_acid_delivery is not None:
            
            _query_params.append(('nucleic_acid_delivery', nucleic_acid_delivery))
            
        if origin_of is not None:
            
            _query_params.append(('origin_of', origin_of))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_material is not None:
            
            _query_params.append(('sample_material', sample_material))
            
        if sample_terms_id is not None:
            
            _query_params.append(('sample_terms.@id', sample_terms_id))
            
        if sample_terms_term_name is not None:
            
            _query_params.append(('sample_terms.term_name', sample_terms_term_name))
            
        if sorted_fractions is not None:
            
            _query_params.append(('sorted_fractions', sorted_fractions))
            
        if sorted_from_id is not None:
            
            _query_params.append(('sorted_from.@id', sorted_from_id))
            
        if sorted_from_accession is not None:
            
            _query_params.append(('sorted_from.accession', sorted_from_accession))
            
        if sorted_from_detail is not None:
            
            _query_params.append(('sorted_from_detail', sorted_from_detail))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if starting_amount is not None:
            
            _query_params.append(('starting_amount', starting_amount))
            
        if starting_amount_units is not None:
            
            _query_params.append(('starting_amount_units', starting_amount_units))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if time_post_library_delivery is not None:
            
            _query_params.append(('time_post_library_delivery', time_post_library_delivery))
            
        if time_post_library_delivery_units is not None:
            
            _query_params.append(('time_post_library_delivery_units', time_post_library_delivery_units))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/technical-samples/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def tissues(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        ccf_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by ccf_id")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        pmi: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by pmi")] = None,
        pmi_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by pmi_units")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        preservation_method: Annotated[Optional[List[StrictStr]], Field(description="Filter by preservation_method")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> TissueResults:
        """List items in the Tissue collection.

        Collection endpoint that accepts various query parameters to filter and sort Tissue items. Supports filtering on fields within Tissue items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param ccf_id: Filter by ccf_id
        :type ccf_id: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param pmi: Filter by pmi
        :type pmi: List[int]
        :param pmi_units: Filter by pmi_units
        :type pmi_units: List[str]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param preservation_method: Filter by preservation_method
        :type preservation_method: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._tissues_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            ccf_id=ccf_id,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            pmi=pmi,
            pmi_units=pmi_units,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            preservation_method=preservation_method,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TissueResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def tissues_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        ccf_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by ccf_id")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        pmi: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by pmi")] = None,
        pmi_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by pmi_units")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        preservation_method: Annotated[Optional[List[StrictStr]], Field(description="Filter by preservation_method")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[TissueResults]:
        """List items in the Tissue collection.

        Collection endpoint that accepts various query parameters to filter and sort Tissue items. Supports filtering on fields within Tissue items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param ccf_id: Filter by ccf_id
        :type ccf_id: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param pmi: Filter by pmi
        :type pmi: List[int]
        :param pmi_units: Filter by pmi_units
        :type pmi_units: List[str]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param preservation_method: Filter by preservation_method
        :type preservation_method: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._tissues_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            ccf_id=ccf_id,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            pmi=pmi,
            pmi_units=pmi_units,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            preservation_method=preservation_method,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TissueResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def tissues_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        ccf_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by ccf_id")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        pmi: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by pmi")] = None,
        pmi_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by pmi_units")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        preservation_method: Annotated[Optional[List[StrictStr]], Field(description="Filter by preservation_method")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Tissue collection.

        Collection endpoint that accepts various query parameters to filter and sort Tissue items. Supports filtering on fields within Tissue items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param ccf_id: Filter by ccf_id
        :type ccf_id: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param pmi: Filter by pmi
        :type pmi: List[int]
        :param pmi_units: Filter by pmi_units
        :type pmi_units: List[str]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param preservation_method: Filter by preservation_method
        :type preservation_method: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._tissues_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            ccf_id=ccf_id,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            pmi=pmi,
            pmi_units=pmi_units,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            preservation_method=preservation_method,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TissueResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _tissues_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        age,
        age_units,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        biomarkers,
        ccf_id,
        cellular_sub_pool,
        classifications,
        collections,
        construct_library_sets,
        creation_timestamp,
        date_obtained,
        dbxrefs,
        description,
        disease_terms_id,
        disease_terms_term_name,
        documents,
        donors,
        embryonic,
        file_sets_id,
        file_sets_accession,
        file_sets_aliases,
        file_sets_assay_term_term_name,
        file_sets_lab_title,
        file_sets_status,
        file_sets_summary,
        institutional_certificates_id,
        institutional_certificates_certificate_identifier,
        lab_id,
        lab_title,
        lot_id,
        lower_bound_age,
        lower_bound_age_in_hours,
        modifications_id,
        modifications_status,
        modifications_summary,
        moi,
        multiplexed_in_id,
        multiplexed_in_accession,
        notes,
        nucleic_acid_delivery,
        origin_of,
        originated_from,
        part_of,
        parts,
        pmi,
        pmi_units,
        pooled_from,
        pooled_in,
        preservation_method,
        product_id,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_terms_id,
        sample_terms_term_name,
        sex,
        sorted_fractions,
        sorted_from_id,
        sorted_from_accession,
        sorted_from_detail,
        sources_id,
        starting_amount,
        starting_amount_units,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        taxa,
        time_post_library_delivery,
        time_post_library_delivery_units,
        treatments_id,
        treatments_purpose,
        treatments_status,
        treatments_summary,
        treatments_treatment_type,
        upper_bound_age,
        upper_bound_age_in_hours,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'age': 'multi',
            'age_units': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biomarkers': 'multi',
            'ccf_id': 'multi',
            'cellular_sub_pool': 'multi',
            'classifications': 'multi',
            'collections': 'multi',
            'construct_library_sets': 'multi',
            'creation_timestamp': 'multi',
            'date_obtained': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'disease_terms.@id': 'multi',
            'disease_terms.term_name': 'multi',
            'documents': 'multi',
            'donors': 'multi',
            'embryonic': 'multi',
            'file_sets.@id': 'multi',
            'file_sets.accession': 'multi',
            'file_sets.aliases': 'multi',
            'file_sets.assay_term.term_name': 'multi',
            'file_sets.lab.title': 'multi',
            'file_sets.status': 'multi',
            'file_sets.summary': 'multi',
            'institutional_certificates.@id': 'multi',
            'institutional_certificates.certificate_identifier': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'lower_bound_age': 'multi',
            'lower_bound_age_in_hours': 'multi',
            'modifications.@id': 'multi',
            'modifications.status': 'multi',
            'modifications.summary': 'multi',
            'moi': 'multi',
            'multiplexed_in.@id': 'multi',
            'multiplexed_in.accession': 'multi',
            'notes': 'multi',
            'nucleic_acid_delivery': 'multi',
            'origin_of': 'multi',
            'originated_from': 'multi',
            'part_of': 'multi',
            'parts': 'multi',
            'pmi': 'multi',
            'pmi_units': 'multi',
            'pooled_from': 'multi',
            'pooled_in': 'multi',
            'preservation_method': 'multi',
            'product_id': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_terms.@id': 'multi',
            'sample_terms.term_name': 'multi',
            'sex': 'multi',
            'sorted_fractions': 'multi',
            'sorted_from.@id': 'multi',
            'sorted_from.accession': 'multi',
            'sorted_from_detail': 'multi',
            'sources.@id': 'multi',
            'starting_amount': 'multi',
            'starting_amount_units': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'time_post_library_delivery': 'multi',
            'time_post_library_delivery_units': 'multi',
            'treatments.@id': 'multi',
            'treatments.purpose': 'multi',
            'treatments.status': 'multi',
            'treatments.summary': 'multi',
            'treatments.treatment_type': 'multi',
            'upper_bound_age': 'multi',
            'upper_bound_age_in_hours': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if age is not None:
            
            _query_params.append(('age', age))
            
        if age_units is not None:
            
            _query_params.append(('age_units', age_units))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biomarkers is not None:
            
            _query_params.append(('biomarkers', biomarkers))
            
        if ccf_id is not None:
            
            _query_params.append(('ccf_id', ccf_id))
            
        if cellular_sub_pool is not None:
            
            _query_params.append(('cellular_sub_pool', cellular_sub_pool))
            
        if classifications is not None:
            
            _query_params.append(('classifications', classifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if construct_library_sets is not None:
            
            _query_params.append(('construct_library_sets', construct_library_sets))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_obtained is not None:
            
            _query_params.append(('date_obtained', date_obtained))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if disease_terms_id is not None:
            
            _query_params.append(('disease_terms.@id', disease_terms_id))
            
        if disease_terms_term_name is not None:
            
            _query_params.append(('disease_terms.term_name', disease_terms_term_name))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors is not None:
            
            _query_params.append(('donors', donors))
            
        if embryonic is not None:
            
            _query_params.append(('embryonic', embryonic))
            
        if file_sets_id is not None:
            
            _query_params.append(('file_sets.@id', file_sets_id))
            
        if file_sets_accession is not None:
            
            _query_params.append(('file_sets.accession', file_sets_accession))
            
        if file_sets_aliases is not None:
            
            _query_params.append(('file_sets.aliases', file_sets_aliases))
            
        if file_sets_assay_term_term_name is not None:
            
            _query_params.append(('file_sets.assay_term.term_name', file_sets_assay_term_term_name))
            
        if file_sets_lab_title is not None:
            
            _query_params.append(('file_sets.lab.title', file_sets_lab_title))
            
        if file_sets_status is not None:
            
            _query_params.append(('file_sets.status', file_sets_status))
            
        if file_sets_summary is not None:
            
            _query_params.append(('file_sets.summary', file_sets_summary))
            
        if institutional_certificates_id is not None:
            
            _query_params.append(('institutional_certificates.@id', institutional_certificates_id))
            
        if institutional_certificates_certificate_identifier is not None:
            
            _query_params.append(('institutional_certificates.certificate_identifier', institutional_certificates_certificate_identifier))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if lower_bound_age is not None:
            
            _query_params.append(('lower_bound_age', lower_bound_age))
            
        if lower_bound_age_in_hours is not None:
            
            _query_params.append(('lower_bound_age_in_hours', lower_bound_age_in_hours))
            
        if modifications_id is not None:
            
            _query_params.append(('modifications.@id', modifications_id))
            
        if modifications_status is not None:
            
            _query_params.append(('modifications.status', modifications_status))
            
        if modifications_summary is not None:
            
            _query_params.append(('modifications.summary', modifications_summary))
            
        if moi is not None:
            
            _query_params.append(('moi', moi))
            
        if multiplexed_in_id is not None:
            
            _query_params.append(('multiplexed_in.@id', multiplexed_in_id))
            
        if multiplexed_in_accession is not None:
            
            _query_params.append(('multiplexed_in.accession', multiplexed_in_accession))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if nucleic_acid_delivery is not None:
            
            _query_params.append(('nucleic_acid_delivery', nucleic_acid_delivery))
            
        if origin_of is not None:
            
            _query_params.append(('origin_of', origin_of))
            
        if originated_from is not None:
            
            _query_params.append(('originated_from', originated_from))
            
        if part_of is not None:
            
            _query_params.append(('part_of', part_of))
            
        if parts is not None:
            
            _query_params.append(('parts', parts))
            
        if pmi is not None:
            
            _query_params.append(('pmi', pmi))
            
        if pmi_units is not None:
            
            _query_params.append(('pmi_units', pmi_units))
            
        if pooled_from is not None:
            
            _query_params.append(('pooled_from', pooled_from))
            
        if pooled_in is not None:
            
            _query_params.append(('pooled_in', pooled_in))
            
        if preservation_method is not None:
            
            _query_params.append(('preservation_method', preservation_method))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_terms_id is not None:
            
            _query_params.append(('sample_terms.@id', sample_terms_id))
            
        if sample_terms_term_name is not None:
            
            _query_params.append(('sample_terms.term_name', sample_terms_term_name))
            
        if sex is not None:
            
            _query_params.append(('sex', sex))
            
        if sorted_fractions is not None:
            
            _query_params.append(('sorted_fractions', sorted_fractions))
            
        if sorted_from_id is not None:
            
            _query_params.append(('sorted_from.@id', sorted_from_id))
            
        if sorted_from_accession is not None:
            
            _query_params.append(('sorted_from.accession', sorted_from_accession))
            
        if sorted_from_detail is not None:
            
            _query_params.append(('sorted_from_detail', sorted_from_detail))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if starting_amount is not None:
            
            _query_params.append(('starting_amount', starting_amount))
            
        if starting_amount_units is not None:
            
            _query_params.append(('starting_amount_units', starting_amount_units))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if time_post_library_delivery is not None:
            
            _query_params.append(('time_post_library_delivery', time_post_library_delivery))
            
        if time_post_library_delivery_units is not None:
            
            _query_params.append(('time_post_library_delivery_units', time_post_library_delivery_units))
            
        if treatments_id is not None:
            
            _query_params.append(('treatments.@id', treatments_id))
            
        if treatments_purpose is not None:
            
            _query_params.append(('treatments.purpose', treatments_purpose))
            
        if treatments_status is not None:
            
            _query_params.append(('treatments.status', treatments_status))
            
        if treatments_summary is not None:
            
            _query_params.append(('treatments.summary', treatments_summary))
            
        if treatments_treatment_type is not None:
            
            _query_params.append(('treatments.treatment_type', treatments_treatment_type))
            
        if upper_bound_age is not None:
            
            _query_params.append(('upper_bound_age', upper_bound_age))
            
        if upper_bound_age_in_hours is not None:
            
            _query_params.append(('upper_bound_age_in_hours', upper_bound_age_in_hours))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/tissues/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def treatments(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by amount")] = None,
        amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by amount_units")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_treated: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_treated")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        depletion: Annotated[Optional[List[StrictBool]], Field(description="Filter by depletion")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        duration: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by duration")] = None,
        duration_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by duration_units")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        p_h: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by pH")] = None,
        post_treatment_time: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by post_treatment_time")] = None,
        post_treatment_time_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by post_treatment_time_units")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by purpose")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        temperature: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by temperature")] = None,
        temperature_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by temperature_units")] = None,
        treatment_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by treatment_term_id")] = None,
        treatment_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatment_term_name")] = None,
        treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatment_type")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> TreatmentResults:
        """List items in the Treatment collection.

        Collection endpoint that accepts various query parameters to filter and sort Treatment items. Supports filtering on fields within Treatment items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param amount: Filter by amount
        :type amount: List[float]
        :param amount_units: Filter by amount_units
        :type amount_units: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_treated: Filter by biosamples_treated
        :type biosamples_treated: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param depletion: Filter by depletion
        :type depletion: List[bool]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param duration: Filter by duration
        :type duration: List[float]
        :param duration_units: Filter by duration_units
        :type duration_units: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param p_h: Filter by pH
        :type p_h: List[float]
        :param post_treatment_time: Filter by post_treatment_time
        :type post_treatment_time: List[float]
        :param post_treatment_time_units: Filter by post_treatment_time_units
        :type post_treatment_time_units: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param purpose: Filter by purpose
        :type purpose: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param temperature: Filter by temperature
        :type temperature: List[float]
        :param temperature_units: Filter by temperature_units
        :type temperature_units: List[str]
        :param treatment_term_id: Filter by treatment_term_id
        :type treatment_term_id: List[str]
        :param treatment_term_name: Filter by treatment_term_name
        :type treatment_term_name: List[str]
        :param treatment_type: Filter by treatment_type
        :type treatment_type: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._treatments_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            amount=amount,
            amount_units=amount_units,
            award_id=award_id,
            award_component=award_component,
            biosamples_treated=biosamples_treated,
            creation_timestamp=creation_timestamp,
            depletion=depletion,
            description=description,
            documents=documents,
            duration=duration,
            duration_units=duration_units,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            notes=notes,
            p_h=p_h,
            post_treatment_time=post_treatment_time,
            post_treatment_time_units=post_treatment_time_units,
            product_id=product_id,
            purpose=purpose,
            release_timestamp=release_timestamp,
            sources_id=sources_id,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            temperature=temperature,
            temperature_units=temperature_units,
            treatment_term_id=treatment_term_id,
            treatment_term_name=treatment_term_name,
            treatment_type=treatment_type,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TreatmentResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def treatments_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by amount")] = None,
        amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by amount_units")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_treated: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_treated")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        depletion: Annotated[Optional[List[StrictBool]], Field(description="Filter by depletion")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        duration: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by duration")] = None,
        duration_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by duration_units")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        p_h: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by pH")] = None,
        post_treatment_time: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by post_treatment_time")] = None,
        post_treatment_time_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by post_treatment_time_units")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by purpose")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        temperature: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by temperature")] = None,
        temperature_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by temperature_units")] = None,
        treatment_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by treatment_term_id")] = None,
        treatment_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatment_term_name")] = None,
        treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatment_type")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[TreatmentResults]:
        """List items in the Treatment collection.

        Collection endpoint that accepts various query parameters to filter and sort Treatment items. Supports filtering on fields within Treatment items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param amount: Filter by amount
        :type amount: List[float]
        :param amount_units: Filter by amount_units
        :type amount_units: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_treated: Filter by biosamples_treated
        :type biosamples_treated: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param depletion: Filter by depletion
        :type depletion: List[bool]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param duration: Filter by duration
        :type duration: List[float]
        :param duration_units: Filter by duration_units
        :type duration_units: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param p_h: Filter by pH
        :type p_h: List[float]
        :param post_treatment_time: Filter by post_treatment_time
        :type post_treatment_time: List[float]
        :param post_treatment_time_units: Filter by post_treatment_time_units
        :type post_treatment_time_units: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param purpose: Filter by purpose
        :type purpose: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param temperature: Filter by temperature
        :type temperature: List[float]
        :param temperature_units: Filter by temperature_units
        :type temperature_units: List[str]
        :param treatment_term_id: Filter by treatment_term_id
        :type treatment_term_id: List[str]
        :param treatment_term_name: Filter by treatment_term_name
        :type treatment_term_name: List[str]
        :param treatment_type: Filter by treatment_type
        :type treatment_type: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._treatments_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            amount=amount,
            amount_units=amount_units,
            award_id=award_id,
            award_component=award_component,
            biosamples_treated=biosamples_treated,
            creation_timestamp=creation_timestamp,
            depletion=depletion,
            description=description,
            documents=documents,
            duration=duration,
            duration_units=duration_units,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            notes=notes,
            p_h=p_h,
            post_treatment_time=post_treatment_time,
            post_treatment_time_units=post_treatment_time_units,
            product_id=product_id,
            purpose=purpose,
            release_timestamp=release_timestamp,
            sources_id=sources_id,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            temperature=temperature,
            temperature_units=temperature_units,
            treatment_term_id=treatment_term_id,
            treatment_term_name=treatment_term_name,
            treatment_type=treatment_type,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TreatmentResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def treatments_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by amount")] = None,
        amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by amount_units")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biosamples_treated: Annotated[Optional[List[StrictStr]], Field(description="Filter by biosamples_treated")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        depletion: Annotated[Optional[List[StrictBool]], Field(description="Filter by depletion")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        duration: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by duration")] = None,
        duration_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by duration_units")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        p_h: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by pH")] = None,
        post_treatment_time: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by post_treatment_time")] = None,
        post_treatment_time_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by post_treatment_time_units")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by purpose")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        temperature: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by temperature")] = None,
        temperature_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by temperature_units")] = None,
        treatment_term_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by treatment_term_id")] = None,
        treatment_term_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatment_term_name")] = None,
        treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatment_type")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Treatment collection.

        Collection endpoint that accepts various query parameters to filter and sort Treatment items. Supports filtering on fields within Treatment items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param amount: Filter by amount
        :type amount: List[float]
        :param amount_units: Filter by amount_units
        :type amount_units: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biosamples_treated: Filter by biosamples_treated
        :type biosamples_treated: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param depletion: Filter by depletion
        :type depletion: List[bool]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param duration: Filter by duration
        :type duration: List[float]
        :param duration_units: Filter by duration_units
        :type duration_units: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param p_h: Filter by pH
        :type p_h: List[float]
        :param post_treatment_time: Filter by post_treatment_time
        :type post_treatment_time: List[float]
        :param post_treatment_time_units: Filter by post_treatment_time_units
        :type post_treatment_time_units: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param purpose: Filter by purpose
        :type purpose: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param temperature: Filter by temperature
        :type temperature: List[float]
        :param temperature_units: Filter by temperature_units
        :type temperature_units: List[str]
        :param treatment_term_id: Filter by treatment_term_id
        :type treatment_term_id: List[str]
        :param treatment_term_name: Filter by treatment_term_name
        :type treatment_term_name: List[str]
        :param treatment_type: Filter by treatment_type
        :type treatment_type: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._treatments_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            amount=amount,
            amount_units=amount_units,
            award_id=award_id,
            award_component=award_component,
            biosamples_treated=biosamples_treated,
            creation_timestamp=creation_timestamp,
            depletion=depletion,
            description=description,
            documents=documents,
            duration=duration,
            duration_units=duration_units,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            notes=notes,
            p_h=p_h,
            post_treatment_time=post_treatment_time,
            post_treatment_time_units=post_treatment_time_units,
            product_id=product_id,
            purpose=purpose,
            release_timestamp=release_timestamp,
            sources_id=sources_id,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            temperature=temperature,
            temperature_units=temperature_units,
            treatment_term_id=treatment_term_id,
            treatment_term_name=treatment_term_name,
            treatment_type=treatment_type,
            uuid=uuid,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "TreatmentResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _treatments_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        amount,
        amount_units,
        award_id,
        award_component,
        biosamples_treated,
        creation_timestamp,
        depletion,
        description,
        documents,
        duration,
        duration_units,
        lab_id,
        lab_title,
        lot_id,
        notes,
        p_h,
        post_treatment_time,
        post_treatment_time_units,
        product_id,
        purpose,
        release_timestamp,
        sources_id,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        temperature,
        temperature_units,
        treatment_term_id,
        treatment_term_name,
        treatment_type,
        uuid,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'amount': 'multi',
            'amount_units': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biosamples_treated': 'multi',
            'creation_timestamp': 'multi',
            'depletion': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'duration': 'multi',
            'duration_units': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'notes': 'multi',
            'pH': 'multi',
            'post_treatment_time': 'multi',
            'post_treatment_time_units': 'multi',
            'product_id': 'multi',
            'purpose': 'multi',
            'release_timestamp': 'multi',
            'sources.@id': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'temperature': 'multi',
            'temperature_units': 'multi',
            'treatment_term_id': 'multi',
            'treatment_term_name': 'multi',
            'treatment_type': 'multi',
            'uuid': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if amount is not None:
            
            _query_params.append(('amount', amount))
            
        if amount_units is not None:
            
            _query_params.append(('amount_units', amount_units))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biosamples_treated is not None:
            
            _query_params.append(('biosamples_treated', biosamples_treated))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if depletion is not None:
            
            _query_params.append(('depletion', depletion))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if duration is not None:
            
            _query_params.append(('duration', duration))
            
        if duration_units is not None:
            
            _query_params.append(('duration_units', duration_units))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if p_h is not None:
            
            _query_params.append(('pH', p_h))
            
        if post_treatment_time is not None:
            
            _query_params.append(('post_treatment_time', post_treatment_time))
            
        if post_treatment_time_units is not None:
            
            _query_params.append(('post_treatment_time_units', post_treatment_time_units))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if purpose is not None:
            
            _query_params.append(('purpose', purpose))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if temperature is not None:
            
            _query_params.append(('temperature', temperature))
            
        if temperature_units is not None:
            
            _query_params.append(('temperature_units', temperature_units))
            
        if treatment_term_id is not None:
            
            _query_params.append(('treatment_term_id', treatment_term_id))
            
        if treatment_term_name is not None:
            
            _query_params.append(('treatment_term_name', treatment_term_name))
            
        if treatment_type is not None:
            
            _query_params.append(('treatment_type', treatment_type))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/treatments/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def users(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        email: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by email")] = None,
        first_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by first_name")] = None,
        groups: Annotated[Optional[List[StrictStr]], Field(description="Filter by groups")] = None,
        job_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by job_title")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        last_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by last_name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submits_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by submits_for")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        viewing_groups: Annotated[Optional[List[StrictStr]], Field(description="Filter by viewing_groups")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> UserResults:
        """List items in the User collection.

        Collection endpoint that accepts various query parameters to filter and sort User items. Supports filtering on fields within User items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param email: Filter by email
        :type email: List[str]
        :param first_name: Filter by first_name
        :type first_name: List[str]
        :param groups: Filter by groups
        :type groups: List[str]
        :param job_title: Filter by job_title
        :type job_title: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param last_name: Filter by last_name
        :type last_name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submits_for: Filter by submits_for
        :type submits_for: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param viewing_groups: Filter by viewing_groups
        :type viewing_groups: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._users_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            email=email,
            first_name=first_name,
            groups=groups,
            job_title=job_title,
            lab=lab,
            last_name=last_name,
            notes=notes,
            status=status,
            submits_for=submits_for,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            viewing_groups=viewing_groups,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "UserResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def users_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        email: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by email")] = None,
        first_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by first_name")] = None,
        groups: Annotated[Optional[List[StrictStr]], Field(description="Filter by groups")] = None,
        job_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by job_title")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        last_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by last_name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submits_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by submits_for")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        viewing_groups: Annotated[Optional[List[StrictStr]], Field(description="Filter by viewing_groups")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[UserResults]:
        """List items in the User collection.

        Collection endpoint that accepts various query parameters to filter and sort User items. Supports filtering on fields within User items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param email: Filter by email
        :type email: List[str]
        :param first_name: Filter by first_name
        :type first_name: List[str]
        :param groups: Filter by groups
        :type groups: List[str]
        :param job_title: Filter by job_title
        :type job_title: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param last_name: Filter by last_name
        :type last_name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submits_for: Filter by submits_for
        :type submits_for: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param viewing_groups: Filter by viewing_groups
        :type viewing_groups: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._users_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            email=email,
            first_name=first_name,
            groups=groups,
            job_title=job_title,
            lab=lab,
            last_name=last_name,
            notes=notes,
            status=status,
            submits_for=submits_for,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            viewing_groups=viewing_groups,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "UserResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def users_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        email: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by email")] = None,
        first_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by first_name")] = None,
        groups: Annotated[Optional[List[StrictStr]], Field(description="Filter by groups")] = None,
        job_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by job_title")] = None,
        lab: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab")] = None,
        last_name: Annotated[Optional[List[StrictStr]], Field(description="Filter by last_name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submits_for: Annotated[Optional[List[StrictStr]], Field(description="Filter by submits_for")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        title: Annotated[Optional[List[StrictStr]], Field(description="Filter by title")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        viewing_groups: Annotated[Optional[List[StrictStr]], Field(description="Filter by viewing_groups")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the User collection.

        Collection endpoint that accepts various query parameters to filter and sort User items. Supports filtering on fields within User items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param email: Filter by email
        :type email: List[str]
        :param first_name: Filter by first_name
        :type first_name: List[str]
        :param groups: Filter by groups
        :type groups: List[str]
        :param job_title: Filter by job_title
        :type job_title: List[str]
        :param lab: Filter by lab
        :type lab: List[str]
        :param last_name: Filter by last_name
        :type last_name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submits_for: Filter by submits_for
        :type submits_for: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param title: Filter by title
        :type title: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param viewing_groups: Filter by viewing_groups
        :type viewing_groups: List[str]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._users_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            aliases=aliases,
            creation_timestamp=creation_timestamp,
            description=description,
            email=email,
            first_name=first_name,
            groups=groups,
            job_title=job_title,
            lab=lab,
            last_name=last_name,
            notes=notes,
            status=status,
            submits_for=submits_for,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            title=title,
            uuid=uuid,
            viewing_groups=viewing_groups,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "UserResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _users_serialize(
        self,
        query,
        limit,
        sort,
        id,
        aliases,
        creation_timestamp,
        description,
        email,
        first_name,
        groups,
        job_title,
        lab,
        last_name,
        notes,
        status,
        submits_for,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        title,
        uuid,
        viewing_groups,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'aliases': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'email': 'multi',
            'first_name': 'multi',
            'groups': 'multi',
            'job_title': 'multi',
            'lab': 'multi',
            'last_name': 'multi',
            'notes': 'multi',
            'status': 'multi',
            'submits_for': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'title': 'multi',
            'uuid': 'multi',
            'viewing_groups': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if email is not None:
            
            _query_params.append(('email', email))
            
        if first_name is not None:
            
            _query_params.append(('first_name', first_name))
            
        if groups is not None:
            
            _query_params.append(('groups', groups))
            
        if job_title is not None:
            
            _query_params.append(('job_title', job_title))
            
        if lab is not None:
            
            _query_params.append(('lab', lab))
            
        if last_name is not None:
            
            _query_params.append(('last_name', last_name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submits_for is not None:
            
            _query_params.append(('submits_for', submits_for))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if title is not None:
            
            _query_params.append(('title', title))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if viewing_groups is not None:
            
            _query_params.append(('viewing_groups', viewing_groups))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/users/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def whole_organisms(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> WholeOrganismResults:
        """List items in the WholeOrganism collection.

        Collection endpoint that accepts various query parameters to filter and sort WholeOrganism items. Supports filtering on fields within WholeOrganism items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._whole_organisms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "WholeOrganismResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def whole_organisms_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[WholeOrganismResults]:
        """List items in the WholeOrganism collection.

        Collection endpoint that accepts various query parameters to filter and sort WholeOrganism items. Supports filtering on fields within WholeOrganism items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._whole_organisms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "WholeOrganismResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def whole_organisms_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        age: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by age")] = None,
        age_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by age_units")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        biomarkers: Annotated[Optional[List[StrictStr]], Field(description="Filter by biomarkers")] = None,
        cellular_sub_pool: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by cellular_sub_pool")] = None,
        classifications: Annotated[Optional[List[StrictStr]], Field(description="Filter by classifications")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        construct_library_sets: Annotated[Optional[List[StrictStr]], Field(description="Filter by construct_library_sets")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        date_obtained: Annotated[Optional[List[StrictStr]], Field(description="Filter by date_obtained")] = None,
        dbxrefs: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by dbxrefs")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        disease_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by disease_terms.@id")] = None,
        disease_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by disease_terms.term_name")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        donors: Annotated[Optional[List[StrictStr]], Field(description="Filter by donors")] = None,
        embryonic: Annotated[Optional[List[StrictBool]], Field(description="Filter by embryonic")] = None,
        file_sets_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.@id")] = None,
        file_sets_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.accession")] = None,
        file_sets_aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.aliases")] = None,
        file_sets_assay_term_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by file_sets.assay_term.term_name")] = None,
        file_sets_lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.lab.title")] = None,
        file_sets_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.status")] = None,
        file_sets_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by file_sets.summary")] = None,
        institutional_certificates_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by institutional_certificates.@id")] = None,
        institutional_certificates_certificate_identifier: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by institutional_certificates.certificate_identifier")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        lot_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by lot_id")] = None,
        lower_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age")] = None,
        lower_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by lower_bound_age_in_hours")] = None,
        modifications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.@id")] = None,
        modifications_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.status")] = None,
        modifications_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by modifications.summary")] = None,
        moi: Annotated[Optional[List[Union[Annotated[float, Field(strict=True, ge=0)], Annotated[int, Field(strict=True, ge=0)]]]], Field(description="Filter by moi")] = None,
        multiplexed_in_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.@id")] = None,
        multiplexed_in_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by multiplexed_in.accession")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        nucleic_acid_delivery: Annotated[Optional[List[StrictStr]], Field(description="Filter by nucleic_acid_delivery")] = None,
        origin_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by origin_of")] = None,
        originated_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by originated_from")] = None,
        part_of: Annotated[Optional[List[StrictStr]], Field(description="Filter by part_of")] = None,
        parts: Annotated[Optional[List[StrictStr]], Field(description="Filter by parts")] = None,
        pooled_from: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_from")] = None,
        pooled_in: Annotated[Optional[List[StrictStr]], Field(description="Filter by pooled_in")] = None,
        product_id: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by product_id")] = None,
        protocols: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by protocols")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        sample_terms_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sample_terms.@id")] = None,
        sample_terms_term_name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by sample_terms.term_name")] = None,
        sex: Annotated[Optional[List[StrictStr]], Field(description="Filter by sex")] = None,
        sorted_fractions: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_fractions")] = None,
        sorted_from_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.@id")] = None,
        sorted_from_accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from.accession")] = None,
        sorted_from_detail: Annotated[Optional[List[StrictStr]], Field(description="Filter by sorted_from_detail")] = None,
        sources_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by sources.@id")] = None,
        starting_amount: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by starting_amount")] = None,
        starting_amount_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by starting_amount_units")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        taxa: Annotated[Optional[List[StrictStr]], Field(description="Filter by taxa")] = None,
        time_post_library_delivery: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by time_post_library_delivery")] = None,
        time_post_library_delivery_units: Annotated[Optional[List[StrictStr]], Field(description="Filter by time_post_library_delivery_units")] = None,
        treatments_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.@id")] = None,
        treatments_purpose: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.purpose")] = None,
        treatments_status: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.status")] = None,
        treatments_summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.summary")] = None,
        treatments_treatment_type: Annotated[Optional[List[StrictStr]], Field(description="Filter by treatments.treatment_type")] = None,
        upper_bound_age: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age")] = None,
        upper_bound_age_in_hours: Annotated[Optional[List[Union[StrictFloat, StrictInt]]], Field(description="Filter by upper_bound_age_in_hours")] = None,
        url: Annotated[Optional[List[StrictStr]], Field(description="Filter by url")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        virtual: Annotated[Optional[List[StrictBool]], Field(description="Filter by virtual")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the WholeOrganism collection.

        Collection endpoint that accepts various query parameters to filter and sort WholeOrganism items. Supports filtering on fields within WholeOrganism items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param age: Filter by age
        :type age: List[str]
        :param age_units: Filter by age_units
        :type age_units: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param biomarkers: Filter by biomarkers
        :type biomarkers: List[str]
        :param cellular_sub_pool: Filter by cellular_sub_pool
        :type cellular_sub_pool: List[str]
        :param classifications: Filter by classifications
        :type classifications: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param construct_library_sets: Filter by construct_library_sets
        :type construct_library_sets: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param date_obtained: Filter by date_obtained
        :type date_obtained: List[str]
        :param dbxrefs: Filter by dbxrefs
        :type dbxrefs: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param disease_terms_id: Filter by disease_terms.@id
        :type disease_terms_id: List[str]
        :param disease_terms_term_name: Filter by disease_terms.term_name
        :type disease_terms_term_name: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param donors: Filter by donors
        :type donors: List[str]
        :param embryonic: Filter by embryonic
        :type embryonic: List[bool]
        :param file_sets_id: Filter by file_sets.@id
        :type file_sets_id: List[str]
        :param file_sets_accession: Filter by file_sets.accession
        :type file_sets_accession: List[str]
        :param file_sets_aliases: Filter by file_sets.aliases
        :type file_sets_aliases: List[str]
        :param file_sets_assay_term_term_name: Filter by file_sets.assay_term.term_name
        :type file_sets_assay_term_term_name: List[str]
        :param file_sets_lab_title: Filter by file_sets.lab.title
        :type file_sets_lab_title: List[str]
        :param file_sets_status: Filter by file_sets.status
        :type file_sets_status: List[str]
        :param file_sets_summary: Filter by file_sets.summary
        :type file_sets_summary: List[str]
        :param institutional_certificates_id: Filter by institutional_certificates.@id
        :type institutional_certificates_id: List[str]
        :param institutional_certificates_certificate_identifier: Filter by institutional_certificates.certificate_identifier
        :type institutional_certificates_certificate_identifier: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param lot_id: Filter by lot_id
        :type lot_id: List[str]
        :param lower_bound_age: Filter by lower_bound_age
        :type lower_bound_age: List[float]
        :param lower_bound_age_in_hours: Filter by lower_bound_age_in_hours
        :type lower_bound_age_in_hours: List[float]
        :param modifications_id: Filter by modifications.@id
        :type modifications_id: List[str]
        :param modifications_status: Filter by modifications.status
        :type modifications_status: List[str]
        :param modifications_summary: Filter by modifications.summary
        :type modifications_summary: List[str]
        :param moi: Filter by moi
        :type moi: List[float]
        :param multiplexed_in_id: Filter by multiplexed_in.@id
        :type multiplexed_in_id: List[str]
        :param multiplexed_in_accession: Filter by multiplexed_in.accession
        :type multiplexed_in_accession: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param nucleic_acid_delivery: Filter by nucleic_acid_delivery
        :type nucleic_acid_delivery: List[str]
        :param origin_of: Filter by origin_of
        :type origin_of: List[str]
        :param originated_from: Filter by originated_from
        :type originated_from: List[str]
        :param part_of: Filter by part_of
        :type part_of: List[str]
        :param parts: Filter by parts
        :type parts: List[str]
        :param pooled_from: Filter by pooled_from
        :type pooled_from: List[str]
        :param pooled_in: Filter by pooled_in
        :type pooled_in: List[str]
        :param product_id: Filter by product_id
        :type product_id: List[str]
        :param protocols: Filter by protocols
        :type protocols: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param sample_terms_id: Filter by sample_terms.@id
        :type sample_terms_id: List[str]
        :param sample_terms_term_name: Filter by sample_terms.term_name
        :type sample_terms_term_name: List[str]
        :param sex: Filter by sex
        :type sex: List[str]
        :param sorted_fractions: Filter by sorted_fractions
        :type sorted_fractions: List[str]
        :param sorted_from_id: Filter by sorted_from.@id
        :type sorted_from_id: List[str]
        :param sorted_from_accession: Filter by sorted_from.accession
        :type sorted_from_accession: List[str]
        :param sorted_from_detail: Filter by sorted_from_detail
        :type sorted_from_detail: List[str]
        :param sources_id: Filter by sources.@id
        :type sources_id: List[str]
        :param starting_amount: Filter by starting_amount
        :type starting_amount: List[float]
        :param starting_amount_units: Filter by starting_amount_units
        :type starting_amount_units: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param taxa: Filter by taxa
        :type taxa: List[str]
        :param time_post_library_delivery: Filter by time_post_library_delivery
        :type time_post_library_delivery: List[float]
        :param time_post_library_delivery_units: Filter by time_post_library_delivery_units
        :type time_post_library_delivery_units: List[str]
        :param treatments_id: Filter by treatments.@id
        :type treatments_id: List[str]
        :param treatments_purpose: Filter by treatments.purpose
        :type treatments_purpose: List[str]
        :param treatments_status: Filter by treatments.status
        :type treatments_status: List[str]
        :param treatments_summary: Filter by treatments.summary
        :type treatments_summary: List[str]
        :param treatments_treatment_type: Filter by treatments.treatment_type
        :type treatments_treatment_type: List[str]
        :param upper_bound_age: Filter by upper_bound_age
        :type upper_bound_age: List[float]
        :param upper_bound_age_in_hours: Filter by upper_bound_age_in_hours
        :type upper_bound_age_in_hours: List[float]
        :param url: Filter by url
        :type url: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param virtual: Filter by virtual
        :type virtual: List[bool]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._whole_organisms_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            age=age,
            age_units=age_units,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            award_id=award_id,
            award_component=award_component,
            biomarkers=biomarkers,
            cellular_sub_pool=cellular_sub_pool,
            classifications=classifications,
            collections=collections,
            construct_library_sets=construct_library_sets,
            creation_timestamp=creation_timestamp,
            date_obtained=date_obtained,
            dbxrefs=dbxrefs,
            description=description,
            disease_terms_id=disease_terms_id,
            disease_terms_term_name=disease_terms_term_name,
            documents=documents,
            donors=donors,
            embryonic=embryonic,
            file_sets_id=file_sets_id,
            file_sets_accession=file_sets_accession,
            file_sets_aliases=file_sets_aliases,
            file_sets_assay_term_term_name=file_sets_assay_term_term_name,
            file_sets_lab_title=file_sets_lab_title,
            file_sets_status=file_sets_status,
            file_sets_summary=file_sets_summary,
            institutional_certificates_id=institutional_certificates_id,
            institutional_certificates_certificate_identifier=institutional_certificates_certificate_identifier,
            lab_id=lab_id,
            lab_title=lab_title,
            lot_id=lot_id,
            lower_bound_age=lower_bound_age,
            lower_bound_age_in_hours=lower_bound_age_in_hours,
            modifications_id=modifications_id,
            modifications_status=modifications_status,
            modifications_summary=modifications_summary,
            moi=moi,
            multiplexed_in_id=multiplexed_in_id,
            multiplexed_in_accession=multiplexed_in_accession,
            notes=notes,
            nucleic_acid_delivery=nucleic_acid_delivery,
            origin_of=origin_of,
            originated_from=originated_from,
            part_of=part_of,
            parts=parts,
            pooled_from=pooled_from,
            pooled_in=pooled_in,
            product_id=product_id,
            protocols=protocols,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            sample_terms_id=sample_terms_id,
            sample_terms_term_name=sample_terms_term_name,
            sex=sex,
            sorted_fractions=sorted_fractions,
            sorted_from_id=sorted_from_id,
            sorted_from_accession=sorted_from_accession,
            sorted_from_detail=sorted_from_detail,
            sources_id=sources_id,
            starting_amount=starting_amount,
            starting_amount_units=starting_amount_units,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            taxa=taxa,
            time_post_library_delivery=time_post_library_delivery,
            time_post_library_delivery_units=time_post_library_delivery_units,
            treatments_id=treatments_id,
            treatments_purpose=treatments_purpose,
            treatments_status=treatments_status,
            treatments_summary=treatments_summary,
            treatments_treatment_type=treatments_treatment_type,
            upper_bound_age=upper_bound_age,
            upper_bound_age_in_hours=upper_bound_age_in_hours,
            url=url,
            uuid=uuid,
            virtual=virtual,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "WholeOrganismResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _whole_organisms_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        age,
        age_units,
        aliases,
        alternate_accessions,
        award_id,
        award_component,
        biomarkers,
        cellular_sub_pool,
        classifications,
        collections,
        construct_library_sets,
        creation_timestamp,
        date_obtained,
        dbxrefs,
        description,
        disease_terms_id,
        disease_terms_term_name,
        documents,
        donors,
        embryonic,
        file_sets_id,
        file_sets_accession,
        file_sets_aliases,
        file_sets_assay_term_term_name,
        file_sets_lab_title,
        file_sets_status,
        file_sets_summary,
        institutional_certificates_id,
        institutional_certificates_certificate_identifier,
        lab_id,
        lab_title,
        lot_id,
        lower_bound_age,
        lower_bound_age_in_hours,
        modifications_id,
        modifications_status,
        modifications_summary,
        moi,
        multiplexed_in_id,
        multiplexed_in_accession,
        notes,
        nucleic_acid_delivery,
        origin_of,
        originated_from,
        part_of,
        parts,
        pooled_from,
        pooled_in,
        product_id,
        protocols,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        sample_terms_id,
        sample_terms_term_name,
        sex,
        sorted_fractions,
        sorted_from_id,
        sorted_from_accession,
        sorted_from_detail,
        sources_id,
        starting_amount,
        starting_amount_units,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        taxa,
        time_post_library_delivery,
        time_post_library_delivery_units,
        treatments_id,
        treatments_purpose,
        treatments_status,
        treatments_summary,
        treatments_treatment_type,
        upper_bound_age,
        upper_bound_age_in_hours,
        url,
        uuid,
        virtual,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'age': 'multi',
            'age_units': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'biomarkers': 'multi',
            'cellular_sub_pool': 'multi',
            'classifications': 'multi',
            'collections': 'multi',
            'construct_library_sets': 'multi',
            'creation_timestamp': 'multi',
            'date_obtained': 'multi',
            'dbxrefs': 'multi',
            'description': 'multi',
            'disease_terms.@id': 'multi',
            'disease_terms.term_name': 'multi',
            'documents': 'multi',
            'donors': 'multi',
            'embryonic': 'multi',
            'file_sets.@id': 'multi',
            'file_sets.accession': 'multi',
            'file_sets.aliases': 'multi',
            'file_sets.assay_term.term_name': 'multi',
            'file_sets.lab.title': 'multi',
            'file_sets.status': 'multi',
            'file_sets.summary': 'multi',
            'institutional_certificates.@id': 'multi',
            'institutional_certificates.certificate_identifier': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'lot_id': 'multi',
            'lower_bound_age': 'multi',
            'lower_bound_age_in_hours': 'multi',
            'modifications.@id': 'multi',
            'modifications.status': 'multi',
            'modifications.summary': 'multi',
            'moi': 'multi',
            'multiplexed_in.@id': 'multi',
            'multiplexed_in.accession': 'multi',
            'notes': 'multi',
            'nucleic_acid_delivery': 'multi',
            'origin_of': 'multi',
            'originated_from': 'multi',
            'part_of': 'multi',
            'parts': 'multi',
            'pooled_from': 'multi',
            'pooled_in': 'multi',
            'product_id': 'multi',
            'protocols': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'sample_terms.@id': 'multi',
            'sample_terms.term_name': 'multi',
            'sex': 'multi',
            'sorted_fractions': 'multi',
            'sorted_from.@id': 'multi',
            'sorted_from.accession': 'multi',
            'sorted_from_detail': 'multi',
            'sources.@id': 'multi',
            'starting_amount': 'multi',
            'starting_amount_units': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'taxa': 'multi',
            'time_post_library_delivery': 'multi',
            'time_post_library_delivery_units': 'multi',
            'treatments.@id': 'multi',
            'treatments.purpose': 'multi',
            'treatments.status': 'multi',
            'treatments.summary': 'multi',
            'treatments.treatment_type': 'multi',
            'upper_bound_age': 'multi',
            'upper_bound_age_in_hours': 'multi',
            'url': 'multi',
            'uuid': 'multi',
            'virtual': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if age is not None:
            
            _query_params.append(('age', age))
            
        if age_units is not None:
            
            _query_params.append(('age_units', age_units))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if biomarkers is not None:
            
            _query_params.append(('biomarkers', biomarkers))
            
        if cellular_sub_pool is not None:
            
            _query_params.append(('cellular_sub_pool', cellular_sub_pool))
            
        if classifications is not None:
            
            _query_params.append(('classifications', classifications))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if construct_library_sets is not None:
            
            _query_params.append(('construct_library_sets', construct_library_sets))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if date_obtained is not None:
            
            _query_params.append(('date_obtained', date_obtained))
            
        if dbxrefs is not None:
            
            _query_params.append(('dbxrefs', dbxrefs))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if disease_terms_id is not None:
            
            _query_params.append(('disease_terms.@id', disease_terms_id))
            
        if disease_terms_term_name is not None:
            
            _query_params.append(('disease_terms.term_name', disease_terms_term_name))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if donors is not None:
            
            _query_params.append(('donors', donors))
            
        if embryonic is not None:
            
            _query_params.append(('embryonic', embryonic))
            
        if file_sets_id is not None:
            
            _query_params.append(('file_sets.@id', file_sets_id))
            
        if file_sets_accession is not None:
            
            _query_params.append(('file_sets.accession', file_sets_accession))
            
        if file_sets_aliases is not None:
            
            _query_params.append(('file_sets.aliases', file_sets_aliases))
            
        if file_sets_assay_term_term_name is not None:
            
            _query_params.append(('file_sets.assay_term.term_name', file_sets_assay_term_term_name))
            
        if file_sets_lab_title is not None:
            
            _query_params.append(('file_sets.lab.title', file_sets_lab_title))
            
        if file_sets_status is not None:
            
            _query_params.append(('file_sets.status', file_sets_status))
            
        if file_sets_summary is not None:
            
            _query_params.append(('file_sets.summary', file_sets_summary))
            
        if institutional_certificates_id is not None:
            
            _query_params.append(('institutional_certificates.@id', institutional_certificates_id))
            
        if institutional_certificates_certificate_identifier is not None:
            
            _query_params.append(('institutional_certificates.certificate_identifier', institutional_certificates_certificate_identifier))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if lot_id is not None:
            
            _query_params.append(('lot_id', lot_id))
            
        if lower_bound_age is not None:
            
            _query_params.append(('lower_bound_age', lower_bound_age))
            
        if lower_bound_age_in_hours is not None:
            
            _query_params.append(('lower_bound_age_in_hours', lower_bound_age_in_hours))
            
        if modifications_id is not None:
            
            _query_params.append(('modifications.@id', modifications_id))
            
        if modifications_status is not None:
            
            _query_params.append(('modifications.status', modifications_status))
            
        if modifications_summary is not None:
            
            _query_params.append(('modifications.summary', modifications_summary))
            
        if moi is not None:
            
            _query_params.append(('moi', moi))
            
        if multiplexed_in_id is not None:
            
            _query_params.append(('multiplexed_in.@id', multiplexed_in_id))
            
        if multiplexed_in_accession is not None:
            
            _query_params.append(('multiplexed_in.accession', multiplexed_in_accession))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if nucleic_acid_delivery is not None:
            
            _query_params.append(('nucleic_acid_delivery', nucleic_acid_delivery))
            
        if origin_of is not None:
            
            _query_params.append(('origin_of', origin_of))
            
        if originated_from is not None:
            
            _query_params.append(('originated_from', originated_from))
            
        if part_of is not None:
            
            _query_params.append(('part_of', part_of))
            
        if parts is not None:
            
            _query_params.append(('parts', parts))
            
        if pooled_from is not None:
            
            _query_params.append(('pooled_from', pooled_from))
            
        if pooled_in is not None:
            
            _query_params.append(('pooled_in', pooled_in))
            
        if product_id is not None:
            
            _query_params.append(('product_id', product_id))
            
        if protocols is not None:
            
            _query_params.append(('protocols', protocols))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if sample_terms_id is not None:
            
            _query_params.append(('sample_terms.@id', sample_terms_id))
            
        if sample_terms_term_name is not None:
            
            _query_params.append(('sample_terms.term_name', sample_terms_term_name))
            
        if sex is not None:
            
            _query_params.append(('sex', sex))
            
        if sorted_fractions is not None:
            
            _query_params.append(('sorted_fractions', sorted_fractions))
            
        if sorted_from_id is not None:
            
            _query_params.append(('sorted_from.@id', sorted_from_id))
            
        if sorted_from_accession is not None:
            
            _query_params.append(('sorted_from.accession', sorted_from_accession))
            
        if sorted_from_detail is not None:
            
            _query_params.append(('sorted_from_detail', sorted_from_detail))
            
        if sources_id is not None:
            
            _query_params.append(('sources.@id', sources_id))
            
        if starting_amount is not None:
            
            _query_params.append(('starting_amount', starting_amount))
            
        if starting_amount_units is not None:
            
            _query_params.append(('starting_amount_units', starting_amount_units))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if taxa is not None:
            
            _query_params.append(('taxa', taxa))
            
        if time_post_library_delivery is not None:
            
            _query_params.append(('time_post_library_delivery', time_post_library_delivery))
            
        if time_post_library_delivery_units is not None:
            
            _query_params.append(('time_post_library_delivery_units', time_post_library_delivery_units))
            
        if treatments_id is not None:
            
            _query_params.append(('treatments.@id', treatments_id))
            
        if treatments_purpose is not None:
            
            _query_params.append(('treatments.purpose', treatments_purpose))
            
        if treatments_status is not None:
            
            _query_params.append(('treatments.status', treatments_status))
            
        if treatments_summary is not None:
            
            _query_params.append(('treatments.summary', treatments_summary))
            
        if treatments_treatment_type is not None:
            
            _query_params.append(('treatments.treatment_type', treatments_treatment_type))
            
        if upper_bound_age is not None:
            
            _query_params.append(('upper_bound_age', upper_bound_age))
            
        if upper_bound_age_in_hours is not None:
            
            _query_params.append(('upper_bound_age_in_hours', upper_bound_age_in_hours))
            
        if url is not None:
            
            _query_params.append(('url', url))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if virtual is not None:
            
            _query_params.append(('virtual', virtual))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/whole-organisms/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )




    @validate_call
    def workflows(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_steps: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_steps")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        source_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by source_url")] = None,
        standards_page_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by standards_page.@id")] = None,
        standards_page_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by standards_page.title")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        workflow_repositories: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow_repositories")] = None,
        workflow_version: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by workflow_version")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> WorkflowResults:
        """List items in the Workflow collection.

        Collection endpoint that accepts various query parameters to filter and sort Workflow items. Supports filtering on fields within Workflow items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_steps: Filter by analysis_steps
        :type analysis_steps: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param standards_page_id: Filter by standards_page.@id
        :type standards_page_id: List[str]
        :param standards_page_title: Filter by standards_page.title
        :type standards_page_title: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param workflow_repositories: Filter by workflow_repositories
        :type workflow_repositories: List[str]
        :param workflow_version: Filter by workflow_version
        :type workflow_version: List[int]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._workflows_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_steps=analysis_steps,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            source_url=source_url,
            standards_page_id=standards_page_id,
            standards_page_title=standards_page_title,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            workflow_repositories=workflow_repositories,
            workflow_version=workflow_version,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "WorkflowResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        ).data


    @validate_call
    def workflows_with_http_info(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_steps: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_steps")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        source_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by source_url")] = None,
        standards_page_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by standards_page.@id")] = None,
        standards_page_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by standards_page.title")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        workflow_repositories: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow_repositories")] = None,
        workflow_version: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by workflow_version")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> ApiResponse[WorkflowResults]:
        """List items in the Workflow collection.

        Collection endpoint that accepts various query parameters to filter and sort Workflow items. Supports filtering on fields within Workflow items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_steps: Filter by analysis_steps
        :type analysis_steps: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param standards_page_id: Filter by standards_page.@id
        :type standards_page_id: List[str]
        :param standards_page_title: Filter by standards_page.title
        :type standards_page_title: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param workflow_repositories: Filter by workflow_repositories
        :type workflow_repositories: List[str]
        :param workflow_version: Filter by workflow_version
        :type workflow_version: List[int]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._workflows_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_steps=analysis_steps,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            source_url=source_url,
            standards_page_id=standards_page_id,
            standards_page_title=standards_page_title,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            workflow_repositories=workflow_repositories,
            workflow_version=workflow_version,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "WorkflowResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        response_data.read()
        return self.api_client.response_deserialize(
            response_data=response_data,
            response_types_map=_response_types_map,
        )


    @validate_call
    def workflows_without_preload_content(
        self,
        query: Annotated[Optional[StrictStr], Field(description="Query string for searching.")] = None,
        limit: Annotated[Optional[Any], Field(description="Maximum number of results to return. Default is 25. Use 'all' for all results.")] = None,
        sort: Annotated[Optional[List[StrictStr]], Field(description="Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.")] = None,
        id: Annotated[Optional[List[StrictStr]], Field(description="Filter by @id")] = None,
        accession: Annotated[Optional[List[StrictStr]], Field(description="Filter by accession")] = None,
        aliases: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by aliases")] = None,
        alternate_accessions: Annotated[Optional[List[StrictStr]], Field(description="Filter by alternate_accessions")] = None,
        analysis_steps: Annotated[Optional[List[StrictStr]], Field(description="Filter by analysis_steps")] = None,
        award_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.@id")] = None,
        award_component: Annotated[Optional[List[StrictStr]], Field(description="Filter by award.component")] = None,
        collections: Annotated[Optional[List[StrictStr]], Field(description="Filter by collections")] = None,
        creation_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by creation_timestamp")] = None,
        description: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by description")] = None,
        documents: Annotated[Optional[List[StrictStr]], Field(description="Filter by documents")] = None,
        lab_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.@id")] = None,
        lab_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by lab.title")] = None,
        name: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by name")] = None,
        notes: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by notes")] = None,
        publications_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by publications.@id")] = None,
        publications_publication_identifiers: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by publications.publication_identifiers")] = None,
        release_timestamp: Annotated[Optional[List[StrictStr]], Field(description="Filter by release_timestamp")] = None,
        revoke_detail: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by revoke_detail")] = None,
        source_url: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by source_url")] = None,
        standards_page_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by standards_page.@id")] = None,
        standards_page_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by standards_page.title")] = None,
        status: Annotated[Optional[List[StrictStr]], Field(description="Filter by status")] = None,
        submitted_by_id: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.@id")] = None,
        submitted_by_title: Annotated[Optional[List[StrictStr]], Field(description="Filter by submitted_by.title")] = None,
        submitter_comment: Annotated[Optional[List[Annotated[str, Field(strict=True)]]], Field(description="Filter by submitter_comment")] = None,
        summary: Annotated[Optional[List[StrictStr]], Field(description="Filter by summary")] = None,
        uuid: Annotated[Optional[List[StrictStr]], Field(description="Filter by uuid")] = None,
        workflow_repositories: Annotated[Optional[List[StrictStr]], Field(description="Filter by workflow_repositories")] = None,
        workflow_version: Annotated[Optional[List[Annotated[int, Field(strict=True, ge=1)]]], Field(description="Filter by workflow_version")] = None,
        _request_timeout: Union[
            None,
            Annotated[StrictFloat, Field(gt=0)],
            Tuple[
                Annotated[StrictFloat, Field(gt=0)],
                Annotated[StrictFloat, Field(gt=0)]
            ]
        ] = None,
        _request_auth: Optional[Dict[StrictStr, Any]] = None,
        _content_type: Optional[StrictStr] = None,
        _headers: Optional[Dict[StrictStr, Any]] = None,
        _host_index: Annotated[StrictInt, Field(ge=0, le=0)] = 0,
    ) -> RESTResponseType:
        """List items in the Workflow collection.

        Collection endpoint that accepts various query parameters to filter and sort Workflow items. Supports filtering on fields within Workflow items.

        :param query: Query string for searching.
        :type query: str
        :param limit: Maximum number of results to return. Default is 25. Use 'all' for all results.
        :type limit: Limit
        :param sort: Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
        :type sort: List[str]
        :param id: Filter by @id
        :type id: List[str]
        :param accession: Filter by accession
        :type accession: List[str]
        :param aliases: Filter by aliases
        :type aliases: List[str]
        :param alternate_accessions: Filter by alternate_accessions
        :type alternate_accessions: List[str]
        :param analysis_steps: Filter by analysis_steps
        :type analysis_steps: List[str]
        :param award_id: Filter by award.@id
        :type award_id: List[str]
        :param award_component: Filter by award.component
        :type award_component: List[str]
        :param collections: Filter by collections
        :type collections: List[str]
        :param creation_timestamp: Filter by creation_timestamp
        :type creation_timestamp: List[str]
        :param description: Filter by description
        :type description: List[str]
        :param documents: Filter by documents
        :type documents: List[str]
        :param lab_id: Filter by lab.@id
        :type lab_id: List[str]
        :param lab_title: Filter by lab.title
        :type lab_title: List[str]
        :param name: Filter by name
        :type name: List[str]
        :param notes: Filter by notes
        :type notes: List[str]
        :param publications_id: Filter by publications.@id
        :type publications_id: List[str]
        :param publications_publication_identifiers: Filter by publications.publication_identifiers
        :type publications_publication_identifiers: List[str]
        :param release_timestamp: Filter by release_timestamp
        :type release_timestamp: List[str]
        :param revoke_detail: Filter by revoke_detail
        :type revoke_detail: List[str]
        :param source_url: Filter by source_url
        :type source_url: List[str]
        :param standards_page_id: Filter by standards_page.@id
        :type standards_page_id: List[str]
        :param standards_page_title: Filter by standards_page.title
        :type standards_page_title: List[str]
        :param status: Filter by status
        :type status: List[str]
        :param submitted_by_id: Filter by submitted_by.@id
        :type submitted_by_id: List[str]
        :param submitted_by_title: Filter by submitted_by.title
        :type submitted_by_title: List[str]
        :param submitter_comment: Filter by submitter_comment
        :type submitter_comment: List[str]
        :param summary: Filter by summary
        :type summary: List[str]
        :param uuid: Filter by uuid
        :type uuid: List[str]
        :param workflow_repositories: Filter by workflow_repositories
        :type workflow_repositories: List[str]
        :param workflow_version: Filter by workflow_version
        :type workflow_version: List[int]
        :param _request_timeout: timeout setting for this request. If one
                                 number provided, it will be total request
                                 timeout. It can also be a pair (tuple) of
                                 (connection, read) timeouts.
        :type _request_timeout: int, tuple(int, int), optional
        :param _request_auth: set to override the auth_settings for an a single
                              request; this effectively ignores the
                              authentication in the spec for a single request.
        :type _request_auth: dict, optional
        :param _content_type: force content-type for the request.
        :type _content_type: str, Optional
        :param _headers: set to override the headers for a single
                         request; this effectively ignores the headers
                         in the spec for a single request.
        :type _headers: dict, optional
        :param _host_index: set to override the host_index for a single
                            request; this effectively ignores the host_index
                            in the spec for a single request.
        :type _host_index: int, optional
        :return: Returns the result object.
        """ # noqa: E501

        _param = self._workflows_serialize(
            query=query,
            limit=limit,
            sort=sort,
            id=id,
            accession=accession,
            aliases=aliases,
            alternate_accessions=alternate_accessions,
            analysis_steps=analysis_steps,
            award_id=award_id,
            award_component=award_component,
            collections=collections,
            creation_timestamp=creation_timestamp,
            description=description,
            documents=documents,
            lab_id=lab_id,
            lab_title=lab_title,
            name=name,
            notes=notes,
            publications_id=publications_id,
            publications_publication_identifiers=publications_publication_identifiers,
            release_timestamp=release_timestamp,
            revoke_detail=revoke_detail,
            source_url=source_url,
            standards_page_id=standards_page_id,
            standards_page_title=standards_page_title,
            status=status,
            submitted_by_id=submitted_by_id,
            submitted_by_title=submitted_by_title,
            submitter_comment=submitter_comment,
            summary=summary,
            uuid=uuid,
            workflow_repositories=workflow_repositories,
            workflow_version=workflow_version,
            _request_auth=_request_auth,
            _content_type=_content_type,
            _headers=_headers,
            _host_index=_host_index
        )

        _response_types_map: Dict[str, Optional[str]] = {
            '200': "WorkflowResults",
            '400': "object",
            '404': "NoResultsResponse",
            '500': None,
        }
        response_data = self.api_client.call_api(
            *_param,
            _request_timeout=_request_timeout
        )
        return response_data.response


    def _workflows_serialize(
        self,
        query,
        limit,
        sort,
        id,
        accession,
        aliases,
        alternate_accessions,
        analysis_steps,
        award_id,
        award_component,
        collections,
        creation_timestamp,
        description,
        documents,
        lab_id,
        lab_title,
        name,
        notes,
        publications_id,
        publications_publication_identifiers,
        release_timestamp,
        revoke_detail,
        source_url,
        standards_page_id,
        standards_page_title,
        status,
        submitted_by_id,
        submitted_by_title,
        submitter_comment,
        summary,
        uuid,
        workflow_repositories,
        workflow_version,
        _request_auth,
        _content_type,
        _headers,
        _host_index,
    ) -> RequestSerialized:

        _host = None

        _collection_formats: Dict[str, str] = {
            'sort': 'multi',
            '@id': 'multi',
            'accession': 'multi',
            'aliases': 'multi',
            'alternate_accessions': 'multi',
            'analysis_steps': 'multi',
            'award.@id': 'multi',
            'award.component': 'multi',
            'collections': 'multi',
            'creation_timestamp': 'multi',
            'description': 'multi',
            'documents': 'multi',
            'lab.@id': 'multi',
            'lab.title': 'multi',
            'name': 'multi',
            'notes': 'multi',
            'publications.@id': 'multi',
            'publications.publication_identifiers': 'multi',
            'release_timestamp': 'multi',
            'revoke_detail': 'multi',
            'source_url': 'multi',
            'standards_page.@id': 'multi',
            'standards_page.title': 'multi',
            'status': 'multi',
            'submitted_by.@id': 'multi',
            'submitted_by.title': 'multi',
            'submitter_comment': 'multi',
            'summary': 'multi',
            'uuid': 'multi',
            'workflow_repositories': 'multi',
            'workflow_version': 'multi',
        }

        _path_params: Dict[str, str] = {}
        _query_params: List[Tuple[str, str]] = []
        _header_params: Dict[str, Optional[str]] = _headers or {}
        _form_params: List[Tuple[str, str]] = []
        _files: Dict[str, Union[str, bytes]] = {}
        _body_params: Optional[bytes] = None

        # process the path parameters
        # process the query parameters
        if query is not None:
            
            _query_params.append(('query', query))
            
        if limit is not None:
            
            _query_params.append(('limit', limit))
            
        if sort is not None:
            
            _query_params.append(('sort', sort))
            
        if id is not None:
            
            _query_params.append(('@id', id))
            
        if accession is not None:
            
            _query_params.append(('accession', accession))
            
        if aliases is not None:
            
            _query_params.append(('aliases', aliases))
            
        if alternate_accessions is not None:
            
            _query_params.append(('alternate_accessions', alternate_accessions))
            
        if analysis_steps is not None:
            
            _query_params.append(('analysis_steps', analysis_steps))
            
        if award_id is not None:
            
            _query_params.append(('award.@id', award_id))
            
        if award_component is not None:
            
            _query_params.append(('award.component', award_component))
            
        if collections is not None:
            
            _query_params.append(('collections', collections))
            
        if creation_timestamp is not None:
            
            _query_params.append(('creation_timestamp', creation_timestamp))
            
        if description is not None:
            
            _query_params.append(('description', description))
            
        if documents is not None:
            
            _query_params.append(('documents', documents))
            
        if lab_id is not None:
            
            _query_params.append(('lab.@id', lab_id))
            
        if lab_title is not None:
            
            _query_params.append(('lab.title', lab_title))
            
        if name is not None:
            
            _query_params.append(('name', name))
            
        if notes is not None:
            
            _query_params.append(('notes', notes))
            
        if publications_id is not None:
            
            _query_params.append(('publications.@id', publications_id))
            
        if publications_publication_identifiers is not None:
            
            _query_params.append(('publications.publication_identifiers', publications_publication_identifiers))
            
        if release_timestamp is not None:
            
            _query_params.append(('release_timestamp', release_timestamp))
            
        if revoke_detail is not None:
            
            _query_params.append(('revoke_detail', revoke_detail))
            
        if source_url is not None:
            
            _query_params.append(('source_url', source_url))
            
        if standards_page_id is not None:
            
            _query_params.append(('standards_page.@id', standards_page_id))
            
        if standards_page_title is not None:
            
            _query_params.append(('standards_page.title', standards_page_title))
            
        if status is not None:
            
            _query_params.append(('status', status))
            
        if submitted_by_id is not None:
            
            _query_params.append(('submitted_by.@id', submitted_by_id))
            
        if submitted_by_title is not None:
            
            _query_params.append(('submitted_by.title', submitted_by_title))
            
        if submitter_comment is not None:
            
            _query_params.append(('submitter_comment', submitter_comment))
            
        if summary is not None:
            
            _query_params.append(('summary', summary))
            
        if uuid is not None:
            
            _query_params.append(('uuid', uuid))
            
        if workflow_repositories is not None:
            
            _query_params.append(('workflow_repositories', workflow_repositories))
            
        if workflow_version is not None:
            
            _query_params.append(('workflow_version', workflow_version))
            
        # process the header parameters
        # process the form parameters
        # process the body parameter

        # Set client side default value of Query Param "frame".
        _query_params.append(('frame', 'object'))

        # set the HTTP header `Accept`
        _header_params['Accept'] = self.api_client.select_header_accept(
            [
                'application/json'
            ]
        )


        # authentication setting
        _auth_settings: List[str] = [
            'basicAuth'
        ]

        return self.api_client.param_serialize(
            method='GET',
            resource_path='/workflows/@@listing',
            path_params=_path_params,
            query_params=_query_params,
            header_params=_header_params,
            body=_body_params,
            post_params=_form_params,
            files=_files,
            auth_settings=_auth_settings,
            collection_formats=_collection_formats,
            _host=_host,
            _request_auth=_request_auth
        )


