# SearchResultItem


## Properties

Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**schema_version** | **str** | The version of the JSON schema that the server uses to validate the object. | [optional] 
**uuid** | **str** | The unique identifier associated with every object. | [optional] 
**notes** | **str** | DACC internal notes. | [optional] 
**aliases** | **List[str]** | Lab specific identifiers to reference an object. | [optional] 
**creation_timestamp** | **str** | The date the object was created. | [optional] 
**submitted_by** | **str** | The user who submitted the object. | [optional] 
**submitter_comment** | **str** | Additional information specified by the submitter to be displayed as a comment on the portal. | [optional] 
**description** | **str** | A plain text description of the object. | [optional] 
**status** | **str** | The status of the metadata object. | [optional] 
**user** | **str** | The user that is assigned to this access key. | [optional] 
**access_key_id** | **str** | An access key. | [optional] 
**secret_access_key_hash** | **str** | A secret access key. | [optional] 
**id** | **str** |  | [optional] 
**type** | **List[str]** |  | [optional] 
**summary** | **str** | A summary of the object. | [optional] 
**release_timestamp** | **str** | The date the object was released. | [optional] 
**lab** | **str** | Lab associated with the submission. | [optional] 
**award** | **str** | Grant associated with the submission. | [optional] 
**analysis_step_types** | **List[str]** | The classification of the software. | [optional] 
**step_label** | **str** | Unique lowercased label of the analysis step that includes the relevant assays, the software used, and the purpose of the step, e.g. rampage-grit-peak-calling-step | [optional] 
**title** | **str** | The full name of the user. | [optional] 
**workflow** | **str** | The workflow used to produce this file. | [optional] 
**parents** | **List[str]** | The precursor steps. | [optional] 
**input_content_types** | **List[str]** | The content types used as input for the analysis step. | [optional] 
**output_content_types** | **List[str]** | The content types produced as output by the analysis step. | [optional] 
**name** | **str** | The preferred viewable name of the workflow. | [optional] 
**analysis_step_versions** | **List[str]** | The analysis step versions associated with this analysis step. | [optional] 
**analysis_step** | **str** | The analysis step which this version belongs to. | [optional] 
**software_versions** | **List[str]** | The software versions associated with this publication. | [optional] 
**url** | **str** | An external resource with additional information. | [optional] 
**start_date** | **str** | The date when the award begins. | [optional] 
**end_date** | **str** | The date when the award concludes. | [optional] 
**pis** | **List[str]** | Principal Investigator(s) of the grant. | [optional] 
**contact_pi** | **str** | The contact Principal Investigator of the grant. | [optional] 
**project** | **str** | The collection of biological data related to a single initiative, originating from a consortium. | [optional] 
**viewing_group** | **str** | The group that determines which set of data the user has permission to view. | [optional] 
**component** | **str** | The project component the award is associated with. | [optional] 
**classification** | **str** | Sample specific biomarker. | [optional] 
**quantification** | **str** | The biomarker association to the biosample, disease or other condition.  This can be the absence of the biomarker or the presence of the biomarker in some low, intermediate or high quantity. | [optional] 
**synonyms** | **List[str]** | Synonyms for the term that have been recorded in an ontology. | [optional] 
**gene** | **str** | Biomarker gene. | [optional] 
**name_quantification** | **str** | A concatenation of the name and quantification of the biomarker. | [optional] 
**biomarker_for** | **List[str]** | The samples which have been confirmed to have this biomarker. | [optional] 
**attachment** | [**Attachment**](Attachment.md) |  | [optional] 
**document_type** | **str** | The category that best describes the document. | [optional] 
**characterization_method** | **str** | The method used for the characterization. | [optional] 
**urls** | **List[str]** | Link to the institutional certification form. | [optional] 
**standardized_file_format** | **bool** | Specifies whether this format is an IGVF-standardized file format defined by Focus Groups or produced by a uniform pipeline. | [optional] 
**taxa** | **str** | The species of the organism. | [optional] 
**publications** | **List[str]** | The publications associated with this object. | [optional] 
**documents** | **List[str]** | Documents that provide additional information (not data file). | [optional] 
**accession** | **str** | A unique identifier to be used to reference the object prefixed with IGVF. | [optional] 
**alternate_accessions** | **List[str]** | Accessions previously assigned to objects that have been merged with this object. | [optional] 
**collections** | **List[str]** | Some samples are part of particular data collections. | [optional] 
**revoke_detail** | **str** | Explanation of why an object was transitioned to the revoked status. | [optional] 
**dbxrefs** | **List[str]** | Biosample identifiers from external resources, such as Biosample database or Cellosaurus. | [optional] 
**sex** | **str** |  | [optional] 
**phenotypic_features** | **List[str]** | A list of associated phenotypic features of the donor. | [optional] 
**virtual** | **bool** | Virtual samples are not representing actual physical entities from experiments, but rather capturing metadata about hypothetical samples that the reported analysis results are relevant for. | [optional] 
**related_donors** | [**List[RelatedDonor]**](RelatedDonor.md) | Familial relations of this donor. | [optional] 
**ethnicities** | **List[str]** | Ethnicity of the donor. | [optional] 
**human_donor_identifiers** | **List[str]** | Identifiers of this human donor. | [optional] 
**sources** | **List[str]** | The originating lab(s) or vendor(s). | [optional] 
**lot_id** | **str** | The lot identifier provided by the originating lab or vendor. | [optional] 
**product_id** | **str** | The product identifier provided by the originating lab or vendor. | [optional] 
**strain_background** | **str** | The specific parent strain designation of a non-human donor. | [optional] 
**strain** | **str** | The specific strain designation of a non-human donor. | [optional] 
**genotype** | **str** | The genotype of the strain according to accepted nomenclature conventions. | [optional] 
**individual_rodent** | **bool** | This rodent donor represents an individual rodent. | [optional] 
**rodent_identifier** | **str** | The identifier for this individual rodent donor. | [optional] 
**controlled_access** | **bool** | Indicator of whether the samples are under controlled access. | [optional] 
**anvil_url** | **str** | URL linking to the controlled access file that has been deposited at AnVIL workspace. | [optional] 
**transcriptome_annotation** | **str** | The annotation and version of the reference resource. | [optional] 
**assembly** | **str** | Genome assembly applicable for the tabular data. | [optional] 
**reference_files** | **List[str]** | Link to the reference files used to generate this file. | [optional] 
**filtered** | **bool** | Indicates whether the file has gone through some filtering step, for example, removal of PCR duplicates or filtering based on significance calling. | [optional] 
**analysis_step_version** | **str** | The analysis step version of the quality metric. | [optional] 
**content_md5sum** | **str** | The MD5sum of the uncompressed file. | [optional] 
**content_type** | **str** | The type of content in the file. | [optional] 
**derived_from** | **List[str]** | The files participating as inputs into software to produce this output file. | [optional] 
**derived_manually** | **bool** | A boolean indicating whether the file has been dervided manually without automated computational methods. | [optional] 
**file_format** | **str** | The file format or extension of the file. | [optional] 
**file_format_specifications** | **List[str]** | Documents that describe the file format and fields of this file. | [optional] 
**file_set** | **str** | The file set that this file belongs to. | [optional] 
**file_size** | **int** | File size specified in bytes. | [optional] 
**md5sum** | **str** | The md5sum of the file being transferred. | [optional] 
**submitted_file_name** | **str** | Original name of the file. | [optional] 
**upload_status** | **str** | The upload/validation status of the file. | [optional] 
**validation_error_detail** | **str** | Explanation of why the file failed the automated content checks. | [optional] 
**checkfiles_version** | **str** | The Checkfiles GitHub version release the file was validated with. | [optional] 
**read_count** | **int** | Number of reads in a fastq file. | [optional] 
**redacted** | **bool** | Indicates whether the alignments data have been sanitized (redacted) to prevent leakage of private and potentially identifying genomic information. | [optional] 
**base_modifications** | **List[str]** | The chemical modifications to bases in a DNA sequence that are detected in this file. | [optional] 
**integrated_in** | **List[str]** | Construct library set(s) that this file was used for in insert design. | [optional] 
**input_file_for** | **List[str]** | The files which are derived from this file. | [optional] 
**gene_list_for** | **List[str]** | File Set(s) that this file is a gene list for. | [optional] 
**loci_list_for** | **List[str]** | File Set(s) that this file is a loci list for. | [optional] 
**quality_metrics** | **List[str]** | The quality metrics that are associated with this file. | [optional] 
**assay_titles** | **List[str]** | Title(s) of assays that produced data analyzed in the analysis set. | [optional] 
**href** | **str** | The download path to obtain file. | [optional] 
**s3_uri** | **str** | The S3 URI of public file object. | [optional] 
**upload_credentials** | **object** | The upload credentials for S3 to submit the file content. | [optional] 
**content_summary** | **str** | A summary of the data in the signal file. | [optional] 
**seqspec_of** | **List[str]** | Sequence files this file is a seqspec of. | [optional] 
**validate_onlist_files** | **bool** | Whether checkfiles will validate the onlist files. | [optional] 
**cell_type_annotation** | **str** | The inferred cell type this file is associated with based on single-cell expression profiling. | [optional] 
**file_format_type** | **str** | The subtype of bed files. | [optional] 
**principal_dimension** | **str** | The principal dimension of the matrix. | [optional] 
**secondary_dimensions** | **List[str]** | The secondary, tertiary....n levels of dimensions of the matrix. | [optional] 
**externally_hosted** | **bool** |  | [optional] 
**external_host_url** | **str** | A link to the resource where the file is externally hosted. | [optional] 
**catalog_adapters** | **List[str]** | IGVF Catalog Adapters that ingests this file | [optional] 
**source_url** | **str** | An external resource to the code base of the workflow in github. | [optional] 
**external** | **bool** | Indicates whether the file was obtained from an external, non-IGVF source. | [optional] 
**flowcell_id** | **str** | The alphanumeric identifier for the flowcell of a sequencing machine. | [optional] 
**lane** | **int** | An integer identifying the lane of a sequencing machine. | [optional] 
**minimum_read_length** | **int** | For high-throughput sequencing, the minimum number of contiguous nucleotides determined by sequencing. | [optional] 
**maximum_read_length** | **int** | For high-throughput sequencing, the maximum number of contiguous nucleotides determined by sequencing. | [optional] 
**mean_read_length** | **float** | For high-throughput sequencing, the mean number of contiguous nucleotides determined by sequencing. | [optional] 
**seqspec_document** | **str** | A seqspec document describing the library and read structure. | [optional] 
**sequencing_platform** | **str** | The measurement device used to produce sequencing data. | [optional] 
**sequencing_kit** | **str** | A reagent kit used with a library to prepare it for sequencing. | [optional] 
**sequencing_run** | **int** | An ordinal number indicating which sequencing run of the associated library that the file belongs to. | [optional] 
**illumina_read_type** | **str** | The read type of the file. Relevant only for files produced using an Illumina sequencing platform. | [optional] 
**index** | **str** | An Illumina index associated with the file. | [optional] 
**read_names** | **List[str]** | The read names of a sequence file based on how it will be used by uniform pipelines. | [optional] 
**seqspecs** | **List[str]** | Link(s) to the associated seqspec YAML configuration file(s). | [optional] 
**strand_specificity** | **str** | The strand-specificity of the sequencing results within Perturb-seq, scCRISPR screen, TAP-seq, and CERES-seq assays. | [optional] 
**normalized** | **bool** | Indicates if the signal file is normalized. | [optional] 
**start_view_position** | **str** | The 0-based coordinate for the default starting position when viewing the signal in a genome browser. | [optional] 
**barcode_map_for** | **List[str]** | Link(s) to the Multiplexed samples using this file as barcode map. | [optional] 
**primer_design_for** | **List[str]** | Link(s) to the MeasurementSets using this file as a primer design. | [optional] 
**input_file_sets** | **List[str]** | The file set(s) required for this prediction set. | [optional] 
**control_type** | **str** | The type of control this file set represents. | [optional] 
**samples** | **List[str]** | The samples associated with this publication. | [optional] 
**donors** | **List[str]** | Donor(s) the sample was derived from. | [optional] 
**file_set_type** | **str** | The category that best describes this prediction set. | [optional] 
**external_image_data_url** | **str** | Links to the external site where images and related data produced by this analysis are stored. | [optional] 
**demultiplexed_samples** | **List[str]** | The sample(s) associated with this analysis set inferred through demultiplexing. | [optional] 
**files** | **List[str]** | The files associated with this file set. | [optional] 
**control_for** | **List[str]** | The file sets for which this file set is a control. | [optional] 
**submitted_files_timestamp** | **str** | The timestamp the first file object in the file_set or associated auxiliary sets was created. | [optional] 
**input_for** | **List[str]** | The file sets that use this file set as an input. | [optional] 
**construct_library_sets** | **List[str]** | The construct library sets of vectors introduced to this sample prior to performing an assay. | [optional] 
**data_use_limitation_summaries** | **List[str]** | The data use limitation summaries of institutional certificates covering the sample associated with this file set which are signed by the same lab (or their partner lab) as the lab that submitted this file set. | [optional] 
**protocols** | **List[str]** | Links to the protocol(s) for preparing the samples on Protocols.io. | [optional] 
**sample_summary** | **str** | A summary of the samples associated with input file sets of this analysis set. | [optional] 
**functional_assay_mechanisms** | **List[str]** | The biological processes measured by this functional assay. For example, a VAMP-seq (MultiSTEP) assay measures the effects of variants on protein carboxylation and secretion processes. | [optional] 
**workflows** | **List[str]** | The workflows associated with this publication. | [optional] 
**barcode_map** | **str** | The link to the barcode mapping tabular file. | [optional] 
**measurement_sets** | **List[str]** | The measurement sets that link to this auxiliary set. | [optional] 
**control_file_sets** | **List[str]** | File sets that can serve as scientific controls for this file set. | [optional] 
**small_scale_loci_list** | [**List[Locus1]**](Locus1.md) | A small scale (&lt;&#x3D;100) list of specific chromosomal region(s) whose functionality is investigated in this prediction set. This property describes the input variables of the prediction set. For example, this list consists of the genetic variants whose functionality is predicted in this prediction set. | [optional] 
**large_scale_loci_list** | **str** | A large scale list (&gt;100) of specific chromosomal regions whose functionality is investigated in this prediction set. This property describes the input variables of the prediction set. For example, this list consists of the genetic variants whose functionality is predicted in this prediction set. | [optional] 
**small_scale_gene_list** | **List[str]** | The specific, small scale list of (&lt;&#x3D;100) gene(s) whose functionality is investigated in this prediction set. This property describes the input variables of the prediction set. For example, this list consists of the genes whose expression level is predicted in this prediction set. It differs from assessed_genes (see more information under assessed_genes). | [optional] 
**large_scale_gene_list** | **str** | The large scale list of (&gt;100 genes) whose functionality is investigated in this prediction set. This property describes the input variables of the prediction set. For example, this list consists of the genes whose expression level is predicted in this prediction set. It differs from assessed_genes (see more information under assessed_genes). | [optional] 
**scope** | **str** | The scope or scale that this prediction set is designed to target. | [optional] 
**selection_criteria** | **List[str]** | The criteria used to select the sequence material cloned into the library. | [optional] 
**integrated_content_files** | **List[str]** | The files containing sequence material of interest either used for insert design or directly cloned into vectors in this library. | [optional] 
**associated_phenotypes** | **List[str]** | Ontological terms for diseases or phenotypes associated with the sequence material cloned in this construct library. | [optional] 
**orf_list** | **List[str]** | List of Open Reading Frame this construct library was designed to target. | [optional] 
**exon** | **str** | An identifier in plain text for the specific exon in an expression vector library. The associated gene must be listed in the small_scale_gene_list property. | [optional] 
**tile** | [**Tile**](Tile.md) |  | [optional] 
**guide_type** | **str** | The design of guides used in a CRISPR library, paired-guide (pgRNA) or single-guide (sgRNA). | [optional] 
**tiling_modality** | **str** | The tiling modality of guides across elements or loci in a CRISPR library. | [optional] 
**average_guide_coverage** | **float** | The average number of guides targeting each element of interest in the library. | [optional] 
**lower_bound_guide_coverage** | **int** | Lower bound of the number of guides targeting each element of interest in the library. | [optional] 
**upper_bound_guide_coverage** | **int** | Upper bound of the number of guides targeting each element of interest in the library. | [optional] 
**average_insert_size** | **float** | The average size of the inserts cloned into vectors in the library. | [optional] 
**lower_bound_insert_size** | **int** | Lower bound of the size of the inserts cloned in vectors in the library. | [optional] 
**upper_bound_insert_size** | **int** | Upper bound of the size of the inserts cloned in vectors in the library. | [optional] 
**targeton** | **str** | An identifier in plain text for the specific targeton in an editing template library. The associated gene must be listed in the small_scale_gene_list property. | [optional] 
**applied_to_samples** | **List[str]** | The samples that link to this construct library set. | [optional] 
**assemblies** | **List[str]** | The genome assemblies to which the referencing files in the file set are utilizing (e.g., GRCh38). | [optional] 
**transcriptome_annotations** | **List[str]** | The annotation versions of the reference resource. | [optional] 
**assay_term** | **str** | The assay used to produce data in this measurement set. | [optional] 
**preferred_assay_title** | **str** | The custom lab preferred label for the experiment performed in this measurement set. | [optional] 
**multiome_size** | **int** | The number of datasets included in the multiome experiment this measurement set is a part of. | [optional] 
**sequencing_library_types** | **List[str]** | Description of the libraries sequenced in this measurement set. | [optional] 
**primer_designs** | **List[str]** | The primer designs used in this measurement set. | [optional] 
**auxiliary_sets** | **List[str]** | The auxiliary sets of files produced alongside raw data from this measurement set. | [optional] 
**external_image_url** | **str** | Links to the external site where images produced by this measurement are stored. | [optional] 
**targeted_genes** | **List[str]** | A list of genes targeted in this assay. For example, TF ChIP-seq attempts to identify binding sites of a protein encoded by a specific gene. In CRISPR FlowFISH, the modified samples are sorted based on expression of a specific gene. This property differs from small_scale_gene_list in Construct Library Set, which describes genes targeted by the content integrated in the constructs (such as guide RNAs.) | [optional] 
**onlist_method** | **str** | The method by which the onlist files will be combined by the seqspec onlist tool to generate the final barcode inclusion list for the single cell uniform pipeline. | [optional] 
**onlist_files** | **List[str]** | The barcode region onlist files listed in associated seqspec yaml files. | [optional] 
**barcode_replacement_file** | **str** | A file containing original barcodes and the new barcodes used to replace the original barcodes. One common application is to use in preprocessing Parse SPLiT-seq data with the single cell uniform pipeline. | [optional] 
**related_multiome_datasets** | **List[str]** | Related datasets included in the multiome experiment this measurement set is a part of. | [optional] 
**model_name** | **str** | The custom lab name given to this predictive model set. | [optional] 
**model_version** | **str** | The semantic version number for this predictive model set. | [optional] 
**prediction_objects** | **List[str]** | The objects this predictive model set is targeting. | [optional] 
**model_zoo_location** | **str** | The link to the model on the Kipoi repository. | [optional] 
**assessed_genes** | **List[str]** | A list of gene(s) assessed in this prediction set. This property is used to describe the gene(s) being investigated, especially how the input variables in the prediction set affect some critical functionality of the gene(s). For example, the effect could be predicted from genetic variants on the binding affinity of a transcription factor encoded by a gene (assessed_genes). It differs from small_scale_gene_list and large_scale_gene_list, as these are used when the input variables of the prediction set are genes. | [optional] 
**external_input_data** | **str** | A tabular file with links to external data utilized for this model. | [optional] 
**geneid** | **str** | ENSEMBL GeneID of official nomenclature approved gene. The GeneID does not include the current version number suffix. | [optional] 
**symbol** | **str** | Gene symbol approved by the official nomenclature. | [optional] 
**study_sets** | **List[str]** | The studies of IGVF that this gene was a part of. | [optional] 
**locations** | [**List[GeneLocation1]**](GeneLocation1.md) | Gene locations specified using 1-based, closed coordinates for different versions of reference genome assemblies. | [optional] 
**version_number** | **str** | Current ENSEMBL GeneID version number of the gene. | [optional] 
**geneid_with_version** | **str** | The ENSEMBL GeneID concatenated with its version number. | [optional] 
**caption** | **str** | The caption of the image. | [optional] 
**download_url** | **str** | Download Url | [optional] 
**certificate_identifier** | **str** | A unique identifier for the certificate. | [optional] 
**data_use_limitation** | **str** | Code indicating the limitations on data use for data generated from the applicable samples. | [optional] 
**data_use_limitation_modifiers** | **List[str]** | Code indicating a modifier on the limitations on data use for data generated from the applicable samples. | [optional] 
**partner_labs** | **List[str]** | Labs which belong to same institution as the signing PI and can share this institutional certificate. | [optional] 
**partner_awards** | **List[str]** | Awards granted to at least one lab that belongs to same institution as the signing PI and can share this institutional certificate. | [optional] 
**data_use_limitation_summary** | **str** | A combination of the data use limitation and its modifiers | [optional] 
**pi** | **str** | Principle Investigator of the lab. | [optional] 
**awards** | **List[str]** | Grants associated with the lab. | [optional] 
**institute_label** | **str** | An abbreviation for the institute the lab is associated with. | [optional] 
**activated** | **bool** | A boolean indicating whether the modification has been activated by a chemical agent. | [optional] 
**activating_agent_term_id** | **str** | The CHEBI identifier for the activating agent of the modification. | [optional] 
**activating_agent_term_name** | **str** | The CHEBI name for the activating agent of the modification. | [optional] 
**modality** | **str** | The purpose or intended effect of a modification. | [optional] 
**tagged_proteins** | **List[str]** | The tagged proteins which are targeted for degradation. | [optional] 
**cas** | **str** | The name of the CRISPR associated protein used in the modification. | [optional] 
**fused_domain** | **str** | The name of the molecule fused to a Cas protein. | [optional] 
**cas_species** | **str** | The originating species of the Cas nuclease. | [optional] 
**biosamples_modified** | **List[str]** | The biosamples which have been modified with this modification. | [optional] 
**degron_system** | **str** | The type of degron system implemented. | [optional] 
**term_id** | **str** | An ontology identifier describing a biological sample | [optional] 
**term_name** | **str** | Ontology term describing a biological sample, assay, trait, or disease. | [optional] 
**deprecated_ntr_terms** | **List[str]** | A list of deprecated NTR terms previously associated with this ontology term. | [optional] 
**is_a** | **List[str]** | A list of ontology terms which are the nearest ancestor to this ontology term. | [optional] 
**preferred_assay_titles** | **List[str]** | The custom lab preferred labels that this assay term may be associated with. | [optional] 
**definition** | **str** | Definition for the term that was recorded in an ontology. | [optional] 
**comments** | **List[str]** | Comment for the term that was recorded in an ontology. | [optional] 
**ancestors** | **List[str]** | List of term names of ontological terms that precede the given term in the ontological tree. These ancestor terms are typically more general ontological terms under which the term is classified. | [optional] 
**ontology** | **str** | The ontology in which the term is recorded. | [optional] 
**assay_slims** | **List[str]** | A broad categorization of the assay term. | [optional] 
**category_slims** | **List[str]** | The type of feature or interaction measured by the assay. | [optional] 
**objective_slims** | **List[str]** | The purpose of the assay. | [optional] 
**company** | **str** | The company that developed and sells the instrument. | [optional] 
**sequencing_kits** | **List[str]** | The available sequencing kits for this platform. | [optional] 
**organ_slims** | **List[str]** | Organs associated with the sample term. | [optional] 
**cell_slims** | **List[str]** | Cells associated with the sample term. | [optional] 
**developmental_slims** | **List[str]** | Developmental stages associated with the sample term. | [optional] 
**system_slims** | **List[str]** | Organ systems associated with the sample term. | [optional] 
**orf_id** | **str** | Open reading frame ID. | [optional] 
**genes** | **List[str]** | ENSEMBL GeneIDs of official nomenclature approved genes. The GeneIDs do not include the current version number suffix. | [optional] 
**protein_id** | **str** | ENSEMBL ProteinID of official nomenclature approved protein. The ProteinID does not include the current version number suffix. | [optional] 
**pct_identical_protein** | **float** | The percentage of identical matches to Ensembl protein. | [optional] 
**pct_coverage_protein** | **float** | The percentage of ORF covered by Ensembl protein. | [optional] 
**pct_coverage_orf** | **float** | The percentage of Ensembl protein covered by ORF. | [optional] 
**parent** | **str** | The parent page associated with this page. | [optional] 
**layout** | [**PageLayout**](PageLayout.md) |  | [optional] 
**canonical_uri** | **str** | The path of the page. | [optional] 
**feature** | **str** | The phenotypic feature observed for the donor. | [optional] 
**quantity** | **float** | A quantity associated with the phenotypic feature, if applicable. | [optional] 
**quantity_units** | **str** | The unit of measurement for a quantity associated with the phenotypic feature. | [optional] 
**quality** | **str** | A quality assessment associated with the phenotypic feature, such as a categorical description. | [optional] 
**observation_date** | **str** | The date the feature was observed or measured. | [optional] 
**abstract** | **str** | Abstract of the publication or communication. | [optional] 
**authors** | **str** | The authors of the publication. | [optional] 
**date_published** | **str** | The date the publication or communication was published; must be in YYYY-MM-DD format. | [optional] 
**date_revised** | **str** | The date the publication was revised. | [optional] 
**issue** | **str** | The issue of the publication. | [optional] 
**page** | **str** | Pagination of the reference | [optional] 
**volume** | **str** | The volume of the publication. | [optional] 
**journal** | **str** | The journal of the publication. | [optional] 
**publication_identifiers** | **List[str]** | The publication identifiers associated with this publication object. | [optional] 
**published_by** | **List[str]** | The affiliation of the lab with a larger organization, such as IGVF. | [optional] 
**publication_year** | **int** | The year the publication was published. | [optional] 
**file_sets** | **List[str]** | The file sets linked to this sample. | [optional] 
**software** | **str** | Unique name of the software package. | [optional] 
**quality_metric_of** | **List[str]** | The file(s) to which this quality metric applies. | [optional] 
**pearson_correlation** | **float** | The correlation of log2 RNA/DNA ratios across tested sequences as a measure of replicable activity signal. Value is the median of replicate comparisons using only oligos with &gt;&#x3D; 10 barcodes. | [optional] 
**median_barcodes_passing_filtering** | **float** | Median number of barcodes across tested sequences that passed filtering to determine if there was sufficient barcode to oligo coverage. Value is the median of all replicates. | [optional] 
**median_rna_read_count** | **float** | Median of RNA read count for oligos that passed filtering to determine sufficient coverage in terms of read count. Value is the median of all replicates. | [optional] 
**fraction_oligos_passing** | **float** | Fraction of tested sequences that passed filtering of the mappable sequences to determine if the designed library was sufficiently recovered. Value is the median of all replicates. | [optional] 
**median_assigned_barcodes** | **float** | Median number of barcodes assigned to tested sequences in mapping as a quality control measure for the assignment step, whether there is sufficient barcode to oligo coverage. | [optional] 
**fraction_assigned_oligos** | **float** | Fraction of assigned tested sequences in mapping to determine if the library during the assignment step was sufficiently recovered. | [optional] 
**total_cells_passing_filters** | **float** | Total Cells Passing Filters | [optional] 
**pct_cells_assigned_guide** | **float** | Percent Cells Assigned Guide | [optional] 
**avg_cells_per_target** | **float** | Average Cells Per Target | [optional] 
**moi** | **float** | The actual multiplicity of infection (MOI) for vectors introduced to this sample. At least one construct library set must be specified in order to specify MOI. This property should capture the actual MOI, and not the targeted MOI. | [optional] 
**avg_umis_per_cell** | **float** | Average UMIs Per Cell | [optional] 
**total_guides** | **float** | Total Guides | [optional] 
**total_targets** | **float** | Total Targets | [optional] 
**guide_diversity** | **float** | Guide diversity (Gini index) | [optional] 
**mean_mitochondrial_reads** | **float** | Mean mitochondrial reads. | [optional] 
**total_reads** | **float** | Total reads (n_processed) reported by Kallisto. | [optional] 
**paired_reads_mapped** | **float** | Paired reads mapped (n_pseudoaligned) reported by Kallisto. | [optional] 
**alignment_percentage** | **float** | Alignment percentage (p_pseudoaligned) reported by Kallisto. | [optional] 
**total_detected_scrna_barcodes** | **float** | Unfiltered total detected scRNA barcodes (numBarcodes) reported by Kallisto. | [optional] 
**pct_duplicates** | **float** |  | [optional] 
**n_reads** | **float** | Total number of input reads. | [optional] 
**n_mapped_reads** | **float** | Reads successfully aligned to the reference genome. | [optional] 
**n_uniquely_mapped_reads** | **float** | Reads aligned to a single location in the genome. | [optional] 
**n_reads_with_multi_mappings** | **float** | Reads aligned to multiple locations in the genome. | [optional] 
**n_candidates** | **float** | Potential mapping locations considered during alignment. | [optional] 
**n_mappings** | **float** | Total successful alignments, including multiple mappings per read. | [optional] 
**n_uni_mappings** | **float** | Total alignments where reads map to a single location. | [optional] 
**n_multi_mappings** | **float** | Total alignments where reads map to multiple locations. | [optional] 
**n_barcodes_on_onlist** | **float** | Barcodes matching the expected list of valid barcodes. | [optional] 
**n_corrected_barcodes** | **float** | Barcodes adjusted to match valid entries in the onlist. | [optional] 
**n_output_mappings** | **float** | Final count of fragments after deduplication and filtering. | [optional] 
**uni_mappings** | **float** | Number of fragments mapping to single location in the genome. | [optional] 
**multi_mappings** | **float** | Number of fragments mappig in multiple locations in the genome. | [optional] 
**total** | **float** | Sum of uni-mappings and multi-mappings. | [optional] 
**atac_fragments_alignment_stats** | [**ATACFragmentsAlignmentStats**](ATACFragmentsAlignmentStats.md) |  | [optional] 
**atac_bam_summary_stats** | [**ATACBamSummaryStats**](ATACBamSummaryStats.md) |  | [optional] 
**atac_fragment_summary_stats** | [**ATACFragmentSummaryStats**](ATACFragmentSummaryStats.md) |  | [optional] 
**n_records** | **float** | Number of records in BUS file. | [optional] 
**n_barcodes** | **float** | Total number of valid cell barcodes detected. | [optional] 
**total_umis** | **float** | Total number of UMIs detected. | [optional] 
**n_barcode_umis** | **float** | Total number of UMIs associated with cell barcodes. | [optional] 
**median_reads_per_barcode** | **float** | Median number of reads per cell barcode. | [optional] 
**mean_reads_per_barcode** | **float** | Mean number of reads per cell barcode. | [optional] 
**median_umis_per_barcode** | **float** | Median number of UMIs per cell barcode. | [optional] 
**mean_umis_per_barcode** | **float** | Mean number of UMIs per cell barcode. | [optional] 
**gt_records** | **float** | Number of BUS records for Good-Toulmin estimation. | [optional] 
**num_barcodes_on_onlist** | **float** | Number of cell barcodes matching an expected list of barcodes (onlist). | [optional] 
**percentage_barcodes_on_onlist** | **float** | Percentage of cell barcodes matching an expected list of barcodes (onlist). | [optional] 
**num_reads_on_onlist** | **float** | Number of reads associated with barcodes on the onlist. | [optional] 
**percentage_reads_on_onlist** | **float** | Percentage of reads associated with barcodes on the onlist. | [optional] 
**rnaseq_kb_info** | [**RNASeqKBInfo**](RNASeqKBInfo.md) |  | [optional] 
**n_targets** | **float** | Total number of target sequences (e.g., transcripts) in the index. | [optional] 
**n_bootstraps** | **float** | Number of bootstrap iterations used to estimate expression uncertainty. | [optional] 
**n_processed** | **float** | Number of valid reads processed by Kallisto. | [optional] 
**n_pseudoaligned** | **float** | Number of reads that could be pseudoaligned to the transcriptome index. | [optional] 
**n_unique** | **float** | Number of reads that could be pseudoaligned to a unique target sequence. | [optional] 
**p_pseudoaligned** | **float** | Percentage of reads that could be pseudoaligned to the transcriptome index. | [optional] 
**p_unique** | **float** | Percentage of reads that could be pseudoaligned to a unique target sequence. | [optional] 
**index_version** | **float** | Version of Kallisto index command used for building the transcriptome index. | [optional] 
**kmer_length** | **float** | Length of k-mers used for building the transcriptome index. | [optional] 
**coverage** | **float** | Coverage of the library. | [optional] 
**coverage_per_basepair** | **float** | Coverage at basepair level. | [optional] 
**rna_correlation_in_peaks** | **float** | Correlation of RNA only over regions called as peaks in DNA. | [optional] 
**annotated_from** | **str** | Links to the sample that was used as a reference for annotating this sample. | [optional] 
**lower_bound_age** | **float** | Lower bound of age of the organism at the time of collection of the sample. | [optional] 
**upper_bound_age** | **float** | Upper bound of age of the organism at the time of collection of the sample. | [optional] 
**age_units** | **str** | The units of time associated with age of the biosample. | [optional] 
**sample_terms** | **List[str]** | Ontology terms identifying a biosample. | [optional] 
**disease_terms** | **List[str]** | Ontology term of the disease associated with the biosample. | [optional] 
**pooled_from** | **List[str]** | The biosamples this biosample is pooled from. | [optional] 
**part_of** | **str** | Links to a biosample which represents a larger sample from which this sample was taken regardless of whether it is a tissue taken from an organism or smaller slices of a piece of tissue or aliquots of a cell growth. | [optional] 
**originated_from** | **str** | Links to a biosample that was originated from due to differentiation, dedifferentiation, reprogramming, or the introduction of a genetic modification. | [optional] 
**treatments** | **List[str]** | A list of treatments applied to the biosample with the purpose of perturbation. | [optional] 
**biomarkers** | **List[str]** | Biological markers that are associated with this sample. | [optional] 
**embryonic** | **bool** | Biosample is embryonic. | [optional] 
**modifications** | **List[str]** | Links to modifications applied to this biosample. | [optional] 
**cellular_sub_pool** | **str** | Cellular sub-pool fraction of the sample. Also known as PKR and sub-library. | [optional] 
**starting_amount** | **float** | The initial quantity of samples obtained. | [optional] 
**starting_amount_units** | **str** | The units used to quantify the amount of samples obtained. | [optional] 
**date_obtained** | **str** | The date the sample was harvested, dissected or created, depending on the type of the sample. | [optional] 
**sorted_from** | **str** | Links to a larger sample from which this sample was obtained through sorting. | [optional] 
**sorted_from_detail** | **str** | Detail for sample sorted into fractions capturing information about sorting. | [optional] 
**nucleic_acid_delivery** | **str** | Method of introduction of nucleic acid into the cell. | [optional] 
**time_post_library_delivery** | **float** | The time that elapsed past the time-point when the construct library sets were introduced. | [optional] 
**time_post_library_delivery_units** | **str** | The units of time that elapsed past the point when the construct library sets were introduced. | [optional] 
**classifications** | **List[str]** | The general category of this type of sample. | [optional] 
**time_post_change** | **float** | The time that elapsed past the time-point when the cell fate change treatments were introduced. | [optional] 
**time_post_change_units** | **str** | The units of time that elapsed past the point when the cell fate change treatments were introduced. | [optional] 
**cell_fate_change_protocol** | **str** | A protocol applied to the biosample with the purpose of differentiation, dedifferentiation, or reprogramming. | [optional] 
**demultiplexed_from** | **str** | The biosample this in vitro system sample was demultiplexed from using computational methods. | [optional] 
**passage_number** | **int** | Number of passages including the passages from the source. | [optional] 
**targeted_sample_term** | **str** | Ontology term identifying the targeted endpoint biosample resulting from differentation or reprogramming. | [optional] 
**growth_medium** | **str** | A growth medium of the in vitro system. | [optional] 
**biosample_qualifiers** | **List[str]** | An array of various cell states. This property provides additional information about a cell at a finer-grained level compared to what ontologies currently capture. For example, exhausted T-cells. | [optional] 
**multiplexed_in** | **List[str]** | The multiplexed samples in which this sample is included. | [optional] 
**sorted_fractions** | **List[str]** | The fractions into which this sample has been sorted. | [optional] 
**origin_of** | **List[str]** | The samples which originate from this sample, such as through a process of cell differentiation. | [optional] 
**institutional_certificates** | **List[str]** | The institutional certificates under which use of this sample is approved. | [optional] 
**age** | **str** | Age of organism at the time of collection of the sample. | [optional] 
**upper_bound_age_in_hours** | **float** | Upper bound of age of organism in hours at the time of collection of the sample. | [optional] 
**lower_bound_age_in_hours** | **float** | Lower bound of age of organism in hours at the time of collection of the sample . | [optional] 
**parts** | **List[str]** | The parts into which this sample has been divided. | [optional] 
**pooled_in** | **List[str]** | The pooled samples in which this sample is included. | [optional] 
**demultiplexed_to** | **List[str]** | The parts into which this sample has been demultiplexed. | [optional] 
**multiplexed_samples** | **List[str]** | The samples multiplexed together to produce this sample. | [optional] 
**multiplexing_methods** | **List[str]** | The methods used for multiplexing and demultiplexing. | [optional] 
**sample_material** | **str** |  | [optional] 
**pmi** | **int** | The amount of time elapsed since death. | [optional] 
**pmi_units** | **str** | The unit in which the PMI time was reported. | [optional] 
**ccf_id** | **str** | HubMap Common Coordinate Framework unique identifier corresponding to the organ, biological structure, and spatial location of the tissue specimen. | [optional] 
**preservation_method** | **str** | The method by which the tissue was preserved: cryopreservation (slow-freeze) or flash-freezing. | [optional] 
**used_by** | **List[str]** | The component(s) of the IGVF consortium that utilize this software. | [optional] 
**categories** | **List[str]** | The general categories of this software. | [optional] 
**versions** | **List[str]** | A list of versions that have been released for this software. | [optional] 
**version** | **str** | The version of a particular software. | [optional] 
**download_id** | **str** | The MD5 checksum, SHA-1 commit ID, image hash, or similar permanent identifier of the particular version of software used. | [optional] 
**amount** | **float** | Specific quantity of the applied treatment (used in conjunction with amount_units). | [optional] 
**amount_units** | **str** | A unit for an amount other than those for time or temperature. | [optional] 
**duration** | **float** | Duration indicates the time elapsed between the start and end of the treatment. | [optional] 
**duration_units** | **str** | A unit of time. | [optional] 
**p_h** | **float** | Final pH of the solution containing a chemical compound (if applicable) | [optional] 
**purpose** | **str** | The intended purpose for treating the samples. | [optional] 
**post_treatment_time** | **float** | Post treatment time in conjunction with post treatment time units is used to specify the time that has passed between the point when biosamples were removed from the treatment solution before being sampled or treated with the next treatment. | [optional] 
**post_treatment_time_units** | **str** | A unit of time. | [optional] 
**temperature** | **float** | The temperature in Celsius to which the sample was exposed | [optional] 
**temperature_units** | **str** | A unit of temperature. | [optional] 
**treatment_type** | **str** | The classification of treatment agent that specifies its exact molecular nature.  | [optional] 
**treatment_term_id** | **str** | Ontology identifier describing a component in the treatment. | [optional] 
**treatment_term_name** | **str** | Ontology term describing a component in the treatment that is the principal component affecting the biosample being treated. Examples: interferon gamma, interleukin-4, Fibroblast growth factor 2, 20-hydroxyecdysone, 5-bromouridine etc. | [optional] 
**depletion** | **bool** | Treatment is depleted. | [optional] 
**biosamples_treated** | **List[str]** | The samples which have been treated using this treatment. | [optional] 
**email** | **str** | The email associated with the user&#39;s account. | [optional] 
**first_name** | **str** | The user&#39;s first (given) name. | [optional] 
**last_name** | **str** | The user&#39;s last (family) name. | [optional] 
**submits_for** | **List[str]** | Labs user is authorized to submit data for. | [optional] 
**groups** | **List[str]** | Additional access control groups | [optional] 
**viewing_groups** | **List[str]** | The group that determines which set of data the user has permission to view. | [optional] 
**job_title** | **str** | The role of the user in their lab or organization. | [optional] 
**workflow_repositories** | **List[str]** | Resources hosting the workflow. | [optional] 
**standards_page** | **str** | A link to a page describing the standards for this workflow. | [optional] 
**workflow_version** | **int** | The version of this workflow. | [optional] 
**uniform_pipeline** | **bool** | Indicates whether the pipeline is developed by the IGVF consortium. | [optional] 
**analysis_steps** | **List[str]** | The analysis steps which are part of this workflow. | [optional] 

## Example

```python
from igvf_client.models.search_result_item import SearchResultItem

# TODO update the JSON string below
json = "{}"
# create an instance of SearchResultItem from a JSON string
search_result_item_instance = SearchResultItem.from_json(json)
# print the JSON string representation of the object
print(SearchResultItem.to_json())

# convert the object into a dict
search_result_item_dict = search_result_item_instance.to_dict()
# create an instance of SearchResultItem from a dict
search_result_item_from_dict = SearchResultItem.from_dict(search_result_item_dict)
```
[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)


